// @adyen/adyen-platform-experience-web/dist/es/core/ConfigContext/context.js
import { jsx as s } from "../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { toChildArray as x, createContext as b } from "../../external/preact/dist/preact.module.js";
import { useContext as g, useMemo as E, useState as e, useEffect as n } from "../../external/preact/hooks/dist/hooks.module.js";
import { ErrorMessageDisplay as v } from "../../components/internal/ErrorMessageDisplay/ErrorMessageDisplay.js";
import P from "./session/utils/sessionAwareComponentAvailability/sessionAwareComponentAvailability.js";
import { isWatchlistUnsubscribeToken as U } from "../../primitives/reactive/watchlist/main.js";
import y from "./session/utils/sessionAwareComponentAvailability/helpers/componentAvailabilityErrors.js";
import { noop as A, asyncNoop as T } from "../../utils/common.js";
import { isUndefined as M } from "../../utils/value/is.js";
import { EMPTY_OBJECT as i } from "../../utils/value/constants.js";
const a = b({
  endpoints: i,
  extraConfig: i,
  hasError: !1,
  http: T,
  isExpired: void 0,
  isFrozen: !1,
  refresh: A,
  refreshing: !1
}), J = ({ children: m, session: r, type: t }) => {
  const { http: c, refresh: f } = E(() => r, [r]), [, p] = e(0), [u, C] = e(0), [o, d] = e();
  return n(() => {
    P(t, r).then(d);
  }, [r, t]), n(() => r.subscribe((l) => {
    (U(l) ? C : p)((h) => h + 1);
  }), [u]), /* @__PURE__ */ s(a.Provider, { value: { ...r.context, http: c, refresh: f }, children: !M(o) && (o ? x(m) : /* @__PURE__ */ s(
    v,
    {
      withImage: !0,
      centered: !0,
      title: "common.errors.somethingWentWrong",
      message: [y(t), "common.errors.contactSupport"]
    }
  )) });
}, K = () => g(a);
export {
  J as ConfigProvider,
  K as default,
  K as useConfigContext
};
