// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/CurrencyInput/CurrencyInput.js
import { jsx as x } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useRef as E, useState as Y, useCallback as D, useEffect as z, useMemo as F } from "../../../../external/preact/hooks/dist/hooks.module.js";
import V from "../InputBase.js";
import j from "../../../../core/Context/useCoreContext.js";
import { CURRENCY_INPUT_BASE_CLASS as k } from "./constants.js";
import { useFetch as G } from "../../../../hooks/useFetch.js";
import { formatAmount as w, getCurrencyExponent as v } from "../../../../utils/currency/main.js";
import { useConfigContext as J } from "../../../../core/ConfigContext/context.js";
import { EMPTY_OBJECT as Q } from "../../../../utils/value/constants.js";
import { uniqueId as A } from "../../../../utils/random/uniqueId.js";
const W = ["EUR", "GBP", "USD"], Z = (a) => {
  const e = new Map(W.map((n, l) => [n, l]));
  return [...a].sort((n, l) => {
    const f = n.id.toUpperCase(), g = l.id.toUpperCase(), s = e.get(f), d = e.get(g);
    return s !== void 0 || d !== void 0 ? s === void 0 ? 1 : d === void 0 ? -1 : s - d : f.localeCompare(g);
  });
}, at = ({
  amount: a,
  currency: e,
  currencyItems: n,
  hideCurrencySelector: l,
  interactionsDisabled: f,
  isInvalid: g = !1,
  maxValue: s,
  name: d,
  onAmountChange: B,
  onCurrencyChange: P,
  selectedCurrencyCode: S
}) => {
  const T = E(e), { i18n: C } = j(), { currencies: c } = J().endpoints, [q, R] = Y(a ? `${w(a, e)}` : ""), M = D(
    (r) => {
      var m;
      const t = ((m = 1.1.toLocaleString(C.locale).match(/\d(.*?)\d/)) == null ? void 0 : m[1]) || ".", o = t === "." ? r : r.replace(t, "."), i = v(e);
      return Math.trunc(+`${parseFloat(o)}e${i}`) || 0;
    },
    [e, C.locale]
  );
  z(() => {
    e !== T.current && (R(a ? `${w(a, e)}` : ""), T.current = e);
  }, [a, e]);
  const O = (r) => {
    var m;
    let t = r.currentTarget.value.trim();
    const o = ((m = 1.1.toLocaleString(C.locale).match(/\d(.*?)\d/)) == null ? void 0 : m[1]) || ".", i = t.split(o);
    if (i.length === 2) {
      const h = v(e), b = i[0];
      let p = i[1];
      p.length >= h && (p = p.substring(0, h), t = b + o + p, r.currentTarget.value = t);
    }
    if (t.endsWith(o) && (t = t.slice(0, -o.length), r.currentTarget.value = t), typeof s == "number") {
      const h = o === "." ? t : t.replace(o, "."), b = parseFloat(h);
      if (Number.isFinite(b) && b > s) {
        const p = v(e), U = s.toFixed(p);
        t = o === "." ? U : U.replace(".", o), r.currentTarget.value = t;
      }
    }
    R(t), B(M(t));
  }, u = G({
    fetchOptions: { enabled: !!c && !(n != null && n.length) },
    queryFn: D(async () => c == null ? void 0 : c(Q), [c])
  }), _ = F(() => {
    var o;
    return (((o = u.data) == null ? void 0 : o.data) ?? []).map((i) => ({
      id: i,
      name: i
    }));
  }, [u.data]), I = F(() => {
    const r = n != null && n.length ? n : _;
    return Z(r);
  }, [_, n]), N = E(A()), $ = E(A()), L = F(() => l ? {} : {
    onDropdownInput: P,
    dropdown: {
      items: I,
      value: S,
      readonly: u.isFetching
    }
  }, [l, I, S, u.isFetching, P]);
  return /* @__PURE__ */ x("div", { className: "adyen-pe-currency-input__container", children: /* @__PURE__ */ x("label", { htmlFor: N.current, "aria-labelledby": $.current, children: /* @__PURE__ */ x(
    V,
    {
      ...L,
      type: "number",
      className: k,
      disabled: f || u.isFetching,
      isInvalid: g,
      lang: C.locale,
      min: 0,
      name: d,
      onInput: O,
      uniqueId: N.current,
      value: q
    }
  ) }) });
};
export {
  at as CurrencyInput
};
