// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/PaymentLinkCreationFormContainer/FormStepRenderer.js
import { jsx as r } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { CustomerDetailsForm as y } from "../Form/CustomerDetailsForm/CustomerDetailsForm.js";
import { PaymentDetailsForm as t } from "../Form/PaymentDetailsForm/PaymentDetailsForm.js";
import { FormSummary as S } from "../Form/Summary/FormSummary.js";
import { StoreForm as x } from "../Form/StoreForm/StoreForm.js";
const P = ({
  setShowTermsAndConditions: u,
  currentFormStep: i,
  settingsData: n,
  storeIds: p,
  storesData: m,
  selectItems: c,
  termsAndConditionsProvisioned: f,
  configurationData: e,
  isSeparateAddress: s,
  setIsSeparateAddress: F,
  countriesData: a,
  isFetchingCountries: d,
  countryDatasetData: o,
  isFetchingCountryDataset: l
}) => {
  switch (i) {
    case "store":
      return /* @__PURE__ */ r(
        x,
        {
          setShowTermsAndConditions: u,
          settingsData: n,
          storeIds: p,
          storesData: m == null ? void 0 : m.data,
          selectItems: c,
          termsAndConditionsProvisioned: f
        }
      );
    case "payment":
      return /* @__PURE__ */ r(t, { configuration: e });
    case "customer":
      return /* @__PURE__ */ r(
        y,
        {
          isSeparateAddress: s,
          setIsSeparateAddress: F,
          countriesData: a,
          isFetchingCountries: d,
          countryDatasetData: o,
          isFetchingCountryDataset: l
        }
      );
    case "summary":
      return /* @__PURE__ */ r(S, { countryDatasetData: o });
    default:
      return /* @__PURE__ */ r(t, { configuration: e });
  }
};
export {
  P as FormStepRenderer
};
