// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/PaymentLinkCreationFormContainer/PaymentLinkCreationFormContainer.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import A from "../../../../internal/Typography/Typography.js";
import Ee from "../../../../../core/Context/useCoreContext.js";
import { TypographyVariant as F, TypographyElement as Re } from "../../../../internal/Typography/types.js";
import { Stepper as De } from "../../../../internal/Stepper/Stepper.js";
import { useState as x, useRef as P, useCallback as m, useEffect as xe, useMemo as h } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { WizardFormProvider as Pe } from "../../../../../hooks/form/wizard/WizardFormContext.js";
import { ButtonVariant as I } from "../../../../internal/Button/types.js";
/* empty css                             */
/* empty css                             */
import Me from "../../../../../hooks/useMutation/useMutation.js";
import { usePaymentLinkFormData as Oe } from "./usePaymentLinkFormData.js";
import { scrollToFirstErrorField as Ve } from "../../utils.js";
import { useResponsiveContainer as We, containerQueries as $e } from "../../../../../hooks/useResponsiveContainer.js";
import { FormStepRenderer as Ye } from "./FormStepRenderer.js";
import je from "../../../PaymentLinkSettings/components/PaymentLinkSettingsContainer/PaymentLinkSettingsContainer.js";
import { AlertTypeOption as U } from "../../../../internal/Alert/types.js";
import { Alert as H } from "../../../../internal/Alert/Alert.js";
import { ErrorMessageDisplay as qe } from "../../../../internal/ErrorMessageDisplay/ErrorMessageDisplay.js";
import { useInvalidFieldsConfig as ze } from "../../hooks/useInvalidFieldsConfig.js";
import Ge from "../../../../internal/CopyText/CopyText.js";
import { Translation as Ke } from "../../../../internal/Translation/Translation.js";
import b from "../../../../internal/Button/Button.js";
import { Fragment as X } from "../../../../../external/preact/dist/preact.module.js";
import { Icon as Ue } from "../../../../internal/Icon/Icon.js";
const Q = 28, He = () => /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__skeleton", children: [
  /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__skeleton-item adyen-pe-payment-link-creation-form__skeleton-item--large" }),
  [...Array(3)].map((i, d) => /* @__PURE__ */ e(X, { children: [
    /* @__PURE__ */ e(
      "div",
      {
        className: "adyen-pe-payment-link-creation-form__skeleton-item adyen-pe-payment-link-creation-form__skeleton-item--small"
      },
      `${d}-small`
    ),
    /* @__PURE__ */ e(
      "div",
      {
        className: "adyen-pe-payment-link-creation-form__skeleton-item adyen-pe-payment-link-creation-form__skeleton-item--large"
      },
      `${d}-large`
    )
  ] }))
] }), St = ({
  fieldsConfig: i,
  storeIds: d,
  onCreationDismiss: g,
  onPaymentLinkCreated: L,
  onContactSupport: p,
  embeddedInOverview: J
}) => {
  var q;
  const [Z, ee] = x(!1), M = P(null), O = P(null), te = !!((q = i == null ? void 0 : i.data) != null && q.billingAddress), [re, ne] = x(te), k = P(""), { i18n: r } = Ee(), v = We($e.down.xs), {
    storesData: u,
    configurationData: B,
    countriesData: oe,
    isFetchingCountries: ae,
    countryDatasetData: ie,
    isFetchingCountryDataset: se,
    settingsData: ce,
    storesSelectorItems: le,
    termsAndConditionsProvisioned: w,
    formSteps: _,
    stepperItems: me,
    formStepsAriaLabel: de,
    wizardForm: y,
    createPaymentLink: pe,
    isDataLoading: ue,
    isFirstLoadDone: ye,
    selectedStore: V,
    setSelectedStore: S
  } = Oe({
    defaultValues: i == null ? void 0 : i.data,
    storeIds: d
  }), { isLastStep: f, isFirstStep: W, currentStep: l, validateStep: $, canGoNext: fe, isStepComplete: he, nextStep: Y, previousStep: ge, goToStep: j } = y, [ke, T] = x(!1), ve = m(
    async (t) => {
      var o;
      if (!f) {
        if (!await $(t)) {
          const n = ((o = O.current) == null ? void 0 : o.getBoundingClientRect().height) ?? 0, c = v ? n + Q : Q;
          Ve(Object.keys(y.formState.errors), c);
          return;
        }
        await Y();
      }
    },
    [f, Y, $, y.formState.errors, v]
  );
  xe(() => {
    var t;
    v && ((t = M.current) == null || t.scrollIntoView({ block: "start" }));
  }, [l, v]);
  const _e = m(
    (t) => {
      j(t);
    },
    [j]
  ), C = h(() => {
    const t = _ == null ? void 0 : _[l];
    return t ? t.id : "store";
  }, [l, _]), Se = () => {
    if (W) {
      g == null || g();
      return;
    }
    s.reset(), ge();
  }, Te = async () => {
    await ve(l);
  }, s = Me({
    queryFn: pe
  }), Ne = async (t) => {
    const { store: o, ...a } = t;
    try {
      const n = await s.mutate(
        {
          body: a,
          contentType: "application/json"
        },
        { path: { storeId: o } }
      );
      L == null || L({ ...t, paymentLink: n });
    } catch (n) {
      throw console.error("Failed to create payment link:", n), n;
    }
  }, Ae = m(() => {
    S(k.current), T(!1), k.current = "";
  }, [T, S]), Fe = (t) => {
    ee(!0);
  }, Ie = s.isLoading || ue, E = h(() => u && u.data && u.data.length === 0, [u]), R = h(() => C !== "store" && !B, [B, C]), be = h(() => !w || E || R, [E, R, w]), { invalidFieldsConfig: D } = ze(), N = m(
    (t) => {
      var o;
      return !t || !("invalidFields" in t) || !((o = t.invalidFields) != null && o.length) ? [] : t.invalidFields.map((a) => {
        var G, K;
        const n = (G = D.fields) == null ? void 0 : G[a.name], c = (K = D.messages) == null ? void 0 : K[a.message];
        if (!n && !c) return null;
        const z = n ? r.get(n) : a.name;
        if (!c) return `${z}`;
        const Ce = r.get(c);
        return `${z} (${Ce})`;
      }).filter((a) => a !== null);
    },
    [r, D]
  ), Le = m(() => {
    k.current = V, S(""), T(!0);
  }, [V, S, T]), Be = m(
    (t) => N(t).length ? r.get("payByLink.creation.form.alert.invalidFields") : r.get("payByLink.creation.form.alert.somethingWentWrong"),
    [N, r]
  ), we = h(() => {
    const t = s.error, o = N(t), a = o.length > 0;
    return /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__error-alert", children: [
      a && /* @__PURE__ */ e("ul", { className: "adyen-pe-payment-link-creation-form__invalid-fields-error", children: o.map((n, c) => /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(A, { variant: F.CAPTION, children: n }) }, c)) }),
      p && /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__contact-support", children: /* @__PURE__ */ e(
        Ke,
        {
          translationKey: "payByLink.creation.form.error.submit.contactSupport",
          fills: {
            contactSupport: /* @__PURE__ */ e(b, { variant: I.TERTIARY, onClick: p, children: r.get("common.actions.contactSupport.labels.reachOut") }),
            errorCode: /* @__PURE__ */ e(Ge, { stronger: !0, textToCopy: (t == null ? void 0 : t.requestId) || (t == null ? void 0 : t.errorCode) })
          }
        }
      ) })
    ] });
  }, [N, r, p, s.error]);
  return ye ? ke ? /* @__PURE__ */ e(
    je,
    {
      hideTitle: !0,
      storeIds: k.current,
      settingsItems: ["termsAndConditions"],
      navigateBack: Ae,
      embeddedInOverview: J
    }
  ) : /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__component", ref: M, children: [
    /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__header", ref: O, children: [
      /* @__PURE__ */ e(A, { variant: F.SUBTITLE, stronger: !0, children: r.get("payByLink.creation.form.title") }),
      /* @__PURE__ */ e(
        De,
        {
          nextStepDisabled: !fe || !he(l),
          variant: "horizontal",
          activeIndex: l,
          ariaLabel: de,
          onChange: _e,
          children: me.map((t) => /* @__PURE__ */ e(X, { children: t.label }))
        }
      )
    ] }),
    /* @__PURE__ */ e(Pe, { ...y, children: /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__container", children: /* @__PURE__ */ e(
      "form",
      {
        className: "adyen-pe-payment-link-creation-form",
        onSubmit: (t) => {
          t.preventDefault(), y.handleSubmit(Ne, Fe)(t);
        },
        children: [
          /* @__PURE__ */ e("div", { children: /* @__PURE__ */ e(
            Ye,
            {
              setShowTermsAndConditions: Le,
              currentFormStep: C,
              settingsData: ce,
              storeIds: d,
              storesData: u,
              selectItems: le,
              termsAndConditionsProvisioned: w,
              configurationData: B,
              isSeparateAddress: re,
              setIsSeparateAddress: ne,
              countriesData: oe,
              isFetchingCountries: ae,
              countryDatasetData: ie,
              isFetchingCountryDataset: se
            }
          ) }),
          (R || Z) && /* @__PURE__ */ e(
            qe,
            {
              condensed: !0,
              title: "common.errors.somethingWentWrong",
              withImage: !0,
              absolutePosition: !1,
              outlined: !1,
              withBackground: !1,
              message: ["payByLink.creation.errors.unavailable", "common.errors.retry"]
            }
          ),
          E && /* @__PURE__ */ e(
            H,
            {
              type: U.WARNING,
              title: r.get("payByLink.common.errors.accountConfiguration"),
              description: /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__warning-alert", children: [
                /* @__PURE__ */ e(A, { variant: F.CAPTION, el: Re.SPAN, children: r.get("common.errors.contactSupport") }),
                p ? /* @__PURE__ */ e(b, { variant: I.TERTIARY, onClick: p, children: r.get("common.actions.contactSupport.labels.reachOut") }) : null
              ] })
            }
          ),
          s.isError && /* @__PURE__ */ e(
            H,
            {
              type: U.CRITICAL,
              title: Be(s.error),
              description: we
            }
          ),
          /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__buttons-container", children: [
            (!W || g) && /* @__PURE__ */ e(b, { variant: I.SECONDARY, onClick: Se, children: r.get("payByLink.creation.form.steps.back") }),
            /* @__PURE__ */ e(
              b,
              {
                className: "adyen-pe-payment-link-creation-form__submit-button",
                type: f ? "submit" : "button",
                variant: I.PRIMARY,
                onClick: f ? void 0 : Te,
                state: Ie ? "loading" : void 0,
                disabled: be,
                iconRight: /* @__PURE__ */ e(Ue, { name: "arrow-right" }),
                children: f ? r.get("payByLink.creation.form.steps.submit") : r.get("payByLink.creation.form.steps.continue")
              }
            )
          ] })
        ]
      }
    ) }) })
  ] }) : /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__component", children: /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__header", children: [
    /* @__PURE__ */ e(A, { variant: F.SUBTITLE, stronger: !0, children: r.get("payByLink.creation.form.title") }),
    /* @__PURE__ */ e(He, {})
  ] }) });
};
export {
  St as PaymentLinkCreationFormContainer
};
