// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/PaymentLinkCreationFormContainer/usePaymentLinkFormData.js
import { useState as j, useCallback as m, useMemo as l, useEffect as q } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { useFetch as h } from "../../../../../hooks/useFetch.js";
import { getFormSteps as J } from "../../utils.js";
import Q from "../../../../../core/Context/useCoreContext.js";
import { useWizardForm as W } from "../../../../../hooks/form/wizard/useWizardForm.js";
import { useConfigContext as Y } from "../../../../../core/ConfigContext/context.js";
import { EMPTY_OBJECT as S } from "../../../../../utils/value/constants.js";
const re = ({ storeIds: p, defaultValues: r }) => {
  const [n, L] = j(""), { i18n: o, getCdnDataset: C } = Q(), {
    countries: y,
    getPayByLinkConfiguration: u,
    createPBLPaymentLink: E,
    getPayByLinkSettings: F,
    getPayByLinkStores: g
  } = Y().endpoints, { data: b, isFetching: k } = h({
    fetchOptions: { enabled: !!g },
    queryFn: m(async () => g == null ? void 0 : g(S, {}), [g])
  }), s = l(() => ((b == null ? void 0 : b.data) ?? []).filter(({ storeId: e }) => p ? Array.isArray(p) && e ? p.includes(e) : p === e : !0).map(({ storeCode: e, storeId: $ }) => ({
    id: $ || "",
    name: e || ""
  })), [b, p]), {
    data: a,
    isFetching: x,
    error: D
  } = h({
    fetchOptions: { enabled: !!u && !!n },
    queryFn: m(async () => u == null ? void 0 : u(S, { path: { storeId: n } }), [u, n])
  }), {
    data: i,
    isFetching: B,
    error: R
  } = h({
    fetchOptions: { enabled: !!F && !!n },
    queryFn: m(async () => F == null ? void 0 : F(S, { path: { storeId: n } }), [F, n])
  }), A = l(() => !!(i != null && i.termsOfServiceUrl), [i == null ? void 0 : i.termsOfServiceUrl]), c = m(
    (t) => a == null ? void 0 : a[t],
    [a]
  ), O = l(() => !!(c("deliveryAddress") || c("billingAddress") || c("countryCode")), [c("deliveryAddress"), c("billingAddress"), c("countryCode")]), { data: P, isFetching: T } = h({
    fetchOptions: { enabled: O && !!y },
    queryFn: m(async () => y == null ? void 0 : y(S), [y])
  }), { data: U, isFetching: w } = h({
    fetchOptions: { enabled: O },
    queryFn: m(async () => {
      const t = `${o.locale ?? "en-US"}`;
      return C ? await C({
        name: t,
        extension: "json",
        subFolder: "countries",
        fallback: []
      }) ?? [] : [];
    }, [C, o.locale, O])
  }), f = l(() => {
    const t = s.length === 1 && A;
    return J({ i18n: o, getFieldConfig: c }).filter((e) => !(e.id === "store" && t));
  }, [c, o, s, A]), z = l(() => f.map((t) => ({
    id: t.id,
    label: o.get(`payByLink.creation.form.steps.${t.id}`)
  })), [f, o]), M = l(() => o.get("payByLink.creation.steps.a11y.label"), [o]), d = W({
    i18n: o,
    steps: f,
    defaultValues: {
      ...r,
      billingAddress: (r == null ? void 0 : r.billingAddress) || (r == null ? void 0 : r.deliveryAddress) || {},
      store: n || (r == null ? void 0 : r.store) || ""
    },
    mode: "all",
    validateBeforeNext: !0
  });
  q(() => {
    var t, e;
    s.length === 1 && (d.setValue("store", (t = s[0]) == null ? void 0 : t.id), d.setFieldDisplayValue("store", (e = s[0]) == null ? void 0 : e.name));
  }, [s]), q(() => d.control.subscribe(() => {
    const e = d.control.getValue("store");
    e && e !== n && L(e);
  }), [d.control, n]);
  const v = x || B || k, N = s.length === 1;
  return {
    // Query data
    storesData: b,
    configurationData: a,
    settingsData: i,
    // Store data
    selectedStore: n,
    storesSelectorItems: s,
    termsAndConditionsProvisioned: A,
    // Countries data
    countriesData: P,
    isFetchingCountries: T,
    // Countries dataset
    countryDatasetData: U,
    isFetchingCountryDataset: w,
    // Form steps
    formSteps: f,
    stepperItems: z,
    formStepsAriaLabel: M,
    // Wizard form
    wizardForm: d,
    // Endpoints
    createPaymentLink: E,
    // Loading state
    isDataLoading: v,
    isFirstLoadDone: !!(!k && (!N || !v && (a || D) && (!v && (i || R)))),
    setSelectedStore: L
  };
};
export {
  re as usePaymentLinkFormData
};
