// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/components/GrantItem/GrantItem.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as p, useCallback as C } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import y from "classnames";
import h from "../../../../../core/Context/useCoreContext.js";
import N from "../../../../../core/Context/analytics/useAnalyticsContext.js";
import E from "../../../../../hooks/useTimezoneAwareDateFormatting.js";
import l from "../../../../internal/Typography/Typography.js";
import { TypographyVariant as c, TypographyElement as A } from "../../../../internal/Typography/types.js";
import { Tag as f } from "../../../../internal/Tag/Tag.js";
import { GRANT_ITEM_CLASS_NAMES as i } from "./constants.js";
import { getGrantConfig as I } from "./utils.js";
/* empty css               */
import { GrantDetails as x } from "../GrantDetails/GrantDetails.js";
import S from "../../../../internal/CopyText/CopyText.js";
import { Tooltip as L } from "../../../../internal/Tooltip/Tooltip.js";
import { Alert as w } from "../../../../internal/Alert/Alert.js";
import { AlertTypeOption as B } from "../../../../internal/Alert/types.js";
import { ButtonVariant as P } from "../../../../internal/Button/types.js";
import G from "../../../../internal/ExpandableCard/ExpandableCard.js";
import { GrantActions as O } from "../GrantActions/GrantActions.js";
import { sharedCapitalOverviewAnalyticsEventProperties as R } from "../../constants.js";
import { Fragment as T } from "../../../../../external/preact/dist/preact.module.js";
import { Translation as V } from "../../../../internal/Translation/Translation.js";
import { DATE_FORMAT_CAPITAL_OVERVIEW as K } from "../../../../../constants/dateFormats.js";
import $ from "../../../../internal/ProgressBar/ProgressBar.js";
import D from "../../../../internal/Button/Button.js";
import { uniqueId as d } from "../../../../../utils/random/uniqueId.js";
const pt = ({ grant: r, showDetails: s }) => {
  const { i18n: a } = h(), { dateFormat: u } = E(), m = N(), e = p(() => I(r), [r]), g = C(() => {
    var o;
    try {
      return s == null ? void 0 : s("unscheduledRepayment");
    } finally {
      (o = m.addEvent) == null || o.call(m, "Clicked button", {
        ...R,
        subCategory: "Grant active",
        label: "Send repayment"
      });
    }
  }, [s, m]), n = p(
    () => ({
      grantAmount: d("elem"),
      grantAmountLabel: d("elem"),
      grantStatus: d("elem"),
      termEnds: d("elem")
    }),
    []
  ), b = p(
    () => {
      var o, v;
      return /* @__PURE__ */ t("div", { className: i.cardContent, children: [
        /* @__PURE__ */ t("div", { className: i.statusContainer, children: [
          /* @__PURE__ */ t(
            l,
            {
              id: n.grantAmountLabel,
              variant: c.CAPTION,
              className: y({ [i.textSecondary]: e.isLabelColorSecondary }),
              testId: "grant-amount-label",
              children: a.get(e.amountLabelKey)
            }
          ),
          /* @__PURE__ */ t("div", { id: n.grantStatus, children: r.status === "Active" ? /* @__PURE__ */ t(T, { children: /* @__PURE__ */ t(l, { id: n.termEnds, variant: c.CAPTION, el: A.SPAN, children: /* @__PURE__ */ t(
            V,
            {
              translationKey: "capital.overview.grants.item.termEnds",
              fills: {
                date: /* @__PURE__ */ t(
                  "time",
                  {
                    "aria-labelledBy": n.termEnds,
                    dateTime: e.repaymentPeriodEndDate.toISOString(),
                    children: /* @__PURE__ */ t(l, { variant: c.CAPTION, stronger: !0, el: A.SPAN, children: u(e.repaymentPeriodEndDate, K) })
                  }
                )
              }
            }
          ) }) }) : e.statusKey ? e.statusTooltipKey ? /* @__PURE__ */ t(L, { content: a.get(e.statusTooltipKey), children: /* @__PURE__ */ t("div", { children: /* @__PURE__ */ t(f, { label: a.get(e.statusKey), variant: e.statusTagVariant }) }) }) : /* @__PURE__ */ t(f, { label: a.get(e.statusKey), variant: e.statusTagVariant }) : null })
        ] }),
        /* @__PURE__ */ t(
          l,
          {
            id: n.grantAmount,
            variant: c.TITLE,
            medium: !0,
            className: y({
              [i.textSecondary]: e.isAmountColorSecondary
            }),
            children: a.amount(e.amount.value, e.amount.currency)
          }
        ),
        e.isProgressBarVisible && /* @__PURE__ */ t(
          $,
          {
            className: i.progressBar,
            value: r.repaidTotalAmount.value,
            max: r.totalAmount.value,
            labels: {
              ariaLabel: a.get("capital.overview.grants.item.progressBar.a11y.label"),
              current: a.get("capital.overview.grants.item.amounts.repaid"),
              max: a.get("capital.overview.grants.item.amounts.remaining")
            },
            tooltips: {
              remaining: `${a.amount(r.remainingTotalAmount.value, r.remainingTotalAmount.currency)} ${(o = a.get("capital.overview.grants.item.amounts.remaining")) == null ? void 0 : o.toLowerCase()}`,
              progress: `${a.amount(r.repaidTotalAmount.value, r.repaidTotalAmount.currency)} ${(v = a.get("capital.overview.grants.item.amounts.repaid")) == null ? void 0 : v.toLowerCase()}`
            }
          }
        ),
        e.isGrantIdVisible ? /* @__PURE__ */ t("div", { className: i.grantID, children: /* @__PURE__ */ t(
          S,
          {
            textToCopy: r.id,
            visibleText: a.get("capital.common.fields.grantID"),
            copyButtonAriaLabelKey: "capital.overview.grants.item.actions.copyGrantID",
            isUnderlineVisible: !0,
            type: "Text",
            "data-testid": "grant-id-copy-text"
          }
        ) }) : null,
        e.hasAlerts ? /* @__PURE__ */ t(T, { children: r.missingActions && r.missingActions.length ? /* @__PURE__ */ t(
          O,
          {
            missingActions: r.missingActions,
            className: i.alert,
            offerExpiresAt: r.offerExpiresAt
          }
        ) : /* @__PURE__ */ t(
          w,
          {
            className: i.alert,
            type: B.HIGHLIGHT,
            title: a.get("capital.overview.grants.item.alerts.processingRequest")
          }
        ) }) : e.hasUnscheduledRepaymentDetails && /* @__PURE__ */ t("div", { className: i.actionsBar, children: /* @__PURE__ */ t(
          D,
          {
            onClick: g,
            className: i.mainActionBtn,
            variant: P.SECONDARY,
            fullWidth: !0,
            children: a.get("capital.overview.grants.item.actions.sendRepayment")
          }
        ) })
      ] });
    },
    [a, u, r, e, g]
  );
  return /* @__PURE__ */ t("div", { className: i.base, children: /* @__PURE__ */ t(
    G,
    {
      "aria-describedby": `${n.grantAmountLabel} ${n.grantAmount} ${n.grantStatus}`,
      "aria-label": a.get("capital.overview.grants.item.details.a11y.label"),
      filled: e.isBackgroundFilled,
      renderContent: b,
      inFlow: !0,
      children: e.hasDetails && /* @__PURE__ */ t(x, { grant: r })
    }
  ) });
};
export {
  pt as GrantItem
};
