// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/components/GrantItem/utils.js
import { TagVariant as a } from "../../../../internal/Tag/types.js";
const o = (e) => e === "Active", r = (e) => e === "Repaid", c = (e) => e === "Active" ? "capital.overview.grants.item.amounts.remaining" : "capital.overview.grants.item.amounts.requestedFunds", u = (e) => e.status === "Active" ? e.remainingTotalAmount : e.grantAmount, d = ({ status: e, missingActions: t }) => {
  switch (e) {
    case "Active":
      return;
    case "Failed":
      return "capital.overview.grants.common.statuses.failed";
    case "Pending":
      return t && t.length ? "capital.overview.grants.common.statuses.actionNeeded" : "capital.overview.grants.common.statuses.pending";
    case "Repaid":
      return "capital.overview.grants.common.statuses.fullyRepaid";
    case "Revoked":
      return "capital.overview.grants.common.statuses.revoked";
    case "WrittenOff":
      return "capital.overview.grants.common.statuses.writtenOff";
  }
}, l = ({ status: e, missingActions: t }) => {
  switch (e) {
    case "Failed":
      return a.ERROR;
    case "Pending":
      return t != null && t.length ? a.WARNING : a.DEFAULT;
    case "Repaid":
      return a.LIGHT;
    case "Revoked":
    case "WrittenOff":
      return a.WARNING;
    default:
      return a.DEFAULT;
  }
}, m = (e) => {
  const t = /* @__PURE__ */ new Date(), s = /* @__PURE__ */ new Date();
  return s.setDate(t.getDate() + e), s;
}, v = (e) => {
  var s, n;
  const t = ((s = e.missingActions) == null ? void 0 : s.some((i) => i.type === "signToS")) || !1;
  switch (e.status) {
    case "Pending":
      return (n = e.missingActions) != null && n.length ? t ? "capital.overview.grants.common.statuses.pending.description.signTerms" : void 0 : "capital.overview.grants.common.statuses.pending.description.awaitingFunds";
    case "Failed":
      return "capital.overview.grants.common.statuses.failed.description";
    case "WrittenOff":
      return "capital.overview.grants.common.statuses.writtenOff.description";
    case "Revoked":
      return "capital.overview.grants.common.statuses.revoked.description";
    default:
      return;
  }
}, g = (e) => {
  var n;
  const t = e.status === "Active", s = e.status === "Pending";
  return {
    amount: u(e),
    amountLabelKey: c(e.status),
    hasAlerts: s,
    hasDetails: o(e.status),
    hasUnscheduledRepaymentDetails: t && !!((n = e.unscheduledRepaymentAccounts) != null && n.length),
    // The grant revocation account details is currently not ready to be rendered.
    // A future iteration of this component might include revocation account details.
    // Only then should the following line be uncommented.
    //
    // hasRevocationDetails: isGrantActive && grant.revocationAccount !== undefined,
    isAmountColorSecondary: !t,
    isBackgroundFilled: r(e.status),
    isGrantIdVisible: !t,
    isLabelColorSecondary: t,
    isProgressBarVisible: t,
    repaymentPeriodEndDate: m(e.repaymentPeriodLeft),
    statusKey: d(e),
    statusTagVariant: l(e),
    statusTooltipKey: v(e)
  };
};
export {
  g as getGrantConfig,
  v as getStatusTooltipKey
};
