// @adyen/adyen-platform-experience-web/dist/es/components/external/PayoutsOverview/components/PayoutsTable/PayoutsTable.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import P from "classnames";
import Y from "../../../../../core/Context/useCoreContext.js";
import { getCurrencyCode as M } from "../../../../../core/Localization/amount/amount-util.js";
import { useMemo as l, useCallback as B } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import N from "../../../../../hooks/useTimezoneAwareDateFormatting.js";
import U from "../../../../internal/DataOverviewError/DataOverviewError.js";
import { TypographyVariant as m, TypographyElement as u } from "../../../../internal/Typography/types.js";
import a from "../../../../internal/Typography/Typography.js";
import { useResponsiveContainer as j, containerQueries as x } from "../../../../../hooks/useResponsiveContainer.js";
import { BASE_CLASS as K, NET_PAYOUT_CLASS as I } from "./constants.js";
/* empty css                  */
import { useTableColumns as R } from "../../../../../hooks/useTableColumns.js";
import v from "../../../../internal/DataGrid/DataGrid.js";
import $ from "../../../../internal/Pagination/Pagination.js";
import { DATE_FORMAT_PAYOUTS_MOBILE as G, DATE_FORMAT_PAYOUTS as q } from "../../../../../constants/dateFormats.js";
import { useConfigContext as z } from "../../../../../core/ConfigContext/context.js";
const h = ["fundsCapturedAmount", "adjustmentAmount", "payoutAmount"], Q = ["createdAt", ...h], S = {
  adjustmentAmount: "payouts.overview.list.fields.adjustmentAmount",
  createdAt: "payouts.overview.list.fields.createdAt",
  fundsCapturedAmount: "payouts.overview.list.fields.fundsCapturedAmount",
  payoutAmount: "payouts.overview.list.fields.payoutAmount"
}, V = (n) => h.includes(n), pe = ({
  error: n,
  loading: c,
  onContactSupport: p,
  onRowClick: E,
  showDetails: k,
  showPagination: _,
  data: r,
  customColumns: b,
  ...D
}) => {
  const { i18n: i } = Y(), { dateFormat: A } = N("UTC"), { refreshing: d } = z(), O = l(() => c || d, [c, d]), o = j(x.up.sm), s = B(
    (e) => {
      var y, C, T, g;
      const f = i.get(S[e]);
      if (V(e))
        return {
          label: (C = (y = r == null ? void 0 : r[0]) == null ? void 0 : y[e]) != null && C.currency ? `${f} (${M((g = (T = r == null ? void 0 : r[0]) == null ? void 0 : T[e]) == null ? void 0 : g.currency)})` : f,
          position: "right"
        };
    },
    [r, i]
  ), w = R({
    customColumns: b,
    fields: Q,
    fieldsKeys: S,
    columnConfig: l(
      () => ({
        fundsCapturedAmount: { ...s("fundsCapturedAmount"), visible: o },
        adjustmentAmount: { ...s("adjustmentAmount"), visible: o },
        payoutAmount: s("payoutAmount")
      }),
      [s, o]
    )
  }), F = {
    title: "payouts.overview.errors.listEmpty",
    message: ["common.errors.updateFilters"]
  }, L = l(
    () => () => /* @__PURE__ */ t(U, { error: n, errorMessage: "payouts.overview.errors.listUnavailable", onContactSupport: p }),
    [n, p]
  );
  return /* @__PURE__ */ t("div", { className: K, children: /* @__PURE__ */ t(
    v,
    {
      errorDisplay: L,
      error: n,
      columns: w,
      data: r,
      loading: O,
      outline: !1,
      onRowClick: { callback: E },
      emptyTableMessage: F,
      customCells: {
        createdAt: ({ value: e }) => e ? o ? e && /* @__PURE__ */ t("time", { dateTime: e, children: /* @__PURE__ */ t(a, { el: u.SPAN, variant: m.BODY, children: A(e, q) }) }) : /* @__PURE__ */ t("time", { dateTime: e, children: A(e, G) }) : null,
        fundsCapturedAmount: ({ value: e }) => e && /* @__PURE__ */ t(a, { el: u.SPAN, variant: m.BODY, children: i.amount(e.value, e.currency, { hideCurrency: !0 }) }),
        adjustmentAmount: ({ value: e }) => e && /* @__PURE__ */ t(a, { el: u.SPAN, variant: m.BODY, children: i.amount(e.value, e.currency, { hideCurrency: !0 }) }),
        payoutAmount: ({ value: e }) => e && /* @__PURE__ */ t(
          a,
          {
            el: u.SPAN,
            variant: m.BODY,
            className: P({ [`${I}--strong`]: !o }),
            children: i.amount(e.value, e.currency, { hideCurrency: o })
          }
        )
      },
      children: _ && /* @__PURE__ */ t(v.Footer, { children: /* @__PURE__ */ t(
        $,
        {
          ...D,
          ariaLabelKey: "payouts.overview.pagination.label",
          limitSelectAriaLabelKey: "payouts.overview.pagination.controls.limitSelect.label"
        }
      ) })
    }
  ) });
};
export {
  Q as PAYOUT_TABLE_FIELDS,
  pe as PayoutsTable
};
