// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/TransactionsTable/TransactionsTable.js
import { jsx as o } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useState as x, useCallback as B, useMemo as $ } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import H from "../../../../internal/DataOverviewError/DataOverviewError.js";
import U from "../../../../../core/Context/useCoreContext.js";
import { getCurrencyCode as E } from "../../../../../core/Localization/amount/amount-util.js";
import K from "../../../../../hooks/useTimezoneAwareDateFormatting.js";
import { Tooltip as G } from "../../../../internal/Tooltip/Tooltip.js";
import { TypographyVariant as m, TypographyElement as c } from "../../../../internal/Typography/types.js";
import l from "../../../../internal/Typography/Typography.js";
import { getTransactionCategory as V, getTransactionCategoryDescription as j } from "../../../../utils/translation/getters.js";
import { useResponsiveContainer as y, containerQueries as T } from "../../../../../hooks/useResponsiveContainer.js";
import { BASE_CLASS as q, DATE_AND_PAYMENT_METHOD_CLASS as z, DATE_METHOD_CLASS as Q, AMOUNT_CLASS as N } from "./constants.js";
/* empty css                      */
import M from "./PaymentMethodCell.js";
import { useTableColumns as X } from "../../../../../hooks/useTableColumns.js";
import { Tag as Z } from "../../../../internal/Tag/Tag.js";
import w from "../../../../internal/DataGrid/DataGrid.js";
import J from "../../../../internal/Pagination/Pagination.js";
import { DATE_FORMAT_TRANSACTIONS_MOBILE as W, DATE_FORMAT_TRANSACTIONS as tt } from "../../../../../constants/dateFormats.js";
const ot = ["createdAt", "paymentMethod", "transactionType", "currency", "netAmount", "grossAmount"], Dt = {
  amount: "netAmount"
}, v = {
  createdAt: "transactions.overview.list.fields.createdAt",
  currency: "transactions.overview.list.fields.currency",
  grossAmount: "transactions.overview.list.fields.grossAmount",
  netAmount: "transactions.overview.list.fields.netAmount",
  paymentMethod: "transactions.overview.list.fields.paymentMethod",
  // status: 'transactions.overview.list.fields.status',
  transactionType: "transactions.overview.list.fields.transactionType"
}, Et = ({
  activeBalanceAccount: d,
  availableCurrencies: n,
  error: A,
  hasMultipleCurrencies: a,
  loading: _,
  onContactSupport: f,
  onRowClick: C,
  showDetails: et,
  showPagination: b,
  transactions: O,
  customColumns: R,
  ...L
}) => {
  const { i18n: r } = U(), { dateFormat: g } = K(d == null ? void 0 : d.timeZone), [F, S] = x(), p = y(T.up.sm), u = y(T.up.md), h = y(T.down.xs), P = X({
    customColumns: R,
    fields: ot,
    fieldsKeys: v,
    columnConfig: {
      netAmount: {
        label: a ? void 0 : `${r.get(v.netAmount)} ${n && n[0] ? `(${E(n[0])})` : ""}`,
        position: "right",
        flex: p ? 1.5 : void 0
      },
      grossAmount: {
        label: a ? void 0 : `${r.get(v.grossAmount)} ${n && n[0] ? `(${E(n[0])})` : ""}`,
        position: "right",
        flex: p ? 1.5 : void 0,
        visible: u
      },
      currency: { visible: u && a },
      transactionType: { visible: u },
      paymentMethod: { visible: p }
    }
  }), I = {
    title: "transactions.overview.errors.listEmpty",
    message: ["common.errors.updateFilters"]
  }, Y = B(
    (t) => {
      S(t ?? void 0);
    },
    [S]
  ), k = $(
    () => () => /* @__PURE__ */ o(H, { error: A, onContactSupport: f, errorMessage: "transactions.overview.errors.listUnavailable" }),
    [A, f]
  );
  return /* @__PURE__ */ o("div", { className: q, children: /* @__PURE__ */ o(
    w,
    {
      autoFitColumns: h,
      errorDisplay: k,
      error: A,
      columns: P,
      data: O,
      loading: _,
      outline: !1,
      onRowClick: { callback: C },
      onRowHover: Y,
      emptyTableMessage: I,
      customCells: {
        // Remove status column temporarily
        // status: ({ item }) => {
        //     return (
        //         <Tag
        //             label={getTransactionStatus(i18n, item.status)}
        //             variant={getTagVariantForTransaction(item)}
        //         />
        //     );
        // },
        transactionType: ({ item: t, rowIndex: e }) => {
          const i = V(r, t.category);
          if (i) {
            const s = j(r, t.category), D = () => /* @__PURE__ */ o(l, { el: c.SPAN, variant: m.BODY, children: i });
            return s ? /* @__PURE__ */ o(G, { content: s, isUnderlineVisible: e === F, children: /* @__PURE__ */ o("span", { children: D() }) }) : D();
          }
          return null;
        },
        currency: ({ item: t }) => /* @__PURE__ */ o(Z, { children: t.amountBeforeDeductions.currency }),
        netAmount: ({ value: t }) => {
          const { value: e, currency: i } = t, s = r.amount(e, i, { hideCurrency: !a });
          return /* @__PURE__ */ o(l, { el: c.SPAN, variant: m.BODY, className: N, children: s });
        },
        grossAmount: ({ item: t }) => {
          const { value: e, currency: i } = t.amountBeforeDeductions, s = r.amount(e, i, { hideCurrency: !a });
          return /* @__PURE__ */ o(l, { el: c.SPAN, variant: m.BODY, className: N, children: s });
        },
        createdAt: ({ item: t, value: e }) => h ? /* @__PURE__ */ o("div", { className: z, children: [
          /* @__PURE__ */ o(M, { paymentMethod: t.paymentMethod, bankAccount: t.bankAccount }),
          /* @__PURE__ */ o("time", { dateTime: t.createdAt, className: Q, children: /* @__PURE__ */ o(l, { el: c.SPAN, variant: m.BODY, children: g(t.createdAt, W) }) })
        ] }) : /* @__PURE__ */ o("time", { dateTime: e, children: /* @__PURE__ */ o(l, { el: c.SPAN, variant: m.BODY, children: g(e, tt) }) }),
        paymentMethod: ({ item: t }) => /* @__PURE__ */ o(M, { paymentMethod: t.paymentMethod, bankAccount: t.bankAccount })
      },
      children: b && /* @__PURE__ */ o(w.Footer, { children: /* @__PURE__ */ o(
        J,
        {
          ...L,
          ariaLabelKey: "transactions.overview.pagination.label",
          limitSelectAriaLabelKey: "transactions.overview.pagination.controls.limitSelect.label"
        }
      ) })
    }
  ) });
};
export {
  ot as TRANSACTION_FIELDS,
  Dt as TRANSACTION_FIELDS_REMAPS,
  Et as TransactionsTable
};
