// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkDetails/components/PaymentLinkExpiration/PaymentLinkExpiration.js
import { jsx as o } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useCallback as L, useMemo as u } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import g from "../../../../../core/Context/useCoreContext.js";
import { TypographyVariant as a, TypographyElement as p } from "../../../../internal/Typography/types.js";
import s from "../../../../internal/Typography/Typography.js";
import k from "../../../../internal/Button/ButtonActions/ButtonActions.js";
import x from "../../../../../hooks/useMutation/useMutation.js";
import { ButtonVariant as m } from "../../../../internal/Button/types.js";
import { useModalContext as B } from "../../../../internal/Modal/Modal.js";
/* empty css                           */
import { Alert as C } from "../../../../internal/Alert/Alert.js";
import { AlertTypeOption as T } from "../../../../internal/Alert/types.js";
import { useConfigContext as E } from "../../../../../core/ConfigContext/context.js";
import { EMPTY_OBJECT as I } from "../../../../../utils/value/constants.js";
const h = {
  root: "adyen-pe-payment-link-expiration"
}, w = ({ paymentLink: n, onCancel: e, onExpirationSuccess: l }) => {
  const { i18n: t } = g(), { withinModal: d } = B(), y = d ? p.H2 : p.DIV, { expirePayByLinkPaymentLink: c } = E().endpoints, i = x({
    queryFn: c,
    options: {
      onSuccess: l
    }
  }), r = L(
    () => i.mutate(I, { path: { paymentLinkId: n.linkInformation.paymentLinkId } }),
    [i, n.linkInformation.paymentLinkId]
  ), f = u(
    () => [
      {
        title: t.get("payByLink.details.expiration.actions.confirmExpiration"),
        event: r,
        variant: m.PRIMARY,
        disabled: i.isLoading,
        state: i.isLoading ? "loading" : "default"
      },
      {
        title: t.get("payByLink.details.expiration.actions.goBack"),
        event: e,
        variant: m.SECONDARY,
        disabled: i.isLoading
      }
    ],
    [i.isLoading, r, t, e]
  );
  return /* @__PURE__ */ o("div", { className: h.root, children: [
    /* @__PURE__ */ o(s, { el: y, variant: a.SUBTITLE, stronger: !0, children: t.get("payByLink.details.expiration.title") }),
    /* @__PURE__ */ o(s, { variant: a.BODY, children: t.get("payByLink.details.expiration.description") }),
    i.error && /* @__PURE__ */ o(
      C,
      {
        type: T.CRITICAL,
        title: t.get("payByLink.details.expiration.errorTitle"),
        description: t.get("payByLink.details.expiration.errorDescription")
      }
    ),
    /* @__PURE__ */ o(k, { actions: f })
  ] });
};
export {
  w as PaymentLinkExpiration
};
