// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputeManagement/components/DefendDisputeFlow/DefendDocumentUpload.js
import { jsx as r } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { FileInput as c } from "../../../../internal/FormFields/FileInput/FileInput.js";
import { getDefenseDocumentContent as y } from "../../utils/index.js";
import h from "../../../../../core/Context/useCoreContext.js";
import { TypographyElement as g, TypographyVariant as p } from "../../../../internal/Typography/types.js";
import d from "../../../../internal/Typography/Typography.js";
import { useMemo as F } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { useDisputeFlow as T } from "../../context/dispute/context.js";
import { ALLOWED_FILE_TYPES as _, DOCUMENT_MAX_SIZE as C } from "./constants.js";
const S = ({
  document: e,
  disabled: l,
  required: m,
  mapError: s
}) => {
  const { i18n: n } = h(), { removeFieldFromDefendPayload: i, addFileToDefendPayload: f, defenseDocumentConfig: a } = T(), { title: u, primaryDescriptionItems: t } = F(() => y(a, n, e), [a, n, e]) || {};
  return /* @__PURE__ */ r("div", { className: "adyen-pe-defend-dispute-document-upload", children: [
    /* @__PURE__ */ r("div", { children: [
      /* @__PURE__ */ r(
        d,
        {
          strongest: !0,
          className: "adyen-pe-defend-dispute-document-upload__title",
          variant: p.BODY,
          el: g.DIV,
          children: u || e
        }
      ),
      t && t.length > 0 ? t.map((o, D) => /* @__PURE__ */ r(
        d,
        {
          className: "adyen-pe-defend-dispute-document-upload__description",
          variant: p.BODY,
          children: o
        },
        `${D}-description`
      )) : null
    ] }),
    /* @__PURE__ */ r(
      c,
      {
        allowedFileTypes: _,
        maxFileSize: (o) => C[o],
        mapError: s,
        onDelete: () => {
          e && i(e);
        },
        disabled: l,
        required: m,
        onChange: (o) => {
          o[0] ? f(e, o[0]) : i(e);
        }
      },
      e
    )
  ] });
};
export {
  S as DefendDocumentUpload
};
