// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputeManagement/components/DefendDisputeFlow/DefendDisputeReason.js
import { jsx as i } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import z from "classnames";
import "../../../../../external/preact/compat/dist/compat.module.js";
import { useState as k, useMemo as l, useEffect as x, useCallback as p } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import J from "../../../../../core/Context/useCoreContext.js";
import { useFetch as U } from "../../../../../hooks/useFetch.js";
import { useResponsiveContainer as K, containerQueries as W } from "../../../../../hooks/useResponsiveContainer.js";
import { Alert as X } from "../../../../internal/Alert/Alert.js";
import { AlertVariantOption as Z, AlertTypeOption as ee } from "../../../../internal/Alert/types.js";
import te from "../../../../internal/Button/ButtonActions/ButtonActions.js";
import { TypographyVariant as R, TypographyElement as S } from "../../../../internal/Typography/types.js";
import A from "../../../../internal/Typography/Typography.js";
import { useDisputeFlow as ne } from "../../context/dispute/context.js";
import { getDefenseReasonContent as E } from "../../utils/index.js";
import oe from "../../../../internal/FormFields/Select/Select.js";
import { Fragment as ie } from "../../../../../external/preact/dist/preact.module.js";
import { useConfigContext as se } from "../../../../../core/ConfigContext/context.js";
import { EMPTY_OBJECT as ae } from "../../../../../utils/value/constants.js";
const b = "adyen-pe-defend-dispute-reason", v = {
  selector: b + "__selector",
  description: b + "__description",
  dropdownList: b + "__dropdown-list",
  dropdownListMobile: b + "__dropdown-list--mobile"
}, Ie = () => {
  var O, T, B, P;
  const { i18n: t } = J(), {
    applicableDocuments: s,
    dispute: n,
    goBack: C,
    setFlowState: u,
    setSelectedDefenseReason: D,
    selectedDefenseReason: r,
    setApplicableDocuments: g,
    defenseReasonConfig: h
  } = ne(), m = (O = n == null ? void 0 : n.dispute) == null ? void 0 : O.allowedDefenseReasons, F = (T = n == null ? void 0 : n.dispute) == null ? void 0 : T.pspReference, [f, y] = k(!1), L = K(W.down.xs), d = l(
    () => Object.freeze(
      m == null ? void 0 : m.map((e) => {
        var o;
        return {
          id: e,
          disabled: m.length === 1,
          name: ((o = E(h, t, e)) == null ? void 0 : o.title) ?? e
        };
      })
    ) ?? [],
    [t, m, h]
  ), a = l(
    () => {
      var e, o;
      return r ? (e = d.find(($) => $.id === r)) == null ? void 0 : e.id : ((o = d == null ? void 0 : d[0]) == null ? void 0 : o.id) ?? null;
    },
    [r, d]
  );
  x(() => {
    a && D(a);
  }, [a, f]);
  const { getApplicableDefenseDocuments: _ } = se().endpoints, M = p(async () => _ == null ? void 0 : _(ae, {
    query: {
      defenseReason: r
    },
    path: {
      disputePspReference: F
    }
  }), [r, F, _]), { error: q, isFetching: w } = U({
    queryFn: M,
    fetchOptions: {
      enabled: f,
      onSuccess: p(
        (e) => {
          y(!1), g((e == null ? void 0 : e.data) ?? null), e != null && e.data && (e == null ? void 0 : e.data.length) > 0 && u("uploadDefenseFilesView");
        },
        [g, y, u]
      )
    }
  });
  x(() => {
    y(!1);
  }, [q]);
  const N = p(() => {
    if (s != null && s.length) return u("uploadDefenseFilesView");
    y(!0);
  }, [s, u]), H = p(
    (e) => {
      var o;
      r !== e.target.value && (s != null && s.length) && g([]), (o = e == null ? void 0 : e.target) != null && o.value && D(e.target.value);
    },
    [s, r, g, D]
  ), V = l(() => [
    {
      title: t.get("disputes.management.defend.common.actions.continue"),
      disabled: f || w,
      event: N
    },
    {
      title: t.get("disputes.management.common.actions.goBack"),
      disabled: f || w,
      event: C
    }
  ], [w, f, t, C, N]), I = l(() => (n == null ? void 0 : n.dispute.type) === "REQUEST_FOR_INFORMATION", [n == null ? void 0 : n.dispute.type]), [G, Y] = k(!I), j = p(() => {
    Y(!1);
  }, []), c = l(
    () => a ? E(h, t, a) : void 0,
    [h, t, a]
  ), Q = l(
    () => I ? t.get("disputes.management.defend.requestForInformation.selectDefenseReason") : t.get("disputes.management.defend.chargeback.selectDefenseReason"),
    [t, I]
  );
  return !d || !a ? null : /* @__PURE__ */ i(ie, { children: [
    /* @__PURE__ */ i("div", { className: v.selector, children: [
      /* @__PURE__ */ i(A, { className: "adyen-pe-defend-dispute__reason-description", variant: R.BODY, children: Q }),
      /* @__PURE__ */ i(
        oe,
        {
          items: d,
          onChange: H,
          selected: a,
          "aria-label": t.get("disputes.management.defend.common.inputs.reasonSelect.a11y.label"),
          popoverClassNameModifiers: [z(v.dropdownList, { [v.dropdownListMobile]: L })],
          fixedPopoverPositioning: !0
        }
      ),
      (B = c == null ? void 0 : c.primaryDescriptionItems) == null ? void 0 : B.map((e, o) => /* @__PURE__ */ i(
        A,
        {
          el: S.PARAGRAPH,
          className: v.description,
          variant: R.BODY,
          children: e
        },
        `description-${o}`
      )),
      ((P = c == null ? void 0 : c.secondaryDescriptionItems) == null ? void 0 : P.length) && /* @__PURE__ */ i("ul", { className: "adyen-pe-defend-dispute-reason__secondary-description-items-container", children: c.secondaryDescriptionItems.map((e, o) => /* @__PURE__ */ i("li", { className: "adyen-pe-defend-dispute-reason__secondary-description-item", children: /* @__PURE__ */ i(
        A,
        {
          el: S.PARAGRAPH,
          className: "adyen-pe-defend-dispute-reason__description",
          variant: R.BODY,
          children: e
        }
      ) }, `description-item-${o}`)) })
    ] }),
    G && /* @__PURE__ */ i(X, { onClose: j, type: ee.HIGHLIGHT, variant: Z.TIP, closeButton: !0, children: /* @__PURE__ */ i(A, { className: "adyen-pe-alert__description", el: S.DIV, variant: R.CAPTION, wide: !0, children: t.get("disputes.management.defend.chargeback.feeInfo") }) }),
    /* @__PURE__ */ i("div", { className: "adyen-pe-defend-dispute__actions", children: /* @__PURE__ */ i(te, { actions: V }) })
  ] });
};
export {
  Ie as DefendDisputeReason
};
