// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputeManagement/components/DefendDisputeFlow/DefendDisputeFileUpload.js
import { jsx as n } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useState as P, useCallback as a, useMemo as g } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import K from "../../../../../core/Context/useCoreContext.js";
import X from "../../../../../hooks/useMutation/useMutation.js";
import Z from "../../../../internal/Button/ButtonActions/ButtonActions.js";
import ee from "../../../../internal/Card/Card.js";
import { TypographyVariant as F } from "../../../../internal/Typography/types.js";
import L from "../../../../internal/Typography/Typography.js";
import { useDisputeFlow as te } from "../../context/dispute/context.js";
import { DefendDocumentUpload as ne } from "./DefendDocumentUpload.js";
import z from "./SelectAndUploadOptionalDoc.js";
import { ButtonVariant as M } from "../../../../internal/Button/types.js";
import { getDefenseDocumentContent as oe } from "../../utils/index.js";
import { validationErrors as R } from "../../../../internal/FormFields/FileInput/constants.js";
/* empty css                       */
import { Icon as V } from "../../../../internal/Icon/Icon.js";
import { Fragment as k } from "../../../../../external/preact/dist/preact.module.js";
import se from "../../../../internal/Button/Button.js";
import { useConfigContext as ie } from "../../../../../core/ConfigContext/context.js";
import { getHumanReadableFileSize as de } from "../../../../../utils/file/size.js";
const ae = [
  "disputes.management.defend.common.documentRequirements.language",
  "disputes.management.defend.common.documentRequirements.recommendedSize",
  "disputes.management.defend.common.documentRequirements.formatAndSize"
], Fe = () => {
  const { i18n: o } = K(), { defendDispute: x } = ie().endpoints, {
    clearFiles: D,
    clearStates: A,
    dispute: _,
    applicableDocuments: B,
    goBack: I,
    defendDisputePayload: r,
    defendResponse: T,
    onDefendSubmit: h,
    removeFieldFromDefendPayload: C,
    setFlowState: c,
    defenseDocumentConfig: $
  } = te(), Y = _ == null ? void 0 : _.dispute.pspReference, [y, H] = P(void 0), [u, E] = P([]), w = a(() => {
    A(), c("details");
  }, [A, c]), v = a(
    (e, t) => {
      var s, d;
      switch (e) {
        case R.DISALLOWED_FILE_TYPE:
          return o.get("common.inputs.file.errors.disallowedType");
        case R.FILE_REQUIRED:
          return o.get("disputes.management.defend.common.inputs.file.errors.required");
        case R.TOO_MANY_FILES:
          return o.get("common.inputs.file.errors.tooMany");
        case R.VERY_LARGE_FILE:
          return o.get("disputes.management.defend.common.inputs.file.errors.tooLarge", {
            values: {
              size: (t == null ? void 0 : t.size) === void 0 ? void 0 : de(t.size),
              type: (d = (s = t == null ? void 0 : t.type) == null ? void 0 : s.replace(/^([^/]+\/)*/gi, "")) == null ? void 0 : d.toUpperCase()
            }
          });
        default:
          return o.get("disputes.management.defend.common.inputs.file.errors.default");
      }
    },
    [o]
  ), { requiredDocuments: m, optionalDocuments: O, oneOrMoreDocuments: i } = g(() => {
    const e = {
      requiredDocuments: [],
      optionalDocuments: [],
      oneOrMoreDocuments: []
    };
    return (B ?? []).forEach(({ documentTypeCode: t, requirementLevel: s }) => {
      var S;
      const d = ((S = oe($, o, t)) == null ? void 0 : S.title) || t;
      switch (s) {
        case "REQUIRED":
          e.requiredDocuments.push(t);
          break;
        case "OPTIONAL":
          e.optionalDocuments.push({
            id: t,
            name: d
          });
          break;
        case "ONE_OR_MORE":
          e.oneOrMoreDocuments.push({
            id: t,
            name: d
          });
          break;
      }
      return e;
    }), e;
  }, [B, o]), Q = g(() => {
    if (!r) return !1;
    let e = m.every((t) => r.get(t) instanceof File);
    return i.length > 0 && e && (e = i.some((t) => r.get(t.id) instanceof File)), e;
  }, [r, i, m]), l = X({
    queryFn: x,
    options: {
      onSuccess: a(() => {
        D(), h("success"), c("defenseSubmitResponseView");
      }, [D, h, c]),
      onError: a(() => {
        D(), h("error"), c("defenseSubmitResponseView");
      }, [D, h, c])
    }
  }), f = T === "success", p = l.isLoading || f, b = r && Q && !p, U = a(() => {
    b && l.mutate(
      { contentType: "multipart/form-data", body: r },
      { path: { disputePspReference: Y } }
    );
  }, [b, Y, l, r]), W = g(() => [
    {
      title: o.get("disputes.management.defend.common.actions.submit"),
      disabled: !b,
      state: l.isLoading ? "loading" : "default",
      variant: M.PRIMARY,
      event: U,
      classNames: f ? ["adyen-pe-defend-dispute__defended-btn"] : void 0,
      renderTitle: (e) => f ? /* @__PURE__ */ n(k, { children: [
        /* @__PURE__ */ n(V, { name: "checkmark-circle-fill", className: "adyen-pe-defend-dispute__defended-icon" }),
        o.get("disputes.management.defend.common.defended")
      ] }) : e
    },
    {
      title: o.get("disputes.management.common.actions.goBack"),
      disabled: l.isLoading,
      variant: M.SECONDARY,
      event: f ? w : I
    }
  ], [o, b, l.isLoading, U, f, w, I]), q = a((e, t) => {
    e === void 0 ? E((s) => [...s, e]) : t !== void 0 && E((s) => {
      if (s[t] === e)
        return s;
      const d = [...s];
      return d[t] = e, d;
    });
  }, []), N = g(() => {
    if (p) return !1;
    const e = O.length + Math.max(0, i.length - 1);
    return !!(e && e !== u.length);
  }, [p, i, O, u]), j = a(() => {
    N && q();
  }, [N, q]), G = g(() => [...i.filter((t) => t.id !== y), ...O].map((t) => ({ ...t, disabled: u.includes(t.id) })), [i, y, O, u]), J = a(
    (e) => {
      E((t) => {
        if (e < 0 || e >= t.length)
          return t;
        const s = t[e];
        return s && C(s), t.filter((d, S) => S !== e);
      });
    },
    [C]
  );
  return /* @__PURE__ */ n(k, { children: /* @__PURE__ */ n(k, { children: [
    /* @__PURE__ */ n(L, { className: "adyen-pe-defend-dispute-file-uploader__subtitle", variant: F.BODY, children: o.get("disputes.management.defend.common.documentUploadInfo") }),
    /* @__PURE__ */ n(
      ee,
      {
        renderHeader: /* @__PURE__ */ n(L, { variant: F.BODY, stronger: !0, className: "adyen-pe-defend-dispute-document-requirements", children: o.get("disputes.management.defend.common.documentRequirements") }),
        filled: !0,
        expandable: !0,
        compact: !0,
        children: /* @__PURE__ */ n("ul", { className: "adyen-pe-defend-dispute-document-requirements--list", children: ae.map((e, t) => /* @__PURE__ */ n("li", { className: "adyen-pe-defend-dispute-document-requirements--item", children: /* @__PURE__ */ n(L, { variant: F.BODY, children: o.get(e) }) }, `${e}-${t}`)) })
      }
    ),
    /* @__PURE__ */ n("div", { className: "adyen-pe-defend-dispute-file-uploader__container", children: [
      m.length || i.length ? /* @__PURE__ */ n("div", { className: "adyen-pe-defend-dispute-document-upload-box", children: [
        m.length ? /* @__PURE__ */ n("div", { className: "adyen-pe-defend-dispute-document-upload-box__required-documents", children: m == null ? void 0 : m.map((e) => /* @__PURE__ */ n(
          ne,
          {
            mapError: v,
            disabled: p,
            document: e,
            required: !0
          },
          e
        )) }) : null,
        i.length ? /* @__PURE__ */ n(
          z,
          {
            mapError: v,
            disabled: p,
            selection: y,
            setSelection: (e) => H(e),
            items: i,
            title: o.get("disputes.management.defend.common.documentTypes.required"),
            required: !0
          }
        ) : null
      ] }) : null,
      u.length ? u.map((e, t) => /* @__PURE__ */ n("div", { className: "adyen-pe-defend-dispute-document-upload-box", children: /* @__PURE__ */ n(
        z,
        {
          mapError: v,
          disabled: p,
          onRemoveOption: J,
          selection: e,
          setSelection: q,
          index: t,
          items: G,
          title: o.get("disputes.management.defend.common.documentTypes.optional"),
          required: !0
        }
      ) }, `optional-doc-${t}`)) : null,
      N && /* @__PURE__ */ n(se, { align: "center", onClick: j, variant: M.SECONDARY, fullWidth: !0, children: [
        /* @__PURE__ */ n(V, { name: "plus" }),
        o.get("disputes.management.defend.common.actions.addOptionalDocument")
      ] })
    ] }),
    /* @__PURE__ */ n("div", { className: "adyen-pe-defend-file-uploader__actions", children: /* @__PURE__ */ n(Z, { actions: W }) })
  ] }) });
};
export {
  Fe as DefendDisputeFileUpload
};
