// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputeManagement/components/DefendDisputeFlow/SelectAndUploadOptionalDoc.js
import { jsx as o } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import E from "classnames";
import { useResponsiveContainer as F, containerQueries as L } from "../../../../../hooks/useResponsiveContainer.js";
import h from "../../../../internal/Typography/Typography.js";
import { TypographyElement as v, TypographyVariant as C } from "../../../../internal/Typography/types.js";
import { FileInput as P } from "../../../../internal/FormFields/FileInput/FileInput.js";
import B from "../../../../../core/Context/useCoreContext.js";
import { useDisputeFlow as M } from "../../context/dispute/context.js";
import { getDefenseDocumentContent as N } from "../../utils/index.js";
import { useMemo as O, useCallback as S, useEffect as R } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import Y from "../../../../internal/Button/Button.js";
import { ButtonVariant as V } from "../../../../internal/Button/types.js";
import { ALLOWED_FILE_TYPES as k, DOCUMENT_MAX_SIZE as U } from "./constants.js";
import { Icon as W } from "../../../../internal/Icon/Icon.js";
import j from "../../../../internal/FormFields/Select/Select.js";
const a = "adyen-pe-defend-dispute-document-upload-box", n = {
  deleteButton: a + "__delete-button-container",
  dropdownList: a + "__dropdown-list",
  dropdownListMobile: a + "__dropdown-list--mobile",
  extraDocuments: a + "__extra-documents-selector",
  extraDocumentsTitle: a + "__extra-documents-selector-title"
}, ne = ({
  items: d,
  selection: t,
  setSelection: p,
  disabled: i,
  required: T,
  title: b,
  index: l,
  onRemoveOption: c,
  mapError: x
}) => {
  var y, _;
  const { i18n: r } = B(), { addFileToDefendPayload: w, moveFieldInDefendPayload: u, removeFieldFromDefendPayload: f, defenseDocumentConfig: D } = M(), g = O(() => r.get("disputes.management.defend.common.inputs.documentSelect.a11y.label"), [r]), I = F(L.down.xs), A = S(
    (e) => N(D, r, e),
    [D, r]
  ), m = S(
    (e) => {
      t && u(t, e), p(e, l);
    },
    [l, u, t, p]
  );
  return R(() => {
    const e = d.filter(({ disabled: s }) => s !== !0);
    e.length === 1 && !t && m(e[0].id);
  }, [d, t, m]), /* @__PURE__ */ o("div", { className: n.extraDocuments, children: [
    /* @__PURE__ */ o("div", { className: n.deleteButton, children: [
      /* @__PURE__ */ o(h, { strongest: !0, className: n.extraDocumentsTitle, variant: C.BODY, el: v.DIV, children: b }),
      c && /* @__PURE__ */ o(
        Y,
        {
          disabled: i,
          "aria-label": r.get("disputes.management.defend.common.actions.deleteOptionalDocument"),
          onClick: () => l !== void 0 && !i && c(l),
          variant: V.TERTIARY,
          fullWidth: !1,
          align: "center",
          children: /* @__PURE__ */ o(W, { name: "trash-can" })
        }
      )
    ] }),
    /* @__PURE__ */ o("div", { children: [
      /* @__PURE__ */ o(
        j,
        {
          onChange: (e) => {
            const s = e.target.value;
            m(s);
          },
          items: d,
          filterable: !1,
          multiSelect: !1,
          showOverlay: !1,
          selected: t,
          readonly: i,
          "aria-label": g,
          placeholder: g,
          popoverClassNameModifiers: [E(n.dropdownList, { [n.dropdownListMobile]: I })],
          fixedPopoverPositioning: !0
        }
      ),
      t && ((_ = (y = A(t)) == null ? void 0 : y.primaryDescriptionItems) == null ? void 0 : _.map((e) => /* @__PURE__ */ o(
        h,
        {
          className: "adyen-pe-defend-dispute-document-upload__description",
          variant: C.BODY,
          el: v.PARAGRAPH,
          children: e
        },
        e
      )))
    ] }),
    /* @__PURE__ */ o(
      P,
      {
        maxFileSize: (e) => U[e],
        allowedFileTypes: k,
        mapError: x,
        onDelete: () => {
          t && f(t);
        },
        disabled: i || !t,
        required: T,
        onChange: (e) => {
          t && (e[0] ? w(t, e[0]) : f(t));
        }
      }
    )
  ] });
};
export {
  ne as default
};
