// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/components/AccountDetails/AccountDetail.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import y from "../../../../internal/CopyText/CopyText.js";
import T from "../../../../../core/Context/useCoreContext.js";
import i from "../../../../internal/Typography/Typography.js";
import { TypographyVariant as o, TypographyElement as p } from "../../../../internal/Typography/types.js";
import h from "classnames";
const v = ({
  className: n,
  content: r,
  contentClassName: l,
  copyButtonLabel: c,
  isPrimary: t,
  textToCopy: a,
  label: m,
  labelClassName: d
}) => {
  const { i18n: s } = T();
  return /* @__PURE__ */ e("div", { className: n, children: [
    /* @__PURE__ */ e("dt", { className: d, children: /* @__PURE__ */ e(i, { el: p.SPAN, variant: o.CAPTION, children: s.get(m) }) }),
    /* @__PURE__ */ e("dd", { className: h(l, { "adyen-pe-capital-account-details__detail-content--primary": t }), children: a ? /* @__PURE__ */ e(
      y,
      {
        copyButtonAriaLabelKey: c,
        visibleText: r,
        textToCopy: a,
        showCopyTextTooltip: !1,
        type: "Text"
      }
    ) : /* @__PURE__ */ e(
      i,
      {
        el: p.SPAN,
        variant: t ? o.SUBTITLE : o.BODY,
        stronger: t,
        children: r
      }
    ) })
  ] });
};
export {
  v as AccountDetail
};
