// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/constants.js
import { TRANSACTION_ANALYTICS_SUBCATEGORY_DETAILS as a, TRANSACTION_ANALYTICS_CATEGORY as T } from "../TransactionsOverview/constants.js";
import { TRANSACTION_FIELDS as r } from "../TransactionsOverview/components/TransactionsTable/TransactionsTable.js";
import { DetailsTab as s } from "./types.js";
const e = "adyen-pe-transaction-data", E = `${e}__amount`, d = `${e}__action-bar`, l = `${e}__container`, N = `${e}__head-container`, o = `${e}__input`, _ = `${o}-container`, S = `${_}--short`, R = `${_}--text-input`, I = `${_}--with-error`, D = `${o}-head`, m = `${e}__label`, O = `${e}__list`, X = `${e}__tags`, f = `${e}__status-box`, p = `${e}__timeline-label`, C = `${e}__timeline-value`, U = `${e}__refund-response`, c = `${e}__refund-response-icon`, $ = `${c}--success`, L = `${c}--error`, b = `${e}__refund-statuses-container`, P = /* @__PURE__ */ new Set([
  "status",
  "category",
  "paymentMethod",
  "bankAccount",
  "balanceAccount",
  "id",
  "balanceAccountId",
  ...r,
  "account",
  "deductedAmount",
  "description",
  "lineItems",
  "merchantReference",
  "originalAmount",
  "paymentPspReference",
  "pspReference",
  "refundDetails",
  "refundMetadata",
  "refundPspReference",
  "refundReason"
]), h = {
  balanceAccount: "account",
  balanceAccountId: (t) => {
    const n = t == null ? void 0 : t.balanceAccount;
    if (n && !n.description) return "account";
  },
  description: (t) => {
    var n;
    if ((n = t == null ? void 0 : t.balanceAccount) != null && n.description) return "account";
  }
}, F = [
  { id: s.SUMMARY, label: "transactions.details.views.summary", content: null },
  { id: s.DETAILS, label: "transactions.details.views.details", content: null },
  { id: s.TIMELINE, label: "transactions.details.views.timeline", content: null }
], y = ["completed", "in_progress", "failed"], M = ["requested_by_customer", "issue_with_item_sold", "fraudulent", "duplicate", "other"], w = Object.freeze({
  requested_by_customer: "transactions.details.common.refundReasons.requestedByCustomer",
  issue_with_item_sold: "transactions.details.common.refundReasons.issueWithItemSold",
  fraudulent: "transactions.details.common.refundReasons.fraudulent",
  duplicate: "transactions.details.common.refundReasons.duplicate",
  other: "transactions.details.common.refundReasons.other"
}), B = {
  category: T,
  subCategory: a
};
export {
  M as REFUND_REASONS,
  w as REFUND_REASONS_KEYS,
  y as REFUND_STATUSES,
  d as TX_DATA_ACTION_BAR,
  E as TX_DATA_AMOUNT,
  e as TX_DATA_CLASS,
  l as TX_DATA_CONTAINER,
  N as TX_DATA_HEAD_CONTAINER,
  o as TX_DATA_INPUT,
  _ as TX_DATA_INPUT_CONTAINER,
  S as TX_DATA_INPUT_CONTAINER_SHORT,
  R as TX_DATA_INPUT_CONTAINER_TEXT,
  I as TX_DATA_INPUT_CONTAINER_WITH_ERROR,
  D as TX_DATA_INPUT_HEAD,
  m as TX_DATA_LABEL,
  O as TX_DATA_LIST,
  X as TX_DATA_TAGS,
  h as TX_DETAILS_FIELDS_REMAPS,
  P as TX_DETAILS_RESERVED_FIELDS_SET,
  F as TX_DETAILS_TABS,
  U as TX_REFUND_RESPONSE,
  L as TX_REFUND_RESPONSE_ERROR_ICON,
  c as TX_REFUND_RESPONSE_ICON,
  $ as TX_REFUND_RESPONSE_SUCCESS_ICON,
  b as TX_REFUND_STATUSES_CONTAINER,
  f as TX_STATUS_BOX,
  p as TX_TIMELINE_LABEL,
  C as TX_TIMELINE_VALUE,
  B as sharedTransactionDetailsEventProperties
};
