// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/PaymentDetails/PaymentDetailsSummary.js
import { jsx as u } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { memo as L } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useMemo as j, useCallback as p } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { TX_DATA_LABEL as y, TX_DATA_LIST as v } from "../../constants.js";
import { TypographyVariant as D, TypographyElement as I } from "../../../../internal/Typography/types.js";
import { getTransactionAmountAdjustmentType as d } from "../../../../utils/translation/getters.js";
import N from "../../../../../core/Context/useCoreContext.js";
import S from "../../../../internal/Typography/Typography.js";
import A from "classnames";
import { Tooltip as _ } from "../../../../internal/Tooltip/Tooltip.js";
import P from "../../../../internal/StructuredList/StructuredList.js";
import { isNullish as V } from "../../../../../utils/value/is.js";
const l = {
  grossAmount: "transactions.details.summary.fields.grossAmount",
  netAmount: "transactions.details.summary.fields.netAmount",
  originalAmount: "transactions.details.summary.fields.originalAmount"
}, b = null, k = (r) => r && typeof r == "object" && "value" in r, C = {
  "transactions.details.summary.adjustments.types.tip": "transactions.details.summary.adjustments.types.tip.information",
  "transactions.details.summary.adjustments.types.surcharge": "transactions.details.summary.adjustments.types.surcharge.information"
}, E = ({ transaction: r }) => {
  const { i18n: o } = N(), f = j(() => {
    const { additions: t, deductions: s, amountBeforeDeductions: e, netAmount: a, originalAmount: c } = r, i = (n) => {
      if (V(n)) return null;
      const { value: m, currency: h } = n;
      return o.amount(m, h);
    };
    return [
      // original amount
      c && (t && t.length > 0 || c.value !== e.value) ? {
        id: "originalAmount",
        key: l.originalAmount,
        value: i(c)
      } : b,
      // additions
      ...(t == null ? void 0 : t.map(({ type: n, ...m }) => ({
        key: d(o, n),
        value: i(m)
      }))) ?? [],
      // amountBeforeDeductions
      {
        id: "grossAmount",
        key: l.grossAmount,
        value: i(e)
      },
      // deductions
      ...(s == null ? void 0 : s.map(({ type: n, ...m }) => ({
        key: `transactions.details.summary.adjustments.types.${n}`,
        value: i(m),
        rawValue: m,
        label: d(o, n)
      }))) ?? [],
      // netAmount
      {
        id: "netAmount",
        key: l.netAmount,
        value: i(a)
      }
    ].filter(Boolean);
  }, [o, r]), g = p(
    (t, s, e) => {
      const a = C[s];
      return e && k(e) && e.value < 0 && a ? /* @__PURE__ */ u(_, { content: o.get(a), isUnderlineVisible: !0, children: /* @__PURE__ */ u("span", { className: A(y), children: t }) }, s) : /* @__PURE__ */ u("div", { className: A(y), children: t });
    },
    [o]
  ), T = p((t, s) => {
    const e = s === l.netAmount, a = D.BODY;
    return /* @__PURE__ */ u(S, { el: I.DIV, variant: a, strongest: e, children: t });
  }, []);
  return /* @__PURE__ */ u(
    P,
    {
      classNames: v,
      items: f,
      layout: "5-7",
      renderLabel: g,
      renderValue: T
    }
  );
}, z = L(E);
export {
  z as default
};
