// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/PaymentDetails/PaymentDetailsProperties.js
import { jsx as r } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { memo as S } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useMemo as u, useCallback as h } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { getTransactionRefundReason as j } from "../../../../utils/translation/getters.js";
import { isCustomDataObject as R } from "../../../../internal/DataGrid/components/TableCells.js";
import { TypographyVariant as N, TypographyElement as _ } from "../../../../internal/Typography/types.js";
import { TX_DETAILS_FIELDS_REMAPS as F, TX_DATA_LABEL as M, TX_DATA_LIST as X, sharedTransactionDetailsEventProperties as w } from "../../constants.js";
import { normalizeCustomFields as Y } from "../../../../utils/customData/normalizeCustomFields.js";
import q from "../../../../../core/Context/analytics/useAnalyticsContext.js";
import $ from "../../../../../core/Context/useCoreContext.js";
import A from "../../../../internal/Typography/Typography.js";
import g from "../../../../internal/DataGrid/components/Icon.js";
import G from "../../../../internal/CopyText/CopyText.js";
import H from "../../../../internal/Link/Link.js";
import T from "classnames";
import J from "../../../../internal/StructuredList/StructuredList.js";
const Q = {
  ...w,
  sectionName: "Details",
  label: "Copy button"
}, c = {
  account: "transactions.details.fields.account",
  id: "transactions.details.fields.referenceID",
  merchantReference: "transactions.details.fields.merchantReference",
  pspReference: "transactions.details.fields.pspReference",
  refundPspReference: "transactions.details.fields.refundPspReference",
  refundReason: "transactions.details.fields.refundReason"
}, P = {
  id: "transactions.details.actions.copyReferenceID",
  merchantReference: "transactions.details.actions.copyMerchantReference",
  pspReference: "transactions.details.actions.copyPspReference"
}, l = null, d = ({ dataCustomization: n, extraFields: f, transaction: o }) => {
  const { i18n: m } = $(), p = u(() => {
    var b;
    const { balanceAccount: e, category: s, id: y, merchantReference: t, paymentPspReference: L, refundMetadata: a } = o, x = (e == null ? void 0 : e.description) || (e == null ? void 0 : e.id), k = s === "Refund", C = Y((b = n == null ? void 0 : n.details) == null ? void 0 : b.fields, F, o), V = [
      // balance account
      x ? { id: "account", key: c.account, value: x } : l,
      // refund reason
      k && (a != null && a.refundReason) ? {
        id: "refundReason",
        key: c.refundReason,
        value: j(m, a.refundReason)
      } : l,
      // reference id
      {
        id: "id",
        key: c.id,
        value: /* @__PURE__ */ r(
          d.CopyText,
          {
            copyButtonAriaLabelKey: P.id,
            trackingName: "Reference ID",
            textToCopy: y
          }
        )
      },
      // merchant reference
      t ? {
        id: "merchantReference",
        key: c.merchantReference,
        value: /* @__PURE__ */ r(
          d.CopyText,
          {
            copyButtonAriaLabelKey: P.merchantReference,
            trackingName: "Merchant reference",
            textToCopy: t
          }
        )
      } : l,
      // payment psp reference
      L ? {
        id: "paymentPspReference",
        key: c.pspReference,
        value: /* @__PURE__ */ r(
          d.CopyText,
          {
            copyButtonAriaLabelKey: P.pspReference,
            trackingName: "PSP reference",
            textToCopy: L
          }
        )
      } : l,
      // refund psp reference
      k && (a != null && a.refundPspReference) ? {
        id: "refundPspReference",
        key: c.refundPspReference,
        value: a.refundPspReference
      } : l
    ], v = C ? (i) => {
      var I;
      return ((I = C.find((O) => O.key === i)) == null ? void 0 : I.visibility) !== "hidden";
    } : () => !0;
    return V.filter((i) => (i == null ? void 0 : i.id) && v(i.id));
  }, [m, n, o]), D = u(
    () => Object.entries(f || {}).filter(([, e]) => (e == null ? void 0 : e.type) !== "button").map(([e, s]) => ({
      key: e,
      value: R(s) ? s.value : s,
      type: R(s) ? s.type : "text",
      config: R(s) ? s.config : void 0
    })),
    [f]
  ), B = u(() => [...p, ...D], [D, p]), E = h(
    (e) => /* @__PURE__ */ r("div", { className: M, children: e }),
    []
  ), K = h((e, s, y, t) => {
    if (t)
      switch (y) {
        case "icon":
          return /* @__PURE__ */ r("div", { className: T(t.className, "adyen-pe-transaction-data__list-icon-value"), children: [
            /* @__PURE__ */ r(g, { url: t.src, alt: t.alt || e }),
            /* @__PURE__ */ r(A, { el: _.SPAN, variant: N.BODY, children: [
              " ",
              e
            ] })
          ] });
        case "link":
          return /* @__PURE__ */ r(H, { classNames: [T(t.className)], href: t.href, target: t.target || "_blank", children: e });
      }
    return /* @__PURE__ */ r(A, { el: _.DIV, variant: N.BODY, className: T(t == null ? void 0 : t.className), children: e });
  }, []);
  return /* @__PURE__ */ r(
    J,
    {
      align: "start",
      layout: "5-7",
      items: B,
      classNames: X,
      renderLabel: E,
      renderValue: K
    }
  );
};
d.CopyText = S(({ trackingName: n, ...f }) => {
  const o = q(), m = h(() => {
    var p;
    n && ((p = o.addEvent) == null || p.call(o, "Clicked button", {
      ...Q,
      subSectionName: n
    }));
  }, [n, o]);
  return /* @__PURE__ */ r(G, { ...f, onCopyText: m, showCopyTextTooltip: !1, type: "Default" });
});
const fe = S(d);
export {
  fe as default
};
