// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/PaymentDetails/PaymentDetailsTimeline.js
import { jsx as a } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { memo as T } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useMemo as m } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { TX_TIMELINE_VALUE as C, TX_TIMELINE_LABEL as D } from "../../constants.js";
import { getTransactionTimelineTxType as _, getTransactionTimelineTxStatus as g } from "../../../../utils/translation/getters.js";
import { TimelineDateFormat as M } from "../../../../internal/Timeline/types.js";
import { TypographyVariant as d, TypographyElement as f } from "../../../../internal/Typography/types.js";
import { TimelineItem as N } from "../../../../internal/Timeline/components/TimelineItem.js";
import w from "../../../../../core/Context/useCoreContext.js";
import y from "../../../../internal/Typography/Typography.js";
import x from "../../../../internal/Timeline/Timeline.js";
import P from "../../../../internal/StructuredList/StructuredList.js";
import { uniqueId as V } from "../../../../../utils/random/uniqueId.js";
const u = ({ transaction: e }) => {
  var r;
  const s = e.createdAt, t = e.events, i = (r = e.balanceAccount) == null ? void 0 : r.timeZone, n = m(
    () => ({
      format: M.FULL_DATE_EXACT_TIME_WITHOUT_SECONDS,
      date: new Date(s),
      timezone: i
    }),
    [s, i]
  ), l = m(() => {
    const o = /* @__PURE__ */ new WeakMap();
    return t == null || t.forEach((c) => o.set(c, V())), o;
  }, [t]);
  return t ? /* @__PURE__ */ a(x, { showMore: { limit: 2, placement: "before-last" }, children: t.map((o) => /* @__PURE__ */ a(u.Item, { event: o, defaultTimestamp: n }, l.get(o))) }) : null;
}, v = (e) => /* @__PURE__ */ a(y, { el: f.DIV, variant: d.CAPTION, className: D, children: e }), S = (e) => /* @__PURE__ */ a(y, { el: f.DIV, variant: d.CAPTION, className: C, children: e });
u.Item = T(({ defaultTimestamp: e, event: s }) => {
  const { amount: t, createdAt: i, status: n, type: l } = s, { i18n: r } = w(), o = m(() => ({ ...e, date: new Date(i) }), [e, i]), c = m(() => _(r, l), [r, l]), I = m(() => {
    const { currency: p, value: E } = t, h = `${r.amount(E, p, { hideCurrency: !0 })} ${p}`, A = g(r, n);
    return [
      { key: "transactions.details.timeline.fields.amount", value: h },
      { key: "transactions.details.timeline.fields.status", value: A }
    ];
  }, [r, t, n]), L = n.toLowerCase().includes("refund") ? "Refund" : n.toLowerCase().includes("auth") ? "Capture" : c;
  return /* @__PURE__ */ a(N, { title: L, timestamp: o, children: /* @__PURE__ */ a(
    P,
    {
      align: "start",
      layout: "3-9",
      items: I,
      renderLabel: v,
      renderValue: S
    }
  ) });
});
const K = T(u);
export {
  K as default
};
