// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/PaymentDetails/PaymentDetailsStatusBox.js
import { jsx as a } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import f from "classnames";
import A from "../../../../internal/StatusBox/StatusBox.js";
import d from "../../../../internal/StatusBox/useStatusBox.js";
import S from "../../../../../core/Context/useCoreContext.js";
import { getTransactionCategory as g } from "../../../../utils/translation/getters.js";
import { getAmountStyleForTransaction as y, getRefundTypeForTransaction as x } from "../../utils.js";
import { TX_DATA_AMOUNT as B, TX_DATA_CONTAINER as C, TX_STATUS_BOX as D, TX_DATA_TAGS as L } from "../../constants.js";
import { RefundType as u, RefundedState as T } from "../../types.js";
import { TagVariant as m } from "../../../../internal/Tag/types.js";
import { Tag as r } from "../../../../internal/Tag/Tag.js";
import { useMemo as c } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { memo as p } from "../../../../../external/preact/compat/dist/compat.module.js";
import { Fragment as U } from "../../../../../external/preact/dist/preact.module.js";
const i = ({ refundedState: e, transaction: t }) => {
  var l;
  const o = d({
    timezone: (l = t.balanceAccount) == null ? void 0 : l.timeZone,
    createdAt: t.createdAt,
    amountData: t.netAmount,
    paymentMethodData: t.paymentMethod,
    bankAccount: t.bankAccount
  }), n = c(
    () => ({
      amount: `${B}--${y(t)}`
    }),
    [t]
  ), s = c(
    () => /* @__PURE__ */ a(i.StatusTags, { transaction: t, refundedState: e }),
    [t, e]
  );
  return /* @__PURE__ */ a("div", { className: f(C, D), children: /* @__PURE__ */ a(A, { ...o, classNames: n, tag: s }) });
};
i.StatusTags = p(({ transaction: e, refundedState: t }) => {
  const { i18n: o } = S(), { category: n } = e, s = x(e);
  return /* @__PURE__ */ a("div", { className: L, children: [
    n && /* @__PURE__ */ a(r, { label: g(o, n), variant: m.DEFAULT }),
    s && /* @__PURE__ */ a(U, { children: [
      s === u.FULL && /* @__PURE__ */ a(r, { label: o.get("transactions.details.common.refundTypes.full"), variant: m.SUCCESS }),
      s === u.PARTIAL && /* @__PURE__ */ a(r, { label: o.get("transactions.details.common.refundTypes.partial"), variant: m.BLUE })
    ] }),
    t === T.FULL && /* @__PURE__ */ a(r, { label: o.get("transactions.details.common.refundedStates.full"), variant: m.SUCCESS }),
    t === T.PARTIAL && /* @__PURE__ */ a(r, { label: o.get("transactions.details.common.refundedStates.partial"), variant: m.BLUE })
  ] });
});
const $ = p(i);
export {
  $ as default
};
