// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/PaymentDetails/PaymentDetails.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import N from "../../../../internal/Tabs/Tabs.js";
import _ from "./PaymentRefundAlerts.js";
import C from "./PaymentDetailsActions.js";
import L from "./PaymentDetailsProperties.js";
import R from "./PaymentDetailsStatusBox.js";
import B from "./PaymentDetailsSummary.js";
import X from "./PaymentDetailsTimeline.js";
import w from "../../../../../core/Context/useCoreContext.js";
import { DetailsTab as o } from "../../types.js";
import { TX_DETAILS_TABS as x, TX_DATA_CLASS as U, TX_DATA_CONTAINER as Y } from "../../constants.js";
import { useState as j, useMemo as p, useEffect as O } from "../../../../../external/preact/hooks/dist/hooks.module.js";
const F = ({
  dataCustomization: i,
  extraFields: s,
  fullRefundFailed: c,
  fullRefundInProgress: v,
  refundAmounts: D,
  refundAvailable: d,
  refundCurrency: y,
  refundDisabled: S,
  refundedAmount: b,
  refundedState: l,
  refundLocked: g,
  setActiveView: h,
  transaction: e,
  transactionNavigator: E
}) => {
  const { i18n: P } = w(), [a, n] = j(), r = p(
    () => x.filter(({ id: m }) => {
      switch (m) {
        case o.SUMMARY:
          const { additions: u, deductions: f, originalAmount: T, amountBeforeDeductions: A, netAmount: M } = e;
          return u && u.length > 0 || f && f.length > 0 || T && T.value !== A.value || M.value !== A.value;
        case o.TIMELINE:
          return e.events && e.events.length > 0;
        default:
          return !0;
      }
    }),
    [e]
  ), I = p(
    () => ({
      [o.DETAILS]: /* @__PURE__ */ t(L, { dataCustomization: i, extraFields: s, transaction: e }),
      [o.SUMMARY]: /* @__PURE__ */ t(B, { transaction: e }),
      [o.TIMELINE]: /* @__PURE__ */ t(X, { transaction: e })
    }),
    [i, s, e]
  );
  return O(() => {
    var m;
    return n((m = r[0]) == null ? void 0 : m.id);
  }, [r]), /* @__PURE__ */ t("div", { className: U, children: [
    /* @__PURE__ */ t(R, { refundedState: l, transaction: e }),
    /* @__PURE__ */ t("div", { className: Y, children: [
      r.length > 1 && /* @__PURE__ */ t(
        N,
        {
          "aria-label": P.get("transactions.details.viewSelect.a11y.label"),
          onChange: ({ id: m }) => n(m),
          tabs: r,
          activeTab: a
        }
      ),
      a && I[a]
    ] }),
    /* @__PURE__ */ t(
      _,
      {
        fullRefundFailed: c,
        fullRefundInProgress: v,
        refundAmounts: D,
        refundCurrency: y,
        refundedAmount: b,
        refundedState: l,
        refundLocked: g
      }
    ),
    /* @__PURE__ */ t(
      C,
      {
        extraFields: s,
        refundAvailable: d,
        refundDisabled: S,
        setActiveView: h,
        transaction: e,
        transactionNavigator: E
      }
    )
  ] });
};
export {
  F as default
};
