// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/PaymentDetails/PaymentRefundAlerts.js
import { jsx as i } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { memo as y } from "../../../../../external/preact/compat/dist/compat.module.js";
import { RefundedState as T } from "../../types.js";
import { useMemo as h, useCallback as A, useRef as P, useEffect as _ } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { TX_REFUND_STATUSES_CONTAINER as I } from "../../constants.js";
import { AlertVariantOption as N, AlertTypeOption as p } from "../../../../internal/Alert/types.js";
import v from "../../../../../core/Context/useCoreContext.js";
import { Alert as F } from "../../../../internal/Alert/Alert.js";
import { TypographyVariant as R, TypographyElement as E } from "../../../../internal/Typography/types.js";
import L from "../../../../internal/Typography/Typography.js";
const b = {
  type: p.HIGHLIGHT,
  variant: N.TIP
}, k = y(
  ({
    fullRefundFailed: m,
    fullRefundInProgress: d,
    refundAmounts: s,
    refundCurrency: l,
    refundedAmount: c,
    refundedState: f,
    refundLocked: u
  }) => {
    const { i18n: t } = v(), a = h(() => {
      const e = new Intl.ListFormat(t.locale, { type: "conjunction" });
      return (o) => e.format(o.map((g) => t.amount(g, l)));
    }, [t, l]), r = A(({ description: e, ...o }) => {
      n.current.push(
        /* @__PURE__ */ i(F, { ...b, ...o, children: /* @__PURE__ */ i(L, { className: "adyen-pe-alert__description", el: E.DIV, variant: R.BODY, wide: !0, children: e }) })
      );
    }, []), n = P([]);
    if (f === T.FULL)
      r({ description: t.get("transactions.details.refund.alerts.refundedFull") });
    else {
      if (c > 0) {
        const e = { amount: a([c]) };
        r({ description: t.get("transactions.details.refund.alerts.refundedAmount", { values: e }) });
      }
      if (u)
        r({ description: t.get("transactions.details.refund.alerts.inProgressBlocked") });
      else {
        if (s.in_progress && s.in_progress.length > 0)
          if (d)
            r({ description: t.get("transactions.details.refund.alerts.inProgress") });
          else {
            const e = { amount: a(s.in_progress) };
            r({ description: t.get("transactions.details.refund.alerts.inProgressAmount", { values: e }) });
          }
        if (s.failed && s.failed.length > 0) {
          const e = p.WARNING;
          if (m)
            r({ type: e, description: t.get("transactions.details.refund.alerts.notPossible") });
          else {
            const o = { amount: a(s.failed) };
            r({ type: e, description: t.get("transactions.details.refund.alerts.notPossibleAmount", { values: o }) });
          }
        }
      }
    }
    return _(() => {
      n.current = [];
    }), n.current.length > 0 ? /* @__PURE__ */ i("div", { className: I, children: n.current }) : null;
  }
);
export {
  k as default
};
