// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/PaymentDetails/PaymentDetailsActions.js
import { jsx as a } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import D from "classnames";
import { memo as P } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useMemo as n } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { ActiveView as h } from "../../types.js";
import { ButtonVariant as u } from "../../../../internal/Button/types.js";
import { ButtonActionsLayoutBasic as E } from "../../../../internal/Button/ButtonActions/types.js";
import { sharedTransactionDetailsEventProperties as O, TX_DATA_CONTAINER as _, TX_DATA_ACTION_BAR as I } from "../../constants.js";
import w from "../../../../internal/Button/ButtonActions/ButtonActions.js";
import x from "../../../../../core/Context/analytics/useAnalyticsContext.js";
import L from "../../../../../core/Context/useCoreContext.js";
import { Icon as M } from "../../../../internal/Icon/Icon.js";
import { Fragment as S } from "../../../../../external/preact/dist/preact.module.js";
const X = ({
  extraFields: f,
  refundAvailable: d,
  refundDisabled: l,
  setActiveView: p,
  transaction: T,
  transactionNavigator: C
}) => {
  const e = x(), { currentTransaction: g, canNavigateBackward: A, canNavigateForward: v, backward: N, forward: y } = C, { i18n: i } = L(), r = n(
    () => ({
      backToRefund: { title: i.get("transactions.details.actions.backToRefund"), eventLabel: "Return to refund" },
      goToPayment: { title: i.get("transactions.details.actions.goToPayment"), eventLabel: "Go to payment" },
      refund: { title: i.get("transactions.details.actions.refund") }
    }),
    [i]
  ), s = n(() => {
    if (g === T.id) {
      if (A) return "backToRefund";
      if (v) return "goToPayment";
    }
  }, [A, v, g, T.id]), R = n(() => {
    if (d)
      return {
        disabled: l,
        event: () => {
          l || p(h.REFUND);
        },
        title: r.refund.title,
        variant: u.PRIMARY
      };
  }, [r, d, l, p]), b = n(() => {
    if (s) {
      const t = r[s], o = s === "backToRefund", c = o ? N : y;
      return {
        disabled: !1,
        event: () => {
          var m;
          try {
            c();
          } finally {
            (m = e.addEvent) == null || m.call(e, "Clicked button", {
              ...O,
              label: t.eventLabel
            });
          }
        },
        renderTitle: (m) => /* @__PURE__ */ a(S, { children: [
          /* @__PURE__ */ a(M, { name: "angle-right", style: { transform: `scaleX(${o ? -1 : 1})` } }),
          /* @__PURE__ */ a("span", { children: m })
        ] }),
        title: t.title,
        variant: u.SECONDARY
      };
    }
  }, [r, N, y, s, e]), B = n(
    () => Object.values(f || []).filter((t) => (t == null ? void 0 : t.type) === "button").map((t) => {
      var o, c;
      return {
        title: t.value,
        variant: u.SECONDARY,
        event: (o = t.config) == null ? void 0 : o.action,
        classNames: (c = t.config) != null && c.className ? [t.config.className] : []
      };
    }),
    [f]
  ), k = n(
    () => [R, b, ...B].filter(Boolean),
    [R, b, B]
  );
  return k.length > 0 ? /* @__PURE__ */ a("div", { className: D(_, I), children: /* @__PURE__ */ a(w, { actions: k, layout: E.BUTTONS_END }) }) : null;
}, W = P(X);
export {
  W as default
};
