// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinksOverview/components/PaymentLinkDetailsModal/PaymentLinkDetailsModal.js
import { jsx as s } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useState as u, useCallback as r } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import f from "../../../../../core/Context/useCoreContext.js";
import { PaymentLinkDetails as c } from "../../../PaymentLinkDetails/components/PaymentLinkDetails/PaymentLinkDetails.js";
import h from "../../../../internal/Modal/Modal.js";
const b = ({
  children: d,
  selectedDetail: i,
  resetDetails: e,
  onUpdate: t
}) => {
  const { i18n: m } = f(), [n, a] = u(!1), l = !!i, o = r(() => {
    n && (a(!1), t()), e();
  }, [n, t, e]), p = r(() => {
    a(!0);
  }, []);
  return /* @__PURE__ */ s("div", { children: [
    d,
    i && /* @__PURE__ */ s(
      h,
      {
        isOpen: l,
        "aria-label": m.get("payByLink.details.title"),
        onClose: o,
        isDismissible: !0,
        headerWithBorder: !1,
        size: i.modalSize || "large",
        children: /* @__PURE__ */ s(
          c,
          {
            id: i.selection.data,
            onUpdate: p,
            onDismiss: o,
            hideTitle: !0,
            isDismissButtonHidden: !0
          }
        )
      }
    )
  ] });
};
export {
  b as PaymentLinkDetailsModal
};
