// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/CustomerDetailsForm/Fields/Address/BillingCountryField.js
import { jsx as w } from "../../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import x from "../../../../../../../../core/Context/useCoreContext.js";
import { useMemo as y } from "../../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { FormSelect as A } from "../../../../../../../internal/FormWrappers/FormSelect.js";
import { useWizardFormContext as h } from "../../../../../../../../hooks/form/wizard/WizardFormContext.js";
const N = ({
  countriesData: e,
  isAddressFieldRequired: n,
  isFetchingCountries: s,
  countryDatasetData: i,
  isFetchingCountryDataset: f
}) => {
  var o;
  const { i18n: p } = x(), { fieldsConfig: b } = h(), g = y(() => {
    var m;
    const r = new Set(((e == null ? void 0 : e.data) ?? []).map(({ countryCode: l }) => l).filter(Boolean));
    return (i != null && i.length ? i : ((m = e == null ? void 0 : e.data) == null ? void 0 : m.map(({ countryCode: l, countryName: d }) => ({ id: l, name: d }))) ?? []).filter(({ id: l }) => !r.size || r.has(l)).sort((l, d) => l.name.localeCompare(d.name));
  }, [e == null ? void 0 : e.data, i]), C = ((o = b["billingAddress.country"]) == null ? void 0 : o.required) || n("billingAddress.country");
  return /* @__PURE__ */ w(
    A,
    {
      clearable: !0,
      filterable: !0,
      className: "adyen-pe-payment-link-creation-form__billing-address-field--medium",
      fieldName: "billingAddress.country",
      label: p.get("payByLink.creation.fields.billingAddress.country.label"),
      items: g,
      readonly: s || f,
      hideOptionalLabel: !0,
      isRequired: C
    }
  );
};
export {
  N as BillingCountryField
};
