// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/CustomerDetailsForm/Fields/Address/ShippingHouseNumberField.js
import { jsx as u } from "../../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import l from "../../../../../../../../core/Context/useCoreContext.js";
import { useWizardFormContext as n } from "../../../../../../../../hooks/form/wizard/WizardFormContext.js";
import { useCallback as N } from "../../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { FormTextInput as p } from "../../../../../../../internal/FormWrappers/FormTextInput.js";
import { PAYMENT_LINK_CREATION_FIELD_LENGTHS as b } from "../../../../../constants.js";
const x = ({
  isSeparateAddress: e,
  isAddressFieldRequired: o
}) => {
  var s;
  const { i18n: m } = l(), { setValue: r, fieldsConfig: i } = n(), t = N(
    (d) => {
      !e && r("billingAddress.houseNumberOrName", d.currentTarget.value);
    },
    [e, r]
  ), a = ((s = i["deliveryAddress.houseNumberOrName"]) == null ? void 0 : s.required) || o("deliveryAddress.houseNumberOrName");
  return /* @__PURE__ */ u(
    p,
    {
      maxLength: b.deliveryAddress.houseNumberOrName.max,
      fieldName: "deliveryAddress.houseNumberOrName",
      label: m.get("payByLink.creation.fields.deliveryAddress.houseNumberOrName.label"),
      onInput: t,
      className: "adyen-pe-payment-link-creation-form__shipping-address-field--small",
      hideOptionalLabel: !0,
      isRequired: a
    }
  );
};
export {
  x as ShippingHouseNumberField
};
