// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/CustomerDetailsForm/Fields/Address/ShippingPostalCodeField.js
import { jsx as n } from "../../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import p from "../../../../../../../../core/Context/useCoreContext.js";
import { useWizardFormContext as m } from "../../../../../../../../hooks/form/wizard/WizardFormContext.js";
import { useCallback as u } from "../../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { FormTextInput as C } from "../../../../../../../internal/FormWrappers/FormTextInput.js";
import { PAYMENT_LINK_CREATION_FIELD_LENGTHS as f } from "../../../../../constants.js";
const L = ({
  isSeparateAddress: e,
  isAddressFieldRequired: s
}) => {
  var r;
  const { i18n: t } = p(), { setValue: o, fieldsConfig: d } = m(), i = u(
    (a) => {
      !e && o("billingAddress.postalCode", a.currentTarget.value);
    },
    [e, o]
  ), l = ((r = d["deliveryAddress.postalCode"]) == null ? void 0 : r.required) || s("deliveryAddress.postalCode");
  return /* @__PURE__ */ n(
    C,
    {
      maxLength: f.deliveryAddress.postalCode.max,
      fieldName: "deliveryAddress.postalCode",
      label: t.get("payByLink.creation.fields.deliveryAddress.postalCode.label"),
      onInput: i,
      className: "adyen-pe-payment-link-creation-form__shipping-address-field--small",
      hideOptionalLabel: !0,
      isRequired: l
    }
  );
};
export {
  L as ShippingPostalCodeField
};
