// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/CustomerDetailsForm/Fields/Address/ShippingStreetField.js
import { jsx as a } from "../../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import m from "../../../../../../../../core/Context/useCoreContext.js";
import { useWizardFormContext as p } from "../../../../../../../../hooks/form/wizard/WizardFormContext.js";
import { useCallback as u } from "../../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { FormTextInput as f } from "../../../../../../../internal/FormWrappers/FormTextInput.js";
import { PAYMENT_LINK_CREATION_FIELD_LENGTHS as c } from "../../../../../constants.js";
const L = ({
  isSeparateAddress: e,
  isAddressFieldRequired: s
}) => {
  var t;
  const { i18n: i } = m(), { setValue: r, fieldsConfig: o } = p(), d = u(
    (n) => {
      !e && r("billingAddress.street", n.currentTarget.value);
    },
    [e, r]
  ), l = ((t = o["deliveryAddress.street"]) == null ? void 0 : t.required) || s("deliveryAddress.street");
  return /* @__PURE__ */ a(
    f,
    {
      maxLength: c.deliveryAddress.street.max,
      fieldName: "deliveryAddress.street",
      label: i.get("payByLink.creation.fields.deliveryAddress.street.label"),
      onInput: d,
      className: "adyen-pe-payment-link-creation-form__shipping-address-field--large",
      hideOptionalLabel: !0,
      isRequired: l
    }
  );
};
export {
  L as ShippingStreetField
};
