// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/CustomerDetailsForm/Fields/Address/ShippingCityField.js
import { jsx as m } from "../../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import a from "../../../../../../../../core/Context/useCoreContext.js";
import { useWizardFormContext as c } from "../../../../../../../../hooks/form/wizard/WizardFormContext.js";
import { useCallback as p } from "../../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { FormTextInput as y } from "../../../../../../../internal/FormWrappers/FormTextInput.js";
import { PAYMENT_LINK_CREATION_FIELD_LENGTHS as u } from "../../../../../constants.js";
const L = ({
  isSeparateAddress: e,
  isAddressFieldRequired: t
}) => {
  var r;
  const { i18n: o } = a(), { setValue: i, fieldsConfig: s } = c(), d = p(
    (l) => {
      !e && i("billingAddress.city", l.currentTarget.value);
    },
    [e, i]
  ), n = ((r = s["deliveryAddress.city"]) == null ? void 0 : r.required) || t("deliveryAddress.city");
  return /* @__PURE__ */ m(
    y,
    {
      maxLength: u.deliveryAddress.city.max,
      fieldName: "deliveryAddress.city",
      label: o.get("payByLink.creation.fields.deliveryAddress.city.label"),
      onInput: d,
      className: "adyen-pe-payment-link-creation-form__shipping-address-field--medium",
      hideOptionalLabel: !0,
      isRequired: n
    }
  );
};
export {
  L as ShippingCityField
};
