// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/CustomerDetailsForm/Fields/Address/ShippingCountryField.js
import { jsx as v } from "../../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import c from "../../../../../../../../core/Context/useCoreContext.js";
import { useMemo as w, useCallback as x } from "../../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { FormSelect as A } from "../../../../../../../internal/FormWrappers/FormSelect.js";
import { useWizardFormContext as k } from "../../../../../../../../hooks/form/wizard/WizardFormContext.js";
const S = ({
  countriesData: e,
  isAddressFieldRequired: p,
  isFetchingCountries: f,
  isSeparateAddress: i,
  countryDatasetData: d,
  isFetchingCountryDataset: C
}) => {
  var n;
  const { i18n: g } = c(), { setValue: m, fieldsConfig: y } = k(), h = w(() => {
    var s;
    const r = new Set(((e == null ? void 0 : e.data) ?? []).map(({ countryCode: l }) => l).filter(Boolean));
    return (d != null && d.length ? d : ((s = e == null ? void 0 : e.data) == null ? void 0 : s.map(({ countryCode: l, countryName: o }) => ({ id: l, name: o }))) ?? []).filter(({ id: l }) => !r.size || r.has(l)).sort((l, o) => l.name.localeCompare(o.name));
  }, [e == null ? void 0 : e.data, d]), b = x(
    (r) => {
      i || m("billingAddress.country", r.target.value);
    },
    [i, m]
  ), u = ((n = y["deliveryAddress.country"]) == null ? void 0 : n.required) || p("deliveryAddress.country");
  return /* @__PURE__ */ v(
    A,
    {
      clearable: !0,
      filterable: !0,
      fieldName: "deliveryAddress.country",
      label: g.get("payByLink.creation.fields.deliveryAddress.country.label"),
      items: h,
      readonly: f || C,
      className: "adyen-pe-payment-link-creation-form__shipping-address-field--medium",
      onChange: b,
      hideOptionalLabel: !0,
      isRequired: u
    }
  );
};
export {
  S as ShippingCountryField
};
