// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/Select/components/SelectButton.js
import { jsx as i } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import L from "classnames";
import { useMemo as N } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import B from "../../../../../core/Context/useCoreContext.js";
import { ButtonVariant as R } from "../../../Button/types.js";
import h from "../../../Typography/Typography.js";
import { TypographyVariant as A, TypographyElement as E } from "../../../Typography/types.js";
import { DROPDOWN_BUTTON_ICON_CLASS as I, DROPDOWN_BUTTON_TEXT_CLASS as P, DROPDOWN_BUTTON_MULTI_SELECT_COUNTER_CLASS as g, DROPDOWN_BUTTON_COLLAPSE_INDICATOR_CLASS as U, DROPDOWN_BUTTON_CLEAR_CLASS as x, DROPDOWN_BUTTON_CLASS as W, DROPDOWN_BUTTON_VALID_CLASS as v, DROPDOWN_BUTTON_INVALID_CLASS as w, DROPDOWN_BUTTON_READONLY_CLASS as V, DROPDOWN_BUTTON_HAS_SELECTION_CLASS as K, DROPDOWN_BUTTON_ACTIVE_CLASS as F, DROPDOWN_BUTTON_CLASSNAME as k } from "../constants.js";
import { InteractionKeyCode as b } from "../../../../types.js";
import { Fragment as M } from "../../../../../external/preact/dist/preact.module.js";
import Y from "../../../Img/Img.js";
import { Icon as u } from "../../../Icon/Icon.js";
import j from "../../../Button/Button.js";
import { boolOrFalse as f } from "../../../../../utils/value/bool.js";
const H = ({
  active: a,
  disabled: T,
  className: t,
  filterable: r,
  toggleButtonRef: n,
  buttonVariant: c,
  ...l
}) => {
  const o = N(() => r ? L(k, t) : t, [t, r]);
  return r ? /* @__PURE__ */ i("div", { ...l, className: o, ref: n }) : /* @__PURE__ */ i(
    j,
    {
      ...l,
      className: o,
      disabled: T,
      variant: c ?? R.SECONDARY,
      ref: n,
      "aria-label": l["aria-label"],
      "aria-labelledby": l["aria-labelledby"]
    }
  );
}, na = (a) => {
  var S, C;
  const { i18n: T } = B(), { active: t, clearable: r, filterable: n, multiSelect: c, placeholder: l, onClear: o, readonly: O, showList: d, withoutCollapseIndicator: y } = a, s = N(() => (l == null ? void 0 : l.trim()) || T.get("common.inputs.select.placeholder"), [T, l]), e = N(() => f(c) ? void 0 : t[0], [t, c]), _ = N(() => {
    var m;
    return ((m = e == null ? void 0 : e.name) == null ? void 0 : m.trim()) || s;
  }, [e, s]), D = N(() => f(r) && !!t.length, [t, r]);
  return /* @__PURE__ */ i(
    H,
    {
      buttonVariant: a.buttonVariant,
      active: t,
      disabled: O,
      "aria-disabled": O,
      "aria-expanded": d,
      "aria-haspopup": "dialog",
      "aria-invalid": a.isInvalid,
      className: L(W, {
        [F]: d,
        [K]: !!t.length,
        [V]: O,
        [w]: a.isInvalid,
        [v]: a.isValid
      }),
      filterable: n,
      name: a.name,
      onClick: O ? void 0 : a.toggleList,
      onKeyDown: O ? void 0 : a.onButtonKeyDown,
      role: !n || d ? "button" : void 0,
      tabIndex: 0,
      title: _,
      toggleButtonRef: a.toggleButtonRef,
      type: n ? void 0 : "button",
      "aria-describedby": a.ariaDescribedBy,
      id: a.id,
      ...d && n ? {} : { "aria-label": a["aria-label"], "aria-labelledby": a["aria-labelledby"] },
      children: [
        a.renderButtonContent ? a.renderButtonContent({ item: e }) : d && n ? /* @__PURE__ */ i(
          "input",
          {
            "aria-autocomplete": "list",
            "aria-label": a["aria-label"],
            "aria-labelledby": a["aria-labelledby"],
            "aria-controls": a.selectListId,
            "aria-expanded": d,
            "aria-owns": a.selectListId,
            "aria-invalid": a.isInvalid,
            autoComplete: "off",
            className: "adyen-pe-filter-input",
            onInput: a.onInput,
            onKeyDown: a.onFilterInputKeyDown,
            placeholder: s,
            ref: a.filterInputRef,
            role: "combobox",
            type: "text"
          }
        ) : /* @__PURE__ */ i(M, { children: [
          (e == null ? void 0 : e.icon) && /* @__PURE__ */ i(Y, { className: I, src: e.icon, alt: ((S = e == null ? void 0 : e.name) == null ? void 0 : S.trim()) ?? "" }),
          /* @__PURE__ */ i("span", { className: P, children: ((C = e == null ? void 0 : e.selectedOptionName) == null ? void 0 : C.trim()) || _ }),
          c && a.appliedFilterNumber > 0 && /* @__PURE__ */ i("div", { className: g, children: /* @__PURE__ */ i(h, { el: E.SPAN, variant: A.BODY, stronger: !0, children: a.appliedFilterNumber }) })
        ] }),
        !y && !D && /* @__PURE__ */ i("span", { className: U, children: /* @__PURE__ */ i(u, { name: "chevron-down" }) }),
        D && /* @__PURE__ */ i(
          "span",
          {
            role: "button",
            tabIndex: 0,
            className: x,
            onClick: o,
            onKeyDown: (m) => {
              (m.code === b.ENTER || m.code === b.SPACE) && (o == null || o(m));
            },
            children: /* @__PURE__ */ i(u, { name: "cross-circle-fill" })
          }
        )
      ]
    }
  );
};
export {
  na as default
};
