// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/Select/components/SelectList.js
import { jsx as o } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useResponsiveContainer as h, containerQueries as W } from "../../../../../hooks/useResponsiveContainer.js";
import w from "../../../Popover/PopoverContainer.js";
import { PopoverContainerPosition as V, PopoverContainerVariant as y, PopoverContainerSize as z } from "../../../Popover/types.js";
import F from "../../../../../core/Context/useCoreContext.js";
import a from "classnames";
import { memo as j } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useMemo as l } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { DROPDOWN_LIST_CLASS as B, DROPDOWN_LIST_ACTIVE_CLASS as K, DROPDOWN_ELEMENT_CLASS as Q, DROPDOWN_ELEMENT_NO_OPTION_CLASS as U } from "../constants.js";
import $, { renderListItemDefault as k } from "./SelectListItem.js";
import { fixedForwardRef as q } from "../../../../../utils/preact/fixedForwardRef.js";
import { isFunction as G } from "../../../../../utils/value/is.js";
import { boolOrFalse as H } from "../../../../../utils/value/bool.js";
const J = q(
  ({
    active: c,
    commitActions: p,
    items: d,
    disableFocusTrap: f,
    multiSelect: n,
    onKeyDown: O,
    onSelect: u,
    renderListItem: i,
    selectListId: S,
    showList: t,
    textFilter: s,
    toggleButtonRef: C,
    dismissPopover: P,
    setToTargetWidth: N,
    popoverClassNameModifiers: v,
    showOverlay: _,
    fitPosition: D,
    fixedPopoverPositioning: E,
    activeIndex: L,
    filterable: T
  }, M) => {
    const { i18n: x } = F(), I = h(W.down.xs), m = d.filter((e) => !s || e.name.toLowerCase().includes(s)), R = a(B, { [K]: t }), A = a(Q, U), b = l(() => G(i) ? i : k, [i]), r = l(() => H(n), [n]);
    return t ? /* @__PURE__ */ o(
      w,
      {
        classNameModifiers: v,
        actions: r ? p : void 0,
        disableFocusTrap: f,
        divider: !0,
        dismiss: P,
        dismissible: !1,
        open: t,
        setToTargetWidth: N,
        containerSize: z.MEDIUM,
        variant: y.POPOVER,
        targetElement: C,
        withContentPadding: !1,
        position: V.BOTTOM,
        showOverlay: _ && I,
        fitPosition: D,
        fixedPositioning: E,
        children: /* @__PURE__ */ o("ul", { className: R, id: S, ref: M, role: "listbox", "aria-multiselectable": r, children: m.length ? m.map((e, g) => /* @__PURE__ */ o(
          $,
          {
            item: e,
            multiSelect: r,
            onKeyDown: O,
            onSelect: u,
            renderListItem: b,
            selected: c.includes(e),
            isKeyboardActive: T && L === g
          },
          e.id
        )) : /* @__PURE__ */ o("div", { className: A, children: x.get("common.inputs.select.errors.noOptions") }) })
      }
    ) : null;
  }
), ce = j(J);
export {
  ce as default
};
