// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/CustomerDetailsForm/CustomerDetailsForm.js
import { jsx as e } from "../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { ShopperEmailField as q } from "./Fields/ShopperEmailField.js";
import { ShopperPhoneField as R } from "./Fields/ShopperPhoneField.js";
import { CountryRegionField as T } from "./Fields/CountryRegionField.js";
import { ShippingStreetField as S } from "./Fields/Address/ShippingStreetField.js";
import { LanguageField as E } from "./Fields/LanguageField.js";
import { BillingAndShippingCheckboxField as P } from "./Fields/BillingAndShippingCheckboxField.js";
import { FormTextInput as s } from "../../../../../internal/FormWrappers/FormTextInput.js";
import { ShippingHouseNumberField as I } from "./Fields/Address/ShippingHouseNumberField.js";
import { ShippingCountryField as V } from "./Fields/Address/ShippingCountryField.js";
import { ShippingCityField as Y } from "./Fields/Address/ShippingCityField.js";
import { ShippingPostalCodeField as H } from "./Fields/Address/ShippingPostalCodeField.js";
import { BillingCountryField as U } from "./Fields/Address/BillingCountryField.js";
import { TypographyVariant as d, TypographyElement as o } from "../../../../../internal/Typography/types.js";
import $ from "../../../../../../core/Context/useCoreContext.js";
import t from "../../../../../internal/Typography/Typography.js";
/* empty css                         */
import { PAYMENT_LINK_CREATION_FIELD_LENGTHS as n } from "../../../constants.js";
import { useWizardFormContext as j } from "../../../../../../hooks/form/wizard/WizardFormContext.js";
import { useAddressChecker as z } from "./useAddressChecker.js";
import { Fragment as G } from "../../../../../../external/preact/dist/preact.module.js";
const ce = ({
  isSeparateAddress: a,
  setIsSeparateAddress: B,
  countriesData: m,
  isFetchingCountries: p,
  countryDatasetData: c,
  isFetchingCountryDataset: b
}) => {
  var y, h, N, u, A, _, L, k, v, x;
  const { i18n: i } = $(), { fieldsConfig: l } = j(), { isAddressFieldRequired: r } = z(), C = ((y = l["shopperName.firstName"]) == null ? void 0 : y.visible) || ((h = l["shopperName.lastName"]) == null ? void 0 : h.visible), O = !((N = l["billingAddress.street"]) != null && N.required), F = !((u = l["deliveryAddress.street"]) != null && u.required), g = (A = l["billingAddress.street"]) == null ? void 0 : A.visible, f = (_ = l["deliveryAddress.street"]) == null ? void 0 : _.visible;
  return /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__fields-container", children: [
    /* @__PURE__ */ e(
      s,
      {
        fieldName: "shopperReference",
        label: i.get("payByLink.creation.fields.shopperReference.label"),
        maxLength: n.shopperReference.max,
        minLength: n.shopperReference.min
      }
    ),
    C && /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__shopper-name-container", children: [
      /* @__PURE__ */ e(
        s,
        {
          fieldName: "shopperName.firstName",
          label: i.get("payByLink.creation.fields.shopperName.label"),
          maxLength: n.shopperName.firstName.max
        }
      ),
      /* @__PURE__ */ e(
        s,
        {
          maxLength: n.shopperName.lastName.max,
          fieldName: "shopperName.lastName",
          label: i.get("payByLink.creation.fields.shopperLastName.label")
        }
      )
    ] }),
    /* @__PURE__ */ e(q, {}),
    /* @__PURE__ */ e(R, {}),
    /* @__PURE__ */ e(
      T,
      {
        countriesData: m,
        isFetchingCountries: p,
        countryDatasetData: c,
        isFetchingCountryDataset: b
      }
    ),
    f && /* @__PURE__ */ e(G, { children: [
      /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__shipping-address-container", children: [
        /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__shipping-address-title-container", children: [
          /* @__PURE__ */ e(
            t,
            {
              el: o.SPAN,
              variant: d.SUBTITLE,
              stronger: !0,
              className: "adyen-pe-payment-link-creation-form__billing-address-title",
              children: i.get("payByLink.creation.sections.deliveryAddress.label")
            }
          ),
          F && /* @__PURE__ */ e(
            t,
            {
              el: o.SPAN,
              variant: d.BODY,
              className: "adyen-pe-payment-link-creation-form__field-label-optional",
              children: `(${i.get("payByLink.common.fields.optional.label")})`
            }
          )
        ] }),
        /* @__PURE__ */ e("div", { children: [
          /* @__PURE__ */ e(S, { isSeparateAddress: a, isAddressFieldRequired: r }),
          /* @__PURE__ */ e(I, { isSeparateAddress: a, isAddressFieldRequired: r })
        ] }),
        /* @__PURE__ */ e("div", { children: [
          /* @__PURE__ */ e(
            V,
            {
              countriesData: m,
              isAddressFieldRequired: r,
              isFetchingCountries: p,
              isSeparateAddress: a,
              countryDatasetData: c,
              isFetchingCountryDataset: b
            }
          ),
          /* @__PURE__ */ e(Y, { isAddressFieldRequired: r, isSeparateAddress: a }),
          /* @__PURE__ */ e(H, { isAddressFieldRequired: r, isSeparateAddress: a })
        ] })
      ] }),
      g && /* @__PURE__ */ e(P, { isSeparateAddress: a, setIsSeparateAddress: B })
    ] }),
    (a || !f && g) && /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__billing-address-container", children: [
      /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__billing-address-title-container", children: [
        /* @__PURE__ */ e(
          t,
          {
            el: o.SPAN,
            variant: d.SUBTITLE,
            stronger: !0,
            className: "adyen-pe-payment-link-creation-form__billing-address-title",
            children: i.get("payByLink.creation.sections.billingAddress.label")
          }
        ),
        O && /* @__PURE__ */ e(
          t,
          {
            el: o.SPAN,
            variant: d.BODY,
            className: "adyen-pe-payment-link-creation-form__field-label-optional",
            children: `(${i.get("payByLink.common.fields.optional.label")})`
          }
        )
      ] }),
      /* @__PURE__ */ e("div", { children: [
        /* @__PURE__ */ e(
          s,
          {
            maxLength: n.billingAddress.street.max,
            className: "adyen-pe-payment-link-creation-form__billing-address-field--large",
            fieldName: "billingAddress.street",
            label: i.get("payByLink.creation.fields.billingAddress.street.label"),
            hideOptionalLabel: !0,
            isRequired: ((L = l["billingAddress.street"]) == null ? void 0 : L.required) || r("billingAddress.street")
          }
        ),
        /* @__PURE__ */ e(
          s,
          {
            maxLength: n.billingAddress.houseNumberOrName.max,
            className: "adyen-pe-payment-link-creation-form__billing-address-field--small",
            fieldName: "billingAddress.houseNumberOrName",
            label: i.get("payByLink.creation.fields.billingAddress.houseNumberOrName.label"),
            hideOptionalLabel: !0,
            isRequired: ((k = l["billingAddress.houseNumberOrName"]) == null ? void 0 : k.required) || r("billingAddress.houseNumberOrName")
          }
        )
      ] }),
      /* @__PURE__ */ e("div", { children: [
        /* @__PURE__ */ e(
          U,
          {
            countriesData: m,
            isAddressFieldRequired: r,
            isFetchingCountries: p,
            countryDatasetData: c,
            isFetchingCountryDataset: b
          }
        ),
        /* @__PURE__ */ e(
          s,
          {
            maxLength: n.billingAddress.city.max,
            className: "adyen-pe-payment-link-creation-form__billing-address-field--medium",
            fieldName: "billingAddress.city",
            label: i.get("payByLink.creation.fields.billingAddress.city.label"),
            hideOptionalLabel: !0,
            isRequired: ((v = l["billingAddress.city"]) == null ? void 0 : v.required) || r("billingAddress.city")
          }
        ),
        /* @__PURE__ */ e(
          s,
          {
            maxLength: n.billingAddress.postalCode.max,
            className: "adyen-pe-payment-link-creation-form__billing-address-field--small",
            fieldName: "billingAddress.postalCode",
            label: i.get("payByLink.creation.fields.billingAddress.postalCode.label"),
            hideOptionalLabel: !0,
            isRequired: ((x = l["billingAddress.postalCode"]) == null ? void 0 : x.required) || r("billingAddress.postalCode")
          }
        )
      ] })
    ] }),
    /* @__PURE__ */ e(E, {})
  ] });
};
export {
  ce as CustomerDetailsForm
};
