// @adyen/adyen-platform-experience-web/dist/es/components/internal/DataGrid/DataGridCell.js
import { jsx as i } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useDataGridContext as n } from "./hooks/useDataGridContext.js";
import { useRef as m, useEffect as d } from "../../../external/preact/hooks/dist/hooks.module.js";
import { cloneElement as f, isValidElement as u } from "../../../external/preact/dist/preact.module.js";
import c from "classnames";
function C({
  children: e,
  column: r,
  position: a,
  ...o
}) {
  const { registerCells: l } = n(), t = m(null);
  return d(() => {
    var s;
    t.current && l({
      column: r,
      width: (s = t.current) == null ? void 0 : s.getBoundingClientRect().width
    });
  }, [r, l]), /* @__PURE__ */ i(
    "div",
    {
      role: "cell",
      className: c("adyen-pe-data-grid__cell", {
        "adyen-pe-data-grid__cell--right": a === "right",
        "adyen-pe-data-grid__cell--center": a === "center"
      }),
      ...o,
      children: e && u(e) ? f(e, {
        ...e == null ? void 0 : e.props,
        ref: t,
        style: { width: "min-content" }
      }) : null
    }
  );
}
export {
  C as default
};
