// @adyen/adyen-platform-experience-web/dist/es/components/internal/DataGrid/DataGrid.js
import { jsx as t } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { toChildArray as w } from "../../../external/preact/dist/preact.module.js";
import u from "classnames";
/* empty css              */
import { TableBody as C } from "./components/TableBody.js";
import { InteractiveBody as _ } from "./components/InteractiveBody.js";
import x from "./components/SkeletonBody.js";
import { ErrorMessageDisplay as T } from "../ErrorMessageDisplay/ErrorMessageDisplay.js";
import { useMemo as r } from "../../../external/preact/hooks/dist/hooks.module.js";
import { DataGridProvider as k } from "./utils/DataGridProvider.js";
import { useDataGridContext as N } from "./hooks/useDataGridContext.js";
import { TableHeaderCell as B } from "./components/TableHeaderCell.js";
import R from "../../../core/Context/useCoreContext.js";
const L = Object.freeze({
  activeIndex: -1,
  index: -1
});
function y({ errorDisplay: e, ...o }) {
  return /* @__PURE__ */ t("div", { style: { width: "100%" }, children: /* @__PURE__ */ t(k, { children: /* @__PURE__ */ t(D, { ...o, errorDisplay: e }) }) });
}
function D({ autoFitColumns: e, errorDisplay: o, ...a }) {
  var c, s;
  const m = r(() => w(a.children), [a.children]), f = r(() => m.find((d) => (d == null ? void 0 : d.type) === g), [m]), n = r(() => {
    var d;
    return ((d = a.data) == null ? void 0 : d.length) === 0;
  }, [a.data]), h = r(() => !a.loading && (n || a.error), [n, a.error, a.loading]), { getMinWidthByColumn: b } = N(), { getImageAsset: l } = R(), i = a.columns.filter((d) => d.visible !== !1).map((d) => ({ ...d, minWidth: b(d.key) })), v = i.map((d) => `minmax(${(d.minWidth || 100) + (a.narrowColumns ? 0 : 40)}px, ${d.flex || 1}fr)`).join(" ");
  return /* @__PURE__ */ t(
    "div",
    {
      className: u("adyen-pe-data-grid", {
        "adyen-pe-data-grid--condensed": a.condensed,
        "adyen-pe-data-grid--outline": a.outline,
        "adyen-pe-data-grid--scrollable": a.scrollable,
        "adyen-pe-data-grid--loading": a.loading,
        "adyen-pe-data-grid--empty": n || a.error
      }),
      style: e ? void 0 : {
        "--adyen-pe-data-grid-cols": i.length,
        "--adyen-pe-data-grid-cells": v
      },
      children: [
        /* @__PURE__ */ t("div", { className: "adyen-pe-data-grid__table-wrapper", children: [
          /* @__PURE__ */ t("div", { role: "table", className: "adyen-pe-data-grid__table", children: [
            /* @__PURE__ */ t("div", { className: "adyen-pe-data-grid__head", role: "rowgroup", children: /* @__PURE__ */ t("div", { role: "rowheader", className: "adyen-pe-data-grid__header", style: a.loading ? { width: "100%" } : {}, children: i.map((d) => /* @__PURE__ */ t(B, { label: d.label, position: d.position, cellKey: d.key }, d.key)) }) }),
            /* @__PURE__ */ t(H, { ...a, columns: i, emptyBody: n })
          ] }),
          h && (n && !a.error ? /* @__PURE__ */ t(
            T,
            {
              withHeaderOffset: !0,
              title: ((c = a.emptyTableMessage) == null ? void 0 : c.title) ?? "common.errors.noResults",
              message: (s = a.emptyTableMessage) == null ? void 0 : s.message,
              imageDesktop: l == null ? void 0 : l({ name: "no-results-found" }),
              imageMobile: l == null ? void 0 : l({ name: "no-results-found", subFolder: "images/small" }),
              centered: !0
            }
          ) : a.error && o ? o() : null)
        ] }),
        f
      ]
    }
  );
}
function H(e) {
  const o = r(() => e.loading || e.emptyBody || e.error, [e.emptyBody, e.error, e.loading]);
  return /* @__PURE__ */ t(
    "div",
    {
      role: "rowgroup",
      className: u("adyen-pe-data-grid__body"),
      style: o && { display: "grid", gridTemplateColumns: "1fr" },
      children: o ? /* @__PURE__ */ t(x, { columnsNumber: e.columns.length, loading: e.loading }) : e.onRowClick ? /* @__PURE__ */ t(
        _,
        {
          onRowHover: e.onRowHover,
          data: e.data,
          columns: e.columns,
          onRowClick: e.onRowClick,
          customCells: e.customCells
        }
      ) : /* @__PURE__ */ t(
        C,
        {
          onRowHover: e.onRowHover,
          data: e.data,
          customCells: e.customCells,
          columns: e.columns
        }
      )
    }
  );
}
y.Footer = g;
function g({ children: e }) {
  return /* @__PURE__ */ t("div", { className: "adyen-pe-data-grid__footer", children: e });
}
y.defaultProps = {
  condensed: !1,
  outline: !0,
  scrollable: !0
};
export {
  L as INITIAL_STATE,
  y as default
};
