// @adyen/adyen-platform-experience-web/dist/es/components/internal/ErrorMessageDisplay/ErrorMessageDisplay.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import A from "classnames";
import _ from "../Typography/Typography.js";
import C from "../../../core/Context/useCoreContext.js";
import { TypographyVariant as u, TypographyElement as M } from "../Typography/types.js";
import { useCallback as S } from "../../../external/preact/hooks/dist/hooks.module.js";
/* empty css                         */
import { Fragment as y } from "../../../external/preact/dist/preact.module.js";
import f from "../Button/Button.js";
const r = "adyen-pe-error-message-display", g = 680, i = {
  base: r,
  base_absolutePosition: r + "--absolute-position",
  base_centered: r + "--centered",
  base_outlined: r + "--outlined",
  base_withBackground: r + "--with-background",
  base_withHeaderOffset: r + "--with-header-offset",
  base_condensed: r + "--condensed",
  button: r + "__button",
  illustration: r + "__illustration"
}, D = () => /* @__PURE__ */ e(y, { children: [
  " ",
  /* @__PURE__ */ e("br", {}),
  " "
] }), R = ({
  title: w,
  message: l,
  imageDesktop: n,
  imageMobile: m,
  withImage: v,
  centered: x,
  refreshComponent: b,
  onContactSupport: s,
  translationValues: a,
  absolutePosition: E = !0,
  outlined: h = !0,
  renderSecondaryButton: c,
  withBackground: T,
  withHeaderOffset: k,
  condensed: O
}) => {
  const { i18n: t, updateCore: B, getImageAsset: o } = C(), N = S(
    (d) => Array.isArray(d) ? d.map((p, P) => /* @__PURE__ */ e(y, { children: [
      P > 0 && /* @__PURE__ */ e(D, {}),
      t.get(p),
      a == null ? void 0 : a[p]
    ] })) : t.get(d),
    [t, a]
  );
  return /* @__PURE__ */ e(
    "div",
    {
      className: A(i.base, {
        [i.base_absolutePosition]: E,
        [i.base_centered]: x,
        [i.base_outlined]: h,
        [i.base_withBackground]: T !== !1 && !h,
        [i.base_withHeaderOffset]: k,
        [i.base_condensed]: O
      }),
      children: [
        (n || m || v) && /* @__PURE__ */ e("div", { className: i.illustration, children: /* @__PURE__ */ e("picture", { children: [
          /* @__PURE__ */ e("source", { type: "image/svg+xml", media: `(min-width: ${g}px)`, srcSet: n }),
          /* @__PURE__ */ e(
            "source",
            {
              type: "image/svg+xml",
              media: `(max-width: ${g}px)`,
              srcSet: m ?? (o == null ? void 0 : o({ name: "wrong-environment", subFolder: "images/small" }))
            }
          ),
          /* @__PURE__ */ e("img", { srcSet: n ?? (o == null ? void 0 : o({ name: "wrong-environment" })), alt: "" })
        ] }) }),
        /* @__PURE__ */ e(_, { el: M.DIV, variant: u.TITLE, children: t.get(w) }),
        l && /* @__PURE__ */ e(_, { variant: u.BODY, children: N(l) }),
        (s || b || c) && /* @__PURE__ */ e("div", { className: i.button, children: [
          c && c(),
          s && /* @__PURE__ */ e(f, { onClick: s, children: t.get("common.actions.contactSupport.labels.reachOut") }),
          !s && b && /* @__PURE__ */ e(f, { onClick: B, children: t.get("common.actions.refresh.labels.default") })
        ] })
      ]
    }
  );
};
export {
  R as ErrorMessageDisplay,
  g as IMAGE_BREAKPOINT_MEDIUM_PX
};
