// @adyen/adyen-platform-experience-web/dist/es/core/Localization/utils.js
import { FALLBACK_LOCALE as p, DEFAULT_TRANSLATIONS as C, SUPPORTED_LOCALES as O } from "./constants/localization.js";
import { hasOwnProperty as f } from "../../utils/struct/property.js";
import { EMPTY_OBJECT as u } from "../../utils/value/constants.js";
import { asPlainObject as m } from "../../utils/struct/main.js";
import { isFunction as g } from "../../utils/value/is.js";
const E = { values: u, count: 0 }, w = /^[a-z]{2}-[A-Z]{2}$/, l = (e) => e.substring(0, 2).toLowerCase();
function T(e, t) {
  if (!e) return null;
  const r = l(e);
  return t.find((n) => l(n) === r) || null;
}
function _(e) {
  const t = e.replace("_", "-");
  if (w.test(t)) return t;
  const [r, n] = t.split("-");
  if (!r || !n) return null;
  const o = `${r.toLowerCase()}-${n.toUpperCase()}`;
  return o.length === 5 ? o : null;
}
function d(e, t) {
  const r = e.trim();
  if (!r || r.length < 1 || r.length > 5) return p;
  const n = _(r);
  return n && t.includes(n) ? n : T(n ?? r, t);
}
function N(e = u, t) {
  return e === u ? e : Object.keys(e).reduce((r, n) => {
    const o = _(n) || d(n, t);
    return o && e[n] && (r[o] = e[n]), r;
  }, {});
}
const L = (e, t) => {
  if (g(t)) {
    const r = /* @__PURE__ */ new Map();
    let n = -1;
    return e.replace(/%{(\w+)}/g, (o, c) => {
      let i = r.get(c) ?? -1;
      const A = t(c, ++n, ++i) ?? "";
      return r.set(c, i), A;
    });
  }
  return e.replace(/%{(\w+)}/g, (r, n) => (t == null ? void 0 : t[n]) ?? "");
}, R = (e, t, r = E) => {
  const n = r.count ?? 0, o = `${t}__${n}`;
  if (f(e, o) && e[o])
    return L(e[o], r.values);
  const c = `${t}__plural`;
  return f(e, c) && n > 1 && e[c] ? L(e[c], r.values) : f(e, t) && e[t] ? L(e[t], r.values) : null;
}, a = async (e, t, r = u) => {
  const n = d(e, O) || p, o = t(n);
  return {
    ...C,
    // Default en-US translations (in case any other translation file is missing any key)
    ...await o ?? u,
    // Merge with our locale file of the locale they are loading
    ...m(r == null ? void 0 : r[e])
    // Merge with their custom locales if available
  };
};
export {
  N as formatCustomTranslations,
  _ as formatLocale,
  R as getTranslation,
  a as loadTranslations,
  T as matchLocale,
  d as parseLocale,
  l as toTwoLetterCode
};
