// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/CustomerDetailsForm/Fields/CountryRegionField.js
import { jsx as w } from "../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import x from "../../../../../../../core/Context/useCoreContext.js";
import { useMemo as F } from "../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { FormSelect as g } from "../../../../../../internal/FormWrappers/FormSelect.js";
import { useWizardFormContext as h } from "../../../../../../../hooks/form/wizard/WizardFormContext.js";
const I = ({ countriesData: e, isFetchingCountries: C, countryDatasetData: l, isFetchingCountryDataset: f }) => {
  const { i18n: s } = x(), { fieldsConfig: m } = h(), b = F(() => {
    var p, n, i;
    const d = new Set(((p = m == null ? void 0 : m.countryCode) == null ? void 0 : p.options) ?? ((n = e == null ? void 0 : e.data) == null ? void 0 : n.map(({ countryCode: o }) => o)) ?? []);
    return (l != null && l.length ? l : ((i = e == null ? void 0 : e.data) == null ? void 0 : i.map(({ countryCode: o, countryName: r }) => ({ id: o, name: r }))) ?? []).filter(({ id: o }) => !d.size || d.has(o)).sort((o, r) => o.name.localeCompare(r.name));
  }, [e == null ? void 0 : e.data, l, m]);
  return /* @__PURE__ */ w(
    g,
    {
      clearable: !0,
      filterable: !0,
      fieldName: "countryCode",
      label: s.get("payByLink.creation.fields.country.label"),
      items: b,
      readonly: C || f
    }
  );
};
export {
  I as CountryRegionField
};
