// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/CustomerDetailsForm/Fields/LanguageField.js
import { jsx as i } from "../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import c from "../../../../../../../core/Context/useCoreContext.js";
import { useMemo as l, useCallback as p } from "../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { useFetch as d } from "../../../../../../../hooks/useFetch.js";
import { FormSelect as g } from "../../../../../../internal/FormWrappers/FormSelect.js";
import { useWizardFormContext as f } from "../../../../../../../hooks/form/wizard/WizardFormContext.js";
const j = () => {
  const { i18n: m, getCdnDataset: n } = c(), { fieldsConfig: t } = f(), e = l(() => {
    var s;
    return (s = t == null ? void 0 : t.shopperLocale) == null ? void 0 : s.options;
  }, [t]), o = d({
    fetchOptions: { enabled: !0 },
    queryFn: p(async () => n ? await n({
      name: "languages",
      extension: "json",
      fallback: []
    }) ?? [] : [], [n])
  }), u = l(() => (o.data ?? []).filter(({ value: a }) => e != null && e.length ? e == null ? void 0 : e.includes(a) : !0).map(({ text: a, value: r }) => ({
    // TODO - Handle 'auto detect' option when submitting information to the BE
    id: r === null ? "auto" : r,
    name: a
  })).sort(({ name: a }, { name: r }) => a.localeCompare(r)), [o.data, e]);
  return /* @__PURE__ */ i(
    g,
    {
      filterable: !0,
      fieldName: "shopperLocale",
      label: m.get("payByLink.creation.fields.language.label"),
      items: u,
      readonly: o.isFetching
    }
  );
};
export {
  j as LanguageField
};
