// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/CustomerDetailsForm/Fields/ShopperEmailField.js
import { jsx as n } from "../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import o from "../../../../../../../core/Context/useCoreContext.js";
import { useMemo as m, useCallback as p } from "../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { FormTextInput as d } from "../../../../../../internal/FormWrappers/FormTextInput.js";
import { PAYMENT_LINK_CREATION_FIELD_LENGTHS as u } from "../../../../constants.js";
const x = () => {
  const { i18n: i } = o(), e = m(
    () => ({
      valid: !1,
      message: i.get("payByLink.creation.fields.shopperEmail.error.validEmail")
    }),
    [i]
  ), l = p(
    (r) => {
      r = String(r).toLowerCase().trim();
      const s = r.split("@");
      if (s.length !== 2)
        return e;
      const [a, t] = s;
      return !a || !t || a.length === 0 || a.startsWith(".") || a.endsWith(".") || a.includes("..") || t.length === 0 || t.startsWith(".") || t.endsWith(".") || t.includes("..") || !t.includes(".") || !// eslint-disable-next-line max-len
      /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/.test(r) ? e : { valid: !0 };
    },
    [e]
  );
  return /* @__PURE__ */ n(
    d,
    {
      fieldName: "shopperEmail",
      label: i.get("payByLink.creation.fields.shopperEmail.label"),
      validate: l,
      maxLength: u.emailAddress.max
    }
  );
};
export {
  x as ShopperEmailField
};
