// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/CustomerDetailsForm/Fields/ShopperPhoneField.js
import { jsx as c } from "../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as l, useEffect as F, useCallback as D } from "../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { useWizardFormContext as P } from "../../../../../../../hooks/form/wizard/WizardFormContext.js";
import { useFetch as q } from "../../../../../../../hooks/useFetch.js";
import { VisibleField as I } from "../../../../../../internal/FormWrappers/VisibleField.js";
import k from "../../../../../../internal/FormFields/InputBase.js";
import w from "../../../../../../internal/FormWrappers/FormField.js";
import T from "../../../../../../../core/Context/useCoreContext.js";
import { PAYMENT_LINK_CREATION_FIELD_LENGTHS as j } from "../../../../constants.js";
import { filterDisallowedCharacters as B } from "../../../../../../internal/FormFields/utils.js";
import { Controller as E } from "../../../../../../../hooks/form/Controller.js";
const Y = () => {
  const { i18n: p, getCdnDataset: b } = T(), { control: V, fieldsConfig: y, getValues: g, setValue: m, setFieldDisplayValue: d, getDisplayValue: C, trigger: $, formState: h } = P(), u = l(() => C("telephoneNumber"), [C]), t = l(() => g("telephoneNumber"), [g]), [n, s] = l(() => {
    if (u) {
      const [e, ...r] = u.split(" ");
      return [e, r.join(" ")];
    }
    if (t) {
      const [e, ...r] = t.split(" ");
      return [e, r.join(" ")];
    }
    return [void 0, void 0];
  }, [u, t]);
  F(() => {
    if (!u && t) {
      const [e, ...r] = t.split(" "), o = r.join(" ");
      m("telephoneNumber", `${e}${o}`), d("telephoneNumber", `${e} ${o}`);
    }
  }, [u, t, m, d]);
  const f = q({
    fetchOptions: { enabled: !0 },
    queryFn: D(async () => b ? await b({
      name: "phonenumbers",
      extension: "json",
      fallback: []
    }) ?? [] : [], [b])
  }), x = l(() => (f.data ?? []).map(({ id: r, prefix: o }) => ({ id: o, name: `${r} (${o})` })).sort(({ name: r }, { name: o }) => r.localeCompare(o)), [f.data]);
  F(() => {
    n && h.touchedFields.telephoneNumber && h.errors.telephoneNumber && $("telephoneNumber");
  }, [n, s, $, h.errors, h.touchedFields]);
  const N = l(() => {
    var e;
    return (e = y.telephoneNumber) == null ? void 0 : e.required;
  }, [y]), L = D(() => !N && !n && !s ? { valid: !0 } : n ? s ? { valid: !0 } : { valid: !1, message: p.get("payByLink.creation.fields.phoneNumber.errors.requiredPhoneNumber") } : { valid: !1, message: p.get("payByLink.creation.fields.phoneNumber.errors.requiredPhoneCode") }, [n, s]);
  return /* @__PURE__ */ c(I, { name: "telephoneNumber", children: /* @__PURE__ */ c(w, { label: p.get("payByLink.creation.fields.shopperPhone.label"), optional: !N, children: /* @__PURE__ */ c(
    E,
    {
      name: "telephoneNumber",
      control: V,
      rules: {
        required: N,
        validate: L
      },
      render: ({ field: e, fieldState: r }) => {
        var v;
        const o = !!r.error && r.isTouched;
        return /* @__PURE__ */ c(
          k,
          {
            ...e,
            onKeyDown: (i) => {
              B({
                event: i,
                inputType: "number"
              });
            },
            onInput: (i) => {
              const a = i.target.value;
              m("telephoneNumber", `${n ?? ""}${a}`), d("telephoneNumber", `${n ?? ""} ${a}`);
            },
            value: s,
            type: "text",
            dropdown: {
              filterable: !0,
              items: x,
              value: n,
              placeholder: p.get("payByLink.creation.fields.shopperPhone.phonePrefix.placeholder"),
              readonly: f.isFetching
            },
            onDropdownInput: (i) => {
              const a = s || "";
              m("telephoneNumber", `${i}${a}`), d("telephoneNumber", `${i} ${a}`);
            },
            isValid: !r.error && !!e.value,
            isInvalid: o,
            errorMessage: (v = r.error) == null ? void 0 : v.message,
            maxLength: j.telephoneNumber.max
          }
        );
      }
    }
  ) }) });
};
export {
  Y as ShopperPhoneField
};
