// @adyen/adyen-platform-experience-web/dist/es/components/internal/Pagination/Pagination.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import T from "../../../core/Context/useCoreContext.js";
import { useMemo as C, useCallback as N } from "../../../external/preact/hooks/dist/hooks.module.js";
import { ButtonVariant as m } from "../Button/types.js";
/* empty css                */
import { Translation as x } from "../Translation/Translation.js";
import A from "../FormFields/Select/Select.js";
import d from "../Button/Button.js";
import { Icon as p } from "../Icon/Icon.js";
import { isNullish as R } from "../../../utils/value/is.js";
import { EMPTY_ARRAY as f } from "../../../utils/value/constants.js";
const i = "adyen-pe-pagination", t = {
  base: i,
  context: i + "__context",
  controls: i + "__controls",
  limit: i + "__limit",
  limitSelector: i + "__limit-selector"
};
function z({
  next: u,
  hasNext: n,
  hasPrev: c,
  prev: b,
  limit: g,
  limitOptions: r,
  onLimitSelection: l,
  ariaLabelKey: h,
  limitSelectAriaLabelKey: _
}) {
  const { i18n: o } = T(), s = C(
    () => r && Object.freeze(
      r.map(
        (a) => ({
          name: a.toLocaleString(o.locale, { style: "decimal" }),
          id: String(a)
        })
      )
    ),
    [o, r]
  ), v = N(
    ({ target: a }) => {
      R(a == null ? void 0 : a.value) || l == null || l(+a.value);
    },
    [l]
  );
  return /* @__PURE__ */ e("div", { role: "group", "aria-label": o.get(h ?? "common.pagination.label"), className: t.base, children: [
    /* @__PURE__ */ e("div", { className: t.context, children: s && l && /* @__PURE__ */ e("div", { className: t.limit, role: "presentation", children: /* @__PURE__ */ e(
      x,
      {
        translationKey: "common.pagination.controls.limitSelect",
        fills: {
          pageLimit: /* @__PURE__ */ e("div", { className: t.limitSelector, children: /* @__PURE__ */ e(
            A,
            {
              setToTargetWidth: !0,
              filterable: !1,
              multiSelect: !1,
              items: s,
              onChange: v,
              selected: `${g ?? ""}`,
              "aria-label": o.get(_ ?? "common.pagination.controls.limitSelect.label")
            }
          ) })
        }
      }
    ) }) }),
    /* @__PURE__ */ e("div", { className: t.controls, children: [
      /* @__PURE__ */ e(
        d,
        {
          variant: m.TERTIARY,
          disabled: !c,
          iconButton: !0,
          "aria-label": o.get("common.pagination.controls.previousPage.label"),
          classNameModifiers: ["circle"].concat(c ? f : "disabled"),
          onClick: b,
          children: /* @__PURE__ */ e(p, { name: "chevron-left" })
        }
      ),
      /* @__PURE__ */ e(
        d,
        {
          variant: m.TERTIARY,
          disabled: !n,
          iconButton: !0,
          "aria-label": o.get("common.pagination.controls.nextPage.label"),
          classNameModifiers: ["circle"].concat(n ? f : "disabled"),
          onClick: u,
          children: /* @__PURE__ */ e(p, { name: "chevron-right" })
        }
      )
    ] })
  ] });
}
export {
  z as default
};
