// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/TransactionTotals/TotalsCard.js
import { jsx as r } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { memo as h } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useState as w, useMemo as i } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import y from "../../../../../core/Context/useCoreContext.js";
import $ from "../../../../internal/ExpandableCard/ExpandableCard.js";
import { TransactionTotalItem as v } from "../TransactionTotalItem/TransactionTotalItem.js";
import { useMaxWidthsState as C } from "../../hooks/useMaxWidths.js";
import { Fragment as M } from "../../../../../external/preact/dist/preact.module.js";
import { uniqueId as I } from "../../../../../utils/random/uniqueId.js";
const z = h(({ totals: p, isLoading: o, hiddenField: t, fullWidth: x, ...E }) => {
  const { i18n: n } = y(), [m, a] = C(), [l, c] = w(!1), d = i(() => n.get("transactions.overview.totals.currency.label"), [n]), [e, ...u] = i(() => p.map((s) => ({
    ...s,
    key: s.currency,
    expensesElemId: I("elem"),
    incomingsElemId: I("elem")
  })), [p]), f = i(() => {
    switch (t) {
      case "expenses":
        return n.get("transactions.overview.totals.lists.incoming");
      case "incomings":
        return n.get("transactions.overview.totals.lists.outgoing");
      default:
        return n.get("transactions.overview.totals.lists.default");
    }
  }, [n, t]), g = i(
    () => /* @__PURE__ */ r(
      "div",
      {
        role: "listitem",
        "aria-label": e ? `${d}: ${e.currency}` : void 0,
        "aria-describedby": e ? `${e.incomingsElemId} ${e.expensesElemId}` : void 0,
        children: /* @__PURE__ */ r(
          v,
          {
            showLabelUnderline: l,
            total: e,
            hiddenField: t,
            widths: m,
            isHeader: !0,
            isSkeleton: o,
            isLoading: o,
            onWidthsSet: a,
            expensesElemId: e == null ? void 0 : e.expensesElemId,
            incomingsElemId: e == null ? void 0 : e.incomingsElemId
          }
        )
      }
    ),
    [l, e, t, m, o, a, d]
  ), b = i(() => !o && u.length ? /* @__PURE__ */ r(M, { children: u.map((s) => /* @__PURE__ */ r(
    "div",
    {
      role: "listitem",
      "aria-label": `${d}: ${s.currency}`,
      "aria-describedby": `${s.incomingsElemId} ${s.expensesElemId}`,
      children: /* @__PURE__ */ r(
        v,
        {
          showLabelUnderline: l,
          total: s,
          hiddenField: t,
          widths: m,
          onWidthsSet: a,
          expensesElemId: s.expensesElemId,
          incomingsElemId: s.incomingsElemId
        }
      )
    },
    s.key
  )) }) : void 0, [o, u, l, t, m, a, d]);
  return /* @__PURE__ */ r(
    $,
    {
      renderContent: ({ collapsibleContent: s }) => /* @__PURE__ */ r("div", { role: "list", "aria-label": f, children: [
        g,
        s
      ] }),
      fullWidth: x,
      onMouseEnter: () => c(!0),
      onFocus: () => c(!0),
      onMouseLeave: () => c(!1),
      onBlur: () => c(!1),
      ...E,
      children: b
    }
  );
});
export {
  z as TotalsCard
};
