// @adyen/adyen-platform-experience-web/dist/es/components/internal/Checkbox/Checkbox.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import o from "classnames";
import { useState as q, useRef as T, useCallback as A, useEffect as C } from "../../../external/preact/hooks/dist/hooks.module.js";
/* empty css              */
import m from "../Typography/Typography.js";
import { TypographyVariant as p, TypographyElement as d } from "../Typography/types.js";
import { Icon as h } from "../Icon/Icon.js";
import { uniqueId as S } from "../../../utils/random/uniqueId.js";
const j = ({ checked: a, error: y, description: s, disabled: u, id: b, label: f, name: _, value: v, onInput: c, className: k, ...x }) => {
  const [l, n] = q(a), N = T(S()), i = b || N.current, g = A(
    (r) => {
      var t;
      c == null || c(r), a === void 0 && n(((t = r == null ? void 0 : r.currentTarget) == null ? void 0 : t.checked) ?? !1);
    },
    [a, c]
  );
  return C(() => {
    n(a);
  }, [a]), /* @__PURE__ */ e("div", { className: o("adyen-pe-checkbox", k), children: [
    /* @__PURE__ */ e(
      "input",
      {
        name: _,
        type: "checkbox",
        checked: l,
        disabled: u,
        className: o("adyen-pe-visually-hidden adyen-pe-checkbox__input"),
        id: i,
        onInput: g,
        ...x
      }
    ),
    /* @__PURE__ */ e("label", { className: "adyen-pe-checkbox__label", htmlFor: i, children: [
      l ? /* @__PURE__ */ e(h, { name: "checkmark-square-fill", className: "adyen-pe-checkbox__icon" }) : /* @__PURE__ */ e(h, { name: "square", className: o("adyen-pe-checkbox__icon", { "adyen-pe-checkbox__icon--error": y }) }),
      /* @__PURE__ */ e("div", { className: "adyen-pe-checkbox__label-content", children: [
        /* @__PURE__ */ e(m, { el: d.SPAN, variant: p.BODY, children: f }),
        s && /* @__PURE__ */ e(m, { el: d.SPAN, variant: p.BODY, className: "adyen-pe-checkbox__description", children: s })
      ] })
    ] })
  ] });
};
export {
  j as Checkbox
};
