// @adyen/adyen-platform-experience-web/dist/es/components/internal/Stepper/Stepper.js
import { jsx as y } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useState as O, useRef as d, useMemo as F, useEffect as K, useCallback as s } from "../../../external/preact/hooks/dist/hooks.module.js";
import M from "classnames";
import { InteractionKeyCode as c } from "../../types.js";
import { Step as P } from "./Step.js";
/* empty css             */
const v = ({ activeIndex: r, onChange: i, variant: h = "vertical", children: n, nextStepDisabled: k, ariaLabel: N }) => {
  const [l, T] = O(r), W = d(null), f = d([]), [p, m] = O(r), S = F(() => (Array.isArray(n) ? n : [n]).filter(Boolean), [n]), A = S.length, a = h === "horizontal";
  K(() => {
    r > l && T(r);
    const e = f.current[r];
    e && e.scrollIntoView({
      block: "nearest",
      inline: "nearest",
      behavior: "smooth"
    });
  }, [r, l]);
  const o = s(
    (e) => k && e > l ? !1 : e <= l + 1,
    [l, k]
  ), w = s(
    (e) => {
      o(e) && (m(e), i(e));
    },
    [o, i]
  ), u = s((e) => {
    m(e);
    const t = f.current[e];
    t && t.focus();
  }, []), R = s(() => {
    const e = Math.min(p + 1, A - 1);
    o(e) && u(e);
  }, [o, u, p, A]), b = s(() => {
    const e = Math.max(p - 1, 0);
    u(e);
  }, [u, p]), E = s(() => {
    const e = document.activeElement, t = f.current.indexOf(e);
    t !== -1 && o(t) && i(t);
  }, [o, i]), z = s(
    (e) => {
      switch (e.code) {
        case c.ARROW_RIGHT:
          a && (e.preventDefault(), R());
          break;
        case c.ARROW_LEFT:
          a && (e.preventDefault(), b());
          break;
        case c.ARROW_DOWN:
          a || (e.preventDefault(), R());
          break;
        case c.ARROW_UP:
          a || (e.preventDefault(), b());
          break;
        case c.TAB:
          m(r);
          break;
        case c.ENTER:
        case c.SPACE:
          e.preventDefault(), E();
          break;
      }
    },
    [a, r, E, R, b]
  );
  return /* @__PURE__ */ y("div", { className: "adyen-pe-stepper", children: /* @__PURE__ */ y(
    "ol",
    {
      "aria-label": N,
      ref: W,
      role: "toolbar",
      "aria-orientation": h,
      className: M("adyen-pe-stepper__list", {
        "adyen-pe-stepper__list--horizontal": a
      }),
      onKeyDown: z,
      children: S.map((e, t) => {
        var _;
        return /* @__PURE__ */ y(
          P,
          {
            totalSteps: A,
            index: t,
            active: t === r,
            completed: t < r,
            disabled: !o(t),
            onClick: () => w(t),
            ref: (D) => {
              D && (f.current[t] = D);
            },
            children: ((_ = e.props) == null ? void 0 : _.children) || e
          },
          t
        );
      })
    }
  ) });
};
export {
  v as Stepper
};
