// @adyen/adyen-platform-experience-web/dist/es/components/internal/Stepper/Step.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { forwardRef as i } from "../../../external/preact/compat/dist/compat.module.js";
import m from "classnames";
import d from "../Typography/Typography.js";
import { TypographyElement as l, TypographyVariant as c } from "../Typography/types.js";
import { Icon as y } from "../Icon/Icon.js";
const f = ({ completed: r, index: a }) => r ? /* @__PURE__ */ e(y, { name: "checkmark-circle-fill" }) : /* @__PURE__ */ e("span", { className: "adyen-pe-step__number", children: a + 1 }), g = i(({ index: r, active: a, completed: p, disabled: t, onClick: n, children: o, totalSteps: u }, s) => /* @__PURE__ */ e(
  "li",
  {
    className: m("adyen-pe-step__item", {
      "adyen-pe-step--active": a,
      "adyen-pe-step--completed": p,
      "adyen-pe-step--disabled": t
    }),
    children: /* @__PURE__ */ e(
      "button",
      {
        "aria-disabled": t,
        "aria-current": a ? "step" : void 0,
        tabIndex: a ? 0 : -1,
        ref: s,
        type: "button",
        className: "adyen-pe-step__button",
        onClick: n,
        disabled: t,
        children: [
          /* @__PURE__ */ e("div", { className: "adyen-pe-step__icon", "aria-hidden": "true", children: /* @__PURE__ */ e(f, { completed: p, index: r }) }),
          /* @__PURE__ */ e(d, { variant: c.BODY, el: l.SPAN, className: "adyen-pe-step__label", children: o })
        ]
      }
    )
  }
));
export {
  g as Step
};
