// @adyen/adyen-platform-experience-web/dist/es/core/core.js
var h = Object.defineProperty;
var c = (n, t, i) => t in n ? h(n, t, { enumerable: !0, configurable: !0, writable: !0, value: i }) : n[t] = i;
var s = (n, t, i) => c(n, typeof t != "symbol" ? t + "" : t, i);
import { FALLBACK_ENV as d, resolveEnvironment as m, getConfigFromCdn as g, getDatasetFromCdn as u } from "./utils.js";
import { AuthSession as C } from "./ConfigContext/session/AuthSession.js";
import { Assets as l } from "./Assets/Assets.js";
import { getCustomTranslationsAnalyticsPayload as p } from "./Analytics/analytics/customTranslations.js";
import y from "./Localization/Localization.js";
import { EMPTY_OBJECT as f } from "../utils/value/constants.js";
class A {
  // [TODO]: Change the error handling strategy.
  constructor(t) {
    s(this, "components", []);
    s(this, "options");
    s(this, "localization");
    s(this, "loadingContext");
    s(this, "analyticsEnabled");
    s(this, "session", new C());
    s(this, "onError");
    s(this, "getImageAsset");
    s(this, "getDatasetAsset");
    s(this, "getCdnConfig");
    s(this, "getCdnDataset");
    s(this, "readyCustomTranslationsAnalytics");
    /**
     * Updates global configurations, resets the internal state and remounts each element.
     * @param options - props to update
     * @returns this - the element instance
     */
    s(this, "update", async (t = f) => (this.setOptions(t), await this.initialize(), this.components.forEach((i) => {
      i.props.core === this && i.update(this.getPropsForComponent(this.options));
    }), this));
    /**
     * Remove the reference of a component
     * @param component - reference to the component to be removed
     * @returns this - the element instance
     */
    s(this, "remove", (t) => (this.components = this.components.filter((i) => i._id !== t._id), t.unmount(), this));
    /**
     * @internal
     * Register components in core to be able to update them all at once
     */
    s(this, "registerComponent", (t) => {
      t.props.core === this && this.components.push(t);
    });
    /**
     * @internal
     * Enhances the config object passed when AdyenPlatformExperience is initialised (environment, clientKey, etc...)
     * (Re)Initializes core properties & processes (i18n, etc...)
     * @param options - the config object passed when AdyenPlatformExperience is initialised
     * @returns this
     */
    s(this, "setOptions", (t) => {
      var i, a;
      return this.options = { ...this.options, ...t }, this.localization.locale = (i = this.options) == null ? void 0 : i.locale, this.localization.customTranslations = (a = this.options) == null ? void 0 : a.translations, this.session.loadingContext = this.loadingContext, this.session.onSessionCreate = this.options.onSessionCreate, this;
    });
    var e;
    this.options = { environment: d, ...t };
    const { cdnTranslationsUrl: i, cdnAssetsUrl: a, cdnConfigUrl: o, apiUrl: r } = m(this.options.environment);
    this.localization = new y(t.locale, t.availableTranslations, i, o), this.loadingContext = t.loadingContext || null || r, this.getImageAsset = new l(a).getAsset({ extension: "svg", subFolder: "images" }), this.getDatasetAsset = new l(a).getAsset({ extension: "json", mainFolder: "datasets" }), this.getCdnConfig = g({ url: o }), this.getCdnDataset = u({ url: `${a}/datasets` }), this.readyCustomTranslationsAnalytics = !1, this.analyticsEnabled = ((e = t == null ? void 0 : t.analytics) == null ? void 0 : e.enabled) ?? !0, this.session.analyticsEnabled = this.analyticsEnabled, this.setOptions(t);
  }
  async initialize() {
    return Promise.all([this.localization.ready]).then(() => {
      if (!this.readyCustomTranslationsAnalytics && this.analyticsEnabled) {
        const t = this.setTranslationsPayload();
        t.length > 0 && (this.session.analyticsPayload = t, this.readyCustomTranslationsAnalytics = !0);
      }
      return this;
    });
  }
  setTranslationsPayload() {
    return this.localization ? p(this.localization.i18n.customTranslations) : [];
  }
  /**
   * @internal
   * @param options - options that will be merged to the global Checkout props
   * @returns props for a new UIElement
   */
  getPropsForComponent(t) {
    return { ...t };
  }
}
s(A, "version", "1.10.0");
export {
  A as default
};
