// @adyen/adyen-platform-experience-web/dist/es/core/utils.js
import { CDN_ENVIRONMENTS as l, API_ENVIRONMENTS as E } from "./constants.js";
import { httpGet as p } from "./Http/http.js";
const $ = "test", i = "live", N = (r) => {
  var s;
  return (s = r == null ? void 0 : r.replace) == null ? void 0 : s.call(r, /([^/])$/, "$1/");
}, h = (r) => r == null ? void 0 : r.replace(/^([^/])/, "/$1"), u = /* @__PURE__ */ (() => {
  const r = E, s = l;
  return (n) => {
    const t = s[(n === "test" ? "live" : n) ?? i] || s[i];
    return {
      apiUrl: r[n ?? $] || r[$],
      cdnTranslationsUrl: `${t}/assets/translations`,
      cdnAssetsUrl: `${t}/assets`,
      cdnConfigUrl: `${t}/config`
    };
  };
})(), y = ({ url: r }) => async ({
  name: s,
  extension: n = "json",
  fallback: e,
  subFolder: t = ""
}) => {
  try {
    return await p({
      loadingContext: `${r}${t ? `/${t}` : ""}`,
      path: `/${s}.${n}`,
      versionless: !0,
      skipContentType: !0,
      errorLevel: "error"
    });
  } catch (o) {
    return console.warn(o), e;
  }
}, L = ({ url: r }) => async ({
  name: s,
  extension: n = "json",
  fallback: e,
  subFolder: t = ""
}) => {
  try {
    return await p({
      loadingContext: `${r}${t ? `/${t}` : ""}`,
      path: `/${s}.${n}`,
      versionless: !0,
      skipContentType: !0,
      errorLevel: "error"
    });
  } catch (o) {
    console.warn(o);
    try {
      const a = await import(`../assets/datasets${t ? `/${t}` : ""}/${s}.${n}`);
      return a.default || a;
    } catch (c) {
      return console.warn(c), e;
    }
  }
};
export {
  i as FALLBACK_CDN_ENV,
  $ as FALLBACK_ENV,
  y as getConfigFromCdn,
  L as getDatasetFromCdn,
  N as normalizeLoadingContext,
  h as normalizeUrl,
  u as resolveEnvironment
};
