// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkSettings/components/PaymentLinkThemeContainer/components/LogoPreview/LogoPreview.js
import { jsx as e } from "../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useUniqueId as l } from "../../../../../../../hooks/useUniqueId.js";
import p from "../../../../../../../core/Context/useCoreContext.js";
import { useCallback as s } from "../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { TypographyVariant as c, TypographyElement as d } from "../../../../../../internal/Typography/types.js";
import f from "../../../../../../internal/Typography/Typography.js";
import { LogoLabel as h } from "../ThemeForm/constants.js";
import { ButtonVariant as y } from "../../../../../../internal/Button/types.js";
import g from "../../../../../../internal/Button/Button.js";
const B = ({
  disabled: i,
  logoType: r,
  logoURL: m,
  onRemoveLogo: o
}) => {
  const { i18n: a } = p(), t = l(), n = s(() => {
    o(r);
  }, [r, o]);
  return /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-theme-form__preview--conteiner", children: [
    /* @__PURE__ */ e("label", { htmlFor: t, "aria-labelledby": t, children: /* @__PURE__ */ e(f, { el: d.SPAN, variant: c.BODY, stronger: !0, children: a.get(h[r]) }) }),
    /* @__PURE__ */ e("img", { id: t, src: m, alt: "full-width-logo", className: "adyen-pe-payment-link-theme-form__preview--image" }),
    /* @__PURE__ */ e(
      g,
      {
        disabled: i,
        variant: y.SECONDARY,
        onClick: n,
        className: "adyen-pe-payment-link-theme-form__preview--remove",
        children: a.get("payByLink.settings.theme.action.logo.remove")
      }
    )
  ] });
};
export {
  B as default
};
