// @adyen/adyen-platform-experience-web/dist/es/components/internal/DataOverviewDisplay/DataDetailsModal.js
import { jsx as r } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useEffect as h } from "../../../external/preact/hooks/dist/hooks.module.js";
import { popoverUtil as M } from "../Popover/utils/popoverUtil.js";
import g from "../../../core/Context/useCoreContext.js";
import s from "../Modal/ModalContent/ModalContent.js";
import v from "../Modal/Modal.js";
const y = ({
  children: i,
  className: p,
  ariaLabelKey: f,
  selectedDetail: o,
  resetDetails: u,
  dataCustomization: C
}) => {
  const { i18n: m } = g(), n = !!o;
  return h(() => {
    n && M.closeAll();
  }, [n]), /* @__PURE__ */ r("div", { className: p, children: [
    i,
    o && /* @__PURE__ */ r(
      v,
      {
        title: o != null && o.title ? m.get(o.title) : void 0,
        isOpen: !!o,
        "aria-label": m.get(f),
        onClose: u,
        isDismissible: !0,
        headerWithBorder: !1,
        size: (o == null ? void 0 : o.modalSize) ?? "large",
        children: o && /* @__PURE__ */ r(s, { dataCustomization: { details: C }, ...o == null ? void 0 : o.selection })
      }
    )
  ] });
};
export {
  y as DataDetailsModal
};
