// @adyen/adyen-platform-experience-web/dist/es/components/internal/StructuredList/StructuredList.js
import { jsx as a } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import L from "classnames";
import { useMemo as E } from "../../../external/preact/hooks/dist/hooks.module.js";
import { TypographyVariant as s } from "../Typography/types.js";
import e from "../Typography/Typography.js";
import { SL_GRID_CLASS as d, SL_LABEL_CLASS as h, SL_CONTENT_CLASS as I, SL_ITEM_CLASS as N, SL_ITEM_WITH_HIGHLIGHT_CLASS as y, SL_BASE_CLASS as D, SL_EXPANDED_CLASS as O, SL_ALIGN_END as v } from "./constants.js";
/* empty css                    */
import { useStructuredListItems as b } from "./useStructuredListItem.js";
const g = "6-6";
function k({
  ["aria-label"]: n,
  items: p,
  highlightable: c,
  renderValue: S,
  renderLabel: t,
  layout: o = g,
  grid: A = !0,
  classNames: i,
  align: u = "end",
  condensed: _ = !0
}) {
  const [C, f] = E(() => o.split("-").map((r) => `${d}--width-${r}-of-12`), [o]), m = b(p), l = _ ? s.CAPTION : s.BODY;
  return /* @__PURE__ */ a("dl", { className: L(D, i, { [v]: u === "end", [O]: !_ }), "aria-label": n, children: m.map((r, T) => /* @__PURE__ */ a(
    "div",
    {
      "data-testid": r.label,
      className: L(N, {
        [y]: c,
        [d]: A
      }),
      children: [
        /* @__PURE__ */ a("dt", { className: L(h, C), children: t ? t(r.label, r.key, r.rawValue) : /* @__PURE__ */ a(e, { variant: l, children: r.label }) }),
        /* @__PURE__ */ a("dd", { className: L(I, f), children: r.render ? r.render(r) : S ? S(r.value, r.key, r.type, r.config) : /* @__PURE__ */ a(e, { variant: l, children: r.value }) })
      ]
    },
    `${T}_${r.id || "0"}`
  )) });
}
export {
  k as default
};
