// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/hooks/useCurrencySelection.js
import F from "../../../../hooks/useAnalytics/useFilterAnalyticsEvent.js";
import { useMemo as O, useCallback as R, useState as M, useRef as x, useEffect as j } from "../../../../external/preact/hooks/dist/hooks.module.js";
const w = ({
  availableCurrencies: t,
  eventCategory: k,
  eventSubCategory: p,
  eventLabel: q,
  onUpdateSelection: m,
  selectedCurrency: f
}) => {
  const { logEvent: n } = F({ category: k, subCategory: p, label: q }), z = O(
    () => Object.freeze((t == null ? void 0 : t.map((c) => ({ id: c, name: c }))) ?? []),
    [t]
  ), d = R(
    (c) => c ? (t == null ? void 0 : t.findIndex((s) => s === c)) ?? -1 : -1,
    [t]
  ), E = R(
    ({ target: c }) => {
      const s = d(c == null ? void 0 : c.value);
      s >= 0 && I(s);
    },
    [d]
  ), [A, I] = M(() => d(f)), o = O(() => t == null ? void 0 : t[A], [t, A]), y = x(), h = x(t), u = x(f);
  return j(() => {
    (h.current !== t || u.current !== f) && (h.current = t, u.current = f, y.current = void 0, I(d(f)));
  }, [d, f]), j(() => {
    const c = y.current;
    c !== o && (y.current = o, c && o && (n == null || n("update", o)), m == null || m(o));
  }, [o, n, m]), { activeCurrency: o, currencySelectionOptions: z, onCurrencySelection: E };
};
export {
  w as default
};
