// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputeManagement/context/dispute/context.js
import { jsx as T } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { createContext as z } from "../../../../../external/preact/dist/preact.module.js";
import { memo as G } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useState as f, useCallback as a, useEffect as H, useContext as J } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import L from "../../../../../core/Context/useCoreContext.js";
import b from "../../../../../config/disputes/defenseDocumentConfig.json.js";
import p from "../../../../../config/disputes/defenseReasonConfig.json.js";
const w = (i) => {
  const l = new FormData();
  for (const [u, o] of i.entries())
    o instanceof File ? l.set(u, o, o.name) : l.set(u, o);
  return l;
}, g = z(void 0), Z = G(({ dispute: i, setDispute: l, children: u }) => {
  const [o, d] = f("details"), [m, C] = f(null), [k, R] = f([]), [y, r] = f(null), [P, F] = f(null), x = a(() => {
    r((e) => {
      if (e) {
        const t = [...e.keys()].filter((n) => n !== "defenseReason");
        if (t.length > 0) {
          const n = w(e);
          return t.forEach((s) => n.delete(s)), n;
        }
      }
      return e;
    });
  }, []), S = a(() => {
    switch (o) {
      case "defendReasonSelectionView":
        d("details");
        break;
      case "accept":
        d("details");
        break;
      case "uploadDefenseFilesView":
        x(), d("defendReasonSelectionView");
        break;
      default:
        d("details");
        break;
    }
  }, [x, o]), h = a(() => {
    C(null), R(null), r(null), F(null), l(void 0);
  }, [l]), V = a((e, t) => {
    r((n) => {
      const s = n ? w(n) : new FormData();
      return s.set(e, t, t.name), s;
    });
  }, []), E = a((e, t) => {
    r((n) => {
      if (n && n.has(e)) {
        const s = n.get(e), D = w(n);
        return D.delete(e), s instanceof File ? D.set(t, s, s.name) : D.set(t, s), D;
      }
      return n;
    });
  }, []), j = a((e) => {
    r((t) => {
      if (t && t.has(e)) {
        const n = w(t);
        return n.delete(e), n;
      }
      return t;
    });
  }, []), q = a((e) => {
    F(e);
  }, []);
  H(() => {
    r(() => {
      if (m) {
        const e = new FormData();
        return e.set("defenseReason", m), e;
      }
      return null;
    });
  }, [m]);
  const { getCdnConfig: c } = L(), [A, B] = f(p), [I, K] = f(b), M = a(async () => {
    const e = await (c == null ? void 0 : c({
      subFolder: "disputes",
      name: "defenseReasonConfig",
      fallback: p
    })), t = await (c == null ? void 0 : c({
      subFolder: "disputes",
      name: "defenseDocumentConfig",
      fallback: b
    }));
    B(e ?? p), K(t ?? b);
  }, [c]);
  return /* @__PURE__ */ T(
    g.Provider,
    {
      value: {
        addFileToDefendPayload: V,
        applicableDocuments: k,
        clearFiles: x,
        clearStates: h,
        defendResponse: P,
        dispute: i,
        flowState: o,
        goBack: S,
        setApplicableDocuments: R,
        setFlowState: d,
        setDispute: l,
        selectedDefenseReason: m,
        setSelectedDefenseReason: C,
        defendDisputePayload: y,
        onDefendSubmit: q,
        moveFieldInDefendPayload: E,
        removeFieldFromDefendPayload: j,
        getDisputesConfig: M,
        defenseReasonConfig: A,
        defenseDocumentConfig: I
      },
      children: u
    }
  );
}), _ = () => {
  const i = J(g);
  if (!i) throw new Error("useDisputeFlow must be used within DisputeFlowProvider");
  return i;
};
export {
  Z as DisputeContextProvider,
  g as DisputeFlowContext,
  _ as useDisputeFlow
};
