// @adyen/adyen-platform-experience-web/dist/es/core/Localization/datetime/restamper/utils.js
import { SYSTEM_TIMEZONE_FORMATTER as c, REGEX_TZ_OFFSET as r } from "./constants.js";
import { EMPTY_ARRAY as i } from "../../../../utils/value/constants.js";
import { mod as p } from "../../../../utils/value/number.js";
const E = /\+(?=-)|([+-]00:00)/g, O = /(^\D?)(\d)$/, S = ([t, e]) => (Math.abs(t * 60) + e) * (t < 0 ? -1 : 1), a = (t) => Object.freeze([
  Math.floor(t / 60),
  // offset hours
  p(t, 60)
  // offset minutes
]), _ = (t) => parseInt(t, 10) || 0, R = (t) => `GMT+${a(t).map((o) => `${o}`.replace(O, "$10$2")).join(":")}`.replace(E, ""), T = (t) => {
  var s;
  const e = ((s = t == null ? void 0 : t.match(r)) == null ? void 0 : s[0].replace("GMT", "").split(":", 2).map(_)) ?? i;
  return S(e.concat(0, 0).slice(0, 2));
}, g = (t, e = c) => {
  var n;
  const s = new Date(t), o = T((n = c) == null ? void 0 : n.format(s));
  return T(e == null ? void 0 : e.format(s)) - o;
}, m = (t, e, s = 1) => {
  const { offset: o, timestamp: f } = t(e);
  return f - o * s * 6e4;
}, u = (t, e) => m(t, e, 1), l = (t, e) => m(t, e, -1);
export {
  S as computeTimezoneOffsetInMinutes,
  a as computeTimezoneOffsetsFromMinutes,
  R as getGMTSuffixForTimezoneOffset,
  g as getTimezoneOffsetForTimestamp,
  T as getTimezoneOffsetFromFormattedDateString,
  _ as parseTimezoneOffset,
  u as systemToTimezone,
  l as timezoneToSystem
};
