// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkSettings/components/PaymentLinkThemeContainer/components/ThemeForm/ThemeForm.js
import { jsx as a } from "../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import j from "../../../../../../internal/FormFields/InputText.js";
import T from "../../../../../../internal/Typography/Typography.js";
import { TypographyVariant as w, TypographyElement as R } from "../../../../../../internal/Typography/types.js";
import { useUniqueId as z } from "../../../../../../../hooks/useUniqueId.js";
import J from "../../../../../../../core/Context/useCoreContext.js";
import { logoOptions as d, logoOptionsList as K } from "./constants.js";
/* empty css               */
import { useState as c, useEffect as L, useCallback as s } from "../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { usePaymentLinkSettingsContext as Q } from "../../../PaymentLinkSettingsContainer/context/context.js";
import { ThemeFormDataRequest as X } from "../../types.js";
import Z from "../LogoPreview/LogoPreview.js";
import $ from "../LogoInput/LogoInput.js";
import { cloneFormData as N } from "../../../PaymentLinkSettingsContainer/utils/getThemePayload.js";
import { Icon as ee } from "../../../../../../internal/Icon/Icon.js";
const ge = ({ theme: A, initialPayload: v }) => {
  const { setPayload: r, saveActionCalled: _, setSaveActionCalled: b, setIsValid: I, isSaving: p, setIsSaveSuccess: O, setIsSaveError: k } = Q(), { brandName: G, logo: W, fullWidthLogo: q } = A, [i, F] = c(G ?? void 0), [B, u] = c(W ?? null), [U, h] = c(q ?? null), [E, m] = c(), [x, f] = c(!1), y = z(), { i18n: g } = J();
  L(() => {
    I(!!i);
  }, [i, I, E, r]), L(() => {
    m(v);
  }, [v]), L(() => {
    _ && (i || f(!0), b(!1), O(!1), k(!1));
  }, [_, b, i, f, k, O]);
  const S = s(
    (e, n) => {
      m((t) => {
        const o = t ? N(t) : new FormData();
        return o.set(e, n, n.name), o.set(e, n, n.name), r(o), o;
      });
    },
    [r]
  ), H = s(
    (e) => {
      var t;
      const n = (t = e == null ? void 0 : e.currentTarget) == null ? void 0 : t.value;
      f(!1), F(n), m((o) => {
        const l = o ? N(o) : new FormData();
        return l.set(X.BRAND, n), r(l), l;
      });
    },
    [f, F, m, r]
  ), C = s(
    (e) => {
      m((n) => {
        if (n && n.has(e)) {
          const t = N(n);
          return t.delete(e), r(t), t;
        }
        return n;
      });
    },
    [r]
  ), P = s(
    (e, n) => {
      const t = new FileReader();
      t.onload = function(o) {
        var D;
        const l = (D = o == null ? void 0 : o.target) == null ? void 0 : D.result;
        e === d.LOGO && u(l), e === d.FULL_WIDTH_LOGO && h(l);
      }, t.readAsDataURL(n);
    },
    [u, h]
  ), M = s(
    (e, n) => {
      const t = n[0];
      t && (S(e, t), P(e, t));
    },
    [S, P]
  ), V = s(
    (e) => {
      C(e), e === d.LOGO && u(null), e === d.FULL_WIDTH_LOGO && h(null);
    },
    [C, u, h]
  ), Y = s(
    (e) => e === d.LOGO ? B : U,
    [U, B]
  );
  return /* @__PURE__ */ a("div", { className: "adyen-pe-payment-link-theme-form-container", children: /* @__PURE__ */ a("div", { className: "adyen-pe-payment-link-theme-form", children: [
    /* @__PURE__ */ a("div", { className: "adyen-pe-payment-link-settings__input-container", children: [
      /* @__PURE__ */ a("label", { htmlFor: y, "aria-labelledby": y, className: "adyen-pe-payment-link-theme-form__brand-input", children: /* @__PURE__ */ a(T, { el: R.SPAN, variant: w.BODY, stronger: !0, children: g.get("payByLink.settings.theme.brandName.input.label") }) }),
      /* @__PURE__ */ a(
        j,
        {
          type: "text",
          disabled: !!p,
          readonly: !!p,
          lang: g.locale,
          uniqueId: y,
          value: i,
          onInput: H,
          placeholder: g.get("payByLink.settings.theme.brandName.input.placeholder"),
          isInvalid: x
        }
      ),
      x && /* @__PURE__ */ a("div", { className: "adyen-pe-payment-link-theme-form__error", children: [
        /* @__PURE__ */ a(ee, { name: "cross-circle-fill", className: "adyen-pe-payment-link-theme-form__error-icon" }),
        /* @__PURE__ */ a(
          T,
          {
            className: "adyen-pe-payment-link-theme-form__error-text",
            el: R.SPAN,
            variant: w.BODY,
            children: g.get("payByLink.settings.theme.inputs.brandName.errors.missing")
          }
        )
      ] })
    ] }),
    K.map((e) => {
      const n = Y(e);
      return /* @__PURE__ */ a("div", { className: "adyen-pe-payment-link-settings__input-container ", children: n ? /* @__PURE__ */ a(Z, { disabled: !!p, logoType: e, logoURL: n, onRemoveLogo: V }) : /* @__PURE__ */ a($, { disabled: !!p, logoType: e, onFileInputChange: M }) }, e);
    })
  ] }) });
};
export {
  ge as ThemeForm
};
