// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/TransactionTotalItem/TransactionTotalItem.js
import { jsx as x } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import V from "../../../../../core/Context/useCoreContext.js";
import { useRef as t, useMemo as h } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { SummaryItem as C } from "../SummaryItem/SummaryItem.js";
const R = ({
  total: e,
  hiddenField: n,
  isHeader: r = !1,
  showLabelUnderline: a = !1,
  isSkeleton: c = !1,
  isLoading: l = !1,
  widths: u,
  onWidthsSet: g,
  expensesElemId: o,
  incomingsElemId: i
}) => {
  const { i18n: s } = V(), m = t(null), f = t(null), p = t(null), y = h(() => {
    const b = {
      elemId: i,
      labelKey: "transactions.overview.totals.tags.incoming",
      ref: m,
      skeletonWidth: 80,
      getValue: () => e && s.amount(e.incomings, e.currency),
      tooltipLabel: "transactions.overview.totals.tags.incoming.description",
      get ariaLabel() {
        return `${s.get(this.labelKey)}: ${this.getValue()}`;
      }
    }, v = {
      elemId: o,
      labelKey: "transactions.overview.totals.tags.outgoing",
      ref: f,
      skeletonWidth: 80,
      getValue: () => e && s.amount(e.expenses, e.currency),
      tooltipLabel: "transactions.overview.totals.tags.outgoing.description",
      get ariaLabel() {
        return `${s.get(this.labelKey)}: ${this.getValue()}`;
      }
    };
    return [
      ...n !== "incomings" ? [b] : [],
      ...n !== "expenses" ? [v] : [],
      {
        ref: p,
        skeletonWidth: 40,
        valueHasLabelStyle: !0,
        getValue: () => e == null ? void 0 : e.currency
      }
    ];
  }, [i, o, n, e, s]);
  return /* @__PURE__ */ x(
    C,
    {
      showLabelUnderline: a,
      isEmpty: !e,
      columnConfigs: y,
      isHeader: r,
      isSkeletonVisible: c,
      isLoading: l,
      widths: u,
      onWidthsSet: g
    }
  );
};
export {
  R as TransactionTotalItem
};
