// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormWrappers/FormCalendarInput.js
import { jsx as r } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as l } from "../../../external/preact/hooks/dist/hooks.module.js";
import { useWizardFormContext as d } from "../../../hooks/form/wizard/WizardFormContext.js";
import c from "./FormField.js";
import { CalendarInput as F } from "../FormFields/CalendarInput/CalendarInput.js";
import { VisibleField as f } from "./VisibleField.js";
import { FieldError as g } from "../FormFields/FieldError/FieldError.js";
import { Controller as h } from "../../../hooks/form/Controller.js";
function j({ clearable: u, fieldName: n, label: p }) {
  const { control: a, fieldsConfig: i } = d(), m = l(() => {
    var o;
    return (o = i[n]) == null ? void 0 : o.required;
  }, [i]);
  return /* @__PURE__ */ r(f, { name: n, children: /* @__PURE__ */ r(c, { label: p, optional: !m, children: /* @__PURE__ */ r(
    h,
    {
      name: n,
      control: a,
      rules: {
        required: m
      },
      render: ({ field: o, fieldState: e }) => {
        var s, t;
        return /* @__PURE__ */ r("div", { children: [
          /* @__PURE__ */ r(
            F,
            {
              value: o.value,
              onInput: o.onInput,
              isInvalid: !!e.error && e.isTouched,
              clearable: u
            }
          ),
          ((s = e.error) == null ? void 0 : s.message) && /* @__PURE__ */ r(g, { errorMessage: (t = e.error) == null ? void 0 : t.message, withTopMargin: !0 })
        ] });
      }
    }
  ) }) });
}
export {
  j as FormCalendarInput
};
