// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormWrappers/FormField.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import c from "../../../core/Context/useCoreContext.js";
import a from "../Typography/Typography.js";
import { TypographyVariant as n, TypographyElement as r } from "../Typography/types.js";
import d from "classnames";
const N = ({
  label: l,
  optional: o,
  supportText: i,
  children: t,
  className: m
}) => {
  const { i18n: p } = c();
  return /* @__PURE__ */ e("div", { className: d("adyen-pe-payment-link-creation-form__field-container", m), children: [
    /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__field-label-container", children: [
      /* @__PURE__ */ e(
        a,
        {
          el: r.SPAN,
          variant: n.BODY,
          stronger: !0,
          className: "adyen-pe-payment-link-creation-form__field-label",
          children: l
        }
      ),
      o && /* @__PURE__ */ e(
        a,
        {
          el: r.SPAN,
          variant: n.BODY,
          className: "adyen-pe-payment-link-creation-form__field-label-optional",
          children: `(${p.get("payByLink.common.fields.optional.label")})`
        }
      )
    ] }),
    t,
    i && /* @__PURE__ */ e("div", { children: /* @__PURE__ */ e(
      a,
      {
        el: r.SPAN,
        variant: n.BODY,
        className: "adyen-pe-payment-link-creation-form__field-support-text",
        children: i
      }
    ) })
  ] });
};
export {
  N as default
};
