// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormWrappers/FormSelect.js
import { jsx as s } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as j, useEffect as h } from "../../../external/preact/hooks/dist/hooks.module.js";
import { useWizardFormContext as w } from "../../../hooks/form/wizard/WizardFormContext.js";
import y from "./FormField.js";
import { VisibleField as z } from "./VisibleField.js";
import { FieldError as C } from "../FormFields/FieldError/FieldError.js";
import { Controller as R } from "../../../hooks/form/Controller.js";
import W from "../FormFields/Select/Select.js";
function L({
  className: v,
  clearable: F,
  fieldName: o,
  filterable: V,
  hideOptionalLabel: f,
  isRequired: i,
  items: e,
  label: q,
  onChange: u,
  preventInvalidState: x,
  readonly: I,
  validate: M
}) {
  const { control: T, fieldsConfig: c, getValues: l, setValue: t } = w(), m = j(() => {
    var r;
    return i ?? ((r = c[o]) == null ? void 0 : r.required);
  }, [c, i]);
  return h(() => {
    if (!e.length) return;
    const r = l(o);
    r && !e.some((n) => n.id === r) && t(o, "");
  }, [l, t, e]), h(() => {
    var r;
    e && e.length === 1 && t(o, (r = e[0]) == null ? void 0 : r.id);
  }, [e, t, o]), /* @__PURE__ */ s(z, { name: o, children: /* @__PURE__ */ s(y, { label: q, optional: !m && !f, className: v, children: /* @__PURE__ */ s(
    R,
    {
      name: o,
      control: T,
      rules: {
        required: m,
        validate: M
      },
      render: ({ field: r, fieldState: n }) => {
        var p, d;
        const E = (g) => {
          const b = g.target.value;
          r.onInput(b), u == null || u(g);
        }, a = !!n.error && n.isTouched;
        return /* @__PURE__ */ s("div", { children: [
          /* @__PURE__ */ s(
            W,
            {
              ...r,
              isInvalid: x ? !1 : a,
              isValid: !n.error,
              items: e,
              filterable: V,
              clearable: F,
              name: o,
              onChange: E,
              readonly: I,
              selected: r.value
            }
          ),
          a && ((p = n.error) == null ? void 0 : p.message) && /* @__PURE__ */ s(C, { errorMessage: (d = n.error) == null ? void 0 : d.message, withTopMargin: !0 })
        ] });
      }
    }
  ) }) });
}
export {
  L as FormSelect
};
