// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormWrappers/FormTextInput.js
import { jsx as n } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as z, useCallback as x } from "../../../external/preact/hooks/dist/hooks.module.js";
import { useWizardFormContext as R } from "../../../hooks/form/wizard/WizardFormContext.js";
import W from "./FormField.js";
import y from "../FormFields/InputBase.js";
import { VisibleField as A } from "./VisibleField.js";
import E from "../../../core/Context/useCoreContext.js";
import { Controller as G } from "../../../hooks/form/Controller.js";
function Z({
  className: F,
  dropdown: d,
  fieldName: s,
  isRequired: a,
  label: h,
  onDropdownInput: c,
  onInput: i,
  supportText: q,
  type: T,
  validate: t,
  min: V,
  maxLength: p,
  minLength: e,
  dropdownPosition: M,
  hideOptionalLabel: b
}) {
  const { control: B, fieldsConfig: l } = R(), { i18n: g } = E(), m = z(() => {
    var r;
    return a ?? ((r = l[s]) == null ? void 0 : r.required);
  }, [l, a]), j = x(
    (r, o) => {
      i == null || i(r), o(r.currentTarget.value);
    },
    [i]
  ), k = x(
    (r) => {
      const o = e && (r == null ? void 0 : r.length) < e;
      return !r && !m ? { valid: !0 } : o ? { valid: !1, message: g.get("common.errors.minLength", { values: { minLength: e } }) } : (t == null ? void 0 : t(r)) ?? { valid: !0 };
    },
    [g, m, e, t]
  );
  return /* @__PURE__ */ n(A, { name: s, children: /* @__PURE__ */ n(W, { label: h, optional: !m && !b, supportText: q, className: F, children: /* @__PURE__ */ n(
    G,
    {
      name: s,
      control: B,
      rules: {
        validate: k,
        required: m
      },
      render: ({ field: r, fieldState: o }) => {
        var C;
        const f = !!o.error && o.isTouched;
        return /* @__PURE__ */ n(
          y,
          {
            ...r,
            value: r.value,
            maxLength: p,
            minLength: e,
            name: s,
            isValid: !o.error,
            isInvalid: f,
            errorMessage: (C = o.error) == null ? void 0 : C.message,
            onDropdownInput: (...u) => {
              c == null || c(...u), f && r.triggerValidation();
            },
            onInput: (u) => j(u, r.onInput),
            type: T,
            min: V,
            dropdown: d,
            dropdownPosition: M
          }
        );
      }
    }
  ) }) });
}
export {
  Z as FormTextInput
};
