// @adyen/adyen-platform-experience-web/dist/es/components/external/PayoutsOverview/components/PayoutsOverview/PayoutsOverview.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { BASE_CLASS as z, EARLIEST_PAYOUT_SINCE_DATE as ee, BASE_CLASS_DETAILS as re } from "./constants.js";
import { PayoutsTable as se, PAYOUT_TABLE_FIELDS as te } from "../PayoutsTable/PayoutsTable.js";
import { LIMIT_OPTIONS as le, DEFAULT_PAGE_LIMIT as oe } from "../../../../internal/Pagination/constants.js";
import ie from "../../../../../hooks/useBalanceAccountSelection.js";
import pe from "../../../../internal/FilterBar/filters/DateFilter/DateFilter.js";
import me from "../../../../../hooks/useModalDetails/useModalDetails.js";
import fe from "../../../../../hooks/useDefaultOverviewFilterParams.js";
import { FilterParam as i } from "../../../../types.js";
import { useCallback as D, useMemo as o, useEffect as de } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { DataDetailsModal as ne } from "../../../../internal/DataOverviewDisplay/DataDetailsModal.js";
/* empty css                     */
import { useCustomColumnsData as De } from "../../../../../hooks/useCustomColumnsData.js";
import { hasCustomField as Ee } from "../../../../utils/customData/hasCustomField.js";
import { mergeRecords as ye } from "../../../../utils/customData/mergeRecords.js";
import { useFilterBarState as Se, FilterBarMobileSwitch as Te, FilterBar as ce } from "../../../../internal/FilterBar/FilterBar.js";
import Fe from "../../../../internal/Pagination/hooks/useCursorPaginatedRecords.js";
import ue from "../../../../internal/FormFields/Select/BalanceAccountSelector/BalanceAccountSelector.js";
import { useConfigContext as Ie } from "../../../../../core/ConfigContext/context.js";
import { isFunction as Le } from "../../../../../utils/value/is.js";
import { Header as Pe } from "../../../../internal/Header/Header.js";
const Qe = ({
  onFiltersChanged: p,
  balanceAccounts: E,
  allowLimitSelection: y = !0,
  preferredLimit: R = oe,
  onRecordSelection: S,
  showDetails: m,
  isLoadingBalanceAccount: g,
  onContactSupport: T,
  hideTitle: h,
  dataCustomization: r
}) => {
  var P, _, b, A;
  const { getPayouts: f } = Ie().endpoints, { activeBalanceAccount: e, balanceAccountSelectionOptions: N, onBalanceAccountSelection: w } = ie({ balanceAccounts: E }), { defaultParams: l, nowTimestamp: U, refreshNowTimestamp: d } = fe("payouts", e), M = D(
    async (s, n) => f({ signal: n, errorLevel: "error" }, {
      query: {
        ...s,
        createdSince: s[i.CREATED_SINCE] ?? l.current.defaultFilterParams[i.CREATED_SINCE],
        createdUntil: s[i.CREATED_UNTIL] ?? l.current.defaultFilterParams[i.CREATED_UNTIL],
        balanceAccountId: (e == null ? void 0 : e.id) ?? ""
      }
    }),
    [e == null ? void 0 : e.id, l, f]
  ), c = Se(), k = o(() => Le(p) ? p : void 0, [p]), K = o(() => y ? le : void 0, [y]), { canResetFilters: x, error: q, fetching: Y, filters: F, limit: j, limitOptions: v, records: u, resetFilters: _e, updateFilters: G, updateLimit: H, ...a } = Fe({
    fetchRecords: M,
    dataField: "data",
    filterParams: l.current.defaultFilterParams,
    initialFiltersSameAsDefault: !0,
    onFiltersChanged: k,
    preferredLimit: R,
    preferredLimitOptions: K,
    enabled: !!(e != null && e.id) && !!f
  });
  de(() => {
    d();
  }, [F, d]);
  const I = o(
    () => ({
      showDetails: m ?? !0,
      callback: S
    }),
    [m, S]
  ), J = D(
    ({ records: s, retrievedData: n }) => ye(s, n, (O, C) => O.createdAt === C.createdAt),
    []
  ), Q = o(() => {
    var s;
    return Ee((s = r == null ? void 0 : r.list) == null ? void 0 : s.fields, te);
  }, [(P = r == null ? void 0 : r.list) == null ? void 0 : P.fields]), { customRecords: V, loadingCustomRecords: W } = De({
    records: u,
    hasCustomColumn: Q,
    onDataRetrieve: (_ = r == null ? void 0 : r.list) == null ? void 0 : _.onDataRetrieve,
    mergeCustomData: J
  }), X = o(() => ({ payout: I }), [I]), { updateDetails: L, resetDetails: Z, selectedDetail: $ } = me(X), B = D(
    (s) => {
      L({
        selection: {
          type: "payout",
          data: { id: e == null ? void 0 : e.id, balanceAccountDescription: (e == null ? void 0 : e.description) || "", date: s.createdAt }
        },
        modalSize: "small"
      }).callback({ balanceAccountId: (e == null ? void 0 : e.id) || "", date: s.createdAt });
    },
    [L, e == null ? void 0 : e.id, e == null ? void 0 : e.description]
  );
  return /* @__PURE__ */ t("div", { className: z, children: [
    /* @__PURE__ */ t(Pe, { hideTitle: h, titleKey: "payouts.overview.title", subtitleKey: "payouts.overview.generateInfo", children: /* @__PURE__ */ t(Te, { ...c }) }),
    /* @__PURE__ */ t(ce, { ...c, ariaLabelKey: "payouts.overview.filters.label", children: [
      /* @__PURE__ */ t(
        ue,
        {
          activeBalanceAccount: e,
          balanceAccountSelectionOptions: N,
          onBalanceAccountSelection: w
        }
      ),
      /* @__PURE__ */ t(
        pe,
        {
          canResetFilters: x,
          defaultParams: l,
          filters: F,
          nowTimestamp: U,
          refreshNowTimestamp: d,
          sinceDate: ee,
          timezone: "UTC",
          updateFilters: G
        }
      )
    ] }),
    /* @__PURE__ */ t(
      ne,
      {
        ariaLabelKey: "payouts.details.title",
        className: re,
        onContactSupport: T,
        selectedDetail: $,
        resetDetails: Z,
        dataCustomization: r == null ? void 0 : r.details,
        children: /* @__PURE__ */ t(
          se,
          {
            loading: Y || g || !E || W,
            data: (b = r == null ? void 0 : r.list) != null && b.onDataRetrieve ? V : u,
            showPagination: !0,
            onRowClick: B,
            showDetails: m,
            limit: j,
            limitOptions: v,
            onContactSupport: T,
            onLimitSelection: H,
            error: q,
            customColumns: (A = r == null ? void 0 : r.list) == null ? void 0 : A.fields,
            ...a
          }
        )
      }
    )
  ] });
};
export {
  Qe as PayoutsOverview
};
