// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/PaymentRefund/PaymentRefund.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { sharedTransactionDetailsEventProperties as C, TX_DATA_CLASS as M, TX_REFUND_RESPONSE as L, TX_REFUND_RESPONSE_ICON as F, TX_REFUND_RESPONSE_SUCCESS_ICON as b, TX_REFUND_RESPONSE_ERROR_ICON as w, REFUND_REASONS as K } from "../../constants.js";
import { memo as U } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useState as A, useCallback as y, useRef as D, useEffect as S, useMemo as h } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import k from "../../../../../core/Context/analytics/useAnalyticsContext.js";
import v from "../../../../../core/Context/useCoreContext.js";
import X from "../../../../../hooks/useComponentTiming.js";
import B from "./PaymentRefundActions.js";
import V from "./PaymentRefundAmount.js";
import H from "./PaymentRefundNotice.js";
import Y from "./PaymentRefundReason.js";
import { Alert as x } from "../../../../internal/Alert/Alert.js";
import G from "../../../../internal/Button/Button.js";
import I from "../../../../internal/Typography/Typography.js";
import { ActiveView as W, RefundMode as m } from "../../types.js";
import { TypographyVariant as g, TypographyElement as j, TypographyModifier as q } from "../../../../internal/Typography/types.js";
import { AlertTypeOption as z, AlertVariantOption as J } from "../../../../internal/Alert/types.js";
import { ButtonVariant as Q } from "../../../../internal/Button/types.js";
import Z from "classnames";
import { Icon as $ } from "../../../../internal/Icon/Icon.js";
import { clamp as ee } from "../../../../../utils/value/number.js";
const _ = ({ disabled: c, refreshTransaction: a, setActiveView: r, setLocked: t, ...f }) => {
  const [s, p] = A(), R = y(() => void (E.current = !0), []), l = y(() => t(!0), [t]), n = y(() => r(W.DETAILS), [r]), { duration: u } = X(), i = D(!1), E = D(!1), o = k();
  return S(() => {
    var d;
    i.current || (i.current = !0, (d = o.addEvent) == null || d.call(o, "Switched to refund view", C));
  }, [o]), S(() => {
    c && !s && n(), s === "done" && l();
  }, [c, l, s, n]), S(() => () => {
    var d;
    u.current !== void 0 && !E.current && ((d = o.addEvent) == null || d.call(o, "Cancelled refund", C));
  }, [u, o]), s ? /* @__PURE__ */ e(_.Result, { result: s, refreshTransaction: a, showDetails: n }) : /* @__PURE__ */ e(_.Form, { ...f, beginRefund: R, setRefundResult: p, showDetails: n });
};
_.Form = U(
  ({
    beginRefund: c,
    currency: a,
    maxAmount: r,
    mode: t,
    refundedAmount: f,
    refundingAmounts: s,
    setRefundResult: p,
    showDetails: R,
    transaction: l
  }) => {
    const [n, u] = A(!1), [i, E] = A(K[0]), [o, d] = A(0), { i18n: N } = v(), O = h(() => {
      switch (t) {
        case m.FULL_AMOUNT:
        case m.PARTIAL_AMOUNT:
        case m.PARTIAL_LINE_ITEMS:
          return r;
        // case RefundMode.PARTIAL_LINE_ITEMS:
        //     return refundingItems.reduce((total, { amount, quantity }) => total + amount * quantity, 0);
        default:
          return 0;
      }
    }, [r, t]), P = h(() => {
      if (r > 0) {
        const T = { amount: N.amount(r, a) };
        switch (t) {
          case m.FULL_AMOUNT:
            return N.get("transactions.details.refund.alerts.refundableAmount", { values: T });
          case m.PARTIAL_AMOUNT:
            return N.get("transactions.details.refund.alerts.refundableMaximum", { values: T });
        }
      }
    }, [N, a, r, t]);
    return /* @__PURE__ */ e("div", { className: M, children: [
      /* @__PURE__ */ e(H, {}),
      /* @__PURE__ */ e(Y, { disabled: n, reason: i, onChange: E }),
      /* @__PURE__ */ e(
        V,
        {
          currency: a,
          disabled: n || t !== m.PARTIAL_AMOUNT,
          onChange: (T) => d(ee(0, T, O)),
          value: O
        }
      ),
      P && /* @__PURE__ */ e(x, { variant: J.TIP, type: z.HIGHLIGHT, children: /* @__PURE__ */ e(I, { className: "adyen-pe-alert__description", el: j.DIV, variant: g.BODY, wide: !0, children: P }) }),
      /* @__PURE__ */ e(
        B,
        {
          beginRefund: c,
          currency: a,
          disabled: n,
          maxAmount: r,
          refundAmount: o,
          refundedAmount: f,
          refundingAmounts: s,
          refundReason: i,
          setRefundInProgress: u,
          setRefundResult: p,
          showDetails: R,
          transactionId: l.id
        }
      )
    ] });
  }
);
_.Result = U(({ result: c, refreshTransaction: a, showDetails: r }) => {
  const { i18n: t } = v(), { isErrorResult: f, iconName: s, titleKey: p, descriptionKey: R } = h(() => {
    let l = "checkmark-circle-fill", n = "transactions.details.refund.alerts.refundSent", u = "transactions.details.refund.alerts.refundSuccess";
    const i = c === "error";
    return i && (l = "cross-circle-fill", n = "common.errors.somethingWentWrong", u = "transactions.details.refund.alerts.refundFailure"), { isErrorResult: i, iconName: l, titleKey: n, descriptionKey: u };
  }, [c]);
  return /* @__PURE__ */ e("div", { className: L, children: [
    /* @__PURE__ */ e(
      $,
      {
        className: Z(F, {
          [w]: f,
          [b]: !f
        }),
        name: s
      }
    ),
    /* @__PURE__ */ e(I, { className: q.MEDIUM, variant: g.TITLE, children: t.get(p) }),
    /* @__PURE__ */ e(I, { variant: g.BODY, children: t.get(R) }),
    /* @__PURE__ */ e(
      G,
      {
        onClick: () => {
          r(), a();
        },
        variant: Q.SECONDARY,
        children: t.get("transactions.details.refund.actions.back")
      }
    )
  ] });
});
export {
  _ as default
};
