// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/PaymentRefund/PaymentRefundActions.js
import { jsx as A } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import E from "classnames";
import { useMemo as s, useEffect as P } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { ButtonVariant as T } from "../../../../internal/Button/types.js";
import { ButtonActionsLayoutBasic as O } from "../../../../internal/Button/ButtonActions/types.js";
import { TX_DATA_CONTAINER as F, TX_DATA_ACTION_BAR as M, sharedTransactionDetailsEventProperties as W } from "../../constants.js";
import j from "../../../../internal/Button/ButtonActions/ButtonActions.js";
import k from "../../../../../core/Context/analytics/useAnalyticsContext.js";
import I from "../../../../../hooks/useMutation/useMutation.js";
import S from "../../../../../core/Context/useCoreContext.js";
import { useConfigContext as X } from "../../../../../core/ConfigContext/context.js";
const tt = ({
  beginRefund: g,
  currency: e,
  disabled: r,
  maxAmount: c,
  refundAmount: n,
  refundedAmount: B,
  refundingAmounts: h,
  refundReason: l,
  setRefundInProgress: u,
  setRefundResult: d,
  showDetails: v,
  transactionId: C
}) => {
  const { initiateRefund: L } = X().endpoints, { isLoading: o, mutate: N } = I({ queryFn: L }), { i18n: t } = S(), m = n > 0 && n <= c, D = B === 0 && h.length === 0 && n === c, f = r || o || !m, R = s(() => t.get("transactions.details.refund.actions.back"), [t]), p = s(
    () => ({
      idle: t.get("transactions.details.refund.actions.refund.labels.payment"),
      inProgress: `${t.get("transactions.details.refund.actions.refund.labels.inProgress")}..`
    }),
    [t]
  ), y = s(() => {
    const a = { amount: t.amount(n, e) };
    return t.get("transactions.details.refund.actions.refund.labels.amount", { values: a });
  }, [t, e, n]), b = o ? p.inProgress : m ? y : p.idle, i = k(), _ = {
    disabled: f,
    event: async () => {
      var a;
      if (!f) {
        g(), (a = i.addEvent) == null || a.call(i, "Completed refund", {
          ...W,
          isFullRefund: D,
          refundReason: l
        });
        try {
          await N({ contentType: "application/json", body: { amount: { currency: e, value: n }, refundReason: l } }, { path: { transactionId: C } }), d("done");
        } catch {
          d("error");
        }
      }
    },
    state: o ? "loading" : void 0,
    title: b,
    variant: T.PRIMARY,
    ariaLabel: o ? y : b
  }, x = {
    disabled: r,
    event: v,
    title: R,
    variant: T.SECONDARY
  };
  return P(() => u(o), [o, u]), /* @__PURE__ */ A("div", { className: E(F, M), children: /* @__PURE__ */ A(j, { actions: [_, x], layout: O.BUTTONS_END }) });
};
export {
  tt as default
};
