// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/PaymentRefund/PaymentRefundReason.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { REFUND_REASONS as p, TX_DATA_CONTAINER as f, TX_DATA_INPUT_HEAD as u, TX_DATA_INPUT_CONTAINER as N, TX_DATA_INPUT_CONTAINER_SHORT as A } from "../../constants.js";
import { useMemo as l, useCallback as _ } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { getTransactionRefundReason as R } from "../../../../utils/translation/getters.js";
import { TypographyVariant as y, TypographyElement as g } from "../../../../internal/Typography/types.js";
import h from "../../../../../core/Context/useCoreContext.js";
import I from "../../../../internal/Typography/Typography.js";
import { useUniqueId as b } from "../../../../../hooks/useUniqueId.js";
import C from "../../../../internal/FormFields/Select/Select.js";
const q = ({ disabled: i, onChange: r, reason: m }) => {
  const { i18n: a } = h(), o = b(), c = l(() => a.get("transactions.details.refund.inputs.reason.label"), [a]), d = l(
    () => p.map((t) => ({
      name: R(a, t),
      id: t
    })),
    [a]
  ), T = _(
    (t) => {
      var n;
      const s = (n = t.target) == null ? void 0 : n.value;
      s && r(s);
    },
    [r]
  );
  return /* @__PURE__ */ e("div", { className: f, children: [
    /* @__PURE__ */ e("div", { id: o, className: u, children: /* @__PURE__ */ e(I, { el: g.SPAN, variant: y.BODY, stronger: !0, children: c }) }),
    /* @__PURE__ */ e("div", { className: `${N} ${A}`, children: /* @__PURE__ */ e(
      C,
      {
        "aria-labelledby": o,
        filterable: !1,
        multiSelect: !1,
        readonly: i,
        items: d,
        onChange: T,
        selected: m
      }
    ) })
  ] });
};
export {
  q as default
};
