// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/PaymentRefund/PaymentRefundAmount.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { TX_DATA_CONTAINER as $, TX_DATA_INPUT_HEAD as B, TX_DATA_INPUT as C, TX_DATA_INPUT_CONTAINER_WITH_ERROR as H, TX_DATA_INPUT_CONTAINER_TEXT as L, TX_DATA_INPUT_CONTAINER_SHORT as V, TX_DATA_INPUT_CONTAINER as Y } from "../../constants.js";
import j from "classnames";
import { memo as k } from "../../../../../external/preact/compat/dist/compat.module.js";
import { Tag as W } from "../../../../internal/Tag/Tag.js";
import { TagVariant as w } from "../../../../internal/Tag/types.js";
import { ARIA_ERROR_SUFFIX as z } from "../../../../../core/Errors/constants.js";
import { getDecimalAmount as G, getDivider as J } from "../../../../../core/Localization/amount/amount-util.js";
import { TypographyVariant as E, TypographyElement as h } from "../../../../internal/Typography/types.js";
import { useState as D, useMemo as P, useRef as b, useCallback as K, useEffect as Q } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import Z from "../../../../../core/Context/useCoreContext.js";
import O from "../../../../internal/Typography/Typography.js";
import ee from "../../../../internal/FormFields/InputBase.js";
import { useUniqueId as x } from "../../../../../hooks/useUniqueId.js";
import { Icon as te } from "../../../../internal/Icon/Icon.js";
const _e = k(({ currency: n, disabled: d, onChange: u, value: U }) => {
  const { i18n: t } = Z(), f = x(), N = x(), i = Math.log10(J(n)), r = parseInt(`${U}`, 10), p = G(r, n).toFixed(i), [X, y] = D(p), [T, S] = D(), q = P(() => {
    const a = { amount: t.amount(r, n) };
    return {
      excess: t.get("transactions.details.refund.inputs.amount.errors.excess", { values: a }),
      negative: t.get("transactions.details.refund.inputs.amount.errors.negative"),
      required: t.get("transactions.details.refund.inputs.amount.errors.required")
    };
  }, [t, n, r]), A = T && q[T], F = P(() => t.get("transactions.details.refund.inputs.amount.label"), [t]), s = b(null), _ = K(
    (a) => {
      var R;
      let l, o = a.value.trim();
      const m = Math.trunc(+`${parseFloat(o)}e${i}`) || 0;
      m || o ? (m < 0 && (l = "negative"), m > r && (l = "excess")) : l = "required";
      const g = ((R = 1.1.toLocaleString(t.locale).match(/\d(.*?)\d/)) == null ? void 0 : R[1]) || ".", I = o.split(g);
      if (I.length === 2) {
        const M = I[0];
        let c = I[1];
        c.length >= i && (c = c.substring(0, i), o = M + g + c, a.value = o);
      }
      y(o), S(l), u == null || u(l ? 0 : m);
    },
    [i, r, u]
  ), v = b();
  return Q(() => {
    if (v.current !== r)
      v.current = r;
    else if (d && T && s.current)
      s.current.value = p;
    else return;
    s.current && _(s.current);
  }, [_, d, p, r]), /* @__PURE__ */ e("div", { className: $, children: [
    /* @__PURE__ */ e("div", { className: B, children: /* @__PURE__ */ e("div", { id: N, children: /* @__PURE__ */ e(O, { el: h.SPAN, variant: E.BODY, stronger: !0, children: F }) }) }),
    /* @__PURE__ */ e(
      "div",
      {
        className: j({
          [Y]: !0,
          [V]: !0,
          [L]: !0,
          [H]: A
        }),
        children: [
          /* @__PURE__ */ e("label", { htmlFor: f, "aria-labelledby": N, children: [
            n && /* @__PURE__ */ e(W, { label: n, variant: w.DEFAULT }),
            /* @__PURE__ */ e(
              ee,
              {
                min: 0,
                type: "number",
                className: C,
                disabled: d,
                lang: t.locale,
                onInput: (a) => _(a.currentTarget),
                value: X,
                uniqueId: f,
                ref: s
              }
            )
          ] }),
          A && /* @__PURE__ */ e("div", { className: "adyen-pe-input__refund-invalid-value", id: `${f}${z}`, children: [
            /* @__PURE__ */ e(te, { name: "cross-circle-fill" }),
            /* @__PURE__ */ e(O, { el: h.SPAN, variant: E.BODY, children: A })
          ] })
        ]
      }
    )
  ] });
});
export {
  _e as default
};
