// @adyen/adyen-platform-experience-web/dist/es/components/internal/FilterBar/FilterBar.js
import { jsx as a } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import b from "classnames";
import { ButtonVariant as c } from "../Button/types.js";
import { useState as C, useEffect as B, useMemo as I } from "../../../external/preact/hooks/dist/hooks.module.js";
import m from "../../../core/Context/useCoreContext.js";
import { useResponsiveContainer as x, containerQueries as E } from "../../../hooks/useResponsiveContainer.js";
/* empty css               */
import { useUniqueId as p } from "../../../hooks/useUniqueId.js";
import d from "../Button/Button.js";
import { Icon as v } from "../Icon/Icon.js";
import { isFunction as y } from "../../../utils/value/is.js";
const s = "adyen-pe-filter-bar-mobile-switch", O = () => {
  const e = x(E.down.xs), [t, i] = C(!e), r = `filters-${p()}`;
  return B(() => {
    i(!e);
  }, [e]), { filterBarElementId: r, isMobileContainer: e, showingFilters: t, setShowingFilters: i };
}, Y = ({
  ariaLabelKey: e,
  filterBarElementId: t,
  isMobileContainer: i,
  showingFilters: r,
  setShowingFilters: o
}) => {
  const { i18n: n } = m(), f = I(() => n.get(e ?? "common.filters.mobile.label"), [n]), l = !y(o), u = !l && !!r;
  return i ? /* @__PURE__ */ a("div", { className: s, children: /* @__PURE__ */ a(
    d,
    {
      iconButton: !0,
      className: `${s}__button`,
      disabled: l,
      "aria-label": f,
      "aria-expanded": u,
      "aria-controls": t,
      onClick: () => o == null ? void 0 : o(!r),
      variant: c.SECONDARY,
      children: /* @__PURE__ */ a(v, { name: r ? "cross" : "filter" })
    }
  ) }) : null;
}, $ = (e) => {
  const { i18n: t } = m();
  return e.showingFilters ? /* @__PURE__ */ a(
    "div",
    {
      role: "group",
      "data-testId": "filter-bar",
      "aria-label": t.get(e.ariaLabelKey ?? "common.filters.label"),
      className: b("adyen-pe-filter-bar", { "adyen-pe-filter-bar--mobile": e.isMobileContainer }),
      id: e.filterBarElementId,
      children: [
        e.children,
        e.canResetFilters && !!e.resetFilters && /* @__PURE__ */ a(d, { variant: c.TERTIARY, onClick: e.resetFilters, children: t.get("common.filters.controls.resetAll.label") })
      ]
    }
  ) : null;
};
export {
  $ as FilterBar,
  Y as FilterBarMobileSwitch,
  $ as default,
  O as useFilterBarState
};
