// @adyen/adyen-platform-experience-web/dist/es/components/internal/Modal/Modal.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { ButtonVariant as w } from "../Button/types.js";
import { useClickOutside as N } from "../../../hooks/element/useClickOutside.js";
import { useResponsiveContainer as g, containerQueries as R } from "../../../hooks/useResponsiveContainer.js";
import { withTabbableRoot as E } from "../../../primitives/dom/tabbableRoot/tabbable.js";
import M from "../../../hooks/element/useFocusTrap.js";
import k from "../../../core/Context/useCoreContext.js";
import { useContext as T, useRef as f, useCallback as B, useEffect as I } from "../../../external/preact/hooks/dist/hooks.module.js";
import { createContext as A, Fragment as F } from "../../../external/preact/dist/preact.module.js";
import s from "classnames";
/* empty css           */
import j from "../Button/Button.js";
import { Icon as q } from "../Icon/Icon.js";
const h = A({ withinModal: !1 }), X = () => T(h);
function Z({
  title: n,
  children: y,
  classNameModifiers: v = [],
  isOpen: o,
  onClose: i,
  isDismissible: t = !0,
  headerWithBorder: x = !0,
  size: r = "fluid",
  ..._
}) {
  const { i18n: b } = k(), C = g(R.down.xs), d = f(null), u = f(null), m = B(() => {
    var a;
    o && t && ((a = u.current) == null || a.focus(), i());
  }, [o, t, i]), c = M(N(null, m), m);
  return I(() => {
    var a, p;
    if (o && c.current) {
      u.current = document.activeElement;
      let l = E();
      return (a = d.current) == null || a.focus(), l.root = c.current, l.current = 1, (p = d.current) == null || p.remove(), () => {
        l.root = null, l = null;
      };
    }
  }, [o]), /* @__PURE__ */ e(F, { children: o && /* @__PURE__ */ e(
    "div",
    {
      className: s(
        "adyen-pe-modal-wrapper",
        v.map((a) => `adyen-pe-modal-wrapper--${a}`),
        { "adyen-pe-modal-wrapper--open": o, "adyen-pe-modal-wrapper--dismissible": t }
      ),
      role: "dialog",
      "aria-modal": "true",
      "aria-hidden": "false",
      ..._,
      children: /* @__PURE__ */ e(h.Provider, { value: { withinModal: !0 }, children: /* @__PURE__ */ e(
        "div",
        {
          className: s("adyen-pe-modal", {
            "adyen-pe-modal--fluid": r === "fluid",
            "adyen-pe-modal--small": r === "small",
            "adyen-pe-modal--large": r === "large",
            "adyen-pe-modal--extra-large": r === "extra-large",
            "adyen-pe-modal--full-screen": r === "full-screen" || C
          }),
          ref: c,
          children: [
            /* @__PURE__ */ e("div", { className: "adyen-pe-visually-hidden", ref: d, tabIndex: -1 }),
            /* @__PURE__ */ e(
              "div",
              {
                className: s("adyen-pe-modal__header", {
                  "adyen-pe-modal__header--with-title": n,
                  "adyen-pe-modal__header--with-border-bottom": x
                }),
                children: [
                  n && /* @__PURE__ */ e("div", { className: "adyen-pe-modal__header-title", children: n }),
                  t && /* @__PURE__ */ e(
                    j,
                    {
                      onClick: m,
                      variant: w.TERTIARY,
                      iconButton: !0,
                      classNameModifiers: ["circle"],
                      className: "adyen-pe-modal__dismiss-button",
                      "aria-label": b.get("common.modal.controls.dismiss.label"),
                      children: /* @__PURE__ */ e(q, { name: "cross" })
                    }
                  )
                ]
              }
            ),
            /* @__PURE__ */ e("div", { className: "adyen-pe-modal__content", children: y })
          ]
        }
      ) })
    }
  ) });
}
export {
  Z as default,
  X as useModalContext
};
