// @adyen/adyen-platform-experience-web/dist/es/core/Context/CoreProvider.js
import { jsx as h } from "../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { toChildArray as v } from "../../external/preact/dist/preact.module.js";
import { useMemo as r, useEffect as P } from "../../external/preact/hooks/dist/hooks.module.js";
import { CoreContext as w } from "./CoreContext.js";
import C from "../../hooks/useBooleanState.js";
import j from "../Localization/Localization.js";
const V = ({
  i18n: o = new j().i18n,
  children: p,
  commonProps: t,
  loadingContext: a,
  updateCore: e,
  externalErrorHandler: m,
  componentRef: s,
  getImageAsset: c,
  getDatasetAsset: u,
  getCdnDataset: f,
  getCdnConfig: i
}) => {
  const [y, d] = C(!1), l = r(() => t || {}, [t]), n = r(() => a ?? "", [a]);
  P(() => {
    (async () => {
      await (o == null ? void 0 : o.ready), d(!0);
    })().catch();
  }, []);
  const x = r(
    () => ({
      i18n: o,
      commonProps: l,
      loadingContext: n,
      updateCore: e,
      externalErrorHandler: m,
      componentRef: s,
      getImageAsset: c,
      getDatasetAsset: u,
      getCdnConfig: i,
      getCdnDataset: f
    }),
    [
      l,
      s,
      m,
      o,
      n,
      c,
      u,
      i,
      f,
      e
    ]
  );
  return y ? /* @__PURE__ */ h(w.Provider, { value: x, children: v(p) }) : null;
};
export {
  V as default
};
