// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputesOverview/components/DisputeManagementModal/DisputeManagementModal.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useState as C, useCallback as u, useEffect as g } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import S from "../../../../../core/Context/useCoreContext.js";
import { popoverUtil as h } from "../../../../internal/Popover/utils/popoverUtil.js";
import { DisputeDetailsContainer as A } from "../../../DisputeManagement/components/DisputeDetailsContainer/DisputeDetailsContainer.js";
import D from "../../../../internal/Modal/Modal.js";
const E = ({
  children: c,
  selectedDetail: e,
  resetDetails: n,
  dataCustomization: p,
  onContactSupport: f,
  refreshDisputesList: a,
  setModalVisible: o
}) => {
  const { i18n: l } = S(), [i, r] = C(!1), s = !!e, d = u(() => {
    i && (r(!1), a("CHARGEBACKS")), o(!1), n();
  }, [i, a, n, o]), m = u(() => {
    r(!0);
  }, []);
  return g(() => {
    s && (o(!0), h.closeAll());
  }, [s, o]), /* @__PURE__ */ t("div", { children: [
    c,
    e && /* @__PURE__ */ t(
      D,
      {
        isOpen: s,
        "aria-label": l.get("disputes.management.common.title"),
        onClose: d,
        isDismissible: !0,
        headerWithBorder: !1,
        size: e.modalSize || "large",
        children: /* @__PURE__ */ t(
          A,
          {
            id: e.selection.data,
            dataCustomization: p,
            onDisputeAccept: m,
            onDisputeDefend: m,
            onContactSupport: f,
            onDismiss: n,
            hideTitle: !0
          }
        )
      }
    )
  ] });
};
export {
  E as DisputeManagementModal
};
