// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/components/PreQualified/PreQualified.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import u from "../PreQualifiedIntro.js";
import { useState as d, useCallback as s, useMemo as C } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { CapitalOffer as g } from "../../../CapitalOffer/components/CapitalOffer/CapitalOffer.js";
import { Fragment as h } from "../../../../../external/preact/dist/preact.module.js";
const q = ({
  hideTitle: f,
  dynamicOffer: o,
  skipPreQualifiedIntro: a,
  onOfferOptionsRequest: r,
  onFundsRequest: n,
  onOfferDismiss: t
}) => {
  const [m, i] = d(a ? "capitalOffer" : "intro"), c = s(() => {
    r ? r() : i("capitalOffer");
  }, [r]), l = C(() => !a || !!t, [t, a]), p = s(() => {
    t ? t() : i("intro");
  }, [t]);
  return /* @__PURE__ */ e(h, { children: m === "intro" ? /* @__PURE__ */ e(u, { hideTitle: f, dynamicOfferConfig: o, onOfferOptionsRequest: c }) : /* @__PURE__ */ e(
    g,
    {
      onFundsRequest: n,
      onOfferDismiss: l ? p : void 0,
      externalDynamicOffersConfig: o
    }
  ) });
};
export {
  q as PreQualified
};
