// @adyen/adyen-platform-experience-web/dist/es/components/internal/CapitalSlider/CapitalSlider.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
/* empty css                   */
import { useMemo as s } from "../../../external/preact/hooks/dist/hooks.module.js";
import { uniqueId as h } from "../../../utils/random/uniqueId.js";
import r from "../Typography/Typography.js";
import { TypographyVariant as i } from "../Typography/types.js";
import b from "../../../core/Context/useCoreContext.js";
import N from "classnames";
import y from "../Slider/Slider.js";
const M = ({
  dynamicOffersConfig: t,
  value: l = t.minAmount.value,
  onValueChange: m,
  onRelease: n,
  className: x
}) => {
  const p = s(() => h("elem"), []), o = s(() => h("elem"), []), { i18n: a } = b(), v = s(
    () => a.amount(l, t.minAmount.currency, { maximumFractionDigits: 0 }),
    [a, l, t.minAmount.currency]
  ), A = (u) => {
    const d = Number(u.target.value);
    m == null || m(d);
  }, c = (u) => {
    const d = Number(u.target.value);
    n == null || n(d);
  };
  return /* @__PURE__ */ e("div", { className: N("adyen-pe-capital-slider", x), children: [
    /* @__PURE__ */ e("label", { id: p, htmlFor: o, className: "adyen-pe-capital-slider__label", role: "presentation", children: /* @__PURE__ */ e(r, { variant: i.BODY, stronger: !0, children: a.get("capital.offer.selection.slider.a11y.label") }) }),
    /* @__PURE__ */ e("div", { children: [
      /* @__PURE__ */ e("output", { "aria-labelledby": p, htmlFor: o, className: "adyen-pe-capital-slider__value", "aria-live": "polite", children: /* @__PURE__ */ e(r, { variant: i.TITLE, strongest: !0, children: v }) }),
      /* @__PURE__ */ e(
        y,
        {
          id: o,
          value: l,
          min: t.minAmount.value,
          max: t.maxAmount.value,
          step: t.step,
          onChange: A,
          onMouseUp: c,
          onTouchEnd: c,
          onKeyUp: c,
          className: "adyen-pe-capital-slider__input",
          "aria-valuetext": v
        }
      )
    ] }),
    /* @__PURE__ */ e("div", { className: "adyen-pe-capital-slider__labels", "aria-hidden": "true", children: [
      /* @__PURE__ */ e("div", { children: [
        /* @__PURE__ */ e(r, { variant: i.CAPTION, children: a.get("capital.offer.selection.slider.markers.min") }),
        /* @__PURE__ */ e(r, { variant: i.BODY, children: a.amount(t.minAmount.value, t.minAmount.currency, { maximumFractionDigits: 0 }) })
      ] }),
      /* @__PURE__ */ e("div", { children: [
        /* @__PURE__ */ e(r, { variant: i.CAPTION, children: a.get("capital.offer.selection.slider.markers.max") }),
        /* @__PURE__ */ e(r, { variant: i.BODY, children: a.amount(t.maxAmount.value, t.maxAmount.currency, { maximumFractionDigits: 0 }) })
      ] })
    ] })
  ] });
};
export {
  M as default
};
