// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputesOverview/components/DisputesOverview/DisputesOverview.js
import { jsx as r } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import Le from "classnames";
import { useState as E, useMemo as a, useCallback as f, useRef as B, useEffect as U } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import W from "../../../../../core/Context/useCoreContext.js";
import Pe, { ALL_BALANCE_ACCOUNTS_SELECTION_ID as he } from "../../../../../hooks/useBalanceAccountSelection.js";
import Ce from "../../../../../hooks/useDefaultOverviewFilterParams.js";
import ve from "../../../../internal/FilterBar/filters/DateFilter/DateFilter.js";
import { EARLIEST_DISPUTES_SINCE_DATE as we, TABS_CONTAINER_CLASS as Me, BASE_CLASS as Ne, BASE_XS_CLASS as ge } from "./constants.js";
import { LIMIT_OPTIONS as ye, DEFAULT_PAGE_LIMIT as Ge } from "../../../../internal/Pagination/constants.js";
import { DISPUTE_REASON_CATEGORIES as $, DISPUTE_PAYMENT_SCHEMES as z, DISPUTE_STATUS_GROUPS as g } from "../../../../utils/disputes/constants.js";
import { useResponsiveContainer as xe, containerQueries as je } from "../../../../../hooks/useResponsiveContainer.js";
import { FilterParam as O } from "../../../../types.js";
import { DisputesTable as ke } from "../DisputesTable/DisputesTable.js";
import { DisputeManagementModal as He } from "../DisputeManagementModal/DisputeManagementModal.js";
import Ve from "../../../../internal/Tabs/Tabs.js";
/* empty css                      */
import Ye from "../../../../internal/Pagination/hooks/useCursorPaginatedRecords.js";
import { useMultiSelectionFilter as q } from "../../../TransactionsOverview/components/MultiSelectionFilter/useMultiSelectionFilter/useMultiSelectionFilter.js";
import Ke from "../../../../../hooks/useModalDetails/useModalDetails.js";
import { useFilterBarState as Be, FilterBarMobileSwitch as qe, FilterBar as Qe } from "../../../../internal/FilterBar/FilterBar.js";
import Xe from "../../../../internal/FormFields/Select/BalanceAccountSelector/BalanceAccountSelector.js";
import Q from "../../../TransactionsOverview/components/MultiSelectionFilter/MultiSelectionFilter.js";
import Ze from "../../../../internal/FormFields/Select/Select.js";
import { useConfigContext as Je } from "../../../../../core/ConfigContext/context.js";
import { listFrom as X } from "../../../../../utils/collection/main.js";
import { isFunction as We } from "../../../../../utils/value/is.js";
import { Header as $e } from "../../../../internal/Header/Header.js";
import { noop as Z } from "../../../../../utils/common.js";
const J = "CHARGEBACKS", M = "schemeCodes", T = "reasonCategories", N = "_t", ze = Object.keys(z), et = Object.keys($), tt = Object.keys(g), ee = Object.entries(g).map(([l, o]) => ({
  id: l,
  label: o,
  content: null
})), rt = ({
  ["aria-label"]: l,
  activeTab: o,
  onChange: _
}) => {
  const { i18n: R } = W(), [A, D] = E(o), L = a(() => Object.entries(g).map(([i, m]) => ({
    id: i,
    name: R.get(m)
  })), [R]);
  return U(() => {
    const i = ee.find((m) => m.id === A);
    i && _(i);
  }, [_, A]), U(() => D(o), [o]), /* @__PURE__ */ r(
    Ze,
    {
      "aria-label": l,
      items: L,
      selected: A,
      onChange: ({ target: i }) => D(i.value),
      showOverlay: !0,
      multiSelect: !1,
      filterable: !1
    }
  );
}, Mt = ({
  onFiltersChanged: l,
  balanceAccounts: o,
  allowLimitSelection: _ = !0,
  preferredLimit: R = Ge,
  isLoadingBalanceAccount: A,
  onContactSupport: D,
  hideTitle: L,
  onRecordSelection: i,
  showDetails: m,
  dataCustomization: c
}) => {
  const { i18n: u } = W(), { getDisputeList: P } = Je().endpoints, { activeBalanceAccount: e, balanceAccountSelectionOptions: te, onBalanceAccountSelection: re } = Pe({
    balanceAccounts: o,
    allowAllSelection: !0
  }), { defaultParams: F, nowTimestamp: h, refreshNowTimestamp: C } = Ce("disputes", e, "last90Days"), [y, se] = E(!1), [ie, oe] = E(), [d, le] = E(J), [ne, G] = E(!1), [x, j] = E(d), k = a(() => u.get("disputes.overview.common.filters.types.statusGroup"), [u]), H = a(
    () => ({
      showDetails: m ?? !0,
      callback: i
    }),
    [m, i]
  ), ae = a(() => ({ dispute: H }), [H]), me = f(
    async ({ [N]: t, ...s }, S) => P({ signal: S, errorLevel: "error" }, {
      query: {
        ...s,
        ...(e == null ? void 0 : e.id) !== he && {
          balanceAccountId: (e == null ? void 0 : e.id) ?? ""
        },
        reasonCategories: X(s[T]),
        schemeCodes: X(s[M]),
        createdSince: s[O.CREATED_SINCE] ?? F.current.defaultFilterParams[O.CREATED_SINCE],
        createdUntil: s[O.CREATED_UNTIL] ?? F.current.defaultFilterParams[O.CREATED_UNTIL]
      }
    }),
    [e == null ? void 0 : e.id, F, P]
  ), V = Be(), ce = a(() => We(l) ? l : void 0, [l]), ue = a(() => _ ? ye : void 0, [_]), v = Object.assign(F.current.defaultFilterParams, {
    [T]: void 0,
    [M]: void 0,
    [N]: performance.now(),
    statusGroup: J
  }), { canResetFilters: de, error: pe, fetching: Se, filters: p, limit: fe, limitOptions: Te, records: Ee, resetFilters: st, updateFilters: n, updateLimit: _e, ...Ae } = Ye({
    fetchRecords: me,
    dataField: "data",
    filterParams: v,
    initialFiltersSameAsDefault: !0,
    onFiltersChanged: ce,
    preferredLimit: R,
    preferredLimitOptions: ue,
    enabled: !!(e != null && e.id) && !!P
  }), w = B(void 0), De = q({
    mapFilterOptionName: f((t) => u.get($[t]), [u]),
    filterParam: T,
    filterValues: et,
    defaultFilters: { ...v, [T]: w.current },
    updateFilters: n,
    filters: p
  }), Fe = q({
    mapFilterOptionName: f((t) => z[t], []),
    filterParam: M,
    filterValues: ze,
    defaultFilters: v,
    updateFilters: n,
    filters: p
  }), { updateDetails: Y, resetDetails: Re, selectedDetail: be } = Ke(ae), Ie = f(
    ({ disputePspReference: t }) => {
      Y({
        selection: {
          type: "dispute",
          data: t
        },
        modalSize: "small"
      }).callback({ id: t });
    },
    [Y]
  ), Oe = a(() => {
    const t = new Date(h), s = t.setFullYear(t.getFullYear() - 1), S = new Date(we).getTime();
    return new Date(Math.max(S, s)).toString();
  }, [h]), b = B(null), K = f(
    ({ id: t }) => {
      b.current && clearTimeout(b.current), b.current = setTimeout(() => {
        requestAnimationFrame(() => G(!1));
        const s = T, S = { statusGroup: t, [s]: void 0 };
        t !== "FRAUD_ALERTS" && (S[s] = w.current), n(S), b.current = null;
      }, 500), le(t), G(!0), j(void 0);
    },
    [n]
  ), Ue = f(
    (t) => {
      t && tt.includes(t) && t !== d ? j(t) : (
        // Refresh the current disputes list status group,
        // by updating the last refresh timestamp filter parameter
        n({ [N]: performance.now() })
      );
    },
    [d, n]
  ), I = xe(je.down.xs);
  return U(() => {
    oe(I && y ? { maxHeight: 0, overflowY: "hidden" } : void 0);
  }, [I, y]), U(() => {
    C(), p.statusGroup !== "FRAUD_ALERTS" && (w.current = p[T]);
  }, [p, C]), /* @__PURE__ */ r("div", { style: ie, className: Le(Ne, { [ge]: I }), children: [
    /* @__PURE__ */ r($e, { hideTitle: L, titleKey: "disputes.overview.common.title", children: /* @__PURE__ */ r(qe, { ...V }) }),
    /* @__PURE__ */ r("div", { children: [
      /* @__PURE__ */ r("div", { children: [
        /* @__PURE__ */ r("div", { className: Me, children: I ? /* @__PURE__ */ r(
          rt,
          {
            "aria-label": k,
            activeTab: x ?? d,
            onChange: K
          }
        ) : /* @__PURE__ */ r(
          Ve,
          {
            "aria-label": k,
            tabs: ee,
            activeTab: x,
            onChange: K
          }
        ) }),
        /* @__PURE__ */ r(Qe, { ...V, ariaLabelKey: "disputes.overview.common.filters.a11y.label", children: [
          /* @__PURE__ */ r(
            Xe,
            {
              activeBalanceAccount: e,
              balanceAccountSelectionOptions: te,
              onBalanceAccountSelection: re
            }
          ),
          /* @__PURE__ */ r(
            ve,
            {
              canResetFilters: de,
              defaultParams: F,
              filters: p,
              nowTimestamp: h,
              refreshNowTimestamp: C,
              sinceDate: Oe,
              timezone: e == null ? void 0 : e.timeZone,
              updateFilters: n
            }
          ),
          /* @__PURE__ */ r(
            Q,
            {
              ...Fe,
              placeholder: u.get("disputes.overview.common.filters.types.paymentMethod"),
              onResetAction: Z
            }
          ),
          d !== "FRAUD_ALERTS" && /* @__PURE__ */ r(
            Q,
            {
              ...De,
              placeholder: u.get("disputes.overview.common.filters.types.disputeReason"),
              onResetAction: Z
            }
          )
        ] })
      ] }),
      /* @__PURE__ */ r(
        He,
        {
          dataCustomization: (c == null ? void 0 : c.details) && { details: c == null ? void 0 : c.details },
          selectedDetail: be,
          resetDetails: Re,
          onContactSupport: D,
          refreshDisputesList: Ue,
          setModalVisible: se,
          children: /* @__PURE__ */ r(
            ke,
            {
              activeBalanceAccount: e,
              balanceAccountId: e == null ? void 0 : e.id,
              loading: ne || Se || A || !o || !e,
              data: Ee,
              showPagination: !0,
              limit: fe,
              limitOptions: Te,
              onContactSupport: D,
              onLimitSelection: _e,
              error: pe,
              onRowClick: Ie,
              statusGroup: d,
              ...Ae
            }
          )
        }
      )
    ] })
  ] });
};
export {
  Mt as DisputesOverview
};
