// @adyen/adyen-platform-experience-web/dist/es/index.js
/* empty css                       */
/* empty css                 */
import t from "./core/core.js";
import { PayoutElement as p } from "./components/external/PayoutDetails/PayoutDetails.js";
import { PayoutsElement as f } from "./components/external/PayoutsOverview/PayoutsOverviewElement.js";
import { TransactionElement as x } from "./components/external/TransactionDetails/TransactionDetails.js";
import { TransactionsElement as y } from "./components/external/TransactionsOverview/TransactionsOverviewElement.js";
import { CapitalOfferElement as P } from "./components/external/CapitalOffer/CapitalOfferElement.js";
import { ReportsElement as w } from "./components/external/ReportsOverview/ReportsOverviewElement.js";
import { CapitalOverviewElement as _ } from "./components/external/CapitalOverview/CapitalOverviewElement.js";
import { DisputesOverviewElement as L } from "./components/external/DisputesOverview/DisputesOverviewElement.js";
import { DisputeManagementElement as k } from "./components/external/DisputeManagement/DisputeManagementElement.js";
import { PaymentLinksOverviewElement as R } from "./components/external/PaymentLinksOverview/PaymentLinksOverviewElement.js";
import { PaymentLinkCreationElement as S } from "./components/external/PaymentLinkCreation/PaymentLinkCreationElement.js";
import { PaymentLinkSettingsElement as M } from "./components/external/PaymentLinkSettings/PaymentLinkSettingsElement.js";
import { PaymentLinkDetailsElement as F } from "./components/external/PaymentLinkDetails/PaymentLinkDetailsElement.js";
import { all_locales as N, da_DK as b, de_DE as z, en_US as B, es_ES as K, fi_FI as U, fr_FR as V, it_IT as h, nl_NL as j, no_NO as q, pt_BR as G, sv_SE as H } from "./translations/index.js";
import { ActiveView as Q, DetailsTab as W, RefundMode as X, RefundType as Y, RefundedState as Z } from "./components/external/TransactionDetails/types.js";
async function i(e) {
  return await new t(e).initialize();
}
export {
  Q as ActiveView,
  i as AdyenPlatformExperience,
  P as CapitalOffer,
  _ as CapitalOverview,
  t as Core,
  W as DetailsTab,
  k as DisputeManagement,
  L as DisputesOverview,
  S as PaymentLinkCreation,
  F as PaymentLinkDetails,
  M as PaymentLinkSettings,
  R as PaymentLinksOverview,
  p as PayoutDetails,
  f as PayoutsOverview,
  X as RefundMode,
  Y as RefundType,
  Z as RefundedState,
  w as ReportsOverview,
  x as TransactionDetails,
  y as TransactionsOverview,
  N as all_locales,
  b as da_DK,
  z as de_DE,
  B as en_US,
  K as es_ES,
  U as fi_FI,
  V as fr_FR,
  h as it_IT,
  j as nl_NL,
  q as no_NO,
  G as pt_BR,
  H as sv_SE
};
