// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOffer/components/CapitalOfferSummary/CapitalOfferSummary.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import E from "../../../../../core/Context/useCoreContext.js";
import { useMemo as p, useCallback as f } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { getExpectedRepaymentDate as D, calculateMaximumRepaymentPeriodInMonths as M, getPercentage as R } from "../utils/utils.js";
import m from "../../../../internal/Typography/Typography.js";
import { TypographyVariant as l, TypographyElement as s } from "../../../../internal/Typography/types.js";
/* empty css                         */
import h from "../../../../internal/Button/Button.js";
import { ButtonVariant as N } from "../../../../internal/Button/types.js";
import B from "../../../../../hooks/useMutation/useMutation.js";
import q from "../../../../../core/Context/analytics/useAnalyticsContext.js";
import { useDurationEvent as L } from "../../../../../hooks/useAnalytics/useDurationEvent.js";
import { Tooltip as _ } from "../../../../internal/Tooltip/Tooltip.js";
import { AlertTypeOption as S } from "../../../../internal/Alert/types.js";
import { Alert as F } from "../../../../internal/Alert/Alert.js";
import { CapitalErrorMessageDisplay as W } from "../utils/CapitalErrorMessageDisplay.js";
import Y from "classnames";
import { sharedCapitalOfferAnalyticsEventProperties as w } from "../CapitalOffer/constants.js";
import { CAPITAL_REPAYMENT_FREQUENCY as b } from "../../../../constants.js";
import { CapitalOfferLegalNotice as V } from "../CapitalOfferLegalNotice/CapitalOfferLegalNotice.js";
import G from "../../../../internal/InfoBox/InfoBox.js";
import { Translation as H } from "../../../../internal/Translation/Translation.js";
import U from "../../../../internal/StructuredList/StructuredList.js";
import { Icon as j } from "../../../../internal/Icon/Icon.js";
import { useConfigContext as J } from "../../../../../core/ConfigContext/context.js";
import { EMPTY_OBJECT as K } from "../../../../../utils/value/constants.js";
const Q = ["30_013"], A = {
  ...w,
  subCategory: "Business financing summary"
}, Ce = ({
  grantOffer: o,
  onBack: g,
  onFundsRequest: d,
  onContactSupport: y
}) => {
  const { i18n: e } = E(), n = q(), T = p(() => {
    const a = D(o.expectedRepaymentPeriodDays);
    return a ? e.date(a, { month: "long" }) : null;
  }, [o, e]), { requestFunds: k } = J().endpoints, r = B({
    queryFn: k,
    options: {
      onSuccess: (a) => d == null ? void 0 : d(a)
    }
  }), v = f(
    (a) => {
      r.mutate(K, { path: { grantOfferId: a } });
    },
    [r]
  ), x = f(() => {
    var a;
    try {
      o.id && v(o.id);
    } finally {
      (a = n.addEvent) == null || a.call(n, "Clicked button", { ...A, label: "Request funds" });
    }
  }, [o.id, v, n]), P = f(() => {
    var a;
    try {
      return g();
    } finally {
      (a = n.addEvent) == null || a.call(n, "Clicked button", { ...A, label: "Back to slider view" });
    }
  }, [g, n]), c = p(
    () => M(o.maximumRepaymentPeriodDays),
    [o.maximumRepaymentPeriodDays]
  ), i = p(() => {
    const a = r.error ? r.error : null;
    if (a && Q.includes(a.errorCode))
      switch (a.errorCode) {
        case "30_013":
          return {
            title: e.get("capital.offer.common.errors.noPrimaryAccount"),
            message: e.get("capital.offer.common.errors.cannotContinueSupport"),
            errorCode: "30_013"
          };
        default:
          return {
            title: e.get("common.errors.somethingWentWrong"),
            message: e.get("capital.offer.common.errors.unavailable")
          };
      }
    return null;
  }, [e, r.error]), I = p(() => {
    const a = [
      {
        key: "capital.common.fields.fees",
        value: e.amount(o.feesAmount.value, o.feesAmount.currency)
      },
      {
        key: "capital.common.fields.totalRepaymentAmount",
        value: e.amount(o.totalAmount.value, o.totalAmount.currency)
      },
      {
        key: "capital.common.fields.repaymentThreshold",
        value: e.amount(o.thresholdAmount.value, o.thresholdAmount.currency)
      },
      {
        key: "capital.common.fields.dailyRepaymentRate",
        value: e.get("capital.common.values.percentage", { values: { percentage: R(o.repaymentRate) } })
      },
      {
        key: "capital.common.fields.expectedRepaymentPeriod",
        value: e.get("capital.common.values.numberOfDays", { values: { days: o.expectedRepaymentPeriodDays } })
      },
      { key: "capital.common.fields.account", value: e.get("capital.common.values.primaryAccount") }
    ];
    return c && a.splice(4, 0, {
      key: "capital.common.fields.maximumRepaymentPeriod",
      value: c === 1 ? e.get("capital.common.values.oneMonth") : e.get("capital.common.values.numberOfMonths", { values: { months: c } })
    }), o.aprBasisPoints && a.splice(1, 0, {
      key: "capital.common.fields.annualPercentageRate",
      value: e.get("capital.common.values.percentage", { values: { percentage: R(o.aprBasisPoints) } })
    }), a;
  }, [o, e, c]);
  return L(A), !i && r.error ? /* @__PURE__ */ t(W, { error: r.error, onBack: P, onContactSupport: y }) : /* @__PURE__ */ t("div", { className: "adyen-pe-capital-offer-summary", children: [
    /* @__PURE__ */ t(G, { className: "adyen-pe-capital-offer-summary__grant-summary", children: [
      /* @__PURE__ */ t(m, { el: s.PARAGRAPH, variant: l.BODY, children: /* @__PURE__ */ t(
        H,
        {
          translationKey: "capital.offer.common.fundingRequestInfo",
          fills: {
            amount: /* @__PURE__ */ t("strong", { children: `${e.amount(o.grantAmount.value, o.grantAmount.currency, { minimumFractionDigits: 0 })}` })
          }
        }
      ) }),
      /* @__PURE__ */ t(m, { el: s.PARAGRAPH, variant: l.CAPTION, children: e.get("capital.offer.common.repaymentInfo", {
        values: {
          amount: e.amount(o.thresholdAmount.value, o.thresholdAmount.currency),
          days: b,
          date: T ?? ""
        }
      }) })
    ] }),
    /* @__PURE__ */ t(
      U,
      {
        classNames: "adyen-pe-capital-offer-summary__details",
        renderLabel: (a, u) => u === "capital.common.fields.repaymentThreshold" ? /* @__PURE__ */ t(
          _,
          {
            isUnderlineVisible: !0,
            content: e.get("capital.common.fields.repaymentThreshold.description", {
              values: { days: b }
            }),
            children: /* @__PURE__ */ t("span", { children: /* @__PURE__ */ t(
              m,
              {
                className: "adyen-pe-capital-offer-summary__list-label",
                el: s.SPAN,
                variant: l.CAPTION,
                children: a
              }
            ) })
          }
        ) : u === "capital.common.fields.annualPercentageRate" ? /* @__PURE__ */ t(_, { isUnderlineVisible: !0, content: e.get("capital.common.fields.annualPercentageRate.description"), children: /* @__PURE__ */ t("span", { children: /* @__PURE__ */ t(
          m,
          {
            className: "adyen-pe-capital-offer-summary__list-label",
            el: s.SPAN,
            variant: l.CAPTION,
            children: a
          }
        ) }) }) : /* @__PURE__ */ t(
          m,
          {
            className: "adyen-pe-capital-offer-summary__list-label",
            el: s.SPAN,
            variant: l.CAPTION,
            children: a
          }
        ),
        renderValue: (a, u) => {
          const C = u === "capital.common.fields.account" && r.error && i && i.errorCode === "30_013";
          return /* @__PURE__ */ t(
            m,
            {
              className: Y({
                "adyen-pe-capital-offer-summary__details--error": C
              }),
              el: s.SPAN,
              variant: l.CAPTION,
              stronger: !0,
              children: [
                C ? /* @__PURE__ */ t(j, { name: "warning-filled", "data-testid": "primary-account-warning-icon" }) : null,
                a
              ]
            }
          );
        },
        items: I
      }
    ),
    i && /* @__PURE__ */ t(
      F,
      {
        className: "adyen-pe-capital-offer-summary__error-alert",
        type: S.WARNING,
        title: i.title,
        description: i.message,
        children: y ? /* @__PURE__ */ t(h, { className: "adyen-pe-capital-offer-summary__error-alert-button", onClick: y, children: e.get("capital.common.actions.contactSupport") }) : null
      }
    ),
    /* @__PURE__ */ t(V, {}),
    /* @__PURE__ */ t("div", { className: "adyen-pe-capital-offer-summary__buttons", children: [
      r.error && !i ? null : /* @__PURE__ */ t(h, { variant: N.SECONDARY, onClick: P, children: e.get("capital.common.actions.goBack") }),
      /* @__PURE__ */ t(
        h,
        {
          variant: N.PRIMARY,
          state: r.isLoading ? "loading" : void 0,
          onClick: x,
          disabled: r.isLoading || !!r.error || !!r.data,
          "aria-label": e.get("capital.offer.summary.actions.requestFunds"),
          children: e.get(
            r.isLoading ? "capital.offer.summary.actions.requestFunds.states.loading" : "capital.offer.summary.actions.requestFunds"
          )
        }
      )
    ] })
  ] });
};
export {
  Ce as CapitalOfferSummary
};
