// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkDetails/components/PaymentLinkError/PaymentLinkError.js
import { jsx as a } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import s from "../../../../../core/Context/useCoreContext.js";
import { useCallback as m, useMemo as u } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { ButtonVariant as l } from "../../../../internal/Button/types.js";
import { ErrorMessageDisplay as c } from "../../../../internal/ErrorMessageDisplay/ErrorMessageDisplay.js";
import f from "../../../../internal/Button/Button.js";
import { getPaymentLinkErrorMessage as p } from "./getPaymentLinkErrorMessage.js";
const L = ({ error: o, onContactSupport: t, onDismiss: r }) => {
  const { i18n: e } = s(), n = m(() => /* @__PURE__ */ a(f, { variant: l.SECONDARY, onClick: r, children: e.get("payByLink.common.actions.goBack") }), [e, r]), i = u(
    () => p(o, "payByLink.details.errors.unavailable", t),
    [o, t]
  );
  return /* @__PURE__ */ a(
    c,
    {
      renderSecondaryButton: r ? n : void 0,
      withImage: !0,
      outlined: !1,
      absolutePosition: !1,
      withBackground: !1,
      ...i
    }
  );
};
export {
  L as PaymentLinkError
};
