// @adyen/adyen-platform-experience-web/dist/es/components/internal/ExpandableCard/ExpandableCard.js
import { jsx as s } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import a from "classnames";
import { useState as T, useMemo as x, useCallback as b, useRef as c, useLayoutEffect as z, useEffect as g } from "../../../external/preact/hooks/dist/hooks.module.js";
import { useClickOutside as U } from "../../../hooks/element/useClickOutside.js";
import { useUniqueId as V } from "../../../hooks/useUniqueId.js";
/* empty css                    */
import { CARD_HEIGHT_PROPERTY as y, BASE_CLASS as Y, CONTENT_CLASS as C, CONTENT_EXPANDABLE_CLASS as H, CHEVRON_CLASS as k, CONTAINER_CLASS as E, CONTAINER_BUTTON_CLASS as B, CONTAINER_FILLED_CLASS as A, CONTAINER_OVERLAY_CLASS as $, CONTAINER_IN_FLOW_CLASS as j, CONTAINER_HIDDEN_CLASS as G } from "./constants.js";
import { Fragment as D } from "../../../external/preact/dist/preact.module.js";
import F from "../BaseButton/BaseButton.js";
import { Icon as P } from "../Icon/Icon.js";
import { isFunction as M } from "../../../utils/value/is.js";
const ae = ({ renderContent: o, children: O, filled: d, fullWidth: S, inFlow: _, ...l }) => {
  const [r, v] = T(!1), [R, h] = T(0), n = x(() => _ === !0, [_]), i = b(() => v((e) => !e), [v]), L = c(null), u = c(null), f = `elem-${V()}`, m = c(!1), I = c(r), p = U(
    void 0,
    b(() => {
      r && (i(), m.current = !0);
    }, [r, i])
  ), N = x(
    () => M(o) ? o : ({ collapsibleContent: e }) => /* @__PURE__ */ s(D, { children: [
      o,
      e && /* @__PURE__ */ s("div", { children: e })
    ] }),
    [o]
  );
  return z(() => {
    const e = L.current;
    e && (n ? e.style.setProperty(y, `${R}px`) : r || e.style.removeProperty(y));
  }, [R, n, r]), g(() => {
    if (!n) return void h(0);
    const e = u.current;
    if (!e) return;
    const t = new ResizeObserver((q) => {
      for (const w of q)
        w.target === e && h(e.offsetHeight || 0);
    });
    return t.observe(e), () => {
      t.unobserve(e), t.disconnect();
    };
  }, [n]), g(() => {
    var e, t;
    r ? (e = p.current) == null || e.focus() : (I.current !== r && !m.current && ((t = u.current) == null || t.focus()), m.current = !1), I.current = r;
  }, [r, p]), /* @__PURE__ */ s("div", { ref: L, className: Y, children: O ? /* @__PURE__ */ s(D, { children: [
    /* @__PURE__ */ s(
      F,
      {
        className: a(E, B, { [A]: d }),
        disabled: r,
        fullWidth: S,
        onClick: i,
        ref: u,
        "data-testid": "expand-button",
        ...r ? { "aria-hidden": !0 } : { "aria-controls": f, "aria-expanded": !1 },
        ...l,
        children: [
          /* @__PURE__ */ s("div", { className: a(C, H), children: N({ isExpanded: r }) }),
          /* @__PURE__ */ s("div", { className: k, children: /* @__PURE__ */ s(P, { name: "chevron-down" }) })
        ]
      }
    ),
    /* @__PURE__ */ s(
      F,
      {
        id: f,
        className: a(E, B, $, {
          [A]: d,
          [G]: !r,
          [j]: n
        }),
        disabled: !r,
        fullWidth: S,
        onClick: i,
        ref: p,
        "data-testid": "collapse-button",
        ...r ? { "aria-controls": f, "aria-expanded": !0 } : { role: "presentation" },
        ...l,
        children: [
          /* @__PURE__ */ s("div", { className: a(C, H), children: N({ isExpanded: r, collapsibleContent: O }) }),
          /* @__PURE__ */ s("div", { className: k, children: /* @__PURE__ */ s(P, { name: "chevron-up" }) })
        ]
      }
    )
  ] }) : /* @__PURE__ */ s("div", { className: a(E, { [A]: d }), ...l, children: /* @__PURE__ */ s("div", { className: C, children: N({ isExpanded: !1 }) }) }) });
};
export {
  ae as default
};
