// @adyen/adyen-platform-experience-web/dist/es/components/internal/Calendar/components/CalendarGrid/CalendarGridDayOfWeek.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { memo as y } from "../../../../../external/preact/compat/dist/compat.module.js";
import { propsProperty as k, property as o } from "./utils.js";
import { CalendarGridRenderToken as L } from "../../types.js";
import { memoComparator as n } from "../../../../../utils/preact/memoComparator/main.js";
import { EMPTY_OBJECT as d } from "../../../../../utils/value/constants.js";
import { getClassName as m } from "../../../../../utils/preact/className.js";
const i = "adyen-pe-calendar__cell adyen-pe-calendar__cell--day-of-week", p = "adyen-pe-calendar__day-of-week", u = (a = d, e) => {
  const s = k.unwrapped(
    {
      childClassName: o.mutable(p),
      childProps: {
        children: o.restricted(),
        className: ""
      },
      className: o.mutable(i),
      props: {
        ...a,
        children: o.restricted(),
        className: ""
      }
    },
    !0
  );
  return e == null || e(L.DAY_OF_WEEK, s), s;
}, A = ({ prepare: a, flags: e, labels: { long: s, short: _ } }) => {
  const l = {
    "aria-label": s,
    "data-first-week-day": e.LINE_START,
    "data-last-week-day": e.LINE_END,
    "data-weekend": e.WEEKEND,
    scope: "col"
  }, r = u(l, a), { children: P, className: N, onCommand: D, ...c } = r.props || d, C = m(r.className, i, N), {
    children: b,
    className: E,
    ...f
  } = r.childProps || d, h = m(r.childClassName, p, E);
  return /* @__PURE__ */ t("th", { ...c, ref: c.ref, ...l, className: C, children: /* @__PURE__ */ t("abbr", { ...f, className: h, children: _ }) });
}, M = y(
  A,
  n({
    block: n.exclude,
    flags: (a) => +a
  })
);
export {
  M as default
};
