// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputeManagement/components/AcceptDisputeFlow/AcceptDisputeFlow.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import p from "../../../../internal/Typography/Typography.js";
import { TypographyElement as h, TypographyVariant as m } from "../../../../internal/Typography/types.js";
import { useModalContext as w } from "../../../../internal/Modal/Modal.js";
import P from "../../../../../core/Context/useCoreContext.js";
import V from "../../../../internal/Button/ButtonActions/ButtonActions.js";
import { ButtonVariant as y } from "../../../../internal/Button/types.js";
import { useRef as a, useState as q, useCallback as _, useMemo as H, useEffect as j } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { useDisputeFlow as J } from "../../context/dispute/context.js";
import Q from "../../../../../hooks/useMutation/useMutation.js";
/* empty css                       */
import { Icon as u } from "../../../../internal/Icon/Icon.js";
import { Fragment as D } from "../../../../../external/preact/dist/preact.module.js";
import U from "../../../../internal/Button/Button.js";
import { useConfigContext as z } from "../../../../../core/ConfigContext/context.js";
import { EMPTY_OBJECT as G } from "../../../../../utils/value/constants.js";
import { uniqueId as K } from "../../../../../utils/random/uniqueId.js";
import { isFunction as W } from "../../../../../utils/value/is.js";
const he = ({ onDisputeAccept: d }) => {
  const { i18n: t } = P(), { acceptDispute: R } = z().endpoints, { dispute: T, clearStates: k, goBack: l } = J(), { withinModal: M } = w(), n = a(T).current ?? T, O = M ? h.H2 : h.DIV, i = n == null ? void 0 : n.dispute.pspReference, b = (n == null ? void 0 : n.dispute.type) === "REQUEST_FOR_INFORMATION", v = a("disputes.management.accept.chargeback.accepted"), I = a("disputes.management.accept.chargeback.disclaimer"), N = a("disputes.management.accept.chargeback.title"), A = a("disputes.management.accept.chargeback.actions.accept"), g = a(b);
  (g.current || (g.current = b)) && (v.current = "disputes.management.accept.requestForInformation.accepted", I.current = "disputes.management.accept.requestForInformation.disclaimer", N.current = "disputes.management.accept.requestForInformation.title", A.current = "disputes.management.accept.requestForInformation.actions.accept");
  const [C, L] = q(!1), [c, S] = q(!1), s = Q({
    queryFn: R,
    options: {
      onSuccess: _(() => {
        k(), S(!0);
      }, [k])
    }
  }), r = s.isLoading || c, o = C && !r, F = _(() => {
    o && s.mutate(G, { path: { disputePspReference: i } });
  }, [o, i, s]), x = _(() => {
    r || L((f) => !f);
  }, [r]), Y = H(() => [
    {
      title: t.get(A.current),
      disabled: !o,
      state: s.isLoading ? "loading" : "default",
      variant: y.PRIMARY,
      event: F,
      classNames: c ? ["adyen-pe-accept-dispute__accepted-btn"] : void 0,
      renderTitle: (f) => c ? /* @__PURE__ */ e(D, { children: [
        /* @__PURE__ */ e(u, { name: "checkmark-circle-fill", className: "adyen-pe-accept-dispute__accepted-icon" }),
        t.get("disputes.management.accept.common.accepted")
      ] }) : f
    },
    {
      title: t.get("disputes.management.common.actions.goBack"),
      disabled: s.isLoading,
      variant: y.SECONDARY,
      event: l
    }
  ], [t, F, s.isLoading, o, l, c]), B = a(!1), E = a(K()).current;
  return j(() => {
    B.current || c && i && W(d) && (B.current = !0, d({ id: i }));
  }, [c, i, d]), /* @__PURE__ */ e("div", { className: "adyen-pe-accept-dispute__container", children: c ? /* @__PURE__ */ e("div", { className: "adyen-pe-accept-dispute__success", children: [
    /* @__PURE__ */ e(u, { name: "checkmark-circle-fill", className: "adyen-pe-accept-dispute__success-icon" }),
    /* @__PURE__ */ e(p, { variant: m.TITLE, children: t.get(v.current) }),
    /* @__PURE__ */ e(U, { variant: y.SECONDARY, onClick: l, children: t.get("disputes.management.common.actions.showDetails") })
  ] }) : /* @__PURE__ */ e(D, { children: [
    /* @__PURE__ */ e(p, { className: "adyen-pe-accept-dispute__title", el: O, variant: m.TITLE, medium: !0, children: t.get(N.current) }),
    /* @__PURE__ */ e(p, { variant: m.BODY, medium: !0, children: t.get(I.current) }),
    /* @__PURE__ */ e("div", { className: "adyen-pe-accept-dispute__input", children: [
      /* @__PURE__ */ e(
        "input",
        {
          type: "checkbox",
          disabled: r,
          className: "adyen-pe-visually-hidden",
          id: E,
          onInput: x
        }
      ),
      /* @__PURE__ */ e("label", { className: "adyen-pe-accept-dispute__label", htmlFor: E, children: [
        C ? /* @__PURE__ */ e(u, { name: "checkmark-square-fill" }) : /* @__PURE__ */ e(u, { name: "square" }),
        /* @__PURE__ */ e(p, { el: h.SPAN, variant: m.BODY, children: t.get("disputes.management.accept.common.agree") })
      ] })
    ] }),
    /* @__PURE__ */ e("div", { className: "adyen-pe-accept-dispute__actions", children: /* @__PURE__ */ e(V, { actions: Y }) })
  ] }) });
};
export {
  he as AcceptDisputeFlow
};
