// @adyen/adyen-platform-experience-web/dist/es/components/internal/DatePicker/DatePicker.js
import { jsx as n } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import _ from "classnames";
import { forwardRef as p } from "../../../external/preact/compat/dist/compat.module.js";
import { useState as O, useMemo as a, useCallback as g } from "../../../external/preact/hooks/dist/hooks.module.js";
import v from "../../../core/Context/useCoreContext.js";
import I from "../../../hooks/useReflex.js";
import N from "../Calendar/hooks/useTimezone.js";
import { DEFAULT_FIRST_WEEK_DAY as D } from "../Calendar/calendar/timerange/presets/shared/offsetWeek.js";
import L from "../Calendar/hooks/useCalendarControlsRendering.js";
/* empty css                */
import M from "./components/TimeRangeSelector/TimeRangeSelector.js";
import P from "../Calendar/Calendar.js";
import u from "../Calendar/calendar/facade/index.js";
import { boolOrTrue as S } from "../../../utils/value/bool.js";
import { EMPTY_OBJECT as x } from "../../../utils/value/constants.js";
import { noop as E } from "../../../utils/common.js";
const $ = p((e, T) => {
  const { i18n: r } = v(), [C, k] = L(e.renderControl), [y, m] = O(performance.now()), t = a(() => S(e.showTimezoneInfo), [e.showTimezoneInfo]), { clockTime: l, GMTOffset: c } = N({ timezone: e.timezone, withClock: t }), z = a(() => _([{ "adyen-pe-datepicker--with-timezone": t }, "adyen-pe-datepicker"]), [t]), R = a(() => t ? { values: { offset: c, time: l } } : x, [c, l, t]), o = I(E, T), w = g(
    (i) => r.get("common.filters.types.date.calendar.label", { values: { monthOfYear: i.label } }),
    [r]
  ), b = g(() => {
    var i, s, d, f, h;
    m(performance.now()), (i = o.current) != null && i.from && ((s = o.current) != null && s.to) && ((h = e.onHighlight) == null || h.call(e, +((d = o.current) == null ? void 0 : d.from), +((f = o.current) == null ? void 0 : f.to)));
  }, [m, e.onHighlight]);
  return /* @__PURE__ */ n("div", { className: z, children: [
    /* @__PURE__ */ n("div", { className: "adyen-pe-datepicker__selector-container", children: /* @__PURE__ */ n(
      M,
      {
        now: e.now,
        calendarRef: o,
        onTimeRangeSelected: e.onPresetOptionSelected,
        options: e.timeRangePresetOptions,
        selectedOption: e.selectedPresetOption,
        timestamp: y,
        timezone: e.timezone,
        "aria-label": e.timeRangeSelectorLabel
      }
    ) }),
    /* @__PURE__ */ n(
      "div",
      {
        ref: k,
        role: "group",
        className: "adyen-pe-datepicker__controls",
        "aria-label": r.get("common.filters.types.date.calendar.navigation.label")
      }
    ),
    /* @__PURE__ */ n(
      P,
      {
        ...e,
        ref: o,
        firstWeekDay: D,
        dynamicBlockRows: !0,
        onlyCellsWithin: !0,
        controls: e.controls ?? u.controls.MINIMAL,
        highlight: e.highlight ?? u.highlight.MANY,
        getGridLabel: w,
        onHighlight: b,
        renderControl: C,
        trackCurrentDay: !0
      }
    ),
    t && /* @__PURE__ */ n("div", { className: "adyen-pe-datepicker__timezone", children: r.get("common.filters.types.date.timezoneInfo", R) })
  ] });
});
export {
  $ as default
};
