// @adyen/adyen-platform-experience-web/dist/es/components/internal/Accordion/Accordion.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import N from "classnames";
import { useState as E, useRef as O, useCallback as p } from "../../../external/preact/hooks/dist/hooks.module.js";
import { useUniqueId as S } from "../../../hooks/useUniqueId.js";
import { ACCORDION_HEADER_CLASS as _, ACCORDION_HEADER_CONTAINER_CLASS as f, ACCORDION_HEADER_CONTROLLER_CLASS as I, ACCORDION_CONTENT_CLASS as R, ACCORDION_BASE_CLASS as h } from "./constants.js";
/* empty css               */
import { Icon as a } from "../Icon/Icon.js";
function T({ children: l, classNames: d, header: m, headerInformation: r }) {
  var s;
  const [n, C] = E(!1), o = O(null), A = p(() => C((u) => !u), []), c = S(), t = `accordion-content-${c}`, i = `accordion-controller-${c}`;
  return /* @__PURE__ */ e("div", { className: N(h, d), children: [
    /* @__PURE__ */ e("div", { className: _, children: [
      /* @__PURE__ */ e(
        "button",
        {
          id: i,
          "aria-controls": t,
          className: f,
          onClick: A,
          "aria-expanded": n,
          children: /* @__PURE__ */ e("div", { className: I, children: [
            m,
            n ? /* @__PURE__ */ e(a, { name: "chevron-up" }) : /* @__PURE__ */ e(a, { name: "chevron-down" })
          ] })
        }
      ),
      r && /* @__PURE__ */ e("div", { children: r })
    ] }),
    /* @__PURE__ */ e(
      "div",
      {
        role: "region",
        id: t,
        "aria-labelledby": i,
        style: { maxHeight: n ? (s = o == null ? void 0 : o.current) == null ? void 0 : s.offsetHeight : 0 },
        className: R,
        children: /* @__PURE__ */ e("div", { ref: o, children: l })
      }
    )
  ] });
}
export {
  T as default
};
