// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkDetails/components/PaymentLinkActivity/PaymentLinkActivity.js
import { jsx as r } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as p } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import a from "../../../../../core/Context/useCoreContext.js";
import { TimelineItem as f } from "../../../../internal/Timeline/components/TimelineItem.js";
import l from "../../../../internal/Timeline/Timeline.js";
import { TimelineDateFormat as T } from "../../../../internal/Timeline/types.js";
import { getStatus as u, getDescriptionKey as d, getTitleKey as g } from "./utils.js";
const E = ({ activities: t }) => {
  const { i18n: m } = a(), i = p(
    () => t.map((e) => ({
      titleKey: g(e),
      descriptionKey: d(e),
      date: e.date,
      status: u(e)
    })),
    [t]
  );
  return /* @__PURE__ */ r(l, { children: i.map(({ titleKey: e, date: o, status: n, descriptionKey: s }) => /* @__PURE__ */ r(
    f,
    {
      title: m.get(e),
      timestamp: {
        date: new Date(o),
        format: T.FULL_DATE_EXACT_TIME
      },
      status: n,
      description: m.get(s)
    },
    o
  )) });
};
export {
  E as PaymentLinkActivity
};
