// @adyen/adyen-platform-experience-web/dist/es/components/internal/ProgressBar/ProgressBar.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as u } from "../../../external/preact/hooks/dist/hooks.module.js";
import _ from "classnames";
/* empty css                 */
import o from "../Typography/Typography.js";
import { TypographyVariant as m, TypographyElement as p } from "../Typography/types.js";
import { Tooltip as h } from "../Tooltip/Tooltip.js";
import { clamp as s } from "../../../utils/value/number.js";
const A = ({ max: c = 1, value: n, labels: r, tooltips: a, className: i }) => {
  const t = u(() => s(0, n * 100 / c, 100), [n, c]), g = !!(r != null && r.current || r != null && r.max), y = (r == null ? void 0 : r.ariaLabel) ?? (r != null && r.current ? `${r.current}: ${n}` : `${n}/${c}`);
  return /* @__PURE__ */ e(
    "div",
    {
      role: "progressbar",
      "aria-valuenow": n,
      "aria-valuemin": 0,
      "aria-valuemax": c,
      "aria-valuetext": `${t}%`,
      "aria-label": y,
      className: _("adyen-pe-progress-bar", i),
      children: [
        /* @__PURE__ */ e("div", { className: "adyen-pe-progress-bar__track", role: "presentation", children: [
          /* @__PURE__ */ e("div", { className: "adyen-pe-progress-bar__track-background" }),
          /* @__PURE__ */ e(
            d,
            {
              tooltipContent: a == null ? void 0 : a.progress,
              title: r == null ? void 0 : r.current,
              percentage: t,
              className: "adyen-pe-progress-bar__track-fill"
            }
          ),
          /* @__PURE__ */ e(
            d,
            {
              tooltipContent: a == null ? void 0 : a.remaining,
              title: r == null ? void 0 : r.max,
              percentage: 100 - t,
              className: "adyen-pe-progress-bar__track-remaining"
            }
          )
        ] }),
        g && /* @__PURE__ */ e("div", { className: "adyen-pe-progress-bar__legend", "aria-hidden": "true", children: [
          r.current && /* @__PURE__ */ e(o, { el: p.SPAN, variant: m.CAPTION, className: "adyen-pe-progress-bar__legend-label", children: r.current }),
          r.max && /* @__PURE__ */ e(o, { el: p.SPAN, variant: m.CAPTION, className: "adyen-pe-progress-bar__legend-label", children: r.max })
        ] })
      ]
    }
  );
}, d = ({ tooltipContent: c, title: n, percentage: r, className: a }) => {
  const i = (t) => /* @__PURE__ */ e("div", { className: a, "aria-label": t, style: { width: `${r}%` } });
  return c ? /* @__PURE__ */ e(h, { content: c, children: i(n) }) : i(n);
};
export {
  A as default
};
