// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/PaymentDetailsForm/Fields/AmountField.js
import { jsx as s } from "../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import N from "../../../../../../../core/Context/useCoreContext.js";
import { useMemo as l, useCallback as v, useEffect as E } from "../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { useWizardFormContext as V } from "../../../../../../../hooks/form/wizard/WizardFormContext.js";
import { CurrencyInput as A } from "../../../../../../internal/FormFields/CurrencyInput/CurrencyInput.js";
import { VisibleField as x } from "../../../../../../internal/FormWrappers/VisibleField.js";
import L from "../../../../../../internal/FormWrappers/FormField.js";
import { FieldError as M } from "../../../../../../internal/FormFields/FieldError/FieldError.js";
import { Controller as k } from "../../../../../../../hooks/form/Controller.js";
import { Fragment as q } from "../../../../../../../external/preact/dist/preact.module.js";
const i = "amount.value", n = "amount.currency", _ = 1e13, X = () => {
  const { i18n: m } = N(), { control: f, setValue: c, getValues: u, fieldsConfig: r, trigger: d } = V(), p = l(() => {
    var e;
    return ((e = r == null ? void 0 : r[n]) == null ? void 0 : e.visible) ?? !1;
  }, [r]), C = v(
    (e) => Number(e) < 0 ? { valid: !1, message: m.get("payByLink.creation.fields.amountValue.error.negativeNumber") } : p && !u(n) ? { valid: !1, message: m.get("payByLink.creation.fields.amountValue.error.currency") } : !e || Number(e) === 0 ? { valid: !1, message: m.get("common.errors.fieldRequired") } : { valid: !0 },
    [m, u, p]
  ), b = v(
    (e, o) => {
      c(n, e), o && d(i);
    },
    [c, d, u]
  ), F = l(() => {
    var e;
    return (e = r == null ? void 0 : r.amount) == null ? void 0 : e.required;
  }, [r]), t = l(() => {
    var o;
    const e = (o = r == null ? void 0 : r[n]) == null ? void 0 : o.options;
    return e == null ? void 0 : e.map((a) => ({ id: a, name: a }));
  }, [r]);
  return E(() => {
    var e;
    (t == null ? void 0 : t.length) === 1 && c(n, (e = t[0]) == null ? void 0 : e.id);
  }, [t]), /* @__PURE__ */ s(x, { name: i, children: /* @__PURE__ */ s(L, { label: m.get("payByLink.creation.fields.amount.label"), optional: !1, supportText: void 0, className: void 0, children: /* @__PURE__ */ s(
    k,
    {
      name: i,
      control: f,
      rules: {
        validate: C,
        required: F
      },
      render: ({ field: e, fieldState: o }) => {
        var g;
        const a = !!o.error && o.isTouched, y = (g = o.error) == null ? void 0 : g.message;
        return /* @__PURE__ */ s(q, { children: [
          /* @__PURE__ */ s(
            A,
            {
              ...e,
              selectedCurrencyCode: u(n),
              onCurrencyChange: (h) => b(h, a),
              currency: u(n),
              currencyItems: t,
              isInvalid: a,
              name: i,
              amount: e.value ? Number(e.value) : void 0,
              onAmountChange: e.onInput,
              maxValue: _
            }
          ),
          a && y && /* @__PURE__ */ s(M, { errorMessage: y })
        ] });
      }
    }
  ) }) });
};
export {
  X as AmountField
};
