// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/PaymentDetailsForm/Fields/ValidityField.js
import { jsx as o } from "../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import H from "../../../../../../../core/Context/useCoreContext.js";
import { useState as D, useCallback as d, useMemo as M, useEffect as J } from "../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { useWizardFormContext as Q } from "../../../../../../../hooks/form/wizard/WizardFormContext.js";
import { VisibleField as Z } from "../../../../../../internal/FormWrappers/VisibleField.js";
import E from "../../../../../../internal/FormWrappers/FormField.js";
import { LINK_VALIDITY_DURATION_UNITS as F } from "../../../../constants.js";
import ii from "../../../../../../internal/FormFields/InputBase.js";
import { FieldError as ti } from "../../../../../../internal/FormFields/FieldError/FieldError.js";
import { transformToMS as S } from "../../../../../../../utils/datetime/main.js";
import { Controller as R } from "../../../../../../../hooks/form/Controller.js";
import ni from "../../../../../../internal/FormFields/Select/Select.js";
const Y = 70, u = "flexible", fi = ({ configuration: p }) => {
  var _, b;
  const [z, V] = D(""), [O, g] = D(void 0), [f, I] = D(""), { i18n: e } = H(), { control: q, fieldsConfig: L, setValue: l, getValues: k, trigger: v } = Q(), a = d(
    () => [k("linkValidity.durationUnit"), k("linkValidity.quantity")],
    [k]
  ), s = M(() => {
    var n;
    return (((n = p == null ? void 0 : p.linkValidity) == null ? void 0 : n.options) ?? []).map(({ quantity: t, durationUnit: r, type: y }) => {
      if (y === u)
        return { id: u, name: e.get("payByLink.creation.fields.validity.linkValidityUnit.custom") };
      const m = `payByLink.creation.fields.validity.linkValidityUnit.${r}`;
      return { id: `${t} ${r}` || "", name: e.get(m, { values: { quantity: t }, count: t }) };
    });
  }, [p, e]), U = d(() => {
    var t;
    if (!s.length) return;
    const [i, n] = a();
    if (!n || !i) {
      const [r, y] = `${(t = s[0]) == null ? void 0 : t.id}`.split(" ");
      l("linkValidity.quantity", r), l("linkValidity.durationUnit", y);
    }
  }, [s, l, a]), h = d(() => {
    const [i, n] = a();
    return !n || !i ? s[0] : s.find((t) => t.id === `${n} ${i}`) || { id: u };
  }, [s, a]);
  J(() => {
    var t;
    const [i, n] = a();
    V(i || ""), g(n || ""), I(((t = h()) == null ? void 0 : t.id) || ""), U();
  }, [a, s, h, U]);
  const C = (_ = L["linkValidity.durationUnit"]) == null ? void 0 : _.required, B = (b = L["linkValidity.quantity"]) == null ? void 0 : b.required, X = d(
    (i) => {
      var r;
      const n = (r = i.target) == null ? void 0 : r.value, t = parseInt(n, 10);
      l("linkValidity.quantity", t), g(t), v("linkValidity.durationUnit");
    },
    [l, v]
  ), j = d(
    (i) => {
      l("linkValidity.durationUnit", i), V(i), v("linkValidity.quantity");
    },
    [l, v]
  ), K = M(
    () => F.map((i) => ({
      id: i,
      name: e.get(`payByLink.creation.fields.validity.linkValidityUnit.${i}__plural`)
    })),
    [e]
  ), T = d(() => {
    if (f !== u) return { valid: !0 };
    const [i, n] = a(), t = parseInt(n, 10);
    return n ? isNaN(t) || t <= 0 ? { valid: !1, message: e.get("payByLink.creation.fields.validity.customDuration.error.invalidDurationValue") } : i ? S(i, t) > S("day", Y) ? {
      valid: !1,
      message: e.get("payByLink.creation.fields.validity.customDuration.error.durationTooLong", {
        values: { maxDays: Y }
      })
    } : { valid: !0 } : { valid: !1, message: e.get("payByLink.creation.fields.validity.customDuration.error.missingDurationUnit") } : { valid: !1, message: e.get("payByLink.creation.fields.validity.customDuration.error.missingDurationValue") };
  }, [f, e, a]);
  return /* @__PURE__ */ o(Z, { name: "linkValidity.durationUnit", children: /* @__PURE__ */ o(
    R,
    {
      name: "linkValidity.durationUnit",
      control: q,
      rules: {
        required: C,
        validate: T
      },
      render: ({ field: i, fieldState: n }) => /* @__PURE__ */ o(
        R,
        {
          name: "linkValidity.quantity",
          control: q,
          rules: { required: B, validate: T },
          render: ({ field: t, fieldState: r }) => {
            var N, w;
            const y = (P) => {
              var A;
              const c = (A = P.target) == null ? void 0 : A.value;
              if (c !== u) {
                const [W, G] = (c == null ? void 0 : c.split(" ")) || [];
                t.onInput(W), i.onInput(G), t.triggerValidation(), i.triggerValidation();
              } else
                i.onInput(""), t.onInput(""), g(void 0), V("");
              I(c);
            }, m = (r.error || n.error) && r.isTouched && n.isTouched, x = !r.error || !n.error, $ = ((N = r.error) == null ? void 0 : N.message) || ((w = n.error) == null ? void 0 : w.message);
            return /* @__PURE__ */ o("div", { children: [
              /* @__PURE__ */ o("div", { className: "adyen-pe-payment-link-creation-form__validity-container", children: [
                /* @__PURE__ */ o(
                  E,
                  {
                    label: e.get("payByLink.creation.fields.validity.label"),
                    supportText: e.get("payByLink.creation.fields.validity.supportText"),
                    optional: !C && !B,
                    children: /* @__PURE__ */ o(
                      ni,
                      {
                        selected: f,
                        onChange: y,
                        items: s,
                        isValid: x,
                        isInvalid: m
                      }
                    )
                  }
                ),
                f === u && /* @__PURE__ */ o(E, { label: e.get("payByLink.creation.fields.validity.customDuration.label"), optional: !1, children: /* @__PURE__ */ o(
                  ii,
                  {
                    ...t,
                    dropdown: { items: K, value: z || "", name: i.name },
                    onDropdownInput: j,
                    dropdownPosition: "end",
                    value: O,
                    type: "number",
                    onInput: X,
                    isValid: x,
                    isInvalid: m
                  }
                ) })
              ] }),
              m && $ && /* @__PURE__ */ o(ti, { errorMessage: $, withTopMargin: !0 })
            ] });
          }
        }
      )
    }
  ) });
};
export {
  fi as ValidityField
};
