// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/SummaryItem/SummaryItem.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { AmountSkeleton as b } from "../AmountSkeleton/AmountSkeleton.js";
import { BASE_CLASS as S, PLACEHOLDER_CLASS as f, AMOUNT_CLASS as T, LABEL_CLASS as E, BODY_CLASS as N } from "./constants.js";
import { SummaryItemLabel as h } from "./SummaryItemLabel.js";
import { Tooltip as B } from "../../../../internal/Tooltip/Tooltip.js";
import { TypographyVariant as n, TypographyElement as I } from "../../../../internal/Typography/types.js";
import O from "../../../../internal/Typography/Typography.js";
import _ from "../../../../../core/Context/useCoreContext.js";
import p from "classnames";
import { useEffect as x, useCallback as D } from "../../../../../external/preact/hooks/dist/hooks.module.js";
/* empty css                 */
import { useResponsiveContainer as H, containerQueries as M } from "../../../../../hooks/useResponsiveContainer.js";
const $ = ({
  columnConfigs: i,
  isHeader: u = !1,
  showLabelUnderline: d = !1,
  isSkeletonVisible: o = !1,
  isLoading: C = !1,
  widths: m,
  onWidthsSet: L,
  isEmpty: A
}) => {
  const { i18n: s } = _();
  x(() => {
    const e = i.map((r) => {
      var a, l;
      return ((l = (a = r.ref) == null ? void 0 : a.current) == null ? void 0 : l.getBoundingClientRect().width) ?? 0;
    });
    L(e);
  }, [L]);
  const c = (e) => ({ width: m && m[e] ? m[e] : "auto" }), y = H(M.only.xs), v = D(
    (e, r) => e.valueHasLabelStyle ? n.CAPTION : r && !y ? n.BODY : n.TITLE,
    [y]
  );
  return /* @__PURE__ */ t("div", { className: p(S, { [N]: !u }), children: i.map((e, r) => {
    const a = e.getValue(), l = !!a && a.length > 15;
    return /* @__PURE__ */ t("div", { children: [
      u && /* @__PURE__ */ t("div", { role: "presentation", children: e.tooltipLabel ? /* @__PURE__ */ t(B, { content: s.get(`${e.tooltipLabel}`), isUnderlineVisible: d, children: /* @__PURE__ */ t(h, { config: e, i18n: s, isSkeletonVisible: o }) }) : /* @__PURE__ */ t(h, { config: e, i18n: s, isSkeletonVisible: o }) }),
      o ? /* @__PURE__ */ t(b, { isLoading: C, hasMargin: e.hasSkeletonMargin, width: e.skeletonWidth + "px" }) : A ? /* @__PURE__ */ t("span", { className: p([S, f]) }) : /* @__PURE__ */ t("div", { id: e.elemId, ref: e.ref, style: c(r), "aria-label": e.ariaLabel, role: "presentation", children: /* @__PURE__ */ t(
        O,
        {
          el: I.SPAN,
          variant: v(e, l),
          className: p({ [E]: e.valueHasLabelStyle, [T]: !e.valueHasLabelStyle }),
          children: a
        }
      ) })
    ] }, r);
  }) });
};
export {
  $ as SummaryItem
};
