// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/components/GrantActions/GrantActions.js
import { jsx as o } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as h, useState as E, useCallback as L } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import M from "../../../../../core/Context/useCoreContext.js";
import R from "../../../../../core/Context/analytics/useAnalyticsContext.js";
import O from "../../../../../hooks/useTimezoneAwareDateFormatting.js";
import { GRANT_ACTION_CLASS_NAMES as n } from "./constants.js";
/* empty css                  */
import { Alert as y } from "../../../../internal/Alert/Alert.js";
import { AlertTypeOption as b } from "../../../../internal/Alert/types.js";
import { setTopWindowHref as _, getTopWindowHref as B } from "./utils.js";
import { ButtonVariant as w } from "../../../../internal/Button/types.js";
import G from "../../../../../hooks/useMutation/useMutation.js";
import K from "../../../../internal/Typography/Typography.js";
import { TypographyVariant as W, TypographyElement as x } from "../../../../internal/Typography/types.js";
import { sharedCapitalOverviewAnalyticsEventProperties as F } from "../../constants.js";
import { DATE_FORMAT_MISSING_ACTION as D } from "../../../../../constants/dateFormats.js";
import f from "../../../../internal/Button/Button.js";
import { useConfigContext as Y } from "../../../../../core/ConfigContext/context.js";
import { EMPTY_OBJECT as q } from "../../../../../utils/value/constants.js";
const st = ({
  missingActions: a = [],
  offerExpiresAt: p,
  className: d
}) => {
  const { i18n: e, updateCore: I } = M(), { dateFormat: T } = O(), { endpoints: C } = Y(), l = R(), A = h(
    () => ({
      signToS: {
        getTitle: (t) => t ? e.get("capital.overview.grants.item.alerts.signTermsAndConditionsBy", {
          values: { date: t }
        }) : e.get("capital.overview.grants.item.alerts.signTermsAndConditions"),
        buttonLabelKey: "capital.overview.grants.item.actions.viewTermsAndConditions",
        eventLabel: "Go to terms & conditions button clicked"
      },
      AnaCredit: {
        getTitle: (t) => t ? e.get("capital.overview.grants.item.alerts.actionNeededBy", {
          values: { date: t }
        }) : e.get("capital.overview.grants.item.alerts.actionNeeded"),
        buttonLabelKey: "capital.overview.grants.item.actions.submitInformation",
        eventLabel: "Submit information for AnaCredit button"
      }
    }),
    [e]
  ), [u, s] = E(null), S = L((t) => {
    t != null && t.url ? _(t.url) : s(null);
  }, []), c = G({
    queryFn: (t) => {
      let i = null;
      switch (t) {
        case "signToS":
          i = C.signToSActionDetails;
          break;
        case "AnaCredit":
          i = C.anaCreditActionDetails;
          break;
        default:
          break;
      }
      const r = i, k = {
        query: { redirectUrl: B(), locale: e.locale }
      };
      return r == null ? void 0 : r(q, k);
    },
    options: {
      onSuccess: S,
      // Reset the loading state when the mutation finishes (success or error)
      onError: () => {
        s(null);
      }
    }
  }), g = h(
    () => p ? T(p, D) : void 0,
    [T, p]
  ), N = L(
    (t) => {
      var i;
      (i = l.addEvent) == null || i.call(l, "Clicked link", {
        ...F,
        subCategory: "Missing action",
        label: t
      });
    },
    [l]
  );
  if (c.error)
    return /* @__PURE__ */ o(
      y,
      {
        className: d,
        type: b.CRITICAL,
        title: e.get("capital.overview.grants.item.alerts.somethingWentWrong"),
        description: /* @__PURE__ */ o(f, { className: n.button, onClick: I, children: e.get("common.actions.refresh.labels.default") })
      }
    );
  if (!a.length)
    return null;
  if (a.length > 1)
    return /* @__PURE__ */ o(
      y,
      {
        className: d,
        type: b.WARNING,
        title: e.get("capital.overview.grants.item.alerts.actionNeededMany"),
        description: /* @__PURE__ */ o("div", { className: n.actionsInformation, children: [
          /* @__PURE__ */ o("ol", { className: n.actionsContainer, children: a.map((t) => {
            const i = A[t.type], r = u === t.type;
            return /* @__PURE__ */ o("li", { children: /* @__PURE__ */ o(
              f,
              {
                className: n.button,
                onClick: () => {
                  try {
                    s(t.type), c.mutate(t.type);
                  } finally {
                    N(i.eventLabel);
                  }
                },
                disabled: r || c.isLoading,
                state: r ? "loading" : void 0,
                variant: w.TERTIARY,
                "aria-label": e.get(i.buttonLabelKey),
                children: /* @__PURE__ */ o("span", { className: n.buttonLabel, children: e.get(i.buttonLabelKey) })
              }
            ) }, t.type);
          }) }),
          g ? /* @__PURE__ */ o(K, { el: x.SPAN, variant: W.BODY, strongest: !0, children: e.get("capital.overview.grants.item.alerts.offerExpiration", { values: { date: g } }) }) : null
        ] })
      }
    );
  const v = a[0], m = A[v.type];
  return /* @__PURE__ */ o(
    y,
    {
      className: d,
      type: b.WARNING,
      title: m.getTitle(g),
      description: /* @__PURE__ */ o(
        f,
        {
          className: n.button,
          onClick: () => {
            try {
              s(v.type), c.mutate(v.type);
            } finally {
              N(m.eventLabel);
            }
          },
          disabled: !!u,
          state: u ? "loading" : void 0,
          variant: w.TERTIARY,
          "aria-label": e.get(m.buttonLabelKey),
          children: /* @__PURE__ */ o("span", { className: n.buttonLabel, children: e.get(m.buttonLabelKey) })
        }
      )
    }
  );
};
export {
  st as GrantActions
};
