// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/components/CapitalOverview/CapitalOverview.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as m, useCallback as d, useState as U } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { isCapitalRegionSupported as w } from "../../../../internal/CapitalHeader/helpers.js";
import { CapitalErrorMessageDisplay as O } from "../../../CapitalOffer/components/utils/CapitalErrorMessageDisplay.js";
import { CAPITAL_OVERVIEW_CLASS_NAMES as s } from "../../constants.js";
import { useFetch as x } from "../../../../../hooks/useFetch.js";
/* empty css                     */
import k from "../Unqualified.js";
import { PreQualified as _ } from "../PreQualified/PreQualified.js";
import { GrantList as D } from "../GrantList/GrantList.js";
import { ErrorMessageDisplay as I } from "../../../../internal/ErrorMessageDisplay/ErrorMessageDisplay.js";
import { getCapitalErrorMessage as L } from "../../../../utils/capital/getCapitalErrorMessage.js";
import { useConfigContext as N } from "../../../../../core/ConfigContext/context.js";
import { EMPTY_OBJECT as P } from "../../../../../utils/value/constants.js";
import { CapitalHeader as R } from "../../../../internal/CapitalHeader/CapitalHeader.js";
const te = ({
  hideTitle: u,
  onContactSupport: g,
  onFundsRequest: c,
  onOfferDismiss: h,
  onOfferOptionsRequest: b,
  skipPreQualifiedIntro: F
}) => {
  var v, A, q;
  const C = (A = (v = N()) == null ? void 0 : v.extraConfig) == null ? void 0 : A.legalEntity, f = m(() => w(C), [C]), { getGrants: a, getDynamicGrantOffersConfiguration: o } = N().endpoints, n = x({
    fetchOptions: { enabled: !!a && f },
    queryFn: d(async () => a == null ? void 0 : a(P), [a])
  }), i = x({
    fetchOptions: { enabled: !!o && f },
    queryFn: d(async () => o == null ? void 0 : o(P), [o])
  }), e = i.data, [p, y] = U(), r = m(
    () => {
      var l, G;
      return p ? [p, ...((l = n.data) == null ? void 0 : l.data) || []] : (G = n.data) == null ? void 0 : G.data;
    },
    [(q = n.data) == null ? void 0 : q.data, p]
  ), M = d(
    (l) => {
      c ? c(l) : y(l);
    },
    [c]
  ), E = m(() => !!(i.error && n.error || i.error && !(r != null && r.length)), [i.error, r == null ? void 0 : r.length, n.error]), Q = m(() => {
    if (f) {
      if (E)
        return "Error";
      if (!a && !o || !e && !r || n.isFetching || i.isFetching)
        return "Loading";
      if (r != null && r.length)
        return "GrantList";
      if (e != null && e.maxAmount && (e != null && e.minAmount))
        return "PreQualified";
    } else return "UnsupportedRegion";
    return "Unqualified";
  }, [
    o,
    e,
    i.isFetching,
    r,
    a,
    n.isFetching,
    E,
    f
  ]), S = m(() => !!(e && e.minAmount && e.maxAmount), [e]);
  return /* @__PURE__ */ t("div", { className: s.base, children: (() => {
    switch (Q) {
      case "Loading":
        return /* @__PURE__ */ t("div", { className: s.skeletonContainer, children: [
          /* @__PURE__ */ t("div", { className: s.headerSkeleton }),
          /* @__PURE__ */ t("div", { className: s.skeleton })
        ] });
      case "Error":
        return /* @__PURE__ */ t("div", { className: s.errorContainer, children: [
          /* @__PURE__ */ t(R, { hideTitle: u, titleKey: "capital.common.title" }),
          /* @__PURE__ */ t(
            I,
            {
              absolutePosition: !1,
              outlined: !1,
              withImage: !0,
              onContactSupport: g,
              ...L(i.error, g)
            }
          )
        ] });
      case "GrantList":
        return r && /* @__PURE__ */ t(
          D,
          {
            externalDynamicOffersConfig: e,
            grantList: r,
            hideTitle: u,
            newOfferAvailable: S,
            onFundsRequest: c,
            onGrantListUpdateRequest: y,
            onOfferDismiss: h
          }
        );
      case "PreQualified":
        return /* @__PURE__ */ t(
          _,
          {
            onOfferDismiss: h,
            onOfferOptionsRequest: b,
            skipPreQualifiedIntro: F,
            hideTitle: u,
            dynamicOffer: e,
            onFundsRequest: M
          }
        );
      case "Unqualified":
        return /* @__PURE__ */ t(k, { hideTitle: u });
      case "UnsupportedRegion":
        return /* @__PURE__ */ t("div", { className: s.errorContainer, children: [
          /* @__PURE__ */ t(R, { hideTitle: u, titleKey: "capital.common.title" }),
          /* @__PURE__ */ t(O, { unsupportedRegion: !0 })
        ] });
      default:
        return null;
    }
  })() });
};
export {
  te as CapitalOverview
};
