// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkSettings/components/PaymentLinkSettingsContainer/components/SettingsError/SettingsError.js
import { jsx as g } from "../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import u from "../../../../../../../core/Context/useCoreContext.js";
import { ErrorMessageDisplay as p } from "../../../../../../internal/ErrorMessageDisplay/ErrorMessageDisplay.js";
import d from "../../utils/getSettingsErrorMessage.js";
const E = ({
  error: r,
  errorMessage: s,
  onContactSupport: t,
  centered: o,
  absolutePosition: a
}) => {
  const { getImageAsset: i } = u(), { title: l, message: m, refreshComponent: n, translationValues: f, images: e } = d(
    r,
    s,
    t,
    i
  );
  return /* @__PURE__ */ g(
    p,
    {
      imageDesktop: e == null ? void 0 : e.desktop,
      imageMobile: e == null ? void 0 : e.mobile,
      withHeaderOffset: !0,
      title: l,
      message: m,
      translationValues: f,
      withImage: !0,
      centered: o ?? !1,
      absolutePosition: a ?? !1,
      refreshComponent: n,
      onContactSupport: t,
      outlined: !1
    }
  );
};
export {
  E as default
};
