// @adyen/adyen-platform-experience-web/dist/es/components/internal/DatePicker/components/TimeRangeSelector/TimeRangeSelector.js
import { jsx as S } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as j, useCallback as w, useRef as a, useEffect as i } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { useTimeRangeSelection as x } from "./useTimeRangeSelection.js";
import D from "../../../FormFields/Select/Select.js";
const z = ({
  ["aria-label"]: b,
  calendarRef: t,
  onTimeRangeSelected: o,
  timestamp: r,
  ...h
}) => {
  const { customSelection: l, from: u, onSelection: f, options: m, selectedOption: s, to: c } = x(h), O = j(() => Object.freeze(m.map((e) => ({ id: e, name: e }))), [m]), C = w(({ target: e }) => f(e == null ? void 0 : e.value), [f]), n = a(!0), p = a(r);
  return i(() => {
    t != null && t.current && u && c && (n.current = !0, t.current.from = new Date(u), t.current.to = new Date(c));
  }, [t, u, c]), i(() => {
    p.current !== r && (p.current = r, n.current ? n.current = !1 : l());
  }, [l, r]), i(() => {
    s && (o == null || o(s));
  }, [s, o]), /* @__PURE__ */ S(
    D,
    {
      setToTargetWidth: !0,
      items: O,
      filterable: !1,
      multiSelect: !1,
      onChange: C,
      selected: s,
      "aria-label": b
    }
  );
};
export {
  z as default
};
