// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkSettings/components/PaymentLinkSettingsContainer/components/SettingsActionButtons/SettingsActionButtons.js
import { jsx as c } from "../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import g from "../../../../../../../core/Context/useCoreContext.js";
import { usePaymentLinkSettingsContext as L } from "../../context/context.js";
import { useResponsiveContainer as A, containerQueries as b } from "../../../../../../../hooks/useResponsiveContainer.js";
import { ButtonActionsLayout as N, ButtonActionsLayoutBasic as v } from "../../../../../../internal/Button/ButtonActions/types.js";
import x from "../../../../../../internal/Button/ButtonActions/ButtonActions.js";
import { useMemo as r } from "../../../../../../../external/preact/hooks/dist/hooks.module.js";
/* empty css                          */
import { ButtonVariant as d } from "../../../../../../internal/Button/types.js";
import B from "classnames";
import { Icon as C } from "../../../../../../internal/Icon/Icon.js";
import { boolOrFalse as p } from "../../../../../../../utils/value/bool.js";
import { noop as k } from "../../../../../../../utils/common.js";
const F = ({ navigateBack: t, closeContent: n }) => {
  const { i18n: s } = g(), { onSave: u, isSaving: a, isLoadingContent: y, isLoadingStores: f, isSaveSuccess: e } = L(), o = A(b.down.xs), i = y || f, m = r(() => ({
    disabled: p(a || i || t && e),
    event: u,
    iconLeft: t && e ? /* @__PURE__ */ c(C, { className: "adyen-pe-payment-link-settings-save-success__cta-icon", name: "checkmark" }) : void 0,
    title: s.get("payByLink.settings.common.action.save"),
    variant: d.PRIMARY,
    state: p(a && !(t && e)) ? "loading" : "default",
    classNames: o ? ["adyen-pe-payment-link-settings__cta--mobile"] : []
  }), [s, u, a, o, t, e, i]), l = r(() => ({
    disabled: p(i),
    event: t ?? n ?? k,
    title: s.get("payByLink.common.actions.goBack"),
    variant: d.SECONDARY,
    classNames: o ? ["adyen-pe-payment-link-settings__cta--mobile"] : []
  }), [t, s, o, n, i]), _ = r(() => !t && !n ? [m] : [m, l], [m, l, t, n]), S = r(() => o ? N.VERTICAL_STACK : v.BUTTONS_END, [o]);
  return /* @__PURE__ */ c(
    "div",
    {
      className: B("adyen-pe-payment-link-settings__cta-container", {
        "adyen-pe-payment-link-settings__cta-container--mobile": o
      }),
      children: /* @__PURE__ */ c(x, { actions: _, layout: S })
    }
  );
};
export {
  F as default
};
