// @adyen/adyen-platform-experience-web/dist/es/components/internal/Header/Header.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import o from "classnames";
import h from "../Typography/Typography.js";
import A from "../../../core/Context/useCoreContext.js";
import { useComponentHeadingElement as i, ComponentHeadingType as E } from "../../../hooks/useComponentHeadingElement.js";
import { TypographyVariant as T, TypographyElement as _ } from "../Typography/types.js";
import { Divider as L } from "../Divider/Divider.js";
/* empty css            */
const N = "adyen-pe-header", V = ({
  baseClassName: m = N,
  forwardedToRoot: d = !0,
  children: p,
  className: t,
  hasDivider: v,
  hideTitle: y,
  titleKey: a,
  subtitleKey: c,
  subtitleConfig: r,
  ...l
}) => {
  const { i18n: n } = A(), { id: S, ref: I } = i({
    headingType: E.TITLE,
    forwardedToRoot: d
  }), { id: $, ref: x } = i({
    headingType: E.SUBTITLE,
    forwardedToRoot: d
  });
  return /* @__PURE__ */ e("div", { ...l, className: o([m, t]), children: [
    /* @__PURE__ */ e("div", { className: `${m}__headings`, children: [
      !y && a && /* @__PURE__ */ e("div", { ref: I, id: S, className: `${m}__title`, children: /* @__PURE__ */ e(h, { el: _.SPAN, variant: T.TITLE, medium: !0, children: n.get(a) }) }),
      c && /* @__PURE__ */ e("div", { ref: x, id: $, className: o(`${m}__subtitle`, r == null ? void 0 : r.classNames), children: /* @__PURE__ */ e(
        h,
        {
          el: (r == null ? void 0 : r.typographyEl) ?? _.SPAN,
          variant: (r == null ? void 0 : r.variant) ?? T.BODY,
          children: n.get(c)
        }
      ) }),
      v && /* @__PURE__ */ e(L, { className: `${m}__divider` })
    ] }),
    p && /* @__PURE__ */ e("div", { className: `${m}__controls`, children: p })
  ] });
};
export {
  N as BASE_CLASS,
  V as Header
};
