// @adyen/adyen-platform-experience-web/dist/es/components/internal/Tooltip/Tooltip.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import T from "../Popover/Popover.js";
import { PopoverContainerVariant as P } from "../Popover/types.js";
import { useTooltipListeners as I } from "./useTooltipListeners.js";
import { TypographyVariant as N, TypographyElement as b } from "../Typography/types.js";
import v from "../Typography/Typography.js";
import x from "../../../hooks/element/useUniqueIdentifier.js";
import E from "classnames";
import { cloneElement as C, Fragment as O } from "../../../external/preact/dist/preact.module.js";
import { useState as S, useMemo as V } from "../../../external/preact/hooks/dist/hooks.module.js";
/* empty css             */
const c = (o) => typeof o == "string", R = ({ content: o, children: t, showTooltip: n, position: f, isUnderlineVisible: a = !1 }) => {
  var m;
  const [r, u] = S(null), { isVisible: l, listeners: g } = I(), s = x(), p = l || !!n, y = V(
    () => {
      var i;
      return E((i = t == null ? void 0 : t.props) == null ? void 0 : i.className, "adyen-pe-tooltip-target", { "adyen-pe-tooltip-target--underlined": a });
    },
    [(m = t == null ? void 0 : t.props) == null ? void 0 : m.className, a]
  );
  return /* @__PURE__ */ e(O, { children: [
    t ? C(t, {
      ...t.props,
      ...g,
      ...p && (r != null && r.id) ? { "aria-describedby": r.id } : {},
      className: y,
      ref: s,
      role: "button",
      tabIndex: 0
    }) : null,
    p && /* @__PURE__ */ e(
      T,
      {
        position: f,
        variant: P.TOOLTIP,
        targetElement: s,
        setPopoverElement: u,
        fitPosition: !0,
        open: !0,
        children: o && c(o) ? /* @__PURE__ */ e(v, { el: b.DIV, variant: N.CAPTION, children: o }) : o
      }
    )
  ] });
};
export {
  R as Tooltip
};
