// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputesOverview/components/DisputesTable/DisputesTable.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import l from "classnames";
import { useState as G, useMemo as S, useCallback as N, useEffect as $ } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import V from "../../../../../core/Context/useCoreContext.js";
import H from "../../../../../hooks/useTimezoneAwareDateFormatting.js";
import { Alert as j } from "../../../../internal/Alert/Alert.js";
import { AlertTypeOption as q } from "../../../../internal/Alert/types.js";
import { DAY_MS as z } from "../../../../internal/Calendar/calendar/constants.js";
import { getDisputeReason as Q } from "../../../../utils/translation/getters.js";
import { isDisputeActionNeededUrgently as W } from "../../../../utils/disputes/actionNeeded.js";
import Z from "../../../../internal/DataOverviewError/DataOverviewError.js";
import { TypographyVariant as v, TypographyElement as y } from "../../../../internal/Typography/types.js";
import A from "../../../../internal/Typography/Typography.js";
import { BASE_CLASS as r } from "./constants.js";
import { useTableColumns as J } from "../../../../../hooks/useTableColumns.js";
import { useResponsiveContainer as X, containerQueries as ee } from "../../../../../hooks/useResponsiveContainer.js";
import f from "../../../TransactionsOverview/components/TransactionsTable/PaymentMethodCell.js";
import te from "./DisputeStatusTag.js";
import { Tag as oe } from "../../../../internal/Tag/Tag.js";
import { Tooltip as ne } from "../../../../internal/Tooltip/Tooltip.js";
/* empty css                   */
import { DATE_FORMAT_RESPONSE_DEADLINE as se, DATE_FORMAT_DISPUTES as R } from "../../../../../constants/dateFormats.js";
import M from "../../../../internal/DataGrid/DataGrid.js";
import re from "../../../../internal/Pagination/Pagination.js";
import { Translation as ie } from "../../../../internal/Translation/Translation.js";
import { Icon as ae } from "../../../../internal/Icon/Icon.js";
import { useConfigContext as le } from "../../../../../core/ConfigContext/context.js";
const L = {
  status: "disputes.overview.common.fields.status",
  respondBy: "disputes.overview.common.fields.respondBy",
  createdAt: "disputes.overview.common.fields.openedOn",
  paymentMethod: "disputes.overview.common.fields.paymentMethod",
  disputeReason: "disputes.overview.common.fields.disputeReason",
  reason: "disputes.overview.common.fields.reason",
  currency: "disputes.overview.common.fields.currency",
  disputedAmount: "disputes.overview.common.fields.disputedAmount",
  totalPaymentAmount: "disputes.overview.common.fields.totalPaymentAmount"
}, O = {
  CHARGEBACKS: { title: "disputes.overview.chargebacks.errors.listEmpty", message: "disputes.overview.chargebacks.errors.updateFilters" },
  FRAUD_ALERTS: { title: "disputes.overview.fraudAlerts.errors.listEmpty", message: "disputes.overview.fraudAlerts.errors.updateFilters" },
  ONGOING_AND_CLOSED: {
    title: "disputes.overview.ongoingAndClosed.errors.listEmpty",
    message: "disputes.overview.ongoingAndClosed.errors.updateFilters"
  }
}, de = Object.keys(L), n = {
  cellContent: `${r}__cell-content`,
  cellContentVStack: `${r}__cell-content--vstack`,
  cellTextGrey: `${r}__cell-text--grey`,
  dateContentUrgent: `${r}__date-content--urgent`,
  statusContent: `${r}__status-content`,
  statusContentUrgent: `${r}__status-content--urgent`
}, Ie = ({
  error: D,
  loading: E,
  balanceAccountId: me,
  onContactSupport: T,
  showPagination: _,
  onRowClick: F,
  data: k,
  customColumns: U,
  activeBalanceAccount: g,
  statusGroup: s,
  ...P
}) => {
  const { i18n: i } = V(), { refreshing: b } = le(), { dateFormat: d } = H(g == null ? void 0 : g.timeZone), [w, x] = G(null), C = S(() => E || b, [E, b]), o = X(ee.down.xs);
  let m;
  if (_)
    switch (s) {
      case "CHARGEBACKS":
        m = "disputes.overview.chargebacks.limitSelect.a11y.label";
        break;
      case "FRAUD_ALERTS":
        m = "disputes.overview.fraudAlerts.limitSelect.a11y.label";
        break;
      case "ONGOING_AND_CLOSED":
        m = "disputes.overview.ongoingAndClosed.limitSelect.a11y.label";
        break;
    }
  const I = J({
    fields: de,
    fieldsKeys: L,
    customColumns: U,
    columnConfig: S(
      () => ({
        status: {
          visible: s === "ONGOING_AND_CLOSED"
        },
        reason: {
          visible: s === "FRAUD_ALERTS" && !o,
          flex: 2
        },
        respondBy: {
          visible: s === "CHARGEBACKS"
        },
        currency: {
          visible: !o,
          flex: 0.5
        },
        disputedAmount: {
          visible: s !== "FRAUD_ALERTS",
          position: "right"
        },
        createdAt: {
          visible: !o || s === "FRAUD_ALERTS"
        },
        paymentMethod: {
          visible: !o
        },
        disputeReason: {
          visible: s !== "FRAUD_ALERTS" && !o
        },
        totalPaymentAmount: {
          visible: s === "FRAUD_ALERTS",
          position: "right"
        }
      }),
      [o, s]
    )
  }), h = N(() => x(null), []), B = N(
    (e) => {
      if (!e) return "";
      const u = new Date(e).getTime() - Date.now(), p = Math.ceil(u / z), a = d(e, { ...se, weekday: void 0 });
      return p <= 1 ? i.get("disputes.overview.common.actionNeeded.respondToday", { values: { date: a } }) : i.get("disputes.overview.common.actionNeeded.respondDays", { values: { days: p, date: a } });
    },
    [d, i]
  ), K = {
    title: O[s].title,
    message: [O[s].message]
  }, Y = S(
    () => () => /* @__PURE__ */ t(Z, { error: D, errorMessage: "disputes.overview.common.errors.listUnavailable", onContactSupport: T }),
    [D, T]
  );
  return $(() => {
    C && h();
  }, [C, h]), /* @__PURE__ */ t("div", { className: r, children: [
    w && /* @__PURE__ */ t(j, { onClose: h, type: q.WARNING, className: "adyen-pe-disputes-table-alert", ...w }),
    /* @__PURE__ */ t(
      M,
      {
        autoFitColumns: o,
        errorDisplay: Y,
        error: D,
        columns: I,
        data: k,
        loading: C,
        outline: !1,
        onRowClick: { callback: F },
        emptyTableMessage: K,
        customCells: {
          status: ({ item: e }) => /* @__PURE__ */ t("div", { className: l(n.cellContent, { [n.cellContentVStack]: o }), children: [
            /* @__PURE__ */ t(te, { dispute: e }),
            o && /* @__PURE__ */ t(f, { paymentMethod: e.paymentMethod })
          ] }),
          reason: ({ item: e }) => e.reason.title,
          respondBy: ({ item: e }) => {
            const c = W(e), u = d(e.dueDate, R), p = !0, a = () => c && p ? /* @__PURE__ */ t(ne, { content: B(e.dueDate), children: /* @__PURE__ */ t("span", { className: n.dateContentUrgent, children: [
              /* @__PURE__ */ t("time", { dateTime: e.dueDate, children: u }),
              /* @__PURE__ */ t(ae, { name: "warning-filled" })
            ] }) }) : /* @__PURE__ */ t("time", { dateTime: e.dueDate, children: u });
            return /* @__PURE__ */ t("div", { className: l(n.cellContent, { [n.cellContentVStack]: o }), children: [
              e.dueDate ? /* @__PURE__ */ t(
                A,
                {
                  el: y.SPAN,
                  variant: v.BODY,
                  className: l(n.statusContent, {
                    [n.cellTextGrey]: o && !c,
                    [n.statusContentUrgent]: c
                  }),
                  children: o ? /* @__PURE__ */ t(
                    ie,
                    {
                      translationKey: "disputes.overview.common.actionNeeded.dueDate",
                      fills: { dueDate: a }
                    }
                  ) : a()
                }
              ) : null,
              o && /* @__PURE__ */ t(f, { paymentMethod: e.paymentMethod })
            ] });
          },
          currency: ({ item: e }) => /* @__PURE__ */ t(oe, { children: e.amount.currency }),
          disputedAmount: ({ item: e }) => e.amount && /* @__PURE__ */ t(A, { el: y.SPAN, variant: v.BODY, stronger: !0, children: i.amount(e.amount.value, e.amount.currency, { hideCurrency: !1 }) }),
          createdAt: ({ item: e }) => /* @__PURE__ */ t("div", { className: l(n.cellContent, { [n.cellContentVStack]: o }), children: [
            /* @__PURE__ */ t(
              "time",
              {
                dateTime: e.createdAt,
                className: l(n.statusContent, {
                  [n.cellTextGrey]: o
                }),
                children: /* @__PURE__ */ t(A, { el: y.SPAN, variant: v.BODY, children: d(e.createdAt, R) })
              }
            ),
            o && /* @__PURE__ */ t(f, { paymentMethod: e.paymentMethod })
          ] }),
          paymentMethod: ({ item: e }) => /* @__PURE__ */ t(f, { paymentMethod: e.paymentMethod }),
          disputeReason: ({ item: e }) => /* @__PURE__ */ t("span", { children: Q(i, e.reason.category) }),
          totalPaymentAmount: ({ item: e }) => e && /* @__PURE__ */ t(A, { el: y.SPAN, variant: v.BODY, stronger: !0, children: i.amount(e.amount.value, e.amount.currency, { hideCurrency: !1 }) })
        },
        children: _ && /* @__PURE__ */ t(M.Footer, { children: /* @__PURE__ */ t(
          re,
          {
            ...P,
            ariaLabelKey: "disputes.overview.common.pagination.a11y.label",
            limitSelectAriaLabelKey: m
          }
        ) })
      }
    )
  ] });
};
export {
  Ie as DisputesTable,
  O as EMPTY_TABLE_MESSAGE_KEYS,
  de as FIELDS,
  L as FIELD_KEYS
};
