// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkSettings/components/TermsAndConditions/TermsAndConditions.js
import { jsx as n } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
/* empty css                        */
import { useRef as w, useState as a, useEffect as f, useCallback as c, useMemo as H } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import J from "../../../../../core/Context/useCoreContext.js";
import Q from "../../../../internal/FormFields/InputText.js";
import C from "../../../../internal/Typography/Typography.js";
import { TypographyVariant as _, TypographyElement as D } from "../../../../internal/Typography/types.js";
import { AlertVariantOption as X, AlertTypeOption as Z } from "../../../../internal/Alert/types.js";
import { Alert as $ } from "../../../../internal/Alert/Alert.js";
import { usePaymentLinkSettingsContext as ee } from "../PaymentLinkSettingsContainer/context/context.js";
import { isTermsAndConditionsData as ne } from "../PaymentLinkThemeContainer/types.js";
import { isValidURL as I } from "../PaymentLinkSettingsContainer/utils/validateTermsAndConditionsURL.js";
import { ButtonVariant as te } from "../../../../internal/Button/types.js";
import { Requirements as se } from "./Requirements/Requirements.js";
import ie from "classnames";
import re from "../../../../internal/Button/Button.js";
import { Translation as ae } from "../../../../internal/Translation/Translation.js";
import { Icon as E } from "../../../../internal/Icon/Icon.js";
import { Checkbox as oe } from "../../../../internal/Checkbox/Checkbox.js";
import { uniqueId as le } from "../../../../../utils/random/uniqueId.js";
const qe = ({ data: h, initialData: A }) => {
  const { i18n: i } = J(), g = w(le()), [t, P] = a((h == null ? void 0 : h.termsOfServiceUrl) ?? ""), [o, m] = a(void 0), [b, N] = a(!1), [T, R] = a(!1), [Y, S] = a(!1), [W, q] = a(!1), d = w(!1), {
    savedData: p,
    setPayload: u,
    saveActionCalled: x,
    setIsValid: k,
    isSaving: v,
    setSaveActionCalled: L,
    setIsSaveError: B,
    setIsSaveSuccess: O,
    isShowingRequirements: U,
    embeddedInOverview: G,
    setIsShowingRequirements: y
  } = ee();
  f(() => {
    o && I(t) ? k(!0) : k(!1);
  }, [o, t, k, u]), f(() => {
    d.current = !1;
  }, [p]), f(() => {
    x && (R(!!(t && !I(t))), N(!o), L(!1), O(!1), B(!1));
  }, [t, o, x, L, u, B, O]), f(() => {
    let e = ne(p) ? p : void 0;
    e || (e = A);
    const s = !e || !e.termsOfServiceUrl;
    e && ((e == null ? void 0 : e.termsOfServiceUrl) === t || !e.termsOfServiceUrl && t === "") ? (s || (q(!0), m(!0)), S(!1)) : (q(!1), S(!0), m(d.current));
  }, [t, p, A]);
  const M = c(
    (e) => {
      var s, r, l;
      e.preventDefault(), R(!1), P((s = e == null ? void 0 : e.currentTarget) == null ? void 0 : s.value), I((r = e == null ? void 0 : e.currentTarget) == null ? void 0 : r.value) && u({ termsOfServiceUrl: (l = e == null ? void 0 : e.currentTarget) == null ? void 0 : l.value });
    },
    [u]
  ), j = c(
    (e) => {
      var s, r, l;
      e.preventDefault(), (s = e.currentTarget) != null && s.checked && N(!1), d.current = (r = e.currentTarget) == null ? void 0 : r.checked, m((l = e.currentTarget) == null ? void 0 : l.checked);
    },
    [d]
  ), V = c(() => {
    y(!0);
  }, [y]), F = H(() => /* @__PURE__ */ n(
    ae,
    {
      translationKey: "payByLink.settings.termsAndConditions.requirement.checkbox.part1",
      fills: {
        requirements: /* @__PURE__ */ n(
          re,
          {
            className: "adyen-pe-payment-link-settings-terms-and-conditions__requirements-link",
            variant: te.TERTIARY,
            onClick: V,
            children: i.get("payByLink.settings.termsAndConditions.requirement.checkbox.part2")
          }
        )
      }
    }
  ), [i, V]), K = c(() => y(!1), [y]), z = c(() => m(!0), []);
  return /* @__PURE__ */ n("section", { className: "adyen-pe-payment-link-settings-terms-and-conditions", children: [
    /* @__PURE__ */ n(
      "div",
      {
        className: ie("adyen-pe-payment-link-settings-terms-and-conditions__content", {
          "adyen-pe-payment-link-settings-terms-and-conditions__content--hidden": U
        }),
        children: [
          /* @__PURE__ */ n("div", { className: "adyen-pe-payment-link-settings__input-container", children: [
            /* @__PURE__ */ n(
              "label",
              {
                htmlFor: g.current,
                "aria-labelledby": g.current,
                className: "adyen-pe-payment-link-settings-terms-and-conditions-input__label",
                children: /* @__PURE__ */ n(
                  C,
                  {
                    variant: _.BODY,
                    stronger: !0,
                    className: "adyen-pe-payment-link-settings-terms-and-conditions-input__label--info-text",
                    children: i.get("payByLink.settings.termsAndConditions.urlInput.label")
                  }
                )
              }
            ),
            /* @__PURE__ */ n(
              Q,
              {
                disabled: !!v,
                readonly: !!v,
                uniqueId: g.current,
                value: t,
                onInput: M,
                maxLength: 2e3,
                isInvalid: T
              }
            ),
            T && /* @__PURE__ */ n("div", { className: "adyen-pe-payment-link-settings-terms-and-conditions__error", children: [
              /* @__PURE__ */ n(E, { name: "cross-circle-fill", className: "adyen-pe-payment-link-settings-terms-and-conditions__error-icon" }),
              /* @__PURE__ */ n(
                C,
                {
                  className: "adyen-pe-payment-link-settings-terms-and-conditions__error-text",
                  el: D.SPAN,
                  variant: _.BODY,
                  children: i.get("payByLink.settings.termsAndConditions.error.urlValidation")
                }
              )
            ] })
          ] }),
          Y && /* @__PURE__ */ n(
            $,
            {
              type: Z.WARNING,
              variant: X.TIP,
              description: i.get("payByLink.settings.termsAndConditions.alert.urlChange"),
              className: "adyen-pe-payment-link-settings-terms-and-conditions__alert"
            }
          ),
          /* @__PURE__ */ n("div", { className: "adyen-pe-payment-link-settings-terms-and-conditions-checkbox__container", children: [
            /* @__PURE__ */ n(
              oe,
              {
                checked: o,
                disabled: W || v,
                className: "adyen-pe-payment-link-settings-terms-and-conditions-checkbox",
                label: F,
                onInput: j,
                error: b
              }
            ),
            b && /* @__PURE__ */ n("div", { className: "adyen-pe-payment-link-settings-terms-and-conditions__error", children: [
              /* @__PURE__ */ n(E, { name: "cross-circle-fill", className: "adyen-pe-payment-link-settings-terms-and-conditions__error-icon" }),
              /* @__PURE__ */ n(
                C,
                {
                  className: "adyen-pe-payment-link-settings-terms-and-conditions__error-text",
                  el: D.SPAN,
                  variant: _.BODY,
                  children: i.get("payByLink.settings.termsAndConditions.error.requirementsNotChecked")
                }
              )
            ] })
          ] })
        ]
      }
    ),
    U && /* @__PURE__ */ n(
      se,
      {
        onGoBack: K,
        termsAndConditionsURL: t,
        acceptRequirements: z,
        embeddedInOverview: G
      }
    )
  ] });
};
export {
  qe as TermsAndConditions
};
