// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/PaymentDetailsForm/PaymentDetailsForm.js
import { jsx as e } from "../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { ValidityField as a } from "./Fields/ValidityField.js";
import { AmountField as m } from "./Fields/AmountField.js";
import { LinkTypeField as o } from "./Fields/LinkTypeField.js";
import { FormTextInput as n } from "../../../../../internal/FormWrappers/FormTextInput.js";
import l from "../../../../../../core/Context/useCoreContext.js";
import { FormCalendarInput as p } from "../../../../../internal/FormWrappers/FormCalendarInput.js";
import { PAYMENT_LINK_CREATION_FIELD_LENGTHS as t } from "../../../constants.js";
/* empty css                        */
const b = ({ configuration: i }) => {
  const { i18n: r } = l();
  return /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form__fields-container", children: [
    /* @__PURE__ */ e(a, { configuration: i }),
    /* @__PURE__ */ e(m, {}),
    /* @__PURE__ */ e(
      n,
      {
        minLength: t.merchantReference.min,
        maxLength: t.merchantReference.max,
        fieldName: "reference",
        label: r.get("payByLink.creation.fields.merchantReference.label")
      }
    ),
    /* @__PURE__ */ e(o, { configuration: i }),
    /* @__PURE__ */ e(
      n,
      {
        maxLength: t.description.max,
        fieldName: "description",
        label: r.get("payByLink.creation.fields.description.label"),
        supportText: r.get("payByLink.creation.fields.description.supportText")
      }
    ),
    /* @__PURE__ */ e(
      p,
      {
        clearable: !0,
        fieldName: "deliverAt",
        label: r.get("payByLink.creation.fields.deliverAt.label")
      }
    )
  ] });
};
export {
  b as PaymentDetailsForm
};
