// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkSettings/components/PaymentLinkThemeContainer/components/LogoInput/LogoInput.js
import { jsx as t } from "../../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import g from "../../../../../../../core/Context/useCoreContext.js";
import { useUniqueId as _ } from "../../../../../../../hooks/useUniqueId.js";
import { useCallback as s, useMemo as l } from "../../../../../../../external/preact/hooks/dist/hooks.module.js";
import { validationErrors as E } from "../../../../../../internal/FormFields/FileInput/constants.js";
import { defaultMapErrors as O } from "../../../../../../internal/FormFields/FileInput/helpers/defaultMapError.js";
import p from "../../../../../../internal/Typography/Typography.js";
import { TypographyVariant as f, TypographyElement as d } from "../../../../../../internal/Typography/types.js";
import { FileInput as y } from "../../../../../../internal/FormFields/FileInput/FileInput.js";
import { LogoLabel as D, THEME_FORM_ALLOWED_FILE_TYPES as I, THEME_FORM_UPLOAD_DOCUMENT_MAX_SIZE as M, logoOptions as u } from "../ThemeForm/constants.js";
import F from "../LogoRequirements/LogoRequirements.js";
import { Fragment as b } from "../../../../../../../external/preact/dist/preact.module.js";
const k = (o) => {
  switch (o) {
    case u.LOGO:
      return { width: 200, height: 200 };
    case u.FULL_WIDTH_LOGO:
    default:
      return { width: 300, height: 30 };
  }
}, R = ({
  disabled: o,
  logoType: e,
  onFileInputChange: a
}) => {
  const { i18n: r } = g(), n = _(), c = s(
    (i) => {
      a(e, i);
    },
    [e, a]
  ), h = l(() => k(e), [e]), m = l(
    () => e === "logo" ? "payByLink.settings.themes.inputs.file.errors.logo.maxDimension" : "payByLink.settings.themes.inputs.file.errors.fullWidthLogo.maxDimension",
    [e]
  ), L = s(
    (i) => {
      switch (i) {
        case E.MAX_DIMENSIONS:
          return r.get(m);
        default:
          return r.get(O(i));
      }
    },
    [r, m]
  );
  return /* @__PURE__ */ t(b, { children: [
    /* @__PURE__ */ t("label", { htmlFor: n, "aria-labelledby": n, className: "adyen-pe-payment-link-theme-form__file-input", children: [
      /* @__PURE__ */ t(p, { el: d.SPAN, variant: f.BODY, stronger: !0, children: r.get(D[e]) }),
      /* @__PURE__ */ t(
        p,
        {
          el: d.SPAN,
          variant: f.BODY,
          className: "adyen-pe-payment-link-theme-form__field-label-optional",
          children: `(${r.get("payByLink.common.fields.optional.label")})`
        }
      )
    ] }),
    /* @__PURE__ */ t(
      y,
      {
        disabled: o,
        maxDimensions: h,
        maxFileSize: M,
        allowedFileTypes: I,
        onChange: c,
        id: n,
        mapError: L
      }
    ),
    /* @__PURE__ */ t(F, { logoType: e })
  ] });
};
export {
  R as default
};
