// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/TransactionsList/TransactionsList.js
import { jsx as l } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useCallback as v } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { TransactionDetailsModal as x } from "./TransactionDetailsModal.js";
import { TransactionsTable as y } from "../TransactionsTable/TransactionsTable.js";
import { sharedTransactionDetailsEventProperties as L } from "../../../TransactionDetails/constants.js";
import P from "../../../../../core/Context/analytics/useAnalyticsContext.js";
import E from "../../../../../hooks/useModalDetails/useModalDetails.js";
const G = ({
  accountBalancesResult: i,
  balanceAccount: a,
  dataCustomization: c,
  loadingBalanceAccounts: m,
  onContactSupport: s,
  onRecordSelection: d,
  showDetails: u,
  transactionsListResult: p
}) => {
  const t = P(), { currencies: f } = i, { updateDetails: r, resetDetails: C, selectedDetail: h } = E({
    transaction: {
      showDetails: u ?? !0,
      callback: d
    }
  }), D = v(
    ({ id: o, category: e }) => {
      var n;
      e && ((n = t.addEvent) == null || n.call(t, "Viewed transaction details", {
        ...L,
        transactionType: e
      })), r({
        selection: {
          data: o,
          type: "transaction",
          balanceAccount: a
        },
        modalSize: "small"
      }).callback({ id: o });
    },
    [a, r, t]
  ), {
    error: T,
    fetching: g,
    fields: k,
    records: w,
    updateLimit: M,
    hasCustomColumn: j,
    ...b
  } = p;
  return /* @__PURE__ */ l(
    x,
    {
      dataCustomization: c,
      onContactSupport: s,
      resetDetails: C,
      selectedDetail: h,
      children: /* @__PURE__ */ l(
        y,
        {
          activeBalanceAccount: a,
          availableCurrencies: f,
          error: T,
          hasMultipleCurrencies: i.isMultiCurrency,
          loading: m || g,
          onContactSupport: s,
          onLimitSelection: M,
          onRowClick: D,
          showPagination: !0,
          transactions: w,
          customColumns: k,
          ...b
        }
      )
    }
  );
};
export {
  G as default
};
