// @adyen/adyen-platform-experience-web/dist/es/components/internal/DataOverviewDetails/DataOverviewDetails.js
import { jsx as o } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
/* empty css                         */
import { useMemo as T, useState as P, useCallback as k, useEffect as x } from "../../../external/preact/hooks/dist/hooks.module.js";
import { useModalContext as I } from "../Modal/Modal.js";
import { useFetch as M } from "../../../hooks/useFetch.js";
import { PayoutData as O } from "../../external/PayoutDetails/components/PayoutData.js";
import B from "../../../hooks/useBalanceAccounts.js";
import { getErrorMessage as q } from "../../utils/getErrorMessage.js";
import { ErrorMessageDisplay as z } from "../ErrorMessageDisplay/ErrorMessageDisplay.js";
import { useDataOverviewDetailsTitle as L } from "./useDataOverviewDetailsTitle.js";
import { TX_DETAILS_RESERVED_FIELDS_SET as N } from "../../external/TransactionDetails/constants.js";
import { PAYOUT_TABLE_FIELDS as R } from "../../external/PayoutsOverview/components/PayoutsTable/PayoutsTable.js";
import { Header as Y } from "../Header/Header.js";
import { useConfigContext as K } from "../../../core/ConfigContext/context.js";
import { EMPTY_OBJECT as j } from "../../../utils/value/constants.js";
const H = {
  payout: "getPayout"
};
function nt(t) {
  var D;
  const s = K().endpoints[H[t.type]], { hideTitle: C, titleKey: g } = L(t), { withinModal: w } = I(), { data: i, error: n, isFetching: A } = M(
    T(
      () => ({
        fetchOptions: { enabled: !!t.id && !!s },
        queryFn: async () => {
          switch (t.type) {
            case "payout":
              const a = { query: { balanceAccountId: t.id, createdAt: t.date } };
              return s(j, { ...a });
          }
        }
      }),
      [s, t]
    )
  ), m = t.id, b = t == null ? void 0 : t.balanceAccountDescription, { balanceAccounts: u } = B(m, !b), d = T(() => {
    if (n) {
      let a;
      switch (t.type) {
        case "payout":
          a = "payouts.details.errors.unavailable";
          break;
      }
      return q(n, a, t.onContactSupport);
    }
  }, [n, t.onContactSupport, t.type]), y = i, [S, _] = P(), l = k(async () => {
    var a, r, E, f, v;
    if (i) {
      const c = await ((E = (r = (a = t.dataCustomization) == null ? void 0 : a.details) == null ? void 0 : r.onDataRetrieve) == null ? void 0 : E.call(r, i));
      _(
        (v = (f = t.dataCustomization) == null ? void 0 : f.details) == null ? void 0 : v.fields.reduce((h, e) => N.has(e.key) || R.includes(e.key) || (e == null ? void 0 : e.visibility) === "hidden" ? h : { ...h, ...c != null && c[e.key] ? { [e.key]: c[e.key] } : {} }, {})
      );
    }
  }, [i, t]), F = t.dataCustomization;
  return x(() => {
    l();
  }, [l]), /* @__PURE__ */ o("div", { className: "adyen-pe-overview-details", children: [
    /* @__PURE__ */ o(Y, { hideTitle: C, titleKey: g, forwardedToRoot: !w }),
    n && d && /* @__PURE__ */ o("div", { className: "adyen-pe-overview-details--error-container", children: /* @__PURE__ */ o(z, { withImage: !0, ...d }) }),
    t.type === "payout" && y && /* @__PURE__ */ o(
      O,
      {
        balanceAccountId: m,
        payout: y,
        balanceAccountDescription: (t == null ? void 0 : t.balanceAccountDescription) || ((D = u == null ? void 0 : u[0]) == null ? void 0 : D.description),
        isFetching: A,
        extraFields: S,
        dataCustomization: F
      }
    )
  ] });
}
export {
  nt as default
};
