// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/TransactionData/TransactionDataContent.js
import { jsx as o } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import x from "../PaymentRefund/PaymentRefund.js";
import S from "../PaymentDetails/PaymentDetails.js";
import C from "../../../../internal/DataOverviewDetails/DataOverviewDetailsSkeleton.js";
import { ActiveView as c } from "../../types.js";
import { useLandedPageEvent as V } from "../../../../../hooks/useAnalytics/useLandedPageEvent.js";
import { useState as l, useRef as j, useMemo as n, useEffect as F } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { useModalContext as N } from "../../../../internal/Modal/Modal.js";
import { sharedTransactionDetailsEventProperties as O } from "../../constants.js";
/* empty css                     */
import { useRefundMetadata as Y } from "../../hooks/useRefundMetadata/useRefundMetadata.js";
import { EMPTY_ARRAY as p } from "../../../../../utils/value/constants.js";
const Z = ({
  dataCustomization: A,
  extraFields: v,
  fetchingTransaction: D,
  refreshTransaction: R,
  transaction: e,
  transactionNavigator: b
}) => {
  const [k, s] = l(c.DETAILS), [r, d] = l(!1), { withinModal: w } = N(), {
    fullRefundFailed: I,
    fullRefundInProgress: L,
    refundableAmount: P,
    refundAmounts: f,
    refundAvailable: E,
    refundCurrency: u,
    refundDisabled: a,
    refundedAmount: i,
    refundedState: M,
    refundMode: T,
    refundLocked: t
  } = Y(e), g = j(t), m = n(() => a || r, [a, r]), y = n(() => t || r, [t, r]), h = n(() => Object.freeze(e.lineItems ?? p), [e.lineItems]);
  if (V({
    ...O,
    ...w && { fromPage: "Transactions overview" }
  }), F(() => {
    (g.current = t) && d(!1);
  }, [t]), D)
    return /* @__PURE__ */ o(C, { skeletonRowNumber: 5 });
  switch (k) {
    case c.REFUND:
      return /* @__PURE__ */ o(
        x,
        {
          currency: u,
          disabled: m,
          lineItems: h,
          maxAmount: P,
          mode: T,
          refreshTransaction: R,
          refundedAmount: i,
          refundingAmounts: f.in_progress ?? p,
          setActiveView: s,
          setLocked: d,
          transaction: e
        }
      );
    default:
      return /* @__PURE__ */ o(
        S,
        {
          dataCustomization: A,
          extraFields: v,
          fullRefundFailed: I,
          fullRefundInProgress: L,
          refundAmounts: f,
          refundAvailable: E,
          refundCurrency: u,
          refundDisabled: m,
          refundedAmount: i,
          refundedState: M,
          refundLocked: y,
          setActiveView: s,
          transaction: e,
          transactionNavigator: b
        }
      );
  }
};
export {
  Z as TransactionDataContent,
  Z as default
};
