// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/TransactionData/TransactionData.js
import { jsx as o } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { TX_DETAILS_RESERVED_FIELDS_SET as M, TX_DETAILS_FIELDS_REMAPS as b } from "../../constants.js";
import I from "../../../../internal/DataOverviewDetails/DataOverviewDetailsSkeleton.js";
import { normalizeCustomFields as x } from "../../../../utils/customData/normalizeCustomFields.js";
import { TransactionDataContent as A } from "./TransactionDataContent.js";
import { useModalContext as H } from "../../../../internal/Modal/Modal.js";
import { useState as u, useMemo as w, useRef as L, useEffect as S } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { ErrorMessageDisplay as N } from "../../../../internal/ErrorMessageDisplay/ErrorMessageDisplay.js";
import { getErrorMessage as P } from "../../../../utils/getErrorMessage.js";
import { useTransaction as O } from "../../hooks/useTransaction/useTransaction.js";
import { boolOrFalse as X } from "../../../../../utils/value/bool.js";
import { Header as j } from "../../../../internal/Header/Header.js";
const C = ({ id: i, dataCustomization: e, hideTitle: m, onContactSupport: f }) => {
  const { error: c, fetchingTransaction: T, refreshTransaction: g, transaction: r, transactionNavigator: k } = O(i), { withinModal: l } = H(), [R, p] = u(), [v, _] = u(!1), F = w(() => v || X(m), [v, m]), n = L(r), E = w(
    () => P(c, "transactions.details.errors.unavailable", f),
    [c, f]
  );
  return !n.current && r && (n.current = r), S(() => {
    _(l);
  }, [l]), S(() => {
    r && r.id === i ? (async () => {
      var a, d, y, h;
      const t = await ((d = (a = e == null ? void 0 : e.details) == null ? void 0 : a.onDataRetrieve) == null ? void 0 : d.call(a, r));
      p(
        (h = x(
          (y = e == null ? void 0 : e.details) == null ? void 0 : y.fields,
          b,
          t
        )) == null ? void 0 : h.reduce((D, s) => !M.has(s.key) && (s == null ? void 0 : s.visibility) !== "hidden" ? {
          ...D,
          ...(t == null ? void 0 : t[s.key]) && { [s.key]: t[s.key] }
        } : D, {})
      );
    })() : p(void 0);
  }, [r, i, e]), /* @__PURE__ */ o("div", { className: "adyen-pe-overview-details", children: [
    /* @__PURE__ */ o(j, { hideTitle: F, titleKey: "transactions.details.title", forwardedToRoot: !l }),
    n.current ? /* @__PURE__ */ o(
      A,
      {
        extraFields: R,
        dataCustomization: e,
        fetchingTransaction: T,
        refreshTransaction: g,
        transaction: r ?? n.current,
        transactionNavigator: k
      }
    ) : T ? /* @__PURE__ */ o(I, { skeletonRowNumber: 5 }) : c && E && /* @__PURE__ */ o("div", { className: "adyen-pe-overview-details--error-container", children: /* @__PURE__ */ o(N, { outlined: !1, absolutePosition: !1, withBackground: !1, withImage: !0, ...E }) })
  ] });
};
export {
  C as TransactionData
};
