// @adyen/adyen-platform-experience-web/dist/es/hooks/useBalanceAccountSelection.js
import { useState as h, useMemo as f, useRef as v, useCallback as C, useEffect as N } from "../external/preact/hooks/dist/hooks.module.js";
import T from "../core/Context/useCoreContext.js";
import b from "./useAnalytics/useFilterAnalyticsEvent.js";
import { capitalize as q } from "../utils/value/string.js";
import { uniqueId as z } from "../utils/random/uniqueId.js";
const x = z(), D = ({
  allowAllSelection: d = !1,
  balanceAccounts: o,
  eventCategory: y,
  eventSubCategory: A,
  eventLabel: B = "Balance account filter",
  onUpdateSelection: r
}) => {
  const { i18n: m } = T(), { logEvent: i } = b({ category: y, subCategory: A, label: B }), [a, I] = h(0), e = f(
    () => o && [
      ...o,
      ...d && o.length > 1 ? [
        {
          ...o[0] ?? {},
          id: x,
          description: void 0
        }
      ] : []
    ],
    [d, o]
  ), c = f(() => e == null ? void 0 : e[a], [e, a]), n = c == null ? void 0 : c.id, p = v(), E = f(
    () => Object.freeze(
      (e == null ? void 0 : e.map(({ description: t, id: u }) => {
        const s = u === x ? m.get("common.filters.types.account.options.all") : q(t);
        return { id: u, name: s };
      })) ?? []
    ),
    [e, m]
  ), L = C(
    ({ target: t }) => {
      const u = t == null ? void 0 : t.value, s = e == null ? void 0 : e.findIndex(({ id: _ }) => _ === u);
      s !== void 0 && s >= 0 && I(s);
    },
    [e]
  ), O = C(() => I(0), []);
  return N(() => {
    const t = p.current;
    t !== n && (p.current = n, t && n && (i == null || i("update", n)), r == null || r(c));
  }, [c, n, i, r]), { activeBalanceAccount: c, balanceAccountSelectionOptions: E, onBalanceAccountSelection: L, resetBalanceAccountSelection: O };
};
export {
  x as ALL_BALANCE_ACCOUNTS_SELECTION_ID,
  D as default
};
