// @adyen/adyen-platform-experience-web/dist/es/hooks/useFetch.js
import { useRef as l, useReducer as p, useCallback as y, useEffect as g } from "../external/preact/hooks/dist/hooks.module.js";
import { boolOrTrue as n } from "../utils/value/bool.js";
function F({
  fetchOptions: { keepPrevData: u, onSuccess: e, enabled: c } = { keepPrevData: !0 },
  queryFn: f
}) {
  const r = l(!1), a = {
    error: void 0,
    data: void 0,
    isFetching: n(c),
    refetch: () => o()
  }, d = (t, s) => {
    switch (s.type) {
      case "loading":
        return { ...a, isFetching: !0, data: u ? t.data : void 0 };
      case "fetched":
        return { ...a, data: s.payload, isFetching: !1 };
      case "error":
        return { ...a, error: s.payload, isFetching: !1 };
      default:
        return t;
    }
  }, [h, i] = p(d, a), o = y(async () => {
    i({ type: "loading" });
    try {
      if (r.current) return;
      const t = await f();
      e == null || e(t), i({ type: "fetched", payload: t });
    } catch (t) {
      if (r.current) return;
      i({ type: "error", payload: t });
    }
  }, [i, f, e]);
  return g(() => (r.current = !1, n(c) && o(), () => {
    r.current = !0;
  }), [c, o]), h;
}
export {
  F as useFetch
};
