// @adyen/adyen-platform-experience-web/dist/es/hooks/useStores.js
import { useState as C, useMemo as s, useEffect as l } from "../external/preact/hooks/dist/hooks.module.js";
import { useFetch as E } from "./useFetch.js";
import { useConfigContext as h } from "../core/ConfigContext/context.js";
import { EMPTY_OBJECT as p } from "../utils/value/constants.js";
const M = (n, d = !0) => {
  const [c, m] = C(void 0), { getPayByLinkStores: i } = h().endpoints, { data: r, error: u, isFetching: f } = E(
    s(
      () => ({
        fetchOptions: {
          enabled: !!i
        },
        queryFn: async () => i == null ? void 0 : i(p, p)
      }),
      [i]
    )
  ), o = s(
    () => {
      var t;
      return (t = r == null ? void 0 : r.data) == null ? void 0 : t.filter((e) => e.storeId ? !n || (typeof n == "string" ? e.storeId === n : n == null ? void 0 : n.includes(e.storeId)) : !1).map((e) => ({
        id: e.storeId || "",
        name: e.storeCode || "",
        storeCode: e.storeCode || "",
        description: e.description || ""
      }));
    },
    [r, n]
  ), a = s(
    () => {
      var t;
      return (t = r == null ? void 0 : r.data) == null ? void 0 : t.map((e) => ({
        id: e.storeId || "",
        name: e.storeCode || "",
        storeCode: e.storeCode || "",
        description: e.description || ""
      }));
    },
    [r]
  );
  return l(() => {
    var t;
    !c && o && (o == null ? void 0 : o.length) > 0 && d && m((t = o == null ? void 0 : o[0]) == null ? void 0 : t.id);
  }, [o, c, d]), { filteredStores: o, selectedStore: c, setSelectedStore: m, isFetching: f, error: u, allStores: a };
};
export {
  M as useStores
};
