// @adyen/adyen-platform-experience-web/dist/es/hooks/useTableColumns.js
import { useMemo as b } from "../external/preact/hooks/dist/hooks.module.js";
import U from "../core/Context/useCoreContext.js";
import { useResponsiveContainer as g, containerQueries as w } from "./useResponsiveContainer.js";
import { EMPTY_OBJECT as A } from "../utils/value/constants.js";
import { isUndefined as O } from "../utils/value/is.js";
function P(t) {
  let i = {};
  for (const r of Object.keys(t))
    O(t[r]) || (i = { ...i, [r]: t[r] });
  return i;
}
const Y = ({
  fields: t,
  customColumns: i,
  columnConfig: r,
  fieldsKeys: k
}) => {
  const { i18n: o } = U(), s = b(() => t.map((y) => ({ key: y })), [t]), v = g(w.up.sm);
  return b(() => {
    const y = (i == null ? void 0 : i.filter((e) => !(t != null && t.some((n) => n === e.key))).map((e) => e.key)) || [], M = [...s, ...(i == null ? void 0 : i.filter((e) => e == null ? void 0 : e.key)) || []], x = (i == null ? void 0 : i.reduce((e, n) => (e[n.key] = n, e), {})) || {}, a = /* @__PURE__ */ new Map();
    return M.forEach((e) => {
      const n = x[e.key];
      if ((n == null ? void 0 : n.visibility) !== "hidden")
        if (a.has(e.key)) {
          const p = a.get(e.key);
          a.set(e.key, {
            ...p,
            ...e,
            visible: e.visibility !== "hidden",
            position: e.align || p.position
          });
        } else {
          const { key: p, flex: T, align: f } = e, h = o.get((k == null ? void 0 : k[p]) || p), E = P((r == null ? void 0 : r[p]) || A);
          a.set(e.key, {
            key: p,
            label: h,
            visible: y.includes(e.key) ? v : !0,
            flex: T,
            position: f,
            ...E
          });
        }
    }), Array.from(a.values());
  }, [r, i, t, o, v, s]);
};
export {
  Y as useTableColumns
};
