// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/StoreForm/StoreForm.js
import { jsx as o } from "../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import f from "../../../../../../core/Context/useCoreContext.js";
import { Alert as d } from "../../../../../internal/Alert/Alert.js";
import { AlertTypeOption as k } from "../../../../../internal/Alert/types.js";
import S from "./Fields/StoreField.js";
import { useMemo as a } from "../../../../../../external/preact/hooks/dist/hooks.module.js";
import { useWizardFormContext as g } from "../../../../../../hooks/form/wizard/WizardFormContext.js";
/* empty css               */
import { useConfigContext as F } from "../../../../../../core/ConfigContext/context.js";
const b = ({ setShowTermsAndConditions: i, settingsData: s, storesData: c, selectItems: m, termsAndConditionsProvisioned: r }) => {
  const { i18n: t } = f(), { savePayByLinkSettings: p } = F().endpoints, { getValues: l } = g(), e = !!p, u = l("store"), n = a(() => {
    if (!r)
      return e ? "payByLink.creation.storeForm.alerts.tcSetupRequired" : "payByLink.creation.storeForm.alerts.tcSetupRequiredWithoutPermissions";
  }, [r, e]), y = a(() => e ? [
    {
      label: t.get("payByLink.creation.storeForm.alerts.tcSetupRequiredAction"),
      onClick: () => {
        i(!0);
      }
    }
  ] : [], [e, i, t]);
  return /* @__PURE__ */ o("div", { className: "adyen-pe-payment-link-creation-form__fields-container", children: [
    /* @__PURE__ */ o(S, { items: m }),
    s && c && u && !r && /* @__PURE__ */ o(
      d,
      {
        className: "adyen-pe-payment-link-creation-form__tc-alert",
        title: t.get("payByLink.creation.storeForm.alerts.tcSetupRequiredTitle"),
        type: k.WARNING,
        description: n && t.get(n),
        actions: y
      }
    )
  ] });
};
export {
  b as StoreForm
};
