// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkSettings/components/PaymentLinkSettingsContainer/utils/getSettingsErrorMessage.js
import { jsx as s } from "../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { UNDEFINED_ERROR as c } from "../../../../../utils/getCommonErrorCode.js";
import i from "../../../../../internal/CopyText/CopyText.js";
const a = "ACCOUNT_MISCONFIGURATION", u = "WRONG_STORE_IDS", d = (r, t, m, o) => {
  if (!r) return c;
  const e = m ? "common.errors.errorCode" : "common.errors.errorCodeSupport", n = {
    [e]: r.requestId ? /* @__PURE__ */ s(i, { copyButtonAriaLabelKey: "common.actions.copy.labels.errorCode", textToCopy: r.requestId }) : null
  };
  switch (r.errorCode) {
    case a:
      return {
        title: "common.errors.somethingWentWrong",
        message: ["payByLink.common.errors.accountConfiguration", "common.errors.contactSupport"],
        translationValues: n
      };
    case u:
      return {
        title: "common.errors.somethingWentWrong",
        message: ["payByLink.common.errors.storeID", "common.errors.contactSupport"],
        translationValues: n,
        onContactSupport: m
      };
    case "00_500":
      return {
        title: "common.errors.somethingWentWrong",
        message: [t, e],
        translationValues: n,
        refreshComponent: !0,
        images: {
          desktop: o == null ? void 0 : o({ name: "wrong-environment", subFolder: "images/small" })
        }
      };
    default:
      return {
        title: "common.errors.somethingWentWrong",
        message: [t, e],
        translationValues: n,
        refreshComponent: !0,
        images: {
          desktop: o == null ? void 0 : o({ name: "wrong-environment", subFolder: "images/small" })
        }
      };
  }
};
export {
  a as ACCOUNT_MISCONFIGURATION,
  u as WRONG_STORE_IDS,
  d as default
};
