// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/Select/constants.js
import { DEFAULT_BUTTON_CLASSNAME as N } from "../../Button/constants.js";
import { ButtonVariant as D } from "../../Button/types.js";
import { getModifierClasses as T } from "../../../../utils/preact/className.js";
const S = "adyen-pe-dropdown", C = T(N, [D.SECONDARY], [N]), _ = `${S}__button`, A = `${_}--active`, c = `${_}-collapse-indicator`, n = `${_}-clear`, s = `${_}--has-selection`, R = `${_}-icon`, e = `${_}-multiselect-counter`, P = `${_}--readonly`, W = `${_}-text`, i = `${_}--valid`, $ = `${_}--invalid`, O = `${S}__element`, a = `${O}--active`, B = `${O}--keyboard-active`, I = `${O}-checkbox`, U = `${O}-checkmark`, r = `${O}-content`, l = `${O}--disabled`, M = `${O}-icon`, d = `${O}--no-option`, t = `${S}__list`, m = `${t}--active`, p = `${S}--multiselect`;
export {
  S as DROPDOWN_BASE_CLASS,
  A as DROPDOWN_BUTTON_ACTIVE_CLASS,
  _ as DROPDOWN_BUTTON_CLASS,
  C as DROPDOWN_BUTTON_CLASSNAME,
  n as DROPDOWN_BUTTON_CLEAR_CLASS,
  c as DROPDOWN_BUTTON_COLLAPSE_INDICATOR_CLASS,
  s as DROPDOWN_BUTTON_HAS_SELECTION_CLASS,
  R as DROPDOWN_BUTTON_ICON_CLASS,
  $ as DROPDOWN_BUTTON_INVALID_CLASS,
  e as DROPDOWN_BUTTON_MULTI_SELECT_COUNTER_CLASS,
  P as DROPDOWN_BUTTON_READONLY_CLASS,
  W as DROPDOWN_BUTTON_TEXT_CLASS,
  i as DROPDOWN_BUTTON_VALID_CLASS,
  a as DROPDOWN_ELEMENT_ACTIVE_CLASS,
  I as DROPDOWN_ELEMENT_CHECKBOX_CLASS,
  U as DROPDOWN_ELEMENT_CHECKMARK_CLASS,
  O as DROPDOWN_ELEMENT_CLASS,
  r as DROPDOWN_ELEMENT_CONTENT_CLASS,
  l as DROPDOWN_ELEMENT_DISABLED_CLASS,
  M as DROPDOWN_ELEMENT_ICON_CLASS,
  B as DROPDOWN_ELEMENT_KEYBOARD_ACTIVE_CLASS,
  d as DROPDOWN_ELEMENT_NO_OPTION_CLASS,
  m as DROPDOWN_LIST_ACTIVE_CLASS,
  t as DROPDOWN_LIST_CLASS,
  p as DROPDOWN_MULTI_SELECT_CLASS
};
