// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/Select/Select.js
import { jsx as W } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import Te from "classnames";
import { useState as N, useRef as d, useMemo as J, useCallback as i, useEffect as D } from "../../../../external/preact/hooks/dist/hooks.module.js";
import { InteractionKeyCode as o } from "../../../types.js";
import { ARIA_ERROR_SUFFIX as Oe } from "../../../../core/Errors/constants.js";
import { useUniqueId as Q } from "../../../../hooks/useUniqueId.js";
import _e from "./components/SelectButton.js";
import Fe from "./components/SelectList.js";
import ye from "./hooks/useSelect.js";
import { DROPDOWN_BASE_CLASS as V, DROPDOWN_MULTI_SELECT_CLASS as Be } from "./constants.js";
/* empty css            */
import We from "../../../../hooks/useCommitAction/useCommitAction.js";
import { CommitAction as Z } from "../../../../hooks/useCommitAction/types.js";
import { EMPTY_ARRAY as K } from "../../../../utils/value/constants.js";
import { boolOrFalse as Ne } from "../../../../utils/value/bool.js";
import { noop as Ke } from "../../../../utils/common.js";
const rt = ({
  className: U,
  classNameModifiers: $ = K,
  clearable: ee = !1,
  popoverClassNameModifiers: te,
  items: g = K,
  filterable: a = !1,
  disableFocusTrap: re = !1,
  disableToggleFocusOnClose: q = !1,
  multiSelect: m = !1,
  readonly: ne = !1,
  onChange: h = Ke,
  selected: se,
  name: I,
  isInvalid: oe,
  isValid: ae,
  placeholder: ce,
  uniqueId: ie,
  renderListItem: le,
  renderButtonContent: ue,
  isCollatingErrors: de,
  setToTargetWidth: fe,
  withoutCollapseIndicator: me = !1,
  showOverlay: pe = !1,
  fitPosition: be,
  fixedPopoverPositioning: Ae,
  onResetAction: ge,
  buttonVariant: Ie,
  ...Ee
}) => {
  const { resetSelection: p, select: w, selection: c } = ye({ items: g, multiSelect: m, selected: se }), [n, f] = N(!1), [E, R] = N(""), [b, u] = N(-1), v = d(null), k = d(null), F = d(null), Re = `elem-${Q()}`, y = ie ?? Re, Y = `select-${Q()}`, M = d(), P = d(!0), B = d(!1), S = d(c), j = d(c), De = J(() => c.length, [c]), he = i(() => {
    R(""), f(!1), u(-1), n && (p(S.current), P.current = !0);
  }, [p, f, R, n]), ke = J(
    () => Te([
      V,
      { [Be]: Ne(m) },
      ...$.map((e) => `${V}--${e}`),
      U
    ]),
    [U, $, m]
  ), { commitAction: H, commitActionButtons: Se, committing: X, resetCommitAction: z } = We({
    resetDisabled: !c.length,
    onResetAction: ge
  }), l = i(() => {
    var e;
    R(""), f(!1), u(-1), z(), P.current ? P.current = !1 : q || (e = F.current) == null || e.focus();
  }, [q, z, f, R]), C = i(() => {
    S.current = c;
    const e = `${c.map(({ id: t }) => t)}`;
    h({ target: { value: e, name: I } });
  }, [I, h, c]), L = i(() => {
    p(), S.current = K, h({ target: { value: "", name: I } });
  }, [I, h, p]);
  D(() => {
    switch (H) {
      case Z.APPLY:
        C();
        break;
      case Z.CLEAR:
        B.current = !0, L();
        break;
    }
  }, [L, H, C, I, h, p]);
  const x = i(
    (e) => {
      var r;
      e.preventDefault();
      const t = e.currentTarget && ((r = k == null ? void 0 : k.current) != null && r.contains(e.currentTarget)) ? e.currentTarget : null;
      if (t && !t.getAttribute("data-disabled")) {
        const A = t.getAttribute("data-value"), T = g.find((O) => O.id === A);
        w(T);
      }
    },
    [g, w]
  );
  D(() => {
    j.current !== c && (j.current = c, (!m || B.current) && n && (C(), l())), B.current = !1;
  }, [l, C, m, c, n]), D(() => {
    X && l();
  }, [X, l]), D(() => {
    n || (S.current = c);
  }, [c, n]);
  const xe = i(
    (e) => {
      switch (e.code) {
        case o.ESCAPE:
        case o.TAB:
          n && l(), P.current = e.key === o.TAB;
          return;
        case o.ENTER:
        case o.SPACE:
          if (a && n) {
            if (e.key === o.ENTER)
              if (E) x(e);
              else break;
            return;
          }
          break;
        case o.ARROW_DOWN:
        case o.ARROW_UP:
          if (a && n)
            return;
          break;
        default:
          return;
      }
      e.preventDefault(), f(!0);
    },
    [l, a, x, n, f, E]
  );
  D(() => {
    n && (cancelAnimationFrame(M.current), M.current = requestAnimationFrame(() => {
      var e;
      e: {
        let t = (e = k.current) == null ? void 0 : e.firstElementChild, r, A = 0;
        for (; t; ) {
          if (!(t.dataset.disabled && t.dataset.disabled === "true")) {
            if (t.getAttribute("aria-selected") === "true") {
              t.tabIndex = 0, a ? u(A) : t.focus();
              break e;
            }
            r = r || t, A++;
          }
          t = t.nextElementSibling;
        }
        r && !a && (r.tabIndex = 0, r.focus());
      }
    }));
  }, [a, n]);
  const we = i(
    (e) => {
      const t = e.target;
      switch (e.code) {
        case o.ESCAPE:
          e.preventDefault(), e.stopPropagation(), l();
          break;
        case o.ENTER:
        case o.SPACE:
          x(e);
          break;
        case o.ARROW_DOWN: {
          e.preventDefault();
          let r = t.nextElementSibling;
          for (; r; ) {
            if (!(r.dataset.disabled && r.dataset.disabled === "true")) {
              t.tabIndex = -1, r.tabIndex = 0, r.focus();
              break;
            }
            r = r.nextElementSibling;
          }
          break;
        }
        case o.ARROW_UP: {
          e.preventDefault();
          e: {
            let r = t.previousElementSibling;
            for (; r; ) {
              if (!(r.dataset.disabled && r.dataset.disabled === "true")) {
                t.tabIndex = -1, r.tabIndex = 0, r.focus();
                break e;
              }
              r = r.previousElementSibling;
            }
            a && v.current && v.current.focus();
          }
          break;
        }
      }
    },
    [l, a, x]
  ), ve = i(
    (e) => {
      var A, T, O;
      if (!a || !n) return;
      const t = g.filter((s) => !E || s.name.toLowerCase().includes(E)), r = t.filter((s) => !s.disabled);
      switch (e.code) {
        case o.ESCAPE:
          e.preventDefault(), l();
          break;
        case o.ENTER:
          if (e.preventDefault(), b >= 0 && b < t.length) {
            const s = t[b];
            s && !(s != null && s.disabled) && w(s);
          }
          break;
        case o.ARROW_DOWN: {
          if (e.preventDefault(), r.length === 0) break;
          let s = b + 1;
          for (; s < t.length; ) {
            if (!((A = t[s]) != null && A.disabled)) {
              u(s);
              break;
            }
            s++;
          }
          if (s >= t.length) {
            for (let _ = 0; _ < t.length; _++)
              if (!((T = t[_]) != null && T.disabled)) {
                u(_);
                break;
              }
          }
          break;
        }
        case o.ARROW_UP: {
          if (e.preventDefault(), r.length === 0) break;
          let s = b - 1;
          for (; s >= 0; ) {
            if (!((O = t[s]) != null && O.disabled)) {
              u(s);
              break;
            }
            s--;
          }
          s < 0 && u(-1);
          break;
        }
      }
    },
    [a, n, g, E, b, l, w]
  ), Pe = i(
    (e) => {
      const t = e.target.value;
      R(t.toLowerCase()), u(-1);
    },
    [R]
  ), G = d(!1), Ce = i(
    (e) => {
      e.preventDefault(), G.current || f((t) => !t), n && p(S.current);
    },
    [f, n, p]
  );
  D(() => {
    var e;
    n && a && ((e = v.current) == null || e.focus(), u(-1)), G.current = n;
  }, [a, n]);
  const Le = i(
    (e) => {
      var t, r;
      (t = e == null ? void 0 : e.preventDefault) == null || t.call(e), (r = e == null ? void 0 : e.stopPropagation) == null || r.call(e), L();
    },
    [L]
  );
  return /* @__PURE__ */ W("div", { className: ke, children: [
    /* @__PURE__ */ W(
      _e,
      {
        buttonVariant: Ie,
        id: y,
        appliedFilterNumber: De,
        active: c,
        clearable: ee,
        filterInputRef: v,
        filterable: a,
        isInvalid: oe,
        isValid: ae,
        name: I,
        onClear: Le,
        onButtonKeyDown: xe,
        onFilterInputKeyDown: ve,
        multiSelect: m,
        placeholder: ce,
        readonly: ne,
        renderButtonContent: ue,
        selectListId: Y,
        showList: n,
        toggleButtonRef: F,
        toggleList: Ce,
        withoutCollapseIndicator: me,
        ariaDescribedBy: !de && y ? `${y}${Oe}` : void 0,
        ...Ee,
        onInput: Pe
      }
    ),
    /* @__PURE__ */ W(
      Fe,
      {
        popoverClassNameModifiers: te,
        setToTargetWidth: fe,
        dismissPopover: he,
        active: c,
        commitActions: Se,
        items: g,
        multiSelect: m,
        disableFocusTrap: re || a,
        onKeyDown: we,
        onSelect: x,
        selectListId: Y,
        ref: k,
        toggleButtonRef: F,
        renderListItem: le,
        showList: n,
        showOverlay: pe,
        textFilter: E,
        activeIndex: a ? b : void 0,
        filterable: a,
        fitPosition: be,
        fixedPopoverPositioning: Ae
      }
    )
  ] });
};
export {
  rt as default
};
