// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/FileInput/components/Dropzone.js
import { jsx as a } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import w from "classnames";
import { useState as v, useMemo as H, useCallback as A } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import O from "../../../Typography/Typography.js";
import { useTrackedRef as J } from "../../../../../hooks/useTrackedRef.js";
import K from "../../../../../core/Context/useCoreContext.js";
import { DEFAULT_MAX_FILE_SIZE as R, DEFAULT_FILE_TYPES as ee, validationErrors as d, BASE_CLASS as n } from "../constants.js";
import { TypographyVariant as $, TypographyElement as M } from "../../../Typography/types.js";
/* empty css                */
import { fixedForwardRef as re } from "../../../../../utils/preact/fixedForwardRef.js";
import { getUploadedFilesFromSource as te } from "../../../../../utils/file/upload.js";
import { Icon as y } from "../../../Icon/Icon.js";
import { Fragment as oe } from "../../../../../external/preact/dist/preact.module.js";
import { uniqueId as ae } from "../../../../../utils/random/uniqueId.js";
import { isFunction as x } from "../../../../../utils/value/is.js";
const o = {
  dropzone: `${n}__dropzone`,
  dropzoneDisabled: `${n}__dropzone--disabled`,
  dropzoneDragOver: `${n}__dropzone--dragover`,
  dropzoneError: `${n}__dropzone--error`,
  label: `${n}__label`,
  labelDefault: `${n}__label-default-content`,
  labelIcon: `${n}__label-icon`,
  labelText: `${n}__label-text`,
  error: `${n}__error`,
  errorIcon: `${n}__error-icon`,
  errorText: `${n}__error-text`
}, Fe = re((C, Y) => {
  const {
    id: D,
    name: g,
    children: q,
    disabled: c = !1,
    required: P = !1,
    maxDimensions: t,
    maxFileSize: m = R,
    allowedFileTypes: E = ee,
    mapError: L,
    uploadFiles: b
  } = C, { i18n: S } = K(), [f, U] = v(""), [h, z] = v(), [V, I] = v(!1), _ = !!f, B = g == null ? void 0 : g.trim(), N = H(() => D || ae(), [D]), F = J(Y), k = (e) => {
    var i;
    Array.from(((i = e.dataTransfer) == null ? void 0 : i.types) ?? []).some((s) => s === "Files") && (e.preventDefault(), I(!0));
  }, X = (e) => {
    e.preventDefault(), I(!1);
  }, j = (e) => {
    e.preventDefault(), I(!1), T(e.dataTransfer);
  }, G = (e) => {
    h && z(void 0), T(e.target);
  }, W = async (e) => {
    var r;
    e.target !== document.activeElement && ((r = e.target) == null || r.checkValidity());
  }, Q = async (e) => {
    !f && e.target.validity.valueMissing && p(d.FILE_REQUIRED);
  }, p = A(
    (e) => {
      const r = F.current;
      if (r) {
        const i = r.required;
        r.required = !1, r.setCustomValidity(e), U(r.validationMessage ?? ""), r.required = i;
      }
    },
    [F]
  ), Z = async (e) => new Promise((r, i) => {
    const s = new Image(), l = URL.createObjectURL(e);
    s.src = l, s.onerror = i, s.onload = function() {
      r({ width: s.width, height: s.height });
    };
  }), T = A(
    async (e) => {
      const r = te(e);
      if (r.length > 1)
        return p(d.TOO_MANY_FILES);
      try {
        const s = (await Promise.all(
          r.map(async (l) => {
            if (!t || !(t != null && t.width) || !(t != null && t.height)) return l;
            const u = await Z(l);
            if ((t == null ? void 0 : t.width) !== u.width || (t == null ? void 0 : t.height) !== u.height)
              throw d.MAX_DIMENSIONS;
            return l;
          })
        )).filter((l) => {
          if (!E.includes(l.type))
            throw d.DISALLOWED_FILE_TYPE;
          const u = x(m) ? m(l.type) ?? R : m;
          if (l.size > u)
            throw z({ type: l.type, limit: u }), d.VERY_LARGE_FILE;
          return !0;
        });
        p(""), b(s);
      } catch (i) {
        switch (i) {
          case d.DISALLOWED_FILE_TYPE:
          case d.VERY_LARGE_FILE:
          case d.MAX_DIMENSIONS:
            return p(i);
        }
      }
    },
    [E, m, p, b, t]
  );
  return /* @__PURE__ */ a(oe, { children: [
    /* @__PURE__ */ a(
      "div",
      {
        role: "region",
        className: w(o.dropzone, {
          [o.dropzoneDisabled]: c,
          [o.dropzoneDragOver]: V,
          [o.dropzoneError]: _
        }),
        onDragOver: c ? void 0 : k,
        onDragLeave: c ? void 0 : X,
        onDrop: c ? void 0 : j,
        children: [
          /* @__PURE__ */ a(
            "input",
            {
              type: "file",
              className: "adyen-pe-visually-hidden",
              id: N,
              ref: F,
              name: B,
              disabled: c,
              required: P,
              accept: String(E),
              onBlur: W,
              onChange: G,
              onInvalid: Q,
              "aria-invalid": _,
              "data-testId": "dropzone-input"
            }
          ),
          /* @__PURE__ */ a("label", { className: o.label, htmlFor: N, children: q ?? /* @__PURE__ */ a("div", {
            className: w(o.labelDefault),
            // prettier-ignore
            children: [
              _ ? /* @__PURE__ */ a(y, { name: "warning-filled", className: o.labelIcon }) : /* @__PURE__ */ a(y, { name: "upload", className: o.labelIcon }),
              /* @__PURE__ */ a(O, { className: o.labelText, el: M.SPAN, variant: $.BODY, stronger: !0, children: S.get("common.inputs.file.labels.default") })
            ]
          }) })
        ]
      }
    ),
    _ && /* @__PURE__ */ a("div", { className: o.error, children: [
      /* @__PURE__ */ a(y, { name: "cross-circle-fill", className: o.errorIcon }),
      /* @__PURE__ */ a(O, { className: o.errorText, el: M.SPAN, variant: $.BODY, children: x(L) ? L(
        f,
        h ? { size: h.limit, type: h.type } : void 0
      ) : S.get(f) })
    ] })
  ] });
});
export {
  Fe as Dropzone,
  Fe as default
};
