// @adyen/adyen-platform-experience-web/dist/es/components/internal/FilterBar/filters/DateFilter/DateFilter.js
import { jsx as b } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import I from "./DateFilterCore.js";
import { FilterParam as e } from "../../../../types.js";
import { useMemo as s, useState as O, useCallback as P } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import _ from "../../../../../core/Context/useCoreContext.js";
import { DateRangeFilterParam as T } from "./types.js";
import { EMPTY_OBJECT as A } from "../../../../../utils/value/constants.js";
const U = ({
  timezone: u,
  canResetFilters: i,
  defaultParams: r,
  filters: m,
  nowTimestamp: c,
  refreshNowTimestamp: l,
  sinceDate: R,
  untilDate: f,
  updateFilters: o
}) => {
  const { i18n: a } = _(), E = s(() => a.get("common.filters.types.date.label"), [a]), t = s(() => a.get(r.current.defaultTimeRange), [a]), [D, C] = O(t), g = P(
    (p = A) => {
      for (const [d, n] of Object.entries(p)) {
        switch (d) {
          case "selectedPresetOption":
            C(n || t);
            break;
          case T.FROM:
            o({
              [e.CREATED_SINCE]: n || r.current.defaultFilterParams[e.CREATED_SINCE]
            });
            break;
          case T.TO:
            o({
              [e.CREATED_UNTIL]: n || r.current.defaultFilterParams[e.CREATED_UNTIL]
            });
            break;
          default:
            return;
        }
        l();
      }
    },
    [t, l, o]
  );
  return s(() => !i && C(t), [i, t]), /* @__PURE__ */ b(
    I,
    {
      label: E,
      "aria-label": E,
      name: e.CREATED_SINCE,
      sinceDate: R,
      untilDate: f ?? new Date(c).toString(),
      from: m[e.CREATED_SINCE],
      to: m[e.CREATED_UNTIL],
      selectedPresetOption: D,
      timeRangePresetOptions: r.current.timeRangeOptions,
      timezone: u,
      onChange: g,
      showTimezoneInfo: !0,
      now: c
    }
  );
};
export {
  U as default
};
