// @adyen/adyen-platform-experience-web/dist/es/components/internal/FilterBar/filters/DateFilter/DateFilterCore.js
import { jsx as R } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import B from "../../../../../hooks/useTimezoneAwareDateFormatting.js";
import { useState as C, useCallback as L, useMemo as P, useEffect as T, useRef as x } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { BASE_LOCALE as z } from "../../../../../core/Localization/datetime/restamper/constants.js";
import M from "../../../../../core/Context/useCoreContext.js";
import { DateRangeFilterParam as A } from "./types.js";
/* empty css                    */
import I from "../BaseFilter/BaseFilter.js";
import V from "../../../DatePicker/DatePicker.js";
import { EMPTY_OBJECT as _ } from "../../../../../utils/value/constants.js";
import { CommitAction as v } from "../../../../../hooks/useCommitAction/types.js";
const Y = {
  month: "short",
  day: "numeric",
  year: "numeric"
}, k = (n, e, o, r) => {
  const m = o && e(o), t = r && e(r);
  if (m && t) return n.get("common.filters.types.date.range.between", { values: { fromDate: m, toDate: t } });
  if (m) return n.get("common.filters.types.date.range.since", { values: { date: m } });
  if (t) return n.get("common.filters.types.date.range.until", { values: { date: t } });
}, s = (n) => {
  try {
    return new Date(n || "").toISOString();
  } catch {
    return "";
  }
}, j = /* @__PURE__ */ (() => {
  const n = ({
    editAction: e,
    from: o,
    to: r,
    now: m,
    onChange: t,
    onValueUpdated: i,
    showTimezoneInfo: b,
    selectedPresetOption: g,
    timeRangePresetOptions: F,
    timeRangeSelectorLabel: c,
    timezone: D,
    sinceDate: l,
    untilDate: w
  }) => {
    const { i18n: p } = M(), { fullDateFormat: d } = B(D), [y, h] = C(g), f = P(() => [new Date(o), new Date(r)], [o, r]), a = x(), O = L(
      (u, S) => {
        i(k(p, d, s(u), s(S)));
      },
      [p, d, i]
    );
    return T(() => {
      var u, S, E;
      switch (e) {
        case v.APPLY:
          t({
            selectedPresetOption: y,
            [A.FROM]: s((u = a.current) == null ? void 0 : u.from),
            [A.TO]: s((S = a.current) == null ? void 0 : S.to)
          });
          break;
        case v.CLEAR:
          (E = a.current) == null || E.clear(), t();
      }
    }, [e, t, y]), /* @__PURE__ */ R(
      V,
      {
        ref: a,
        now: m,
        originDate: f,
        onHighlight: O,
        onPresetOptionSelected: h,
        selectedPresetOption: g,
        timeRangePresetOptions: F,
        timeRangeSelectorLabel: c,
        timezone: D,
        showTimezoneInfo: b,
        sinceDate: s(l),
        untilDate: s(w)
      }
    );
  };
  return (e) => /* @__PURE__ */ R(n, { ...e });
})(), q = (n, e, o) => n.formatRange(e, o);
function tt({
  title: n,
  from: e,
  to: o,
  selectedPresetOption: r,
  timeRangeSelectorLabel: m,
  ...t
}) {
  const { i18n: i } = M(), { fullDateFormat: b } = B(t.timezone), [g, F] = C(), [c, D] = C(), [l, w] = C(), p = L(
    (f) => {
      const { from: a, to: O, selectedPresetOption: u } = f ?? _;
      try {
        F(g ?? u), D(s(c ?? a)), w(s(l ?? O));
      } finally {
        t.onChange({ from: a, to: O, selectedPresetOption: u });
      }
    },
    [g, c, l, t]
  ), d = P(() => i.get("common.filters.types.date.rangeSelect.options.custom"), [i]), y = P(() => {
    const f = { ...Y, timeZone: t.timezone };
    let a = new Intl.DateTimeFormat(z, f);
    try {
      a = new Intl.DateTimeFormat(i.locale, f);
    } catch {
    }
    return a;
  }, [i, t.timezone]);
  T(() => F(r), [r]), T(() => D(s(e || Date.now())), [e]), T(() => w(s(o || Date.now())), [o]);
  const h = P(() => r === d && c && l ? q(y, new Date(c), new Date(l)) : r ?? t.label, [d, y, c, l, r, t.label]);
  return /* @__PURE__ */ R(
    I,
    {
      ...t,
      from: e,
      to: o,
      type: "date",
      label: h,
      onChange: p,
      render: j,
      selectedPresetOption: r,
      timeRangeSelectorLabel: m ?? i.get("common.filters.types.date.rangeSelect.label"),
      value: k(i, b, e, o),
      withContentPadding: !1
    }
  );
}
export {
  tt as default
};
