// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/InsightsTotals/InsightsTotals.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import u from "classnames";
import { memo as b } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useMemo as v } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { classes as r } from "./constants.js";
import { getTransactionCategory as N } from "../../../../utils/translation/getters.js";
import { TypographyVariant as p, TypographyElement as g } from "../../../../internal/Typography/types.js";
import y from "../AmountDisplay/AmountDisplay.js";
import f from "../../../../../core/Context/useCoreContext.js";
import k from "../../../../internal/Typography/Typography.js";
/* empty css                    */
import { ErrorMessageDisplay as A } from "../../../../internal/ErrorMessageDisplay/ErrorMessageDisplay.js";
import { Fragment as T } from "../../../../../external/preact/dist/preact.module.js";
import L from "../../../../internal/StructuredList/StructuredList.js";
const l = ({ currency: s, loadingTotals: i, totals: m, error: n }) => {
  const { i18n: a } = f(), o = s && m[s] || void 0;
  if (i || !o) {
    const t = Array.from({ length: 2 }), c = Array.from({ length: 3 }), d = u(r.skeleton, r.skeletonLoading);
    return /* @__PURE__ */ e("div", { className: r.root, children: [
      /* @__PURE__ */ e("span", { className: u(d, r.skeletonAmount, r.skeletonAmountLarge) }),
      /* @__PURE__ */ e("div", { className: r.breakdowns, children: t.map((x, h) => /* @__PURE__ */ e("div", { className: r.breakdown, children: [
        /* @__PURE__ */ e("span", { className: u(d, r.skeletonAmount) }),
        /* @__PURE__ */ e("div", { className: r.breakdownList, children: c.map((C, w) => /* @__PURE__ */ e("span", { className: d }, `breakdown-${w}`)) })
      ] }, `breakdown-${h}`)) })
    ] });
  }
  return /* @__PURE__ */ e("div", { className: r.root, children: n ? /* @__PURE__ */ e("div", { className: r.errorContainer, children: /* @__PURE__ */ e(
    A,
    {
      title: "common.errors.somethingWentWrong",
      message: ["common.errors.retry"],
      absolutePosition: !1,
      outlined: !1,
      withImage: !0,
      refreshComponent: !0
    }
  ) }) : /* @__PURE__ */ e(T, { children: [
    /* @__PURE__ */ e(
      y,
      {
        amount: o.total,
        currency: o.currency,
        label: a.get("transactions.overview.totals.tags.periodResult"),
        large: !0
      }
    ),
    /* @__PURE__ */ e("div", { className: r.breakdowns, children: [
      /* @__PURE__ */ e(
        l.Breakdown,
        {
          amount: o.incomings,
          breakdown: o.breakdown.incomings,
          currency: o.currency,
          label: a.get("transactions.overview.totals.tags.incoming")
        }
      ),
      /* @__PURE__ */ e(
        l.Breakdown,
        {
          amount: o.expenses,
          breakdown: o.breakdown.expenses,
          currency: o.currency,
          label: a.get("transactions.overview.totals.tags.outgoing")
        }
      )
    ] })
  ] }) });
};
l.Breakdown = ({ ariaLabel: s, breakdown: i, ...m }) => {
  const { currency: n } = m, { i18n: a } = f(), o = v(
    () => i.map(
      ({ category: t, value: c }) => ({
        key: N(a, t),
        value: `${a.amount(c, n)} ${n}`
      })
    ),
    [a, i, n]
  );
  return /* @__PURE__ */ e("div", { className: r.breakdown, children: [
    /* @__PURE__ */ e(y, { ...m }),
    /* @__PURE__ */ e(
      L,
      {
        items: o,
        "aria-label": s,
        classNames: r.breakdownList,
        renderLabel: (t) => /* @__PURE__ */ e(k, { el: g.SPAN, variant: p.CAPTION, children: t }),
        renderValue: (t) => /* @__PURE__ */ e(k, { el: g.SPAN, variant: p.CAPTION, stronger: !0, children: t })
      }
    )
  ] });
};
const R = b(l);
export {
  R as default
};
