// @adyen/adyen-platform-experience-web/dist/es/config/disputes/defenseReasonConfig.json.js
const e = { title: "atmDispute", help: "invalidChargebackReasonCode" }, t = { title: "accountTakeover", help: "transactionFromAccountTakeover" }, i = { title: "additionalInformation", help: "merchantCanProvideAdditionalInformation" }, n = { title: "airlineCompellingEvidence", help: "evidenceCardholderAuthorizedTransaction" }, o = { title: "airlineFlightProvided", help: "shopperUsedAirlineTicket" }, a = { title: "responseToRfiAirlines", help: "provideTransactionalInfoTravelDetails" }, r = { title: "transactionWasAuthorised", help: "authorisationReceivedForAmount" }, c = { title: "autoResponseToRfi", help: "autoResponseSentToIssuer" }, s = { title: "cvc2ValidationProgram", help: "transactionFallsCvc2ValidationProgram" }, l = { title: "cancellationOrReturns", help: "useThisDefenseReasonIfApplicable", helpitems: ["noCreditSlip", "returnNotAccepted", "merchandiseNotReturned"] }, d = { title: "cancellationContractFailed", help: "cardholderDidNotMeetCancellationTerms" }, h = { title: "responseToRfiCarRental", help: "provideTransactionalInfoTravelDetails" }, p = { title: "accountNumberNotListedInExceptions", help: "accountNumberNotListedInExceptionsFor60Days" }, u = { title: "reponseToRfi", help: "provideTransactionalInformation" }, C = { title: "fraudEvidence" }, m = { title: "cbCodeNotApplicable", help: "firstChargebackCodeNotApplicable" }, f = { title: "reponseToRfiCardPresent", help: "provideTransactionalInformation" }, v = { title: "cardholderNotifiedButNoReply", help: "cardholderNotifiedBeforeRecurringTransaction" }, T = { title: "cardholderParticipated", help: "evidenceThatCardholderParticipated" }, g = { title: "membershipRenewedAfterCancellation", help: "cardholderCancelledRecurringServiceButRenewedLater" }, R = { title: "chipLiabilityShift", help: "transactionBetweenChipLiabilityShiftProgramCustomers" }, A = { title: "chipTransactionReportedSAFE", help: "intraregionalTransactionReportedSAFE" }, P = { title: "reponseToRfi", help: "provideTransactionalInformation" }, S = { title: "compellingEvidence", help: "evidenceCardholderAuthorizedTransaction" }, b = { title: "compellingEvidenceRecurringTransactions", help: "evidenceServicesProvidedAndReceivedWrittenCorrespondence" }, I = { title: "correctTransactionAmount", help: "transactionInformationDocument" }, N = { title: "correctTransactionCurrency", help: "chargebackInvalidCorrectAmountAndCurrencyProvided" }, D = { title: "correctTransactionDateProvided", help: "acquirerProvidesCorrectDateWithinTimeLimit" }, y = { title: "creditOrCancellationPolicyDisclosed", help: "creditOrCancellationPolicyDisclosedToCardholder" }, F = { title: "creditPreviouslyIssued", help: "transactionAmountAlreadyRefunded" }, M = { title: "deficiencyCorrected", help: "proofDeficiencyCorrected" }, k = { title: "disputedSurcharge", help: "surchargeCorrectlyProcessed" }, O = { title: "disputedSurchargeIncorrectCalculation", help: "surchargeIncorrectlyCalculatedByIssuer" }, E = { title: "duplicateChargeback", help: "issuerProcessedChargebackMoreThanOnce" }, W = { title: "emvLiabilityShift", help: "emvDeviceNonEmvCard" }, w = { title: "emergencyPaymentAuthorizationService", help: "transactioAuthorisedEmergencyPaymentAuthorizationService" }, B = { title: "faceToFaceTransactionNoConflictingInformation", help: "transactionWasInFaceToFaceEnvironment" }, L = { title: "faceToFaceTransactionWithConflictingInformation", help: "de22DontMatchInFaceToFaceTransaction" }, x = { title: "goodsNotReturned", help: "cardholderShouldHaveReturnedGoods" }, q = { title: "goodsOrServicesProvided", help: "proofCardholderReceivedMerchandise", helpitems: ["signedImprintedSalesSlipInvoicePosReceipt", "parcelServiceReceipt", "waiverFormAbsolvingMerchantResponsibility"] }, V = { title: "goodsRepairedOrReplaced", help: "goodsWereRepairedOrReplaced" }, G = { title: "goodsWereAsDescribed", help: "evidenceGoodsNotDamaged" }, U = { title: "identifiedAddendum", help: ["cardholderResponsibleForAddendumTransaction", "cardholderResponsibleForDisputedAmount", "cardholderBilledForSeparateAdditionalAmount"] }, z = { title: "documentationReceivedWasIllegible", help: "firstChargebackInvalid" }, J = { title: "increasedTransactionAmount", help: "documentationIncreasedDebitCardholderAccount" }, H = { title: "invalidAcquirerReferenceData", help: "invalidReferenceDataDocumentationNotRequired" }, j = { title: "invalidCancellationCode", help: "cancellationCodeWasInvalidAsShownByExplanationProvided" }, Q = { title: "invalidChargeback", help: ["firstChargebackDoesNotMeetPrerequisites", "thisReasonIsTechnical"] }, X = { title: "invalidChargeback", help: ["firstChargebackDoesNotMeetPrerequisites", "thisReasonIsTechnical"] }, K = { title: "chargebackBundlingInvalid", help: "chargebackBundlingInvalid" }, Y = { title: "invalidMessageText", help: "formatCbmmddyyArdXxxExpected" }, Z = { title: "invalidMessageText", help: "formatMultipleTransactionsNnnExpected" }, _ = { title: "invalidMessageText", help: "formatRpcsMmddyyExpected" }, $ = { title: "invalidChargebackBasedOnMcc", help: "merchantCantReceiveThisChargebackBasedOnTheirMcc" }, ee = { title: "reponseToRfi", help: "provideTransactionalInformation" }, te = { title: "justifiedDelayedPresentment", help: "delayInSubmittingTransaction" }, ie = { title: "liabilityShiftFullAuthenticated", help: "transactionIssuerApproved3dSecureAuthenticated" }, ne = { title: "3dSecureFullyAuthenticatedTransaction", help: "transactionApproved3dSecureAuthenticated" }, oe = { title: "3dSecureLiabilityShiftNotFullyAuthenticated", help: "transactionNotFully3dSecureAuthenticated" }, ae = { title: "merchandiseWasNotCounterfeit" }, re = { title: "merchandiseWasReceivedByShopper", help: "proofCardholderReceivedMerchandise", helpitems: ["signedImprintedSalesSlipInvoicePosReceipt", "parcelServiceReceipt", "waiverFormAbsolvingMerchantResponsibility"] }, ce = { title: "provideMissingInformation", help: "merchantProvideMissingDocuments" }, se = { title: "invalidMessageText", help: "issuerDidNotIncludeTwoSetsOfArns" }, le = { title: "noShowTransaction", help: "transactionWasResultOfGuaranteedReservation" }, de = { title: "requiredDocumentationNotReceived", help: "issuerDidNotProvideRequiredDocumentation" }, he = { title: "requiredDocumentationNotReceived", help: "issuerDidNotProvideRequiredDocumentation" }, pe = { title: "not3dSecureTransaction", help: "chargebackCodeNA3dSecureNotOffered" }, ue = { title: "serviceNotCancelled", help: "serviceNotCancelledEvidence" }, Ce = { title: "noPriorChargebackReceived", help: "claimNotJustifiedSinceTransactionNotChargebacked" }, me = { title: "noPriorChargebackReceived", help: "transactionNotChargebackedAutoDefense" }, fe = { title: "merchantNotNotifiedOfCancellation", help: "serviceCancelledButMerchantNotNotified" }, ve = { title: "additionalInformation", help: "docToProveNoPaymentByOtherMean" }, Te = { title: "transactionNotRecurring", help: "transactionNotRecurringTransactionInstallment" }, ge = { title: "pinTransaction", help: "pinUsedToVerifyCardholder" }, Re = { title: "pastChargebackTimeLimit", help: "chargebackReceivedAfterMastercardTimeLimit" }, Ae = { title: "paypassTransaction", help: "paypassAmountBelowProtectionAmount" }, Pe = { title: "paymentByOtherMeans", help: "validityOfTransactionChargedExplanation" }, Se = { title: "transactionSettledWithinTimeLimit", help: "timeBetweenAuthorisationAndSettlementWithinTimeLimits" }, be = { title: "noPriorChargebackReceived", help: "previousChargebackCancellingRecurringContract" }, Ie = { title: "chipTransaction", help: "cardPresentAndChipUsedNoPinProvided" }, Ne = { title: "transactionsInitiatedDueToLossTheftOrDamages", help: "receiptWithItemsBilledBecauseOfLossTheftDamages" }, De = { title: "proofOfCardPresenceMastercardCorporateFleet", help: "cardAndSignatureForCorporateFleetCardTransaction" }, ye = { title: "proofOfCardPresenceAvailableNotAuthorisedMastercardNetwork", help: "cardWithSignatureNotProcessedMastercardNetwork" }, Fe = { title: "signedTerminalReceiptAvailable", help: "posTransactionAndSignedTerminalReceiptAvailable" }, Me = { title: "cardPresenceProof", help: "posTransactionCardholderWasPresentProof" }, ke = { title: "provideInformationForCollectionOrFraudCase" }, Oe = { title: "purchaseProperlyPosted", help: "transactionWasProperlyIdentifiedAsInstallmentTransaction" }, Ee = { title: "qpsTransaction", help: "qpsTransactionWithAmountLessThanChargebackProtectionAmount" }, We = { title: "recurringTransactionCompellingEvidence", help: "autoGeneratedRecurringTransactionDetails" }, we = { title: "recurringTransactionCompellingMerchantEvidence", help: "merchantCanProvideAdditionalCompellingEvidence" }, Be = { title: "notAgreeWithRefund", help: "shopperClaimsRefundWasPromisedButNotExecuted" }, Le = { title: "retrievalRequestFulfilled", help: "documentsSentToFulfillRetrievalRequest" }, xe = { title: "buyerExceededWindowToFileReturn", help: "uploadReturnPolicyDocument" }, qe = { title: "termsOfSaleWereNotMisrepresented" }, Ve = { title: "scanningError", help: "defenseReasonIfProblemsWithScans" }, Ge = { title: "goodsOrServicesProvided", help: "merchantProvidedServicesToCardholderProof" }, Ue = { title: "servicesProvidedAfterCancellation", help: "servicesProvidedAndUsedAfterCancellation" }, ze = { title: "addressVerificationService", help: "addressMerchandiseSentAvsConfirmed" }, Je = { title: "addressVerificationServiceAdditionalInformation", help: "addressMerchandiseSentAvsConfirmed" }, He = { title: "supplyDefenseMaterial", help: "fallbackDefenseReasonCode" }, je = { title: "supplyDefenseMaterial", help: "fallbackDefenseReasonCode" }, Qe = { title: "manyFraudulentChargebacks", help: "issuerHasInitiatedTooManyFraudulentDisputes" }, Xe = { title: "authorisedOnline", help: "transactionWasAuthorisedOnline" }, Ke = { title: "notPosTransaction", help: "transactionWasECommerceGivenChargebackReasonNotApplicable" }, Ye = { title: "twoDifferentTransactions", help: "twoSeparateTransactions" }, Ze = { title: "twoDifferentTidsNonAtm", help: "twoDifferentTidsWithSameCardholderAccount" }, _e = { title: "pinTransaction", help: "pinForOneOrBothTransactions" }, $e = { title: "twoPreviousFraudRelatedChargebacks", help: "prevChargedbackTwoTransactionsWithSameAccount" }, et = { title: "unfoundedCardholderDispute", help: "acquirerDefendChargebackIfIssuerRequested" }, tt = { title: "unreasonableAmount", help: "cardholderAgreedToAmountAsReasonable" }, it = {
  ATMDispute: e,
  AccountTakeover: t,
  AdditionalInformation: i,
  AirlineCompellingEvidence: n,
  AirlineFlightProvided: o,
  AirlineRFIresponse: a,
  AuthorisedTransaction: r,
  AutomaticRFIresponse: c,
  CVC2ValidationProgram: s,
  CancellationOrReturns: l,
  CancellationTermsFailed: d,
  CarRentalRFIresponse: h,
  CardNotOnExceptionFile: p,
  CardNotPresentRFIresponse: u,
  CardPresentFraud: C,
  CardPresentNotApplicable: m,
  CardPresentRFIresponse: f,
  CardholderNotReplied: v,
  CardholderParticipated: T,
  CardholderRenewedMembership: g,
  ChipLiabilityShift: R,
  ChipTransactionReportedToSAFE: A,
  CieloRFIresponse: P,
  CompellingEvidence: S,
  CompellingEvidenceRecurringTransactions: b,
  CorrectTransactionAmount: I,
  CorrectTransactionCurrency: N,
  CorrectTransactionDate: D,
  CreditOrCancellationPolicyProperlyDisclosed: y,
  CreditPreviouslyIssued: F,
  DeficiencyCorrected: M,
  DisputedSurcharge: k,
  DisputedSurchargeIncorrectProratedCalculation: O,
  DuplicateChargeback: E,
  EMVLiabilityShift: W,
  EmergencyPaymentAuthorizationService: w,
  FaceToFace: B,
  FaceToFaceConflictingMessages: L,
  GoodsNotReturned: x,
  GoodsOrServicesProvided: q,
  GoodsRepairedOrReplaced: V,
  GoodsWereAsDescribed: G,
  IdentifiedAddendum: U,
  IllegibleDocumentationReceived: z,
  IncreasedTransactionAmount: J,
  InvalidARD: H,
  InvalidCancellationCode: j,
  InvalidChargeback: Q,
  InvalidChargebackAllocation: X,
  InvalidChargebackBundling: K,
  InvalidDE72CB: Y,
  InvalidDE72MultipleTransactions: Z,
  InvalidDE72RPCS: _,
  InvalidReasonCodeForMCC: $,
  JCBRFIResponse: ee,
  JustifiedDelayedPresentment: te,
  LiabilityShiftFullAuthenticated: ie,
  LiabilityShiftFullUCAF: ne,
  LiabilityShiftMerchantUCAF: oe,
  MerchandiseNotCounterfeit: ae,
  MerchandiseReceived: re,
  MissingInformation: ce,
  NoARDprovidedInMessageText: se,
  NoShowTransaction: le,
  NonReceiptOfRequiredDocuments: de,
  NonReceiptOfRequiredDocumentsAutoDefense: he,
  Not3DSecure: pe,
  NotCancelled: ue,
  NotChargebacked: Ce,
  NotChargebackedAuto: me,
  NotNotified: fe,
  NotPaidByOtherMeans: ve,
  NotRecurring: Te,
  PINTransaction: ge,
  PastChargebackTimeLimit: Re,
  PayPassTransaction: Ae,
  PaymentByOtherMeans: Pe,
  PresentmentWithinTimeLimit: Se,
  PreviousTransactionNotChargebacked: be,
  ProofOfCardPresenceAndSignatureChipNoPIN: Ie,
  ProofOfCardPresenceAndSignatureLossTheftOrDamage: Ne,
  ProofOfCardPresenceAndSignatureMasterCardCorporateFleetCard: De,
  ProofOfCardPresenceAndSignatureNotMasterCardWorldWideNetwork: ye,
  ProofOfCardPresenceAndSignatureWithTerminalReceipt: Fe,
  ProofOfCardPresenceAndSignatureWithoutTerminalReceipt: Me,
  ProvideInformation: ke,
  PurchaseProperlyPosted: Oe,
  QPSTransaction: Ee,
  RecurringTransactionsCompellingEvidence: We,
  RecurringTransactionsCompellingMerchantEvidence: we,
  RefundNotAgreed: Be,
  RetrievalRequestFulfilled: Le,
  ReturnPeriodExpired: xe,
  SaleTermsNotMisrepresented: qe,
  ScanningError: Ve,
  ServicesProvided: Ge,
  ServicesProvidedAfterCancellation: Ue,
  ShippedToAVS: ze,
  ShippedToAVSAdditionalInformation: Je,
  SupplyDefenseDocument: He,
  SupplyDefenseMaterial: je,
  TooManyFraudulentChargebacks: Qe,
  TransactionAuthorisedOnline: Xe,
  TransactionNotPOS: Ke,
  TwoDifferentTIDs: Ye,
  TwoDifferentTIDsPOS: Ze,
  TwoPINTransactions: _e,
  TwoPriorFraudChargebacks: $e,
  UnfoundedCardholderDispute: et,
  UnreasonableAmount: tt
};
export {
  e as ATMDispute,
  t as AccountTakeover,
  i as AdditionalInformation,
  n as AirlineCompellingEvidence,
  o as AirlineFlightProvided,
  a as AirlineRFIresponse,
  r as AuthorisedTransaction,
  c as AutomaticRFIresponse,
  s as CVC2ValidationProgram,
  l as CancellationOrReturns,
  d as CancellationTermsFailed,
  h as CarRentalRFIresponse,
  p as CardNotOnExceptionFile,
  u as CardNotPresentRFIresponse,
  C as CardPresentFraud,
  m as CardPresentNotApplicable,
  f as CardPresentRFIresponse,
  v as CardholderNotReplied,
  T as CardholderParticipated,
  g as CardholderRenewedMembership,
  R as ChipLiabilityShift,
  A as ChipTransactionReportedToSAFE,
  P as CieloRFIresponse,
  S as CompellingEvidence,
  b as CompellingEvidenceRecurringTransactions,
  I as CorrectTransactionAmount,
  N as CorrectTransactionCurrency,
  D as CorrectTransactionDate,
  y as CreditOrCancellationPolicyProperlyDisclosed,
  F as CreditPreviouslyIssued,
  M as DeficiencyCorrected,
  k as DisputedSurcharge,
  O as DisputedSurchargeIncorrectProratedCalculation,
  E as DuplicateChargeback,
  W as EMVLiabilityShift,
  w as EmergencyPaymentAuthorizationService,
  B as FaceToFace,
  L as FaceToFaceConflictingMessages,
  x as GoodsNotReturned,
  q as GoodsOrServicesProvided,
  V as GoodsRepairedOrReplaced,
  G as GoodsWereAsDescribed,
  U as IdentifiedAddendum,
  z as IllegibleDocumentationReceived,
  J as IncreasedTransactionAmount,
  H as InvalidARD,
  j as InvalidCancellationCode,
  Q as InvalidChargeback,
  X as InvalidChargebackAllocation,
  K as InvalidChargebackBundling,
  Y as InvalidDE72CB,
  Z as InvalidDE72MultipleTransactions,
  _ as InvalidDE72RPCS,
  $ as InvalidReasonCodeForMCC,
  ee as JCBRFIResponse,
  te as JustifiedDelayedPresentment,
  ie as LiabilityShiftFullAuthenticated,
  ne as LiabilityShiftFullUCAF,
  oe as LiabilityShiftMerchantUCAF,
  ae as MerchandiseNotCounterfeit,
  re as MerchandiseReceived,
  ce as MissingInformation,
  se as NoARDprovidedInMessageText,
  le as NoShowTransaction,
  de as NonReceiptOfRequiredDocuments,
  he as NonReceiptOfRequiredDocumentsAutoDefense,
  pe as Not3DSecure,
  ue as NotCancelled,
  Ce as NotChargebacked,
  me as NotChargebackedAuto,
  fe as NotNotified,
  ve as NotPaidByOtherMeans,
  Te as NotRecurring,
  ge as PINTransaction,
  Re as PastChargebackTimeLimit,
  Ae as PayPassTransaction,
  Pe as PaymentByOtherMeans,
  Se as PresentmentWithinTimeLimit,
  be as PreviousTransactionNotChargebacked,
  Ie as ProofOfCardPresenceAndSignatureChipNoPIN,
  Ne as ProofOfCardPresenceAndSignatureLossTheftOrDamage,
  De as ProofOfCardPresenceAndSignatureMasterCardCorporateFleetCard,
  ye as ProofOfCardPresenceAndSignatureNotMasterCardWorldWideNetwork,
  Fe as ProofOfCardPresenceAndSignatureWithTerminalReceipt,
  Me as ProofOfCardPresenceAndSignatureWithoutTerminalReceipt,
  ke as ProvideInformation,
  Oe as PurchaseProperlyPosted,
  Ee as QPSTransaction,
  We as RecurringTransactionsCompellingEvidence,
  we as RecurringTransactionsCompellingMerchantEvidence,
  Be as RefundNotAgreed,
  Le as RetrievalRequestFulfilled,
  xe as ReturnPeriodExpired,
  qe as SaleTermsNotMisrepresented,
  Ve as ScanningError,
  Ge as ServicesProvided,
  Ue as ServicesProvidedAfterCancellation,
  ze as ShippedToAVS,
  Je as ShippedToAVSAdditionalInformation,
  He as SupplyDefenseDocument,
  je as SupplyDefenseMaterial,
  Qe as TooManyFraudulentChargebacks,
  Xe as TransactionAuthorisedOnline,
  Ke as TransactionNotPOS,
  Ye as TwoDifferentTIDs,
  Ze as TwoDifferentTIDsPOS,
  _e as TwoPINTransactions,
  $e as TwoPriorFraudChargebacks,
  et as UnfoundedCardholderDispute,
  tt as UnreasonableAmount,
  it as default
};
