// @adyen/adyen-platform-experience-web/dist/es/config/disputes/defenseDocumentConfig.json.js
const e = { title: "clearingText", help: "descriptionOfDisputeReason" }, t = { title: "latestRecurringTransactions", help: "latestRecurringTransactions" }, o = { title: "acquirerRepresentmentForm", help: "acquirerRepresentmentFormAutomaticallyGenerated" }, n = { title: "acquirerRetrievalFulfilmentForm", help: "acquirerRetrievalFulfilmentFormAutomaticallyGenerated" }, i = { title: "automaticallyGeneratedTransactionDetails", help: "automaticallyGeneratedTransactionDetails" }, r = { title: "proofOfAddendum", help: "cardholderResponsibleForAddendumTransaction" }, a = { title: "additionalInformation", help: "docIdentifyTransaction" }, l = { title: "additionalTransactions", help: "additionalTransactionsConnectedWithDisputedFlight" }, s = { title: "alternativeDefenseMaterial", help: "docRemediesChargeback" }, c = { title: "amexFaxCover", help: "coverPageForAmexDisputes" }, d = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, p = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, h = { title: "rebuttalCancellationNotAccepted", help: "rebuttalStatingCancellationNotAccepted" }, m = { title: "evidenceCardHolderParticipation", help: "docProvingCardHolderParticipated", helpitems: ["receiptOrOther", "customerWrittenConfirmation", "writtenCorrespondenceExchanged"] }, f = { title: "proofRenewedMembership", help: "proofShowingCardholderRenewedMembership" }, u = { title: "proofCardPresence", help: "proofCardAndCardholderSignature" }, g = { title: "docChargebackCodeNotApplicable" }, T = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, C = { title: "defenseMaterial", help: "docViaFraudReporter" }, R = { title: "CopyOfSalesDraftTitle", help: "CopyOfSalesDraftDetails" }, S = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, A = { title: "defenseMaterial", help: "docCorrectCurrency" }, b = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, v = { title: "documentShipmentDate", help: "documentContainingImportantShipmentData" }, P = { title: "defenseMaterial", help: "docRemediesChargeback" }, O = { title: "disclosureAtPointOfInteraction", help: "rebuttalGivenToCardholder" }, M = { title: "firstChargebackNumberAndDate", help: "numberAndDateOfOriginalChargeback" }, N = { title: "compellingEvidence", help: "evidenceProofingParticipationOfCardholder" }, D = { title: "flightManifest", help: "flightManifestShowingCardholderName" }, F = { title: "flightTicket", help: "flightTicketWithCardholderName" }, x = { title: "deliveryOfFlightTicketAtAddress", help: "receiptOfFlightTicketAtBillingAddress" }, I = { title: "flightTicketUsed", help: "proofAirlineTicketsWereUsed" }, k = { title: "flightTookPlace", help: "proofFlightTookPlace" }, y = { title: "frequentFlyerInformation", help: "creditsOfMilesShowingConnectionToCardholder" }, H = { title: "merchantWrittenRebuttal", help: "statementMerchantDidNotReceiveGoods" }, w = { title: "merchantWrittenRebuttal", help: "writtenRebutalGoodsRepairedOrReplaced" }, B = { title: "merchantWrittenRebuttal", help: "proofGoodsWereDeliveredAsDescribed" }, W = { title: "proofValidIncreaseOfAmount", help: "cardholderResponsibleForDisputedAmount" }, E = { title: "explanation", help: "explanationWhyCancellationCodeInvalid" }, q = { title: "proofOfInvalidChargeback", help: "chargebackRemediedOrInvalid" }, G = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, L = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, z = { title: "merchandiseDescription", help: "descriptionOfMerchandiseOrServices" }, U = { title: "defenseMaterial", help: "docMerchandiseNotCounterfeit" }, V = { title: "proofOfRecurringTransaction", help: "docAllOfFollowing", helpitems: ["evidenceTransactionWasRecurring", "moreThanOneTransactionProcessed", "previousTransactionsNotDisputed"] }, J = { title: "mpiData", help: "resultsOfMpiCalls" }, X = { title: "proofOfNoCancellation", help: "proofRecurringContractNotCancelledAtTimeOfSettlement" }, Y = { title: "proofOfNoChargebackReceived", help: "proofMerchantHadNotReceivedPreviousChargeback" }, j = { title: "proofOfNoChargebackReceived", help: "proofMerchantHadNotReceivedPreviousChargebackAutoGenerated" }, K = { title: "noNotification", help: "proofMerchantNotNotifiedOfCancellation" }, Q = { title: "originalAmount", help: "originalAmountIfDisputedRepresentsPartialShipment" }, Z = { title: "paperAirlineTicket", help: "docCardholderIssuedPaperAirlineTickets" }, _ = { title: "passengerId", help: "passengerIdLinkedToCardholder" }, $ = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, ee = { title: "appropriateExplanation", help: "appropriateExplanationAndDocTwoSeparateTransactions" }, te = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, oe = { title: "avsDocumentation", help: "documentationOfPositiveAvsResponseXOrY" }, ne = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, ie = { title: "signedTerminalReceipt", help: "printedSignedReceipt" }, re = { title: "proofOfDelayedDelivery", help: "ifDelayedProofShowingMerchantAbleToProvideServices" }, ae = { title: "proofOfAccountTakeover", help: "provingTransactionResultedFromAccountTakeover" }, le = { title: "proofCardPresence", help: "docTransactionOccurredOnPos" }, se = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, ce = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, de = { title: "signatureOrChipPinEvidence" }, pe = { title: "proofProvidedMerchandise", help: "proofGoodsServicesWereProvided" }, he = { title: "proofOfNoShow", help: "merchantObtainedCardAtTimeReservationMade" }, me = { title: "proofOfRecurringTransaction", help: "docAllOfFollowing", helpitems: ["evidenceTransactionWasRecurring", "moreThanOneTransactionProcessed", "previousTransactionsNotDisputed"] }, fe = { title: "proofOfRetailSale", help: "copyOfTid" }, ue = { title: "reasonForInvalidation", help: "reasonForInvalidationOfChargeback" }, ge = { title: "retrievalRequestFulfilled", help: "autoGeneratedDocRetrievalRequestFulfilled" }, Te = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, Ce = { title: "defenseMaterial", help: "docTermsOfSaleNotMisrepresented" }, Re = { title: "shipmentDocumentation", help: "merchandiseSentToAvsConfirmedBillingAddress" }, Se = { title: "shipToAddress", help: "shipToAddressIfApplicable" }, Ae = { title: "SupplementalDocumentsTitle", help: "SupplementalDocumentsDetails" }, be = { title: "TEDocumentTitle", help: "TEDocumentDetails" }, ve = { title: "copyOfInvoice", help: "copyOfInvoiceOrOtherRelevantTransactionDetails" }, Pe = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, Oe = { title: "docTwoDifferentTransactions", help: "docsTwoTransactionsWithSameShopper" }, Me = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, Ne = { title: "memberMessageText", help: "memberMessageTextSentToSchemes" }, De = { title: "reasonableAmount", help: "cardholderAgreedToAmountRange" }, Fe = { title: "duplicateProcessing", help: "duplicateProcessingHelp" }, xe = { title: "chargeToWrongAccountNumber", help: "chargeToWrongAccountNumberHelp" }, Ie = { title: "SplitSales", help: "SplitSalesHelp" }, ke = { title: "missingSignature", help: "missingSignatureHelp" }, ye = { title: "processingError", help: "processingErrorHelp" }, He = { title: "expiredCard", help: "expiredCardHelp" }, we = { title: "latePresentment", help: "latePresentmentgHelp" }, Be = { title: "falseTransaction", help: "falseTransactionHelp" }, We = { title: "authorizationNotObtained", help: "authorizationNotObtainedHelp" }, Ee = { title: "transactionNotRecognizedContactlessAndCardNotPresented", help: "transactionNotRecognizedContactlessAndCardNotPresentedHelp" }, qe = { title: "transactionNotRecognizedChipLiabilityShift", help: "transactionNotRecognizedChipLiabilityShiftHelp" }, Ge = { title: "deffectiveMerchandise", help: "deffectiveMerchandiseHelp" }, Le = { title: "merchandiseNotReceived", help: "merchandiseNotReceivedHelp" }, ze = { title: "serviceNotRendered", help: "serviceNotRenderedHelp" }, Ue = { title: "cardRecoveryBulletinOrExceptionFile", help: "cardRecoveryBulletinOrExceptionFileHelp" }, Ve = { title: "transactionAfterReservationCancelled", help: "transactionAfterReservationCancelledHelp" }, Je = { title: "differentSignature", help: "differentSignatureHelp" }, Xe = { title: "retrievalRequestNotHonored", help: "retrievalRequestNotHonoredHelp" }, Ye = { title: "creditNotProcessed", help: "creditNotProcessedHelp" }, je = { title: "paidByOtherMeansThanJCBCard", help: "paidByOtherMeansThanJCBCardHelp" }, Ke = { title: "nonMatchingAccountNumber", help: "nonMatchingAccountNumberHelp" }, Qe = { title: "merchantFraudPerformanceProgram", help: "merchantFraudPerformanceProgramHelp" }, Ze = { title: "other", help: "otherHelp" }, _e = { title: "proofOfFulfillment", help: "proofShowingTrackingInformation" }, $e = { title: "proofOfRefund", help: "uploadListOfRefundIds" }, et = { title: "merchandiseNotAsDescribed", help: "documentCanBetUrlReturnPolicy" }, tt = { title: "cancelledRecurringBilling", help: "documentCanBeSubscriptionAgreement" }, ot = { title: "incorrectAmount", help: "provideRefundId" }, nt = { title: "incorrectAmount", help: "provideRefundId" }, it = { title: "unauthorisedReason", help: "documentCanBeProofOfDelivery" }, rt = { title: "creditNotProcessedReason", help: "documentCanBeCreditReasonDue" }, at = {
  AcquirerMemberMessageText: e,
  AcquirerRecurringTransactionDetails: t,
  AcquirerRepresentmentForm: o,
  AcquirerRetrievalFulfilmentForm: n,
  AcquirerRFItransactionDetails: i,
  AddendumDocumentation: r,
  AdditionalInformation: a,
  AdditionalTransactions: l,
  AlternativeDefenseMaterial: s,
  AmexFaxCover: c,
  ARDnotProvided: d,
  AuthorisedTransaction: p,
  CancellationNeverAccepted: h,
  CardholderIdentification: m,
  CardholderRenewedMembership: f,
  CardPresent: u,
  ChargebackCodeNotApplicable: g,
  ChipLiabilityShift: T,
  ChipTransactionReportedToSAFE: C,
  CopyOfSalesDraft: R,
  CompellingEvidence: S,
  CorrectTransactionCurrency: A,
  CreditPreviouslyIssued: b,
  DateMerchandiseShipped: v,
  DefenseMaterial: P,
  DisclosureAtPointOfInteraction: O,
  DuplicateChargeback: M,
  EvidenceOfCardholderParticipation: N,
  FlightManifest: D,
  FlightTicket: F,
  FlightTicketAtBillingAddress: x,
  FlightTicketUsed: I,
  FligthTookPlace: k,
  FrequentFlyer: y,
  GoodsNotReturned: H,
  GoodsRepairedOrReplaced: w,
  GoodsWereAsDescribed: B,
  IncreasedTransactionAmount: W,
  InvalidCancellationCode: E,
  InvalidChargeback: q,
  LiabilityShiftFullUCAF: G,
  LiabilityShiftMerchantUCAF: L,
  MerchandiseDescription: z,
  MerchandiseNotCounterfeit: U,
  MerchantProofOfRecurringTransaction: V,
  MPIdata: J,
  NotCancelled: X,
  NotChargebacked: Y,
  NotChargebackedAuto: j,
  NotNotified: K,
  OriginalAmount: Q,
  PaperAirlineTicket: Z,
  PassengerIdentification: _,
  PastChargebackTimeLimit: $,
  PaymentByOtherMeans: ee,
  PINTransaction: te,
  PositiveAVS: oe,
  PresentmentWithinTimeLimit: ne,
  PrintedSignedTerminalReceipt: ie,
  ProofOfAbilityToProvideServices: re,
  ProofOfAccountTakeover: ae,
  ProofOfAttendedPOSTerminal: le,
  ProofOfCardPresenceAndSignatureChipNoPIN: se,
  ProofOfCardPresenceAndSignatureWithTerminalReceipt: ce,
  ProofOfCardPresenceSignatureOrChipPIN: de,
  ProofOfGoodsOrServicesProvided: pe,
  ProofOfNoShow: he,
  ProofOfRecurringTransaction: me,
  ProofOfRetailSaleRatherThanCredit: fe,
  ReasonForReturn: ue,
  RetrievalRequestFulfilled: ge,
  ReturnedMerchandiseNotReceived: Te,
  SaleTermsNotMisrepresented: Ce,
  ShippedToAVS: Re,
  ShipToAddress: Se,
  SupplementalDocuments: Ae,
  TEDocument: be,
  TIDorInvoice: ve,
  TransactionNotPOS: Pe,
  TwoDifferentTIDs: Oe,
  TwoPINTransactions: Me,
  TwoPriorFraudChargebacks: Ne,
  UnreasonableAmount: De,
  DuplicateProcessing: Fe,
  ChargeToWrongAccountNumber: xe,
  SplitSales: Ie,
  MissingSignature: ke,
  ProcessingError: ye,
  ExpiredCard: He,
  LatePresentment: we,
  FalseTransaction: Be,
  AuthorizationNotObtained: We,
  TransactionNotRecognizedContactlessAndCardNotPresented: Ee,
  TransactionNotRecognizedChipLiabilityShift: qe,
  DeffectiveMerchandise: Ge,
  MerchandiseNotReceived: Le,
  ServiceNotRendered: ze,
  CardRecoveryBulletinOrExceptionFile: Ue,
  TransactionAfterReservationCancelled: Ve,
  DifferentSignature: Je,
  RetrievalRequestNotHonored: Xe,
  CreditNotProcessed: Ye,
  PaidByOtherMeansThanJCBCard: je,
  NonMatchingAccountNumber: Ke,
  MerchantFraudPerformanceProgram: Qe,
  Other: Ze,
  ProofOfFulfillment: _e,
  ProofOfRefund: $e,
  MerchandiseNotAsDescribedReason: et,
  CancelledRecurringBillingReason: tt,
  IncorrectAmountReason: ot,
  ProofOfRefundForAmountDifference: nt,
  UnauthorisedReason: it,
  CreditNotProcessedReason: rt
};
export {
  d as ARDnotProvided,
  e as AcquirerMemberMessageText,
  i as AcquirerRFItransactionDetails,
  t as AcquirerRecurringTransactionDetails,
  o as AcquirerRepresentmentForm,
  n as AcquirerRetrievalFulfilmentForm,
  r as AddendumDocumentation,
  a as AdditionalInformation,
  l as AdditionalTransactions,
  s as AlternativeDefenseMaterial,
  c as AmexFaxCover,
  p as AuthorisedTransaction,
  We as AuthorizationNotObtained,
  h as CancellationNeverAccepted,
  tt as CancelledRecurringBillingReason,
  u as CardPresent,
  Ue as CardRecoveryBulletinOrExceptionFile,
  m as CardholderIdentification,
  f as CardholderRenewedMembership,
  xe as ChargeToWrongAccountNumber,
  g as ChargebackCodeNotApplicable,
  T as ChipLiabilityShift,
  C as ChipTransactionReportedToSAFE,
  S as CompellingEvidence,
  R as CopyOfSalesDraft,
  A as CorrectTransactionCurrency,
  Ye as CreditNotProcessed,
  rt as CreditNotProcessedReason,
  b as CreditPreviouslyIssued,
  v as DateMerchandiseShipped,
  P as DefenseMaterial,
  Ge as DeffectiveMerchandise,
  Je as DifferentSignature,
  O as DisclosureAtPointOfInteraction,
  M as DuplicateChargeback,
  Fe as DuplicateProcessing,
  N as EvidenceOfCardholderParticipation,
  He as ExpiredCard,
  Be as FalseTransaction,
  D as FlightManifest,
  F as FlightTicket,
  x as FlightTicketAtBillingAddress,
  I as FlightTicketUsed,
  k as FligthTookPlace,
  y as FrequentFlyer,
  H as GoodsNotReturned,
  w as GoodsRepairedOrReplaced,
  B as GoodsWereAsDescribed,
  ot as IncorrectAmountReason,
  W as IncreasedTransactionAmount,
  E as InvalidCancellationCode,
  q as InvalidChargeback,
  we as LatePresentment,
  G as LiabilityShiftFullUCAF,
  L as LiabilityShiftMerchantUCAF,
  J as MPIdata,
  z as MerchandiseDescription,
  et as MerchandiseNotAsDescribedReason,
  U as MerchandiseNotCounterfeit,
  Le as MerchandiseNotReceived,
  Qe as MerchantFraudPerformanceProgram,
  V as MerchantProofOfRecurringTransaction,
  ke as MissingSignature,
  Ke as NonMatchingAccountNumber,
  X as NotCancelled,
  Y as NotChargebacked,
  j as NotChargebackedAuto,
  K as NotNotified,
  Q as OriginalAmount,
  Ze as Other,
  te as PINTransaction,
  je as PaidByOtherMeansThanJCBCard,
  Z as PaperAirlineTicket,
  _ as PassengerIdentification,
  $ as PastChargebackTimeLimit,
  ee as PaymentByOtherMeans,
  oe as PositiveAVS,
  ne as PresentmentWithinTimeLimit,
  ie as PrintedSignedTerminalReceipt,
  ye as ProcessingError,
  re as ProofOfAbilityToProvideServices,
  ae as ProofOfAccountTakeover,
  le as ProofOfAttendedPOSTerminal,
  se as ProofOfCardPresenceAndSignatureChipNoPIN,
  ce as ProofOfCardPresenceAndSignatureWithTerminalReceipt,
  de as ProofOfCardPresenceSignatureOrChipPIN,
  _e as ProofOfFulfillment,
  pe as ProofOfGoodsOrServicesProvided,
  he as ProofOfNoShow,
  me as ProofOfRecurringTransaction,
  $e as ProofOfRefund,
  nt as ProofOfRefundForAmountDifference,
  fe as ProofOfRetailSaleRatherThanCredit,
  ue as ReasonForReturn,
  ge as RetrievalRequestFulfilled,
  Xe as RetrievalRequestNotHonored,
  Te as ReturnedMerchandiseNotReceived,
  Ce as SaleTermsNotMisrepresented,
  ze as ServiceNotRendered,
  Se as ShipToAddress,
  Re as ShippedToAVS,
  Ie as SplitSales,
  Ae as SupplementalDocuments,
  be as TEDocument,
  ve as TIDorInvoice,
  Ve as TransactionAfterReservationCancelled,
  Pe as TransactionNotPOS,
  qe as TransactionNotRecognizedChipLiabilityShift,
  Ee as TransactionNotRecognizedContactlessAndCardNotPresented,
  Oe as TwoDifferentTIDs,
  Me as TwoPINTransactions,
  Ne as TwoPriorFraudChargebacks,
  it as UnauthorisedReason,
  De as UnreasonableAmount,
  at as default
};
