// @adyen/adyen-platform-experience-web/dist/es/components/internal/Popover/Popover.js
import { jsx as r } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import io from "../Button/ButtonActions/ButtonActions.js";
import { ButtonActionsLayoutBasic as to } from "../Button/ButtonActions/types.js";
import { DEFAULT_POPOVER_CLASSNAME as i, POPOVER_HEADER_TITLE_CLASSNAME as so, TOOLTIP_CONTENT_CLASSNAME as W, POPOVER_CONTENT_CLASSNAME as L, POPOVER_FOOTER_CLASSNAME as uo, DEFAULT_TOOLTIP_CLASSNAME as ao, POPOVER_CONTAINER_CLASSNAME as fo, POPOVER_HEADER_CLASSNAME as w } from "./constants.js";
import mo from "./PopoverDismissButton/PopoverDismissButton.js";
import po from "./PopoverTitle/PopoverTitle.js";
import { PopoverContainerVariant as t, PopoverContainerSize as q } from "./types.js";
import { InteractionKeyCode as Oo } from "../../types.js";
import { useClickOutside as lo, ClickOutsideVariant as To, CONTROL_ELEMENT_PROPERTY as d } from "../../../hooks/element/useClickOutside.js";
import Po from "../../../hooks/element/useFocusTrap.js";
import Ao from "../../../hooks/element/usePopoverPositioner.js";
import C from "../../../hooks/element/useUniqueIdentifier.js";
import Eo from "../../../hooks/useReflex.js";
import { SELECTORS as No, isFocusable as Lo } from "../../../primitives/dom/tabbableRoot/tabbable.js";
import x from "classnames";
import { createPortal as Co } from "../../../external/preact/compat/dist/compat.module.js";
import { useMemo as U, useRef as _, useCallback as y, useEffect as K } from "../../../external/preact/hooks/dist/hooks.module.js";
/* empty css             */
import { Fragment as R } from "../../../external/preact/dist/preact.module.js";
import { isFunction as _o } from "../../../utils/value/is.js";
import { boolOrTrue as yo } from "../../../utils/value/bool.js";
import { getModifierClasses as Ro } from "../../../utils/preact/className.js";
const So = (s) => {
  var e;
  let f;
  const O = (e = s.querySelector(`:scope .${W}`)) == null ? void 0 : e.querySelectorAll(No);
  return O ? (Array.prototype.some.call(O, (u) => {
    if (Lo(u)) return f = u;
  }), f) : null;
}, Io = (s) => s === t.TOOLTIP ? [10, 3, 5, 5] : [8, 8, 8, 8];
function Xo({
  actions: s,
  disableFocusTrap: f = !1,
  actionsLayout: O = to.SPACE_BETWEEN,
  variant: e = t.TOOLTIP,
  title: u,
  open: m,
  dismissible: S,
  modifiers: H,
  divider: I,
  fitContent: F,
  withoutSpace: v,
  containerSize: T,
  position: j,
  targetElement: o,
  setToTargetWidth: G,
  dismiss: n,
  children: M,
  withContentPadding: Y,
  classNameModifiers: J,
  showOverlay: p = !1,
  fitPosition: Q,
  fixedPositioning: X = !1,
  additionalStyle: Z,
  setPopoverElement: P,
  ...z
}) {
  var b;
  const V = U(() => _o(n) && yo(S), [n, S]), $ = C(), h = C(), D = _(), g = y(
    (c) => {
      var a;
      n && n(), c && ((a = o == null ? void 0 : o.current) == null || a.focus());
    },
    [n, o]
  ), A = y(
    (c) => {
      c.code === Oo.ESCAPE && (n && n(), (o == null ? void 0 : o.current).focus());
    },
    [n, o]
  ), E = _(A), B = _(), k = lo(
    Ao(
      Io(e),
      o,
      e,
      j,
      $,
      G,
      p,
      Q,
      X,
      Z,
      void 0,
      h
    ),
    n,
    e === t.TOOLTIP && !m,
    To.POPOVER
  ), oo = Po(f ? null : k, g), l = Eo(
    y(
      (c, a) => {
        a instanceof Element && (a[d] = void 0, delete a[d]), c instanceof Element && (c[d] = o.current, cancelAnimationFrame(B.current), B.current = requestAnimationFrame(() => {
          if (D.current === m || !(D.current = m)) return;
          const N = So(c);
          N == null || N.focus();
        }));
      },
      [m, o]
    ),
    f ? k : oo
  ), ro = C(l), eo = U(
    () => ({
      [`${i}--medium`]: T === q.MEDIUM,
      [`${i}--with-divider`]: !!I,
      [`${i}--wide`]: T === q.WIDE,
      [`${i}--fit-content`]: F,
      [`${i}--without-space`]: v,
      [`${i}--auto-width`]: p
    }),
    [T, I, v, F, p]
  );
  K(() => {
    l.current && (l.current[d] = o.current);
  }, [l, o]), K(() => (document.removeEventListener("keydown", E.current), document.addEventListener("keydown", E.current = A), () => document.removeEventListener("keydown", E.current)), [A]);
  const no = e === t.TOOLTIP ? ao : `${i} ${fo}`, co = e === t.TOOLTIP ? W : `${L}`;
  return Co(
    /* @__PURE__ */ r(R, { children: m ? /* @__PURE__ */ r(R, { children: [
      p && /* @__PURE__ */ r("div", { className: "adyen-pe-popover__overlay" }),
      /* @__PURE__ */ r(
        "div",
        {
          ...z,
          ref: (c) => {
            ro(c), P == null || P(c);
          },
          "aria-labelledby": (b = o.current) == null ? void 0 : b.id,
          className: x(no, eo, J),
          role: e === t.POPOVER ? "dialog" : "tooltip",
          style: { visibility: "hidden" },
          children: [
            (u || V) && /* @__PURE__ */ r("div", { className: Ro(w, H, [w]), children: [
              u && /* @__PURE__ */ r("div", { className: so, children: /* @__PURE__ */ r(po, { title: u }) }),
              V && /* @__PURE__ */ r(mo, { onClick: n })
            ] }),
            M && /* @__PURE__ */ r(R, { children: [
              /* @__PURE__ */ r(
                "div",
                {
                  className: x(co, {
                    [`${L}--with-padding`]: Y,
                    [`${L}--overlay`]: p
                  }),
                  ref: h,
                  children: M
                }
              ),
              e === t.TOOLTIP && /* @__PURE__ */ r("span", { "data-popover-placement": "hidden", ref: $, className: "adyen-pe-tooltip__arrow" })
            ] }),
            s && /* @__PURE__ */ r("div", { className: uo, children: /* @__PURE__ */ r(io, { actions: s, layout: O }) })
          ]
        }
      )
    ] }) : null }),
    document.getElementsByTagName("body")[0]
  );
}
export {
  Xo as default
};
