// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputeManagement/components/DisputesData/constants.js
const _ = "adyen-pe-dispute-data", t = "adyen-pe-dispute-data--mobile", T = `${_}__status-box`, s = `${_}__action-bar`, E = `${_}__label`, n = `${_}__list`, S = `${_}__list--evidence`, a = `${_}__list-evidence-error-message`, A = `${_}__issuer-comment`, e = `${_}__issuer-comments`, o = `${e}--expanded`, D = `${e}--truncated`, d = `${_}__issuer-comments-alert`, I = `${_}__issuer-comments-group`, c = "adyen-pe-dispute-data__error-container", U = "adyen-pe-dispute-data__status-box-skeleton", p = "adyen-pe-dispute-data__properties-skeleton", P = "adyen-pe-dispute-data__properties-skeleton-element", i = "adyen-pe-dispute-data__properties-skeleton-container", u = "adyen-pe-dispute-data__status-box-status-container", O = "adyen-pe-dispute-data__status-box-status", r = "adyen-pe-dispute-data__status-box-amount", R = "adyen-pe-dispute-data__status-box-payment-method", N = "adyen-pe-dispute-data__skeleton-container", M = "adyen-pe-dispute-data__status-box-payment-method-container", m = /* @__PURE__ */ new Set([
  "allowedDefenseReasons",
  "balanceAccount",
  "amount",
  "createdAt",
  "defensibility",
  "dueDate",
  "id",
  "latestDefense",
  "paymentMerchantReference",
  "paymentMethod",
  "paymentPspReference",
  "reasonCode",
  "reasonGroup",
  "status"
]);
export {
  s as DISPUTE_DATA_ACTION_BAR,
  _ as DISPUTE_DATA_CLASS,
  c as DISPUTE_DATA_ERROR_CONTAINER,
  A as DISPUTE_DATA_ISSUER_COMMENT,
  e as DISPUTE_DATA_ISSUER_COMMENTS,
  d as DISPUTE_DATA_ISSUER_COMMENTS_ALERT,
  o as DISPUTE_DATA_ISSUER_COMMENTS_EXPANDED,
  I as DISPUTE_DATA_ISSUER_COMMENTS_GROUP,
  D as DISPUTE_DATA_ISSUER_COMMENTS_TRUNCATED,
  E as DISPUTE_DATA_LABEL,
  n as DISPUTE_DATA_LIST,
  S as DISPUTE_DATA_LIST_EVIDENCE,
  a as DISPUTE_DATA_LIST_EVIDENCE_ERROR_MESSAGE,
  t as DISPUTE_DATA_MOBILE_CLASS,
  p as DISPUTE_DATA_PROPERTIES_SKELETON,
  i as DISPUTE_DATA_PROPERTIES_SKELETON_CONTAINER,
  P as DISPUTE_DATA_PROPERTIES_SKELETON_ELEMENT,
  N as DISPUTE_DATA_SKELETON_CONTAINER,
  r as DISPUTE_DATA_STATUS_BOX_AMOUNT,
  R as DISPUTE_DATA_STATUS_BOX_PAYMENT_METHOD,
  M as DISPUTE_DATA_STATUS_BOX_PAYMENT_METHOD_CONTAINER,
  U as DISPUTE_DATA_STATUS_BOX_SKELETON,
  O as DISPUTE_DATA_STATUS_BOX_STATUS,
  u as DISPUTE_DATA_STATUS_BOX_STATUS_CONTAINER,
  m as DISPUTE_DETAILS_RESERVED_FIELDS_SET,
  T as DISPUTE_STATUS_BOX
};
