// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputeManagement/components/DisputesData/DisputeDataProperties.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import A from "classnames";
import { useState as q, useCallback as j, useEffect as W, useMemo as X } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import Y from "../../../../../hooks/useTimezoneAwareDateFormatting.js";
import G from "../../../../internal/Button/DownloadButton/DownloadButton.js";
import I from "../../../../internal/CopyText/CopyText.js";
import H from "../../../../internal/DataGrid/components/Icon.js";
import { isCustomDataObject as O } from "../../../../internal/DataGrid/components/TableCells.js";
import Z from "../../../../internal/Link/Link.js";
import { Tag as J } from "../../../../internal/Tag/Tag.js";
import { TypographyElement as Q, TypographyVariant as _ } from "../../../../internal/Typography/types.js";
import S from "../../../../internal/Typography/Typography.js";
import { isDisputeActionNeeded as z } from "../../../../utils/disputes/actionNeeded.js";
import { DISPUTE_DETAILS_RESERVED_FIELDS_SET as g, DISPUTE_DATA_LIST_EVIDENCE as ee, DISPUTE_DATA_LIST_EVIDENCE_ERROR_MESSAGE as te, DISPUTE_DATA_LIST as se, DISPUTE_DATA_LABEL as ne } from "./constants.js";
import ae from "../../../../../core/Context/useCoreContext.js";
import { getDisputeReason as re } from "../../../../utils/translation/getters.js";
import { getDefenseReasonContent as de, getDefenseDocumentContent as ie } from "../../utils/index.js";
import { useDisputeFlow as oe } from "../../context/dispute/context.js";
import { DATE_FORMAT_DISPUTE_DETAILS as D } from "../../../../../constants/dateFormats.js";
import { Icon as pe } from "../../../../internal/Icon/Icon.js";
import { Fragment as le } from "../../../../../external/preact/dist/preact.module.js";
import me from "../../../../internal/StructuredList/StructuredList.js";
const r = {
  acceptedOn: "disputes.management.details.fields.acceptedOn",
  account: "disputes.management.details.fields.account",
  defendedOn: "disputes.management.details.fields.defendedOn",
  defenseReason: "disputes.management.details.fields.defenseReason",
  disputeEvidence: "disputes.management.details.fields.evidence",
  disputeReason: "disputes.management.details.fields.disputeReason",
  disputeReference: "disputes.management.details.fields.disputeReference",
  expiredOn: "disputes.management.details.fields.expiredOn",
  merchantReference: "disputes.management.details.fields.merchantReference",
  openedOn: "disputes.management.details.fields.openedOn",
  paymentReference: "disputes.management.details.fields.paymentReference",
  reasonCode: "disputes.management.details.fields.reasonCode",
  respondBy: "disputes.management.details.fields.respondBy"
}, ce = ["ACCEPTED", "EXPIRED"], we = ({ dispute: d, dataCustomization: a, defenseReasonConfig: k }) => {
  var P;
  const { i18n: u } = ae(), { dateFormat: y } = Y(d.payment.balanceAccount.timeZone), { defenseDocumentConfig: v } = oe(), [N, B] = q(), x = j(async () => {
    var l, f, T;
    if (d) {
      const m = await ((f = (l = a == null ? void 0 : a.details) == null ? void 0 : l.onDataRetrieve) == null ? void 0 : f.call(l, d));
      B(
        (T = a == null ? void 0 : a.details) == null ? void 0 : T.fields.reduce((o, i) => g.has(i.key) || (i == null ? void 0 : i.visibility) === "hidden" ? o : { ...o, ...m != null && m[i.key] ? { [i.key]: m[i.key] } : {} }, {})
      );
    }
  }, [d, a]);
  return W(() => {
    x();
  }, [x]), X(() => {
    var F;
    const { pspReference: l, reason: f, acceptedDate: T, createdAt: m, dueDate: o, status: i, type: w } = d.dispute, { pspReference: U, merchantReference: b, balanceAccount: M } = d.payment, { reason: R, defendedOn: E, suppliedDocuments: h } = d.defense || {}, L = w === "NOTIFICATION_OF_FRAUD", C = i === "EXPIRED" || i === "LOST" && !L && !E, K = z(d.dispute) && d.dispute.defensibility !== "NOT_ACTIONABLE", p = null, V = [
      ...[
        // dispute reason
        {
          key: r.disputeReason,
          value: `${re(u, f.category)} - ${f.title}`,
          // [NOTE]: Not translated at the moment (maybe in the future)
          id: "disputeReason"
        },
        // reason code
        L ? p : {
          key: r.reasonCode,
          value: f.code,
          id: "reasonCode"
        },
        // created at
        {
          key: r.openedOn,
          value: /* @__PURE__ */ t("time", { dateTime: m, children: y(m, D) }),
          id: "openedOn"
        },
        // respond by
        o && K ? {
          key: r.respondBy,
          value: /* @__PURE__ */ t("time", { dateTime: o, children: y(o, D) }),
          id: "respondBy"
        } : p,
        // dispute reference
        {
          key: r.disputeReference,
          value: /* @__PURE__ */ t(
            I,
            {
              copyButtonAriaLabelKey: "disputes.management.details.actions.copyDisputeReference",
              type: "Default",
              textToCopy: l,
              showCopyTextTooltip: !1
            }
          ),
          id: "disputeId"
        },
        // balance account
        {
          key: r.account,
          value: M.description,
          id: "account"
        },
        // psp reference
        {
          key: r.paymentReference,
          value: /* @__PURE__ */ t(
            I,
            {
              copyButtonAriaLabelKey: "disputes.management.details.actions.copyPaymentReference",
              type: "Default",
              textToCopy: U,
              showCopyTextTooltip: !1
            }
          ),
          id: "paymentPspReference"
        },
        // merchant reference
        b ? {
          key: r.merchantReference,
          value: /* @__PURE__ */ t(
            I,
            {
              copyButtonAriaLabelKey: "disputes.management.details.actions.copyMerchantReference",
              type: "Default",
              textToCopy: b,
              showCopyTextTooltip: !1
            }
          ),
          id: "paymentMerchantReference"
        } : p,
        // defense reason
        R ? {
          key: r.defenseReason,
          value: ((F = de(k, u, R)) == null ? void 0 : F.title) ?? R,
          id: "defenseReason"
        } : p,
        // defended on
        E ? {
          key: r.defendedOn,
          value: /* @__PURE__ */ t("time", { dateTime: E, children: y(E, D) }),
          id: "defendedOn"
        } : p,
        // evidence
        h && h.length > 0 ? {
          key: r.disputeEvidence,
          value: /* @__PURE__ */ t(le, { children: h.map((s, n) => {
            var e;
            const c = {
              path: { disputePspReference: l },
              query: { documentType: s }
            };
            return /* @__PURE__ */ t("div", { className: ee, children: [
              /* @__PURE__ */ t(J, { label: ((e = ie(v, u, s)) == null ? void 0 : e.title) ?? s }),
              /* @__PURE__ */ t(
                G,
                {
                  className: "adyen-pe-dispute-document-download",
                  endpointName: "downloadDefenseDocument",
                  disabled: !1,
                  requestParams: c,
                  iconButton: !0,
                  errorMessage: () => /* @__PURE__ */ t("div", { className: te, children: [
                    /* @__PURE__ */ t(pe, { name: "info-filled" }),
                    /* @__PURE__ */ t(S, { variant: _.CAPTION, el: Q.SPAN, children: u.get("disputes.management.details.errors.downloadFailure") })
                  ] }),
                  setError: () => console.warn("Download failed for", s),
                  "aria-label": u.get("disputes.management.details.actions.downloadEvidence")
                }
              )
            ] }, `${s}-${n}`);
          }) }),
          id: "disputeEvidence"
        } : p,
        // accepted on
        T && ce.includes(i) ? {
          key: r.acceptedOn,
          value: /* @__PURE__ */ t("time", { dateTime: T, children: y(T, D) }),
          id: "acceptedOn"
        } : p,
        // expired on
        o && C ? {
          key: r.expiredOn,
          value: /* @__PURE__ */ t("time", { dateTime: o, children: y(o, D) }),
          id: "expiredOn"
        } : p
      ].filter(Boolean).filter((s) => {
        var n, c;
        return !((c = (n = a == null ? void 0 : a.details) == null ? void 0 : n.fields) != null && c.some((e) => e.key === s.id && e.visibility === "hidden"));
      }),
      ...Object.entries(N || {}).filter(
        ([s, n]) => !g.has(s) && n.type !== "button" && n.visibility !== "hidden"
      ).map(([s, n]) => ({
        key: s,
        value: O(n) ? n.value : n,
        type: O(n) ? n.type : "text",
        config: O(n) ? n.config : void 0
      })) || {}
    ];
    return /* @__PURE__ */ t(
      me,
      {
        classNames: se,
        items: V,
        layout: "4-8",
        align: "start",
        renderLabel: (s) => /* @__PURE__ */ t("div", { className: ne, children: s }),
        renderValue: (s, n, c, e) => {
          if (c === "link" && e)
            return /* @__PURE__ */ t(Z, { classNames: [A(e == null ? void 0 : e.className)], href: e.href, target: e.target || "_blank", children: s });
          if (c === "icon" && e) {
            const $ = { url: e == null ? void 0 : e.src, alt: e.alt || s };
            return /* @__PURE__ */ t("div", { className: A("adyen-pe-dispute-data__list-icon-value", e == null ? void 0 : e.className), children: [
              /* @__PURE__ */ t(H, { ...$ }),
              /* @__PURE__ */ t(S, { variant: _.BODY, children: s })
            ] });
          }
          return /* @__PURE__ */ t(S, { className: A(e == null ? void 0 : e.className), variant: _.BODY, children: s });
        }
      }
    );
  }, [
    d.dispute,
    d.payment,
    d.defense,
    u,
    y,
    k,
    N,
    v,
    (P = a == null ? void 0 : a.details) == null ? void 0 : P.fields
  ]);
};
export {
  we as default
};
