// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputeManagement/components/DisputesData/DisputeIssuerComments.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import y from "classnames";
import { useState as o, useRef as M, useLayoutEffect as C } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import R from "../../../../../core/Context/useCoreContext.js";
import { Alert as U } from "../../../../internal/Alert/Alert.js";
import u from "../../../../internal/Typography/Typography.js";
import { AlertTypeOption as x } from "../../../../internal/Alert/types.js";
import { ButtonVariant as H } from "../../../../internal/Button/types.js";
import { TypographyVariant as l, TypographyElement as T } from "../../../../internal/Typography/types.js";
import { DISPUTE_DATA_ISSUER_COMMENTS_ALERT as P, DISPUTE_DATA_ISSUER_COMMENTS_GROUP as N, DISPUTE_DATA_ISSUER_COMMENT as h, DISPUTE_DATA_ISSUER_COMMENTS as O, DISPUTE_DATA_ISSUER_COMMENTS_TRUNCATED as v, DISPUTE_DATA_ISSUER_COMMENTS_EXPANDED as G } from "./constants.js";
/* empty css                 */
import B from "../../../../internal/Button/Button.js";
const J = ({ issuerComments: E }) => {
  const { i18n: a } = R(), [S, d] = o(0), [g, _] = o(0), [c, A] = o(!1), [n, I] = o(!1), s = M(null), f = (r) => {
    var t;
    r.preventDefault(), I(!n), (t = s.current) == null || t.style.setProperty("max-height", `${n ? S : g}px`);
  };
  return C(() => {
    var r;
    if (s.current) {
      const t = s.current, i = t.querySelector(`:scope .${h}`), D = i ? parseInt(getComputedStyle(i).getPropertyValue("line-height")) : 0, m = Math.min((i == null ? void 0 : i.clientHeight) || 1 / 0, D * 3), p = t.scrollHeight;
      d(m), _(p), A(p > m), (r = s.current) == null || r.style.setProperty("max-height", `${m}px`);
    }
  }, []), /* @__PURE__ */ e(
    U,
    {
      type: x.HIGHLIGHT,
      description: /* @__PURE__ */ e("div", { className: P, children: [
        /* @__PURE__ */ e(u, { el: T.DIV, variant: l.BODY, strongest: !0, children: a.get("disputes.management.details.issuerComment") }),
        /* @__PURE__ */ e(
          "div",
          {
            className: y(O, {
              [G]: n,
              [v]: c
            }),
            children: /* @__PURE__ */ e("ul", { ref: s, className: N, children: E.map((r, t) => /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(
              u,
              {
                className: h,
                el: T.PARAGRAPH,
                variant: l.BODY,
                children: r
              }
            ) }, t)) })
          }
        ),
        c && /* @__PURE__ */ e(B, { variant: H.TERTIARY, onClick: f, children: a.get(
          n ? "disputes.management.details.issuerComment.showLess" : "disputes.management.details.issuerComment.showMore"
        ) })
      ] })
    }
  );
};
export {
  J as DisputeIssuerComments
};
