// @adyen/adyen-platform-experience-web/dist/es/components/external/DisputeManagement/components/DisputesData/DisputeData.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import j from "classnames";
import { useMemo as A, useCallback as N, useState as pe, useEffect as de } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import $ from "../../../../../core/Context/useCoreContext.js";
import { useFetch as Te } from "../../../../../hooks/useFetch.js";
import { useResponsiveContainer as ue, containerQueries as Ae } from "../../../../../hooks/useResponsiveContainer.js";
/* empty css                 */
import { Alert as S } from "../../../../internal/Alert/Alert.js";
import { AlertVariantOption as O, AlertTypeOption as I } from "../../../../internal/Alert/types.js";
import Ee from "../../../../internal/Button/ButtonActions/ButtonActions.js";
import { ButtonVariant as h } from "../../../../internal/Button/types.js";
import De from "../../../../internal/StatusBox/StatusBox.js";
import fe from "../../../../internal/StatusBox/useStatusBox.js";
import { Tag as _e } from "../../../../internal/Tag/Tag.js";
import Ne from "../../../DisputesOverview/components/DisputesTable/DisputeStatusTag.js";
import { useDisputeFlow as Se } from "../../context/dispute/context.js";
import { isDisputeActionNeeded as Oe } from "../../../../utils/disputes/actionNeeded.js";
import { DisputeIssuerComments as Ie } from "./DisputeIssuerComments.js";
import he from "./DisputeDataProperties.js";
import { DISPUTE_DATA_SKELETON_CONTAINER as ge, DISPUTE_DATA_STATUS_BOX_SKELETON as Re, DISPUTE_DATA_STATUS_BOX_STATUS_CONTAINER as Pe, DISPUTE_DATA_STATUS_BOX_STATUS as q, DISPUTE_DATA_STATUS_BOX_AMOUNT as ye, DISPUTE_DATA_STATUS_BOX_PAYMENT_METHOD_CONTAINER as Be, DISPUTE_DATA_STATUS_BOX_PAYMENT_METHOD as Q, DISPUTE_DATA_PROPERTIES_SKELETON_CONTAINER as Fe, DISPUTE_DATA_PROPERTIES_SKELETON as Ue, DISPUTE_DATA_PROPERTIES_SKELETON_ELEMENT as Z, DISPUTE_DATA_CLASS as J, DISPUTE_DATA_MOBILE_CLASS as W, DISPUTE_DATA_ERROR_CONTAINER as ve, DISPUTE_STATUS_BOX as Ce, DISPUTE_DATA_ACTION_BAR as Le } from "./constants.js";
import be from "../../../../internal/Typography/Typography.js";
import { TypographyElement as Me, TypographyVariant as ke } from "../../../../internal/Typography/types.js";
import xe from "../../../../../hooks/useTimezoneAwareDateFormatting.js";
import { getDisputeType as He } from "../../../../utils/translation/getters.js";
import { ErrorMessageDisplay as Xe } from "../../../../internal/ErrorMessageDisplay/ErrorMessageDisplay.js";
import { getDisputesErrorMessage as Ye } from "../../../../utils/disputes/getDisputesErrorMessage.js";
import we from "../../../../internal/Button/Button.js";
import { Fragment as g } from "../../../../../external/preact/dist/preact.module.js";
import { Translation as Ke } from "../../../../internal/Translation/Translation.js";
import { useConfigContext as Ge } from "../../../../../core/ConfigContext/context.js";
import { EMPTY_OBJECT as Ve } from "../../../../../utils/value/constants.js";
import { isFunction as F } from "../../../../../utils/value/is.js";
import { DATE_FORMAT_RESPONSE_DEADLINE as je } from "../../../../../constants/dateFormats.js";
const qe = (c) => !!c && typeof c == "object" && c.type === "button", Qe = ({
  alertMode: c,
  dispute: o,
  timeZone: d
}) => {
  const { i18n: m } = $(), { dateFormat: r } = xe(d);
  switch (c) {
    case "contactSupport": {
      const { dueDate: n, type: T } = o, u = T === "REQUEST_FOR_INFORMATION" ? "disputes.management.details.alerts.contactSupport.requestForInformation" : T === "NOTIFICATION_OF_FRAUD" ? "disputes.management.details.alerts.contactSupport.notificationOfFraud" : "disputes.management.details.alerts.contactSupport.chargeback";
      return /* @__PURE__ */ t(
        S,
        {
          type: I.WARNING,
          variant: O.TIP,
          description: /* @__PURE__ */ t(g, { children: [
            m.get(u),
            T !== "NOTIFICATION_OF_FRAUD" && !!n && /* @__PURE__ */ t(g, { children: [
              " ",
              /* @__PURE__ */ t(
                Ke,
                {
                  translationKey: "disputes.management.details.alerts.responseDeadline",
                  fills: {
                    date: /* @__PURE__ */ t("time", { dateTime: n, children: /* @__PURE__ */ t(be, { variant: ke.CAPTION, el: Me.SPAN, stronger: !0, children: r(n, je) }) })
                  }
                }
              )
            ] })
          ] })
        }
      );
    }
    case "autoDefended":
      return /* @__PURE__ */ t(
        S,
        {
          type: I.HIGHLIGHT,
          variant: O.TIP,
          description: m.get("disputes.management.details.alerts.autoDefended")
        }
      );
    case "notDefended": {
      const n = o.status === "EXPIRED" ? "disputes.management.details.alerts.notDefendedExpired" : "disputes.management.details.alerts.notDefendedLost";
      return /* @__PURE__ */ t(S, { type: I.HIGHLIGHT, variant: O.TIP, description: m.get(n) });
    }
    case "notDefendable":
      return /* @__PURE__ */ t(
        S,
        {
          type: I.HIGHLIGHT,
          variant: O.TIP,
          description: m.get("disputes.management.details.alerts.notDefendable")
        }
      );
  }
  return null;
}, yt = ({
  disputeId: c,
  dataCustomization: o,
  onContactSupport: d,
  onDismiss: m
}) => {
  var w, K, G;
  const { i18n: r } = $(), { dispute: n, setDispute: T, setFlowState: u, defenseReasonConfig: z } = Se(), { getDisputeDetail: R, getApplicableDefenseDocuments: ee, acceptDispute: te } = Ge().endpoints, ne = F(te), se = F(ee), U = ue(Ae.up.sm), { data: E, isFetching: ae, error: P } = Te(
    A(
      () => ({
        fetchOptions: {
          enabled: !!c && !!R && !n,
          onSuccess: (s) => {
            T(s);
          }
        },
        queryFn: async () => R(Ve, {
          path: {
            disputePspReference: c
          }
        })
      }),
      [n, c, R, T]
    )
  ), e = n || E, l = (w = e == null ? void 0 : e.dispute) == null ? void 0 : w.defensibility, re = fe({
    amountData: e == null ? void 0 : e.dispute.amount,
    paymentMethodData: e == null ? void 0 : e.payment.paymentMethod
  }), v = A(() => {
    const { chargeback: s, preArbitration: a } = (e == null ? void 0 : e.dispute.issuerExtraData) ?? {}, i = [];
    return [a, s].forEach((_) => {
      _ && ["LIABILITY_NOT_ACCEPTED_FULLY", "PRE_ARB_REASON", "NOTE"].forEach((le) => {
        const B = _[le], V = B == null ? void 0 : B.trim();
        V && i.push(V);
      });
    }), i.filter(Boolean);
  }, [e]), C = A(() => He(r, e == null ? void 0 : e.dispute.type), [r, e]), D = !!l && ["ACCEPTABLE", "DEFENDABLE_EXTERNALLY"].includes(l) || (e == null ? void 0 : e.dispute.type) === "NOTIFICATION_OF_FRAUD", L = !!l && l === "DEFENDABLE" && se, b = !!l && ["ACCEPTABLE", "DEFENDABLE"].includes(l) && ne, M = N(() => {
    u("accept");
  }, [u]), k = N(() => {
    u("defendReasonSelectionView");
  }, [u]), [f, oe] = pe([]), x = N(async () => {
    var a, i;
    const s = E && await ((i = (a = o == null ? void 0 : o.details) == null ? void 0 : a.onDataRetrieve) == null ? void 0 : i.call(a, E));
    if (s)
      return oe(
        Object.values(s).filter((_) => qe(_))
      );
  }, [E, o == null ? void 0 : o.details]);
  de(() => {
    x();
  }, [x]);
  const H = A(() => (n == null ? void 0 : n.dispute.type) === "REQUEST_FOR_INFORMATION" ? r.get("disputes.management.details.actions.submitInformation") : r.get("disputes.management.details.actions.defendChargeback"), [r, n == null ? void 0 : n.dispute.type]), X = A(() => {
    const s = [];
    return L && s.push({
      title: H,
      event: k
    }), b && s.push({
      title: r.get("disputes.management.details.actions.accept"),
      event: M,
      variant: h.SECONDARY
    }), D && F(d) && s.push({
      title: r.get("disputes.management.details.actions.contactSupport"),
      event: d,
      variant: h.SECONDARY
    }), f && f.length && f.forEach((a) => {
      var i;
      s.push({
        title: a == null ? void 0 : a.value,
        event: (i = a == null ? void 0 : a.config) == null ? void 0 : i.action,
        variant: h.SECONDARY
      });
    }), s;
  }, [L, H, k, b, D, d, f, r, M]), Y = A(() => !!e && Oe(e.dispute), [e]), ie = N(() => /* @__PURE__ */ t(we, { variant: h.SECONDARY, onClick: m, children: r.get("disputes.management.common.actions.goBack") }), [r, m]);
  if (!e && !P || ae) {
    const s = Array.from({ length: 5 });
    return /* @__PURE__ */ t("div", { className: j(J, { [W]: !U }), children: /* @__PURE__ */ t("div", { className: ge, children: [
      /* @__PURE__ */ t("div", { className: Re, children: [
        /* @__PURE__ */ t("div", { className: Pe, children: [
          /* @__PURE__ */ t("span", { className: q }),
          /* @__PURE__ */ t("span", { className: q })
        ] }),
        /* @__PURE__ */ t("span", { className: ye }),
        /* @__PURE__ */ t("div", { className: Be, children: [
          /* @__PURE__ */ t("span", { className: Q }),
          /* @__PURE__ */ t("span", { className: Q })
        ] })
      ] }),
      /* @__PURE__ */ t("div", { className: Fe, children: s.map((a, i) => /* @__PURE__ */ t("div", { className: Ue, children: [
        /* @__PURE__ */ t("span", { className: Z }),
        /* @__PURE__ */ t("span", { className: Z })
      ] }, `skeleton-${i}`)) })
    ] }) });
  }
  const y = (e == null ? void 0 : e.dispute.type) === "NOTIFICATION_OF_FRAUD", ce = !!(e != null && e.defense && e.defense.defendedOn);
  let p;
  ((K = e == null ? void 0 : e.defense) == null ? void 0 : K.autodefended) === !0 ? p = "autoDefended" : Y && l === "NOT_ACTIONABLE" ? p = "notDefendable" : Y && D || D && y ? p = "contactSupport" : ((e == null ? void 0 : e.dispute.status) === "EXPIRED" || (e == null ? void 0 : e.dispute.status) === "LOST" && !(y || ce)) && (p = "notDefended");
  const me = Ye(
    P,
    "disputes.management.common.errors.unavailable",
    d
  );
  return /* @__PURE__ */ t("div", { className: j(J, { [W]: !U }), children: P ? /* @__PURE__ */ t("div", { className: ve, children: /* @__PURE__ */ t(
    Xe,
    {
      renderSecondaryButton: m ? ie : void 0,
      withImage: !0,
      outlined: !1,
      absolutePosition: !1,
      withBackground: !1,
      ...me
    }
  ) }) : e ? /* @__PURE__ */ t(g, { children: [
    /* @__PURE__ */ t("div", { className: Ce, children: /* @__PURE__ */ t(
      De,
      {
        ...re,
        tag: /* @__PURE__ */ t(g, { children: [
          C && /* @__PURE__ */ t(_e, { label: C }),
          !y && /* @__PURE__ */ t(Ne, { dispute: e.dispute })
        ] })
      }
    ) }),
    v.length > 0 && /* @__PURE__ */ t(Ie, { issuerComments: v }),
    p && /* @__PURE__ */ t(
      Qe,
      {
        alertMode: p,
        dispute: e.dispute,
        timeZone: (G = e.payment.balanceAccount) == null ? void 0 : G.timeZone
      }
    ),
    /* @__PURE__ */ t(he, { dispute: e, dataCustomization: o, defenseReasonConfig: z }),
    X.length > 0 ? /* @__PURE__ */ t("div", { className: Le, children: /* @__PURE__ */ t(Ee, { actions: X }) }) : null
  ] }) : null });
};
export {
  yt as DisputeData,
  yt as default
};
