// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/components/GrantList/GrantsDisplay.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { BaseList as A } from "../../../../internal/BaseList/BaseList.js";
import { GrantItem as E } from "../GrantItem/GrantItem.js";
import { getGrantConfig as P } from "../GrantItem/utils.js";
import { GRANT_ADJUSTMENT_DETAILS as T } from "../GrantAdjustmentDetails/constants.js";
import { GrantRepaymentDetails as k } from "../GrantRepaymentDetails/GrantRepaymentDetails.js";
import { sharedCapitalOverviewAnalyticsEventProperties as R } from "../../constants.js";
import { useLandedPageEvent as x } from "../../../../../hooks/useAnalytics/useLandedPageEvent.js";
import B from "../../../../internal/SegmentedControl/SegmentedControl.js";
import U from "../../../../../core/Context/analytics/useAnalyticsContext.js";
import I from "../../../../../core/Context/useCoreContext.js";
import { useState as b, useMemo as c, useCallback as f } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import M from "../../../../internal/Button/Button.js";
import { ButtonVariant as O } from "../../../../internal/Button/types.js";
import { CapitalHeader as j } from "../../../../internal/CapitalHeader/CapitalHeader.js";
const w = {
  ...R,
  subCategory: "Grants overview"
}, h = ({ grants: t, showDetails: m }) => /* @__PURE__ */ e(A, { classNames: "adyen-pe-grant-list__items", children: t.map((s) => /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(E, { grant: s, showDetails: m.bind(null, s) }) }, s.id)) }), te = ({ grantList: t, hideTitle: m, newOfferAvailable: s, onNewOfferRequest: v }) => {
  const [D, g] = b(), [n, C] = b(), { i18n: y } = I(), r = U(), [i, d] = c(() => {
    const a = [], o = [];
    return t == null || t.forEach((l) => {
      l.status === "Active" || l.status === "Pending" ? a.push(l) : o.push(l);
    }), [a, o];
  }, [t]), G = c(() => t.length > 1 && i.length && d.length ? "SegmentedGrants" : "UnifiedGrants", [i.length, t.length, d.length]), S = c(() => s && !i.some((a) => a.status === "Pending"), [i, s]), N = f(() => {
    var a;
    try {
      return v();
    } finally {
      (a = r.addEvent) == null || a.call(r, "Clicked button", { ...w, label: "See new offer" });
    }
  }, [v, r]), p = c(() => n && P(n), [n]), _ = f(() => g(void 0), []), u = f((a, o) => {
    g(o), C(a);
  }, []);
  if (x({ ...w, label: "Capital overview" }), n)
    switch (D) {
      case T.unscheduledRepayment: {
        if (p != null && p.hasUnscheduledRepaymentDetails)
          return /* @__PURE__ */ e(k, { grant: n, onDetailsClose: _ });
        break;
      }
    }
  return /* @__PURE__ */ e("div", { className: "adyen-pe-grant-list", children: [
    /* @__PURE__ */ e("div", { className: "adyen-pe-grant-list__header-container", children: [
      /* @__PURE__ */ e(j, { hideTitle: m, titleKey: "capital.common.title" }),
      S ? /* @__PURE__ */ e(M, { onClick: N, className: "adyen-pe-grant-list__offer-button", variant: O.SECONDARY, children: y.get("capital.overview.grants.list.actions.newOffer") }) : null
    ] }),
    G === "UnifiedGrants" && /* @__PURE__ */ e(h, { grants: t, showDetails: u }),
    G === "SegmentedGrants" && /* @__PURE__ */ e(
      B,
      {
        items: [
          {
            label: "capital.overview.grants.list.tabs.labels.inProgress",
            content: /* @__PURE__ */ e(h, { grants: i, showDetails: u }),
            id: "active"
          },
          {
            label: "capital.overview.grants.list.tabs.labels.closed",
            content: /* @__PURE__ */ e(h, { grants: d, showDetails: u }),
            id: "inactive"
          }
        ],
        activeItem: "active",
        "aria-label": y.get("capital.overview.grants.list.tabs.a11y.label")
      }
    )
  ] });
};
export {
  te as GrantsDisplay
};
