// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/components/GrantList/GrantList.js
import { jsx as a } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
/* empty css               */
import { useState as k, useCallback as t } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { CapitalOffer as C } from "../../../CapitalOffer/components/CapitalOffer/CapitalOffer.js";
import { GrantsDisplay as b } from "./GrantsDisplay.js";
import { Fragment as x } from "../../../../../external/preact/dist/preact.module.js";
const V = ({
  externalDynamicOffersConfig: l,
  grantList: p,
  newOfferAvailable: m,
  onFundsRequest: e,
  onGrantListUpdateRequest: i,
  onOfferDismiss: o
}) => {
  const [c, s] = k(!1), r = t(() => s(!1), []), n = t(() => s(!0), []), u = t(() => {
    o ? o(r) : r();
  }, [r, o]), g = t(
    (f) => {
      e ? e(f) : (i(f), s(!1));
    },
    [e, i]
  );
  return /* @__PURE__ */ a(x, { children: c ? /* @__PURE__ */ a(
    C,
    {
      externalDynamicOffersConfig: l,
      onFundsRequest: g,
      onOfferDismiss: u
    }
  ) : /* @__PURE__ */ a(b, { grantList: p, newOfferAvailable: m, onNewOfferRequest: n }) });
};
export {
  V as GrantList
};
