// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkDetails/components/PaymentLinkTabs/PaymentLinkTabs.js
import { jsx as i } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import oe from "../../../../../core/Context/useCoreContext.js";
import { useCallback as b, useMemo as X } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import ae from "../../../../internal/Tabs/Tabs.js";
import ne from "../../../../../hooks/useTimezoneAwareDateFormatting.js";
import { TypographyVariant as Z } from "../../../../internal/Typography/types.js";
import ee from "../../../../internal/Typography/Typography.js";
import { PaymentLinkActivity as de } from "../PaymentLinkActivity/PaymentLinkActivity.js";
/* empty css                     */
import pe from "../../../../internal/CopyText/CopyText.js";
import { BACKEND_REDACTED_DATA_MARKER as ie, FRONTEND_REDACTED_DATA_MARKER as g } from "../../../../constants.js";
import ye from "../../../../internal/Link/Link.js";
import { DATE_FORMAT_PAYMENT_LINK_DETAILS_TABS as se } from "../../../../../constants/dateFormats.js";
import m from "../../../../internal/StructuredList/StructuredList.js";
import { Fragment as A } from "../../../../../external/preact/dist/preact.module.js";
const l = {
  root: "adyen-pe-payment-link-tabs",
  list: "adyen-pe-payment-link-tabs__list",
  listHeading: "adyen-pe-payment-link-tabs__list-heading",
  listLabel: "adyen-pe-payment-link-tabs__list-label",
  listValue: "adyen-pe-payment-link-tabs__list-value"
}, _e = ({ paymentLink: e }) => {
  const { i18n: a } = oe(), { dateFormat: h } = ne(), n = b((s) => Object.values(s).some((y) => y === ie), []), t = X(() => {
    var u, r, o, f, k, I, v, B, N, C, T, _, L, D, E, O, R, S, V, x, M, j, F, P, K, H, U, $, q, w, z, Y, G, J, Q, W;
    const s = e.linkInformation.linkType, y = {
      linkInformation: [
        {
          key: "payByLink.details.fields.paymentLinkId",
          value: e.linkInformation.paymentLinkId,
          config: { isCopyable: !0, linkUrl: e.linkInformation.paymentLink }
        },
        {
          key: "payByLink.details.fields.store",
          value: e.linkInformation.storeCode
        },
        {
          key: "payByLink.details.fields.merchantReference",
          value: e.linkInformation.merchantReference
        },
        {
          key: "payByLink.details.fields.createdOn",
          value: h(e.linkInformation.creationDate, se)
        },
        {
          key: "payByLink.details.fields.expiresOn",
          value: h(e.linkInformation.expirationDate, se)
        },
        {
          key: "payByLink.details.fields.linkType",
          value: a.has(`payByLink.common.linkType.${s}`) ? a.get(`payByLink.common.linkType.${s}`) : s
        },
        {
          key: "payByLink.details.fields.description",
          value: e.linkInformation.description
        }
      ],
      shopperInformation: [
        {
          key: "payByLink.details.fields.shopper.reference",
          value: (u = e.shopperInformation) == null ? void 0 : u.shopperReference
        },
        {
          key: "payByLink.details.fields.shopper.fullName",
          value: [(o = (r = e.shopperInformation) == null ? void 0 : r.shopperName) == null ? void 0 : o.firstName, (k = (f = e.shopperInformation) == null ? void 0 : f.shopperName) == null ? void 0 : k.lastName].filter(Boolean).join(" "),
          config: { isCopyable: !0 }
        },
        {
          key: "payByLink.details.fields.shopper.email",
          value: (I = e.shopperInformation) == null ? void 0 : I.shopperEmail,
          config: { isCopyable: !0 }
        },
        {
          key: "payByLink.details.fields.shopper.phone",
          value: (v = e.shopperInformation) == null ? void 0 : v.telephoneNumber,
          config: { isCopyable: !0 }
        },
        {
          key: "payByLink.details.fields.shopper.country",
          value: (B = e.shopperInformation) == null ? void 0 : B.shopperCountry
        },
        ...(N = e.shopperInformation) != null && N.shippingAddress && n((C = e.shopperInformation) == null ? void 0 : C.shippingAddress) ? [
          {
            key: "payByLink.details.fields.shippingAddress.title",
            value: g
          }
        ] : [],
        ...(T = e.shopperInformation) != null && T.billingAddress && n((_ = e.shopperInformation) == null ? void 0 : _.billingAddress) ? [
          {
            key: "payByLink.details.fields.billingAddress.title",
            value: g
          }
        ] : []
      ],
      shippingAddress: !((L = e.shopperInformation) != null && L.shippingAddress) || n((D = e.shopperInformation) == null ? void 0 : D.shippingAddress) ? [] : [
        {
          key: "payByLink.details.fields.shippingAddress.street",
          value: (O = (E = e.shopperInformation) == null ? void 0 : E.shippingAddress) == null ? void 0 : O.street,
          config: { isCopyable: !0 }
        },
        {
          key: "payByLink.details.fields.shippingAddress.houseNumberOrName",
          value: (S = (R = e.shopperInformation) == null ? void 0 : R.shippingAddress) == null ? void 0 : S.houseNumberOrName,
          config: { isCopyable: !0 }
        },
        {
          key: "payByLink.details.fields.shippingAddress.country",
          value: (x = (V = e.shopperInformation) == null ? void 0 : V.shippingAddress) == null ? void 0 : x.country
        },
        {
          key: "payByLink.details.fields.shippingAddress.city",
          value: (j = (M = e.shopperInformation) == null ? void 0 : M.shippingAddress) == null ? void 0 : j.city,
          config: { isCopyable: !0 }
        },
        {
          key: "payByLink.details.fields.shippingAddress.postalCode",
          value: (P = (F = e.shopperInformation) == null ? void 0 : F.shippingAddress) == null ? void 0 : P.postalCode,
          config: { isCopyable: !0 }
        }
      ],
      billingAddress: !((K = e.shopperInformation) != null && K.billingAddress) || n((H = e.shopperInformation) == null ? void 0 : H.billingAddress) ? [] : [
        {
          key: "payByLink.details.fields.billingAddress.street",
          value: ($ = (U = e.shopperInformation) == null ? void 0 : U.billingAddress) == null ? void 0 : $.street,
          config: { isCopyable: !0 }
        },
        {
          key: "payByLink.details.fields.billingAddress.houseNumberOrName",
          value: (w = (q = e.shopperInformation) == null ? void 0 : q.billingAddress) == null ? void 0 : w.houseNumberOrName,
          config: { isCopyable: !0 }
        },
        {
          key: "payByLink.details.fields.billingAddress.country",
          value: (Y = (z = e.shopperInformation) == null ? void 0 : z.billingAddress) == null ? void 0 : Y.country
        },
        {
          key: "payByLink.details.fields.billingAddress.city",
          value: (J = (G = e.shopperInformation) == null ? void 0 : G.billingAddress) == null ? void 0 : J.city,
          config: { isCopyable: !0 }
        },
        {
          key: "payByLink.details.fields.billingAddress.postalCode",
          value: (W = (Q = e.shopperInformation) == null ? void 0 : Q.billingAddress) == null ? void 0 : W.postalCode,
          config: { isCopyable: !0 }
        }
      ]
    };
    return Object.fromEntries(
      Object.entries(y).map(([le, te]) => [
        le,
        te.filter((c) => c.value != null && c.value !== "" && c.value !== void 0)
      ])
    );
  }, [e, h, a, n]), d = b((s) => /* @__PURE__ */ i("div", { className: l.listLabel, children: s }), []), p = b((s, y, u, r) => {
    let o;
    if (s && s.toString().includes(ie))
      o = g;
    else if (r != null && r.isCopyable && s && s !== "") {
      const f = r != null && r.linkUrl ? /* @__PURE__ */ i(ye, { href: r.linkUrl, target: "_blank", children: s.toString() }) : void 0;
      o = /* @__PURE__ */ i(pe, { textToCopy: s.toString(), visibleText: f, type: "Default" });
    } else
      o = s;
    return /* @__PURE__ */ i("div", { className: l.listValue, children: o });
  }, []), re = X(
    () => [
      {
        id: "linkInformation",
        label: "payByLink.details.tabs.linkInformation",
        content: /* @__PURE__ */ i(
          m,
          {
            classNames: l.list,
            items: t.linkInformation,
            align: "start",
            layout: "4-8",
            renderLabel: d,
            renderValue: p
          }
        )
      },
      {
        id: "shopperInformation",
        label: "payByLink.details.tabs.shopperInformation",
        content: /* @__PURE__ */ i(A, { children: [
          /* @__PURE__ */ i(
            m,
            {
              classNames: l.list,
              items: t.shopperInformation,
              align: "start",
              layout: "4-8",
              renderLabel: d,
              renderValue: p
            }
          ),
          t.shippingAddress.length > 0 && /* @__PURE__ */ i(A, { children: [
            /* @__PURE__ */ i(ee, { variant: Z.CAPTION, stronger: !0, className: l.listHeading, children: a.get("payByLink.details.fields.shippingAddress.title") }),
            /* @__PURE__ */ i(
              m,
              {
                classNames: l.list,
                items: t.shippingAddress,
                align: "start",
                layout: "4-8",
                renderLabel: d,
                renderValue: p
              }
            )
          ] }),
          t.billingAddress.length > 0 && /* @__PURE__ */ i(A, { children: [
            /* @__PURE__ */ i(ee, { variant: Z.CAPTION, stronger: !0, className: l.listHeading, children: a.get("payByLink.details.fields.billingAddress.title") }),
            /* @__PURE__ */ i(
              m,
              {
                classNames: l.list,
                items: t.billingAddress,
                align: "start",
                layout: "4-8",
                renderLabel: d,
                renderValue: p
              }
            )
          ] })
        ] })
      },
      {
        id: "activity",
        label: "payByLink.details.tabs.activity",
        content: /* @__PURE__ */ i(de, { activities: e.paymentLinkActivities ?? [] })
      }
    ],
    [t, d, p, a, e.paymentLinkActivities]
  );
  return /* @__PURE__ */ i("div", { className: l.root, children: /* @__PURE__ */ i(ae, { tabs: re }) });
};
export {
  _e as PaymentLinkTabs
};
