// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/utils.js
const x = (u, e) => {
  const s = u.map((r) => document.querySelector(`[name="${r}"]`)).filter((r) => r !== null).sort((r, d) => {
    const l = r.getBoundingClientRect(), a = d.getBoundingClientRect();
    return l.top - a.top;
  })[0];
  if (!s) return;
  const i = s.getBoundingClientRect();
  if (!(i.top >= e && i.bottom <= window.innerHeight)) {
    const r = Math.max(0, window.scrollY + i.top - e);
    window.scrollTo({ top: r, behavior: "smooth" });
  }
}, D = ({
  i18n: u,
  getFieldConfig: e
}) => {
  var m, s, i, y, r, d, l, a, n, p, o, c, t, b, q, v, N, A, k, h, L, B, f, T, V, S, R, w, E, O;
  return [
    {
      id: "store",
      fields: [
        {
          fieldName: "store",
          required: !0,
          visible: !0,
          label: "payByLink.creation.summary.fields.store"
        }
      ],
      isOptional: !1
    },
    {
      id: "payment",
      fields: [
        {
          fieldName: "linkValidity.quantity",
          required: !!((m = e("linkValidity")) != null && m.required),
          visible: !!e("linkValidity"),
          label: "payByLink.creation.summary.fields.linkValidity"
        },
        {
          fieldName: "linkValidity.durationUnit",
          required: !!((s = e("linkValidity")) != null && s.required),
          visible: !!e("linkValidity")
        },
        {
          fieldName: "amount.value",
          required: !!((i = e("amountValue")) != null && i.required),
          visible: !!e("amountValue"),
          label: "payByLink.creation.summary.fields.amountValue"
        },
        {
          fieldName: "amount.currency",
          required: !!((y = e("currency")) != null && y.required),
          visible: !!e("currency"),
          label: "payByLink.creation.summary.fields.currency",
          options: (r = e("currency")) == null ? void 0 : r.options
        },
        {
          fieldName: "reference",
          required: !!((d = e("merchantReference")) != null && d.required),
          visible: !!e("merchantReference"),
          label: "payByLink.creation.summary.fields.merchantReference"
        },
        {
          fieldName: "linkType",
          required: !!((l = e("linkType")) != null && l.required),
          visible: !!e("linkType"),
          label: "payByLink.creation.summary.fields.linkType"
        },
        {
          fieldName: "description",
          required: !!((a = e("description")) != null && a.required),
          visible: !!e("description"),
          label: "payByLink.creation.summary.fields.description"
        },
        {
          fieldName: "deliverAt",
          required: !!((n = e("deliveryDate")) != null && n.required),
          visible: !!e("deliveryDate")
        }
      ],
      isOptional: !1
    },
    {
      id: "customer",
      fields: [
        {
          fieldName: "shopperReference",
          required: !!((p = e("shopperReference")) != null && p.required),
          visible: !!e("shopperReference"),
          label: "payByLink.creation.summary.fields.shopperReference"
        },
        {
          fieldName: "shopperName.firstName",
          required: !!((o = e("shopperName")) != null && o.required),
          visible: !!e("shopperName"),
          label: "payByLink.creation.summary.fields.shopperName"
        },
        {
          fieldName: "shopperName.lastName",
          required: !!((c = e("shopperName")) != null && c.required),
          visible: !!e("shopperName"),
          label: "payByLink.creation.summary.fields.shopperLastName"
        },
        {
          fieldName: "shopperEmail",
          required: !!((t = e("emailAddress")) != null && t.required),
          visible: !!e("emailAddress"),
          label: "payByLink.creation.summary.fields.emailAddress"
        },
        {
          fieldName: "sendSuccessEmailToShopper",
          required: !!((b = e("sendSuccessEmailToShopper")) != null && b.required),
          visible: !!e("sendSuccessEmailToShopper")
        },
        {
          fieldName: "sendLinkToShopper",
          required: !!((q = e("sendLinkToShopper")) != null && q.required),
          visible: !!e("sendLinkToShopper"),
          label: "payByLink.creation.summary.fields.emailAddress"
        },
        {
          fieldName: "telephoneNumber",
          required: !!((v = e("phoneNumber")) != null && v.required),
          visible: !!e("phoneNumber"),
          label: "payByLink.creation.summary.fields.phoneNumber"
        },
        {
          fieldName: "countryCode",
          required: !!((N = e("countryCode")) != null && N.required),
          visible: !!e("countryCode"),
          label: "payByLink.creation.summary.fields.countryCode",
          options: (A = e("countryCode")) == null ? void 0 : A.options
        },
        {
          fieldName: "deliveryAddress.street",
          required: !!((k = e("deliveryAddress")) != null && k.required),
          visible: !!e("deliveryAddress"),
          label: "payByLink.creation.summary.fields.deliveryAddress.street"
        },
        {
          fieldName: "deliveryAddress.houseNumberOrName",
          required: !!((h = e("deliveryAddress")) != null && h.required),
          visible: !!e("deliveryAddress"),
          label: "payByLink.creation.summary.fields.deliveryAddress.houseNumberOrName"
        },
        {
          fieldName: "deliveryAddress.postalCode",
          required: !!((L = e("deliveryAddress")) != null && L.required),
          visible: !!e("deliveryAddress"),
          label: "payByLink.creation.summary.fields.deliveryAddress.postalCode"
        },
        {
          fieldName: "deliveryAddress.city",
          required: !!((B = e("deliveryAddress")) != null && B.required),
          visible: !!e("deliveryAddress"),
          label: "payByLink.creation.summary.fields.deliveryAddress.city"
        },
        {
          fieldName: "deliveryAddress.country",
          required: !!((f = e("deliveryAddress")) != null && f.required),
          visible: !!e("deliveryAddress"),
          label: "payByLink.creation.summary.fields.deliveryAddress.country"
        },
        {
          fieldName: "billingAddress.street",
          required: !!((T = e("billingAddress")) != null && T.required),
          visible: !!e("billingAddress"),
          label: "payByLink.creation.summary.fields.billingAddress.street"
        },
        {
          fieldName: "billingAddress.houseNumberOrName",
          required: !!((V = e("billingAddress")) != null && V.required),
          visible: !!e("billingAddress"),
          label: "payByLink.creation.summary.fields.billingAddress.houseNumberOrName"
        },
        {
          fieldName: "billingAddress.postalCode",
          required: !!((S = e("billingAddress")) != null && S.required),
          visible: !!e("billingAddress"),
          label: "payByLink.creation.summary.fields.billingAddress.postalCode"
        },
        {
          fieldName: "billingAddress.city",
          required: !!((R = e("billingAddress")) != null && R.required),
          visible: !!e("billingAddress"),
          label: "payByLink.creation.summary.fields.billingAddress.city"
        },
        {
          fieldName: "billingAddress.country",
          required: !!((w = e("billingAddress")) != null && w.required),
          visible: !!e("billingAddress"),
          label: "payByLink.creation.summary.fields.billingAddress.country"
        },
        {
          fieldName: "shopperLocale",
          required: !!((E = e("shopperLocale")) != null && E.required),
          visible: !!e("shopperLocale"),
          options: (O = e("shopperLocale")) == null ? void 0 : O.options
        }
      ],
      isOptional: !1
    },
    {
      id: "summary",
      title: u.get("payByLink.creation.form.steps.customer"),
      fields: [],
      isOptional: !0
    }
  ];
};
export {
  D as getFormSteps,
  x as scrollToFirstErrorField
};
