// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOffer/components/CapitalOfferSelection/CapitalOfferSelection.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import P from "../../../../internal/Typography/Typography.js";
import { TypographyVariant as S, TypographyElement as b } from "../../../../internal/Typography/types.js";
import E from "../../../../internal/Button/Button.js";
import { ButtonVariant as T } from "../../../../internal/Button/types.js";
import M from "../../../../../core/Context/useCoreContext.js";
import j from "../../../../../core/Context/analytics/useAnalyticsContext.js";
import { useDurationEvent as G } from "../../../../../hooks/useAnalytics/useDurationEvent.js";
import { useMemo as d, useState as B, useCallback as s, useEffect as H } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import I from "../../../../../hooks/useMutation/useMutation.js";
/* empty css                           */
import { sharedCapitalOfferAnalyticsEventProperties as Q } from "../CapitalOffer/constants.js";
import { getExpectedRepaymentDate as U, getPercentage as $ } from "../utils/utils.js";
import { CapitalErrorMessageDisplay as z } from "../utils/CapitalErrorMessageDisplay.js";
import { calculateSliderAdjustedMidValue as J } from "../../../../internal/Slider/Slider.js";
import { CAPITAL_REPAYMENT_FREQUENCY as K } from "../../../../constants.js";
import { debounce as W } from "../../../../utils/utils.js";
import { Fragment as X } from "../../../../../external/preact/dist/preact.module.js";
import Z from "../../../../internal/CapitalSlider/CapitalSlider.js";
import C from "../../../../internal/InfoBox/InfoBox.js";
import O from "../../../../internal/StructuredList/StructuredList.js";
import { useConfigContext as ee } from "../../../../../core/ConfigContext/context.js";
const x = {
  ...Q,
  subCategory: "Business financing offer"
}, te = () => /* @__PURE__ */ t("div", { className: "adyen-pe-capital-offer-selection__loading-container", children: [...Array(4)].map((e, r) => /* @__PURE__ */ t("div", { className: "adyen-pe-capital-offer-selection__loading-skeleton" }, r)) }), ae = ({ data: e }) => {
  const { i18n: r } = M(), u = d(() => {
    const n = e.expectedRepaymentPeriodDays && U(e.expectedRepaymentPeriodDays);
    return n ? r.date(n, { month: "long" }) : null;
  }, [e, r]);
  return /* @__PURE__ */ t("div", { className: "adyen-pe-capital-offer-selection__information", children: [
    /* @__PURE__ */ t(P, { el: b.SPAN, variant: S.BODY, wide: !0, children: u && r.get("capital.offer.common.repaymentInfo", {
      values: {
        amount: r.amount(e.thresholdAmount.value, e.thresholdAmount.currency),
        days: K,
        date: u
      }
    }) }),
    /* @__PURE__ */ t(
      O,
      {
        renderValue: (n) => /* @__PURE__ */ t(P, { el: b.SPAN, stronger: !0, variant: S.CAPTION, children: n }),
        renderLabel: (n) => /* @__PURE__ */ t(P, { el: b.SPAN, variant: S.CAPTION, children: n }),
        items: [
          { key: "capital.common.fields.fees", value: r.amount(e.feesAmount.value, e.feesAmount.currency) },
          {
            key: "capital.common.fields.dailyRepaymentRate",
            value: `${r.get("capital.common.values.percentage", {
              values: { percentage: $(e.repaymentRate) }
            })}`
          },
          {
            key: "capital.common.fields.expectedRepaymentPeriod",
            value: r.get("capital.common.values.numberOfDays", { values: { days: e.expectedRepaymentPeriodDays } })
          }
        ]
      }
    )
  ] });
}, _e = ({
  dynamicOffersConfig: e,
  dynamicOffersConfigError: r,
  emptyGrantOffer: u,
  onContactSupport: n,
  onOfferDismiss: h,
  onOfferSelect: V,
  requestedAmount: A
}) => {
  const { i18n: R } = M(), l = j(), g = d(() => {
    if (e)
      return J(
        e.minAmount.value,
        e.maxAmount.value,
        e.step
      );
  }, [e]), [c, _] = B(void 0), p = d(() => e == null ? void 0 : e.minAmount.currency, [e == null ? void 0 : e.minAmount.currency]), { createGrantOffer: f, getDynamicGrantOffer: w } = ee().endpoints, o = I({
    queryFn: w,
    options: {
      retry: 1,
      shouldRetry: s((a) => a.status === 500, []),
      onSettled: s(() => {
        k(!1);
      }, [])
    }
  }), i = I({
    queryFn: f,
    options: {
      onSuccess: (a) => V(a)
    }
  }), Y = s(() => {
    var a, m, D;
    try {
      i.mutate({
        body: {
          amount: ((a = o.data) == null ? void 0 : a.grantAmount.value) || c,
          currency: ((m = o.data) == null ? void 0 : m.grantAmount.currency) || p
        },
        contentType: "application/json"
      });
    } finally {
      (D = l.addEvent) == null || D.call(l, "Clicked button", { ...x, label: "Review offer" });
    }
  }, [p, o.data, c, i, l]), v = s(
    (a) => o.mutate({}, { query: { amount: a, currency: p } }),
    [p, o]
  ), [N, k] = B(!1), y = d(() => W(v, 300), [v]), q = s(
    (a) => {
      y.cancel(), k(!0), _(a);
    },
    [y]
  ), F = s(
    (a) => {
      var m;
      try {
        return y(a);
      } finally {
        (m = l.addEvent) == null || m.call(l, "Changed capital offer slider", {
          ...x,
          label: "Slider changed",
          currency: p,
          value: a
        });
      }
    },
    [y, l]
  );
  H(() => {
    e && !o.data && !c && (_(
      (a) => a || (A ? Number(A) : g || e.minAmount.value)
    ), v(c || g || e.minAmount.value));
  }, [e, o.data, v, g, A, c]);
  const L = d(
    () => i.isLoading || o.isLoading || N,
    [o.isLoading, N, i.isLoading]
  );
  return G(x), /* @__PURE__ */ t("div", { className: "adyen-pe-capital-offer-selection", children: i.error || o.error || u || r ? /* @__PURE__ */ t(
    z,
    {
      error: i.error || o.error || r,
      onBack: h,
      onContactSupport: n,
      emptyGrantOffer: u
    }
  ) : /* @__PURE__ */ t(X, { children: [
    e && /* @__PURE__ */ t(
      Z,
      {
        value: c,
        dynamicOffersConfig: e,
        onValueChange: q,
        onRelease: F
      }
    ),
    /* @__PURE__ */ t(C, { className: "adyen-pe-capital-offer-selection__details", children: !o.data || o.isLoading || N ? /* @__PURE__ */ t(te, {}) : o.data ? /* @__PURE__ */ t(ae, { data: o.data }) : null }),
    /* @__PURE__ */ t("div", { className: "adyen-pe-capital-offer-selection__buttons", children: [
      h && /* @__PURE__ */ t(E, { variant: T.SECONDARY, onClick: h, children: R.get("capital.common.actions.goBack") }),
      /* @__PURE__ */ t(
        E,
        {
          variant: T.PRIMARY,
          state: L ? "loading" : void 0,
          onClick: Y,
          disabled: i.isLoading || !(e != null && e.minAmount),
          "aria-label": R.get("capital.offer.selection.actions.reviewOffer"),
          children: R.get(
            L ? "capital.offer.selection.actions.reviewOffer.states.loading" : "capital.offer.selection.actions.reviewOffer"
          )
        }
      )
    ] })
  ] }) });
};
export {
  _e as CapitalOfferSelection
};
