// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOffer/components/CapitalOfferLegalNotice/CapitalOfferLegalNotice.js
import { jsx as r } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import m from "../../../../internal/AnchorButton/AnchorButton.js";
import { ButtonVariant as f } from "../../../../internal/Button/types.js";
import { SupportedLocation as p } from "../../../../internal/CapitalHeader/constants.js";
import c from "../../../../internal/Card/Card.js";
import { TypographyVariant as e } from "../../../../internal/Typography/types.js";
import i from "../../../../internal/Typography/Typography.js";
import { SUMMARY_TEXT_EMAIL as l } from "../CapitalOfferSummary/constants.js";
/* empty css                             */
import { Translation as n } from "../../../../internal/Translation/Translation.js";
import { Fragment as s } from "../../../../../external/preact/dist/preact.module.js";
import { useConfigContext as u } from "../../../../../core/ConfigContext/context.js";
const O = () => {
  var t, o, a;
  return ((a = (o = (t = u()) == null ? void 0 : t.extraConfig) == null ? void 0 : o.legalEntity) == null ? void 0 : a.countryCode) === p.US ? /* @__PURE__ */ r(c, { filled: !0, noOutline: !0, children: [
    /* @__PURE__ */ r(i, { variant: e.CAPTION, className: "adyen-pe-capital-offer-legal-notice--title", children: /* @__PURE__ */ r(n, { translationKey: "capital.offer.summary.legalNotice.US.title", fills: { break: /* @__PURE__ */ r("br", {}) } }) }),
    /* @__PURE__ */ r("br", {}),
    /* @__PURE__ */ r(i, { variant: e.CAPTION, className: "adyen-pe-capital-offer-legal-notice--description", children: /* @__PURE__ */ r(
      n,
      {
        translationKey: "capital.offer.summary.legalNotice.US.note",
        fills: {
          email: /* @__PURE__ */ r(
            m,
            {
              href: `mailto:${l}`,
              variant: f.TERTIARY,
              className: "adyen-pe-capital-offer-summary__info--email",
              children: l
            }
          ),
          break: /* @__PURE__ */ r(s, { children: [
            /* @__PURE__ */ r("br", {}),
            /* @__PURE__ */ r("br", {})
          ] })
        }
      }
    ) })
  ] }) : null;
};
export {
  O as CapitalOfferLegalNotice
};
