// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/constants.js
import { TransactionsView as e } from "./types.js";
import n from "../../internal/Calendar/calendar/timerange/presets/yearToDate.js";
import a from "../../internal/Calendar/calendar/timerange/presets/lastMonth.js";
import r from "../../internal/Calendar/calendar/timerange/presets/thisMonth.js";
import c from "../../internal/Calendar/calendar/timerange/presets/lastWeek.js";
import A from "../../internal/Calendar/calendar/timerange/presets/thisWeek.js";
import o from "../../internal/Calendar/calendar/timerange/presets/lastNDays.js";
const s = "adyen-pe-transactions", t = s + "-overview", i = s + "-details", y = {
  root: t,
  rootSmall: t + "--xs",
  summary: t + "__summary",
  summaryItem: t + "__summary-item",
  toolbar: t + "__toolbar",
  details: i,
  filterBarSmall: t + "__filter-bar-small",
  totalsError: t + "__totals-error"
}, T = Object.freeze({
  "common.filters.types.date.rangeSelect.options.last7Days": o(7),
  "common.filters.types.date.rangeSelect.options.last30Days": o(30),
  "common.filters.types.date.rangeSelect.options.thisWeek": A(),
  "common.filters.types.date.rangeSelect.options.lastWeek": c(),
  "common.filters.types.date.rangeSelect.options.thisMonth": r(),
  "common.filters.types.date.rangeSelect.options.lastMonth": a(),
  "common.filters.types.date.rangeSelect.options.yearToDate": n()
}), E = "common.filters.types.date.rangeSelect.options.custom", m = "common.filters.types.date.rangeSelect.options.last30Days", S = T[m], d = 2, O = [
  "ATM",
  "Capital",
  "Chargeback",
  "Correction",
  "Payment",
  "Refund",
  "Transfer",
  "Other"
], f = [
  { id: e.TRANSACTIONS, label: "transactions.overview.views.transactions", content: null },
  { id: e.INSIGHTS, label: "transactions.overview.views.insights", content: null }
], u = {
  balanceAccount: void 0,
  categories: [],
  createdDate: S,
  currencies: [],
  paymentPspReference: void 0,
  statuses: ["Booked"]
}, D = [
  "id",
  "balanceAccountId",
  "createdAt",
  "status",
  "paymentMethod",
  "category",
  "paymentPspReference",
  "currency",
  "netAmount",
  "amountBeforeDeductions"
], L = [
  "createdAt",
  "paymentMethod",
  "category",
  "currency",
  "netAmount",
  "amountBeforeDeductions"
], g = "Transaction component", h = "Transaction details", M = "Transactions insights", G = "Transactions list";
export {
  t as BASE_CLASS,
  L as DEFAULT_EXPORT_COLUMNS,
  i as DETAILS_CLASS,
  D as EXPORT_COLUMNS,
  u as INITIAL_FILTERS,
  f as TRANSACTIONS_VIEW_TABS,
  g as TRANSACTION_ANALYTICS_CATEGORY,
  h as TRANSACTION_ANALYTICS_SUBCATEGORY_DETAILS,
  M as TRANSACTION_ANALYTICS_SUBCATEGORY_INSIGHTS,
  G as TRANSACTION_ANALYTICS_SUBCATEGORY_LIST,
  O as TRANSACTION_CATEGORIES,
  T as TRANSACTION_DATE_RANGES,
  E as TRANSACTION_DATE_RANGE_CUSTOM,
  m as TRANSACTION_DATE_RANGE_DEFAULT,
  S as TRANSACTION_DATE_RANGE_DEFAULT_TIMESTAMPS,
  d as TRANSACTION_DATE_RANGE_MAX_YEARS,
  y as classes
};
