// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/Balances/BalancesCard.js
import { jsx as r } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { memo as w } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useState as y, useMemo as t } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import x from "../../../../../core/Context/useCoreContext.js";
import $ from "../../../../internal/ExpandableCard/ExpandableCard.js";
import { BalanceItem as u } from "../BalanceItem/BalanceItem.js";
import { useMaxWidthsState as C } from "../../hooks/useMaxWidths.js";
import { Fragment as M } from "../../../../../external/preact/dist/preact.module.js";
import { uniqueId as B } from "../../../../../utils/random/uniqueId.js";
const q = w(({ balances: b, isLoading: n, hiddenField: l, fullWidth: I, ...E }) => {
  const { i18n: s } = x(), [c, o] = C(), [i, m] = y(!1), d = t(() => s.get("transactions.overview.balances.currency.label"), [s]), [e, ...v] = t(() => b.map((a) => ({
    ...a,
    key: a.currency,
    availableBalanceElemId: B("elem"),
    reservedBalanceElemId: B("elem")
  })), [b]), f = t(() => {
    switch (l) {
      case "available":
        return s.get("transactions.overview.balances.lists.reserved");
      case "reserved":
        return s.get("transactions.overview.balances.lists.available");
      default:
        return s.get("transactions.overview.balances.lists.default");
    }
  }, [s, l]), p = t(
    () => /* @__PURE__ */ r(
      "div",
      {
        role: "listitem",
        "aria-label": e ? `${d}: ${e.currency}` : void 0,
        "aria-describedby": e ? `${e.availableBalanceElemId} ${e.reservedBalanceElemId}` : void 0,
        children: /* @__PURE__ */ r(
          u,
          {
            showLabelUnderline: i,
            balance: e,
            hiddenField: l,
            widths: c,
            isHeader: !0,
            isSkeleton: n,
            isLoading: n,
            onWidthsSet: o,
            reservedBalanceElemId: e == null ? void 0 : e.reservedBalanceElemId,
            availableBalanceElemId: e == null ? void 0 : e.availableBalanceElemId
          }
        )
      }
    ),
    [i, e, l, c, n, o, d]
  ), h = t(() => !n && v.length ? /* @__PURE__ */ r(M, { children: v.map((a) => /* @__PURE__ */ r(
    "div",
    {
      role: "listitem",
      "aria-label": `${d}: ${a.currency}`,
      "aria-describedby": `${a.availableBalanceElemId} ${a.reservedBalanceElemId}`,
      children: /* @__PURE__ */ r(
        u,
        {
          showLabelUnderline: i,
          balance: a,
          hiddenField: l,
          widths: c,
          onWidthsSet: o,
          reservedBalanceElemId: a.reservedBalanceElemId,
          availableBalanceElemId: a.availableBalanceElemId
        }
      )
    },
    a.key
  )) }) : void 0, [n, v, i, l, c, o, d]);
  return /* @__PURE__ */ r(
    $,
    {
      renderContent: ({ collapsibleContent: a }) => /* @__PURE__ */ r("div", { role: "list", "aria-label": f, children: [
        p,
        a
      ] }),
      fullWidth: I,
      onMouseEnter: () => m(!0),
      onFocus: () => m(!0),
      onMouseLeave: () => m(!1),
      onBlur: () => m(!1),
      ...E,
      children: h
    }
  );
});
export {
  q as BalancesCard
};
