// @adyen/adyen-platform-experience-web/dist/es/components/external/UIElement/UIElement.js
var m = Object.defineProperty;
var p = (n, e, t) => e in n ? m(n, e, { enumerable: !0, configurable: !0, writable: !0, value: t }) : n[e] = t;
var s = (n, e, t) => p(n, typeof e != "symbol" ? e + "" : e, t);
import { jsx as o } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import c from "classnames";
import { createRef as l } from "../../../external/preact/dist/preact.module.js";
import { AnalyticsProvider as d } from "../../../core/Context/analytics/AnalyticsProvider.js";
import h from "../../../core/Context/CoreProvider.js";
import u from "../BaseElement.js";
/* empty css               */
import { ConfigProvider as f } from "../../../core/ConfigContext/context.js";
class A extends u {
  constructor(t) {
    super(t);
    s(this, "componentRef", null);
    s(this, "componentToRender", null);
    s(this, "compRef");
    s(this, "customClassNames");
    s(this, "elementRef");
    s(this, "onContactSupport");
    s(this, "setUIElementStatus");
    this.setState = this.setState.bind(this), this.onContactSupport = t.onContactSupport, this.elementRef = this, this.compRef = l();
  }
  get isValid() {
    return !!this.state.isValid;
  }
  /**
   * Get the element's displayable name
   */
  get displayName() {
    return this.type;
  }
  /**
   * Get the element accessible name, used in the aria-label of the button that controls selected component
   */
  get accessibleName() {
    return this.displayName;
  }
  /**
   * Return the type of an element
   */
  get type() {
    var t;
    return (t = this.constructor) == null ? void 0 : t.type;
  }
  formatProps(t) {
    return t;
  }
  /**
   * Formats the component data output
   */
  formatData() {
    return {
      ...this.state
    };
  }
  setState(t) {
    this.state = { ...this.state, ...t };
  }
  setStatus(t, r) {
    var a, i;
    return (a = this.componentRef) != null && a.setStatus ? this.componentRef.setStatus(t, r) : (i = this.setUIElementStatus) == null || i.call(this, t), this;
  }
  render() {
    const t = this.props.core, r = this.props.onError || t.onError || null, a = t.update.bind(t);
    return t.session.errorHandler = r, /* @__PURE__ */ o(f, { type: this.type, session: t.session, children: /* @__PURE__ */ o(
      h,
      {
        componentRef: this.compRef,
        i18n: t.localization.i18n,
        getCdnConfig: t.getCdnConfig,
        getImageAsset: t.getImageAsset,
        getDatasetAsset: t.getDatasetAsset,
        getCdnDataset: t.getCdnDataset,
        loadingContext: t.loadingContext,
        externalErrorHandler: r,
        updateCore: a,
        children: /* @__PURE__ */ o(d, { componentName: this.displayName, analyticsEnabled: (t == null ? void 0 : t.analyticsEnabled) ?? !0, children: this.componentToRender && /* @__PURE__ */ o("section", { ref: this.compRef, className: c("adyen-pe-component", this.customClassNames), children: /* @__PURE__ */ o("div", { className: "adyen-pe-component__container", children: this.componentToRender() }) }) })
      }
    ) }, performance.now());
  }
}
export {
  A as UIElement,
  A as default
};
