// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/TransactionFilters/TransactionFilters.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { INITIAL_FILTERS as q, TRANSACTION_ANALYTICS_SUBCATEGORY_LIST as z, TRANSACTION_ANALYTICS_SUBCATEGORY_INSIGHTS as H, TRANSACTION_ANALYTICS_CATEGORY as K, TRANSACTION_CATEGORIES as Q } from "../../constants.js";
import { compareTransactionsFilters as Z } from "../utils.js";
import { useRef as J, useState as n, useMemo as o, useCallback as W, useEffect as C } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { useResponsiveContainer as X, containerQueries as $ } from "../../../../../hooks/useResponsiveContainer.js";
import V from "../../../../../hooks/useBalanceAccountSelection.js";
import _ from "./TransactionMultiSelectionFilter.js";
import ee from "./TransactionPspReferenceFilter.js";
import te from "../../../../../core/Context/useCoreContext.js";
import re from "../../hooks/useCurrencySelection.js";
import ce from "./TransactionDateFilter.js";
import { selectionOptionsFor as P } from "../MultiSelectionFilter/useMultiSelectionFilter/utils.js";
import ne from "../../../../internal/FormFields/Select/BalanceAccountSelector/BalanceAccountSelector.js";
import { Fragment as h } from "../../../../../external/preact/dist/preact.module.js";
import oe from "../../../../internal/FormFields/Select/Select.js";
import { FilterBar as se } from "../../../../internal/FilterBar/FilterBar.js";
const s = K, ve = ({
  availableCurrencies: a,
  balanceAccounts: B,
  isTransactionsView: T,
  onChange: i,
  setInsightsCurrency: E,
  ...w
}) => {
  var L;
  const { i18n: l } = te(), c = T ? z : H, e = J({ ...q }), U = X($.down.xs), [y, Y] = n(e.current.statuses), [m, F] = n(e.current.categories), [p, f] = n(e.current.currencies), [A, R] = n(e.current.createdDate), [S, I] = n(e.current.paymentPspReference), [r, O] = n(e.current.balanceAccount), u = o(
    () => ({ balanceAccount: r, categories: m, createdDate: A, currencies: p, paymentPspReference: S, statuses: y }),
    [r, m, A, p, S, y]
  ), D = o(() => Z(u, e.current), [u]), G = W(() => {
    Y(e.current.statuses), F(e.current.categories), f(e.current.currencies), R(e.current.createdDate), I(e.current.paymentPspReference), O(e.current.balanceAccount);
  }, []), x = o(() => P(Q), []), k = o(() => l.get("transactions.overview.filters.types.category.label"), [l]), M = o(() => P(a ?? []), [a]), d = o(() => l.get("transactions.overview.filters.types.currency.label"), [l]), v = "Currency filter", { resetBalanceAccountSelection: ae, ...b } = V({
    allowAllSelection: !1,
    onUpdateSelection: O,
    balanceAccounts: B,
    eventCategory: s,
    eventSubCategory: c
  }), { activeCurrency: N, currencySelectionOptions: g, onCurrencySelection: j } = re({
    eventLabel: v,
    onUpdateSelection: E,
    selectedCurrency: r == null ? void 0 : r.defaultCurrencyCode,
    availableCurrencies: a,
    eventCategory: s,
    eventSubCategory: c
  });
  return C(() => f(e.current.currencies), [a]), C(() => i == null ? void 0 : i(u), [i, u]), C(() => {
    !e.current.balanceAccount && r && (e.current.balanceAccount = r);
  }, [r]), /* @__PURE__ */ t(
    se,
    {
      ...w,
      ariaLabelKey: "transactions.overview.filters.label",
      canResetFilters: D && !1,
      resetFilters: G,
      children: [
        /* @__PURE__ */ t(ne, { ...b }),
        /* @__PURE__ */ t(
          ce,
          {
            createdDate: A,
            eventCategory: s,
            eventSubCategory: c,
            setCreatedDate: R,
            timezone: (L = b.activeBalanceAccount) == null ? void 0 : L.timeZone
          }
        ),
        T ? /* @__PURE__ */ t(h, { children: [
          /* @__PURE__ */ t(
            _,
            {
              eventLabel: "Category filter",
              eventCategory: s,
              eventSubCategory: c,
              placeholder: k,
              selectionOptions: x,
              selection: m,
              onUpdateFilter: F
            }
          ),
          /* @__PURE__ */ t(
            _,
            {
              eventCategory: s,
              eventSubCategory: c,
              eventLabel: v,
              placeholder: d,
              selectionOptions: M,
              selection: p,
              onUpdateFilter: f
            }
          ),
          /* @__PURE__ */ t(
            ee,
            {
              eventCategory: s,
              eventSubCategory: c,
              onChange: I,
              value: S
            }
          )
        ] }) : /* @__PURE__ */ t(h, { children: g.length > 1 ? /* @__PURE__ */ t(
          oe,
          {
            filterable: !1,
            multiSelect: !1,
            items: g,
            onChange: j,
            selected: N,
            "aria-label": d,
            placeholder: N || d,
            showOverlay: U,
            withoutCollapseIndicator: !0
          }
        ) : null })
      ]
    }
  );
};
export {
  ve as default
};
