// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/TransactionFilters/TransactionPspReferenceFilter.js
import { jsx as r } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useState as T, useMemo as l, useRef as g, useCallback as P, useEffect as A } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { TypographyVariant as x, TypographyElement as F } from "../../../../internal/Typography/types.js";
import k from "../../../../../hooks/useAnalytics/useFilterAnalyticsEvent.js";
import S from "../../../../../core/Context/useCoreContext.js";
import q from "../../../../internal/Typography/Typography.js";
import B from "../../../../internal/FormFields/InputBase.js";
/* empty css                                   */
import L from "../../../../internal/FilterBar/filters/TextFilter/TextFilter.js";
import { uniqueId as v } from "../../../../../utils/random/uniqueId.js";
import { CommitAction as E } from "../../../../../hooks/useCommitAction/types.js";
const R = "adyen-pe-psp-reference-filter", h = {
  root: R,
  input: R + "__input",
  title: R + "__title"
}, V = ({ eventCategory: p, eventSubCategory: c, onChange: a, value: t }) => {
  const { i18n: u } = S(), { logEvent: e } = k({ category: p, subCategory: c, label: "PSP reference filter" }), [s, n] = T(!1), f = l(() => u.get("transactions.overview.filters.types.paymentPspReference.label"), [u]), o = g(t), m = P(
    (i) => {
      o.current !== i && (o.current = i, e == null || e("update", null), a == null || a(i));
    },
    [e, a]
  ), b = P(() => n(!0), []);
  return A(() => {
    s && (n(!1), o.current !== t && (e == null || e("reset")));
  }, [s, t, e]), /* @__PURE__ */ r(
    L,
    {
      "aria-label": f,
      label: t ?? f,
      onChange: m,
      onResetAction: b,
      value: t,
      render: (i) => /* @__PURE__ */ r(V.EditModal, { ...i }),
      name: "pspReference"
    }
  );
};
V.EditModal = ({
  editAction: p,
  onChange: c,
  onValueUpdated: a,
  name: t,
  type: u,
  value: e
}) => {
  const { i18n: s } = S(), [n, f] = T(e), o = g(null), m = l(v, []), b = l(v, []), i = l(() => s.get("transactions.overview.filters.types.paymentPspReference.label"), [s]), C = l(() => s.get("transactions.overview.filters.types.paymentPspReference.placeholder"), [s]), _ = (w) => {
    const d = w.currentTarget, I = d.selectionEnd, y = d.value.replace(/[^a-z\d]/gi, "").slice(0, 16).toUpperCase();
    d.value = y, d.setSelectionRange(I, I), y !== n && (f(y || void 0), a(y || void 0));
  };
  return A(() => {
    switch (p) {
      case E.APPLY:
        c(n);
        break;
      case E.CLEAR:
        c();
        break;
    }
  }, [p, c, n]), A(() => {
    o.current && o.current.focus();
  }, []), /* @__PURE__ */ r("div", { className: h.root, children: [
    /* @__PURE__ */ r("div", { className: h.title, children: /* @__PURE__ */ r("label", { id: b, htmlFor: m, children: /* @__PURE__ */ r(q, { el: F.DIV, variant: x.BODY, strongest: !0, children: i }) }) }),
    /* @__PURE__ */ r("div", { className: h.input, children: /* @__PURE__ */ r(
      B,
      {
        autoComplete: "off",
        uniqueId: m,
        ref: o,
        placeholder: C,
        "data-testid": t,
        name: t,
        type: u,
        value: n,
        onInput: _
      }
    ) })
  ] });
};
export {
  V as default
};
