// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkCreation/components/Form/Summary/FormSummary.js
import { jsx as e } from "../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import U from "classnames";
import { Alert as q } from "../../../../../internal/Alert/Alert.js";
import G from "../../../../../../core/Context/useCoreContext.js";
import y from "../../../../../internal/Typography/Typography.js";
import { AlertTypeOption as M, AlertVariantOption as Y } from "../../../../../internal/Alert/types.js";
import { useResponsiveContainer as $, containerQueries as z } from "../../../../../../hooks/useResponsiveContainer.js";
import { Divider as Q } from "../../../../../internal/Divider/Divider.js";
import { TypographyElement as b, TypographyVariant as p } from "../../../../../internal/Typography/types.js";
import { invisibleFields as C } from "./constants.js";
import { useWizardFormContext as R } from "../../../../../../hooks/form/wizard/WizardFormContext.js";
import { useMemo as x } from "../../../../../../external/preact/hooks/dist/hooks.module.js";
import { Tag as O } from "../../../../../internal/Tag/Tag.js";
/* empty css                 */
import A from "../../../../../internal/StructuredList/StructuredList.js";
import { Fragment as j } from "../../../../../../external/preact/dist/preact.module.js";
const ce = ({ countryDatasetData: v }) => {
  var N, I, B;
  const { getSummaryData: P, getValues: T, getDisplayValue: _ } = R(), g = P(), { i18n: t } = G(), W = $(z.down.xs), E = x(() => {
    const { payment: r } = g, s = r == null ? void 0 : r.fields.filter(({ id: m }) => !C.includes(m)), c = {
      id: "store",
      label: "payByLink.creation.summary.fields.store",
      value: T("store"),
      displayValue: _("store")
    }, u = [...c ? [c] : [], ...s || []];
    return (u == null ? void 0 : u.map(({ label: m, value: f, id: d, displayValue: k }) => ({
      key: m || d,
      value: k || f,
      id: d,
      render: (a) => {
        var h, n;
        switch (a.id) {
          case "linkValidity.quantity": {
            const i = (h = r == null ? void 0 : r.fields) == null ? void 0 : h.find((o) => o.id === "linkValidity.durationUnit");
            return t.get(`payByLink.creation.fields.validity.linkValidityUnit.${i == null ? void 0 : i.value}`, {
              values: { quantity: a.value },
              count: Number(a.value)
            });
          }
          case "amount.value": {
            const i = (n = r == null ? void 0 : r.fields) == null ? void 0 : n.find((o) => o.id === "amount.currency");
            return t.amount(a.value, i == null ? void 0 : i.value);
          }
          case "linkType":
            return t.get(`payByLink.creation.form.linkTypes.${a.value}`);
          default:
            return a.value;
        }
      }
    }))) || [];
  }, [g, _, T, t]), l = x(() => {
    var k, a, h;
    const { customer: r } = g, s = r == null ? void 0 : r.fields.filter(({ id: n }) => !C.includes(n)), c = ({
      label: n,
      value: i,
      id: o,
      displayValue: w
    }) => ({
      key: n || o,
      value: w || i,
      id: o,
      render: (L) => {
        var V;
        switch (L.id) {
          case "countryCode":
          case "deliveryAddress.country":
          case "billingAddress.country":
            return (V = v == null ? void 0 : v.find((H) => H.id === L.value)) == null ? void 0 : V.name;
          default:
            return L.value;
        }
      }
    }), u = (k = s == null ? void 0 : s.filter((n) => {
      var i;
      return (i = n.id) == null ? void 0 : i.startsWith("deliveryAddress.");
    })) == null ? void 0 : k.map(c), S = (a = s == null ? void 0 : s.filter((n) => {
      var i;
      return (i = n.id) == null ? void 0 : i.startsWith("billingAddress.");
    })) == null ? void 0 : a.map(c), m = (h = s == null ? void 0 : s.filter((n) => {
      var i, o;
      return !((i = n == null ? void 0 : n.id) != null && i.startsWith("deliveryAddress.")) && !((o = n == null ? void 0 : n.id) != null && o.startsWith("billingAddress."));
    })) == null ? void 0 : h.map(c), f = r == null ? void 0 : r.fields.find((n) => n.id === "sendLinkToShopper" && n.value === !0), d = r == null ? void 0 : r.fields.find((n) => n.id === "sendSuccessEmailToShopper" && n.value === !0);
    return (f || d) && (m == null || m.splice(3, 0, {
      key: "payByLink.creation.summary.fields.emailNotifications",
      value: [d, f].filter(Boolean),
      render: () => /* @__PURE__ */ e("div", { className: "adyen-pe-payment-link-creation-form-summary__tags-container", children: [
        f && /* @__PURE__ */ e(O, { children: /* @__PURE__ */ e(y, { variant: p.CAPTION, el: b.SPAN, stronger: !0, children: t.get("payByLink.creation.summary.fields.emailNotifications.emailCreation") }) }),
        d && /* @__PURE__ */ e(O, { children: /* @__PURE__ */ e(y, { variant: p.CAPTION, el: b.SPAN, stronger: !0, children: t.get("payByLink.creation.summary.fields.emailNotifications.paymentSuccess") }) })
      ] })
    })), { nonAddressItems: m, deliveryAddressItems: u, billingAddressItems: S };
  }, [v, g, t]);
  return /* @__PURE__ */ e("section", { className: U("adyen-pe-payment-link-creation-form-summary", { "adyen-pe-payment-link-creation-form-summary--mobile": W }), children: [
    /* @__PURE__ */ e("section", { className: "adyen-pe-payment-link-creation-form-summary__section", children: [
      /* @__PURE__ */ e(y, { variant: p.SUBTITLE, className: "adyen-pe-payment-link-creation-form-summary__section-title", children: t.get("payByLink.creation.summary.paymentDetails") }),
      /* @__PURE__ */ e("div", { children: /* @__PURE__ */ e(A, { layout: "5-7", align: "start", condensed: !1, items: E }) })
    ] }),
    ((N = l.nonAddressItems) == null ? void 0 : N.length) && /* @__PURE__ */ e(j, { children: [
      /* @__PURE__ */ e(Q, {}),
      /* @__PURE__ */ e("section", { className: "adyen-pe-payment-link-creation-form-summary__section", children: [
        /* @__PURE__ */ e(y, { variant: p.SUBTITLE, className: "adyen-pe-payment-link-creation-form-summary__section-title", children: t.get("payByLink.creation.summary.shopperInformation") }),
        /* @__PURE__ */ e("div", { children: /* @__PURE__ */ e(A, { layout: "5-7", align: "start", condensed: !1, items: l.nonAddressItems }) })
      ] })
    ] }),
    !!((I = l.deliveryAddressItems) != null && I.length) && /* @__PURE__ */ e("section", { className: "adyen-pe-payment-link-creation-form-summary__section", children: [
      /* @__PURE__ */ e(y, { variant: p.BODY, className: "adyen-pe-payment-link-creation-form-summary__section-title", children: t.get("payByLink.creation.summary.deliveryAddress") }),
      /* @__PURE__ */ e("div", { children: /* @__PURE__ */ e(A, { layout: "5-7", align: "start", condensed: !1, items: l.deliveryAddressItems || [] }) })
    ] }),
    !!((B = l.billingAddressItems) != null && B.length) && /* @__PURE__ */ e("section", { className: "adyen-pe-payment-link-creation-form-summary__section", children: [
      /* @__PURE__ */ e(y, { variant: p.BODY, className: "adyen-pe-payment-link-creation-form-summary__section-title", children: t.get("payByLink.creation.summary.billingAddress") }),
      /* @__PURE__ */ e("div", { children: /* @__PURE__ */ e(A, { layout: "5-7", align: "start", condensed: !1, items: l.billingAddressItems || [] }) })
    ] }),
    /* @__PURE__ */ e(
      q,
      {
        className: "adyen-pe-payment-link-creation-form-summary__alert",
        variant: Y.TIP,
        type: M.HIGHLIGHT,
        description: t.get("payByLink.creation.summary.alertDescription")
      }
    )
  ] });
};
export {
  ce as FormSummary
};
