// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkDetails/components/PaymentLinkSkeleton/PaymentLinkSkeleton.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
/* empty css                         */
const a = {
  root: "adyen-pe-payment-link-skeleton",
  summaryContainer: "adyen-pe-payment-link-skeleton__summary-container",
  status: "adyen-pe-payment-link-skeleton__status",
  amount: "adyen-pe-payment-link-skeleton__amount",
  expirationDate: "adyen-pe-payment-link-skeleton__expiration-date",
  tabsContainer: "adyen-pe-payment-link-skeleton__tabs-container",
  tabLabelsContainer: "adyen-pe-payment-link-skeleton__tab-labels-container",
  tabLabel: "adyen-pe-payment-link-skeleton__tab-label",
  fieldsContainer: "adyen-pe-payment-link-skeleton__fields-container",
  fieldContainer: "adyen-pe-payment-link-skeleton__field-container",
  fieldLabel: "adyen-pe-payment-link-skeleton__field-label",
  fieldValue: "adyen-pe-payment-link-skeleton__field-value"
}, s = () => /* @__PURE__ */ e("div", { className: a.root, children: [
  /* @__PURE__ */ e("div", { className: a.summaryContainer, children: [
    /* @__PURE__ */ e("div", { className: a.status }),
    /* @__PURE__ */ e("div", { className: a.amount }),
    /* @__PURE__ */ e("div", { className: a.expirationDate })
  ] }),
  /* @__PURE__ */ e("div", { className: a.tabsContainer, children: [
    /* @__PURE__ */ e("div", { className: a.tabLabelsContainer, children: [...Array(3)].map((t, n) => /* @__PURE__ */ e("div", { className: a.tabLabel }, n)) }),
    /* @__PURE__ */ e("div", { className: a.fieldsContainer, children: [...Array(6)].map((t, n) => /* @__PURE__ */ e("div", { className: a.fieldContainer, children: [
      /* @__PURE__ */ e("div", { className: a.fieldLabel }),
      /* @__PURE__ */ e("div", { className: a.fieldValue })
    ] }, n)) })
  ] })
] });
export {
  s as PaymentLinkSkeleton
};
