// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkSettings/components/PaymentLinkSettingsContainer/PaymentLinkSettings.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { CONTAINER_CLASS_NAME as b, SIDEBAR_CONTAINER_CLASS_NAME as k, SECONDARY_NAV_CLASS_NAME as V, CONTENT_CONTAINER_CLASS_NAME as x, CONTENT_CONTAINER_MOBILE_CLASS_NAME as B } from "./constants.js";
/* empty css                                  */
import { usePaymentLinkSettingsContext as q } from "./context/context.js";
import f from "./components/PaymentLinkSettingsContent/PaymentLinkSettingsContent.js";
import w from "./components/SettingsActionButtons/SettingsActionButtons.js";
import { useState as D, useEffect as F, useCallback as A, useMemo as C } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { useResponsiveContainer as G, containerQueries as P } from "../../../../../hooks/useResponsiveContainer.js";
import H from "classnames";
import K from "./components/LoadingSkeleton/LoadingSkeleton.js";
import U from "./components/SettingsError/SettingsError.js";
import { ACCOUNT_MISCONFIGURATION as Y, WRONG_STORE_IDS as j } from "./utils/getSettingsErrorMessage.js";
import { Fragment as Q } from "../../../../../external/preact/dist/preact.module.js";
import { SecondaryNav as W } from "../../../../internal/SecondaryNav/SecondaryNav.js";
import { StoreSelector as z } from "../../../../internal/StoreSelector/StoreSelector.js";
import { Header as J } from "../../../../internal/Header/Header.js";
const X = "payByLink.settings.errors.couldNotLoadSettings", ue = ({
  navigateBack: a,
  ...c
}) => {
  const {
    activeMenuItem: u,
    setSelectedMenuItem: g,
    selectedStore: _,
    isLoadingStores: n,
    isShowingRequirements: h,
    storesError: d,
    setSelectedStore: I,
    filteredStores: i,
    menuItems: t,
    isLoadingContent: N,
    allStores: r,
    themeError: L,
    termsAndConditionsError: y
  } = q(), o = G(P.down.xs), [m, l] = D(!1);
  F(() => {
    const s = !!(!o || o && t && (t == null ? void 0 : t.length) === 1);
    l(s);
  }, [o, t]);
  const O = A(
    (s) => {
      l(s);
    },
    [l]
  ), R = A(() => {
    o && l(!1);
  }, [o]), E = C(() => {
    if (!(!r || r.length > 0 || n))
      return {
        errorCode: Y,
        type: "error",
        requestId: ""
      };
  }, [r, n]), p = C(() => {
    if (!(r && (r == null ? void 0 : r.length) > 0 && (i == null ? void 0 : i.length) !== 0 || n))
      return {
        errorCode: j,
        type: "error",
        requestId: ""
      };
  }, [r, i, n]), S = d ?? E ?? p, T = C(() => S ? /* @__PURE__ */ e(U, { error: S, onContactSupport: c.onContactSupport, errorMessage: X }) : null, [c.onContactSupport, S]), M = !!E || !!d || !!p;
  if (!t || t.length === 0) return null;
  const v = m && !L && !d && !y && !h;
  return /* @__PURE__ */ e("div", { className: b, children: [
    (!o || !m) && /* @__PURE__ */ e(J, { hideTitle: c.hideTitle, titleKey: "payByLink.settings.title" }),
    M ? T : /* @__PURE__ */ e(Q, { children: [
      /* @__PURE__ */ e("div", { className: H(x, { [B]: o && m }), children: t.length > 1 ? /* @__PURE__ */ e("div", { className: k, children: /* @__PURE__ */ e(
        W,
        {
          renderHeader: () => /* @__PURE__ */ e(
            z,
            {
              stores: i,
              selectedStoreId: _,
              setSelectedStoreId: I
            }
          ),
          contentVisible: m,
          loading: n,
          className: V,
          items: t,
          activeValue: u,
          onValueChange: g,
          onContentVisibilityChange: O,
          renderContent: (s) => /* @__PURE__ */ e(
            f,
            {
              activeMenuItem: s,
              isLoadingContent: N,
              navigateBack: a
            }
          ),
          renderLoadingContent: () => /* @__PURE__ */ e(K, { rowNumber: 3 })
        }
      ) }) : /* @__PURE__ */ e(
        f,
        {
          activeMenuItem: u,
          isLoadingContent: N,
          navigateBack: a
        }
      ) }),
      v && /* @__PURE__ */ e(
        w,
        {
          navigateBack: a || void 0,
          closeContent: o ? R : void 0
        }
      )
    ] })
  ] });
};
export {
  ue as default
};
