// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkSettings/components/PaymentLinkSettingsContainer/PaymentLinkSettingsContainer.js
import { jsx as m } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
/* empty css                                  */
import { PaymentLinkSettingsProvider as d } from "./context/context.js";
import k from "./PaymentLinkSettings.js";
import { useMemo as l } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { MENU_ITEMS as r } from "./context/constants.js";
import M from "../../../../../core/Context/useCoreContext.js";
const x = ({
  settingsItems: e,
  storeIds: c,
  embeddedInOverview: u,
  ...n
}) => {
  const { i18n: a } = M(), t = l(
    () => e && (e == null ? void 0 : e.length) > 0 ? r.filter((o) => e.includes(o.value)) : r,
    [e]
  ), f = l(() => (t.length > 0 ? t : r).map((i) => ({ ...i, label: a.get(i.label) })), [a, t]);
  return /* @__PURE__ */ m(
    d,
    {
      embeddedInOverview: u,
      selectedMenuItems: f,
      storeIds: c,
      navigateBack: n == null ? void 0 : n.navigateBack,
      children: /* @__PURE__ */ m(k, { ...n })
    }
  );
};
export {
  x as default
};
