// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinksOverview/utils/getPaymentLinksErrorMessage.js
import { jsx as a } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { UNDEFINED_ERROR as c } from "../../../utils/getCommonErrorCode.js";
import l from "../../../internal/CopyText/CopyText.js";
const u = "ACCOUNT_MISCONFIGURATION", y = "WRONG_STORE_IDS", N = (o, t, e, r) => {
  var m;
  if (!o) return c;
  const s = e ? "common.errors.errorCode" : "common.errors.errorCodeSupport", n = {
    [s]: o.requestId ? /* @__PURE__ */ a(l, { copyButtonAriaLabelKey: "common.actions.copy.labels.errorCode", textToCopy: o.requestId }) : null
  };
  switch (o.errorCode) {
    case u:
      return {
        title: "common.errors.somethingWentWrong",
        message: ["payByLink.common.errors.accountConfiguration", "common.errors.contactSupport"],
        translationValues: n,
        onContactSupport: e
      };
    case y:
      return {
        title: "common.errors.somethingWentWrong",
        message: ["payByLink.common.errors.storeID", "common.errors.contactSupport"],
        translationValues: n,
        onContactSupport: e
      };
    case "29_001":
      return (m = o.invalidFields) != null && m.some((i) => i.name === "paymentLinkId") ? {
        title: "payByLink.overview.errors.listEmpty",
        message: ["payByLink.overview.errors.listEmpty.message"],
        images: {
          desktop: r == null ? void 0 : r({ name: "no-results-found" }),
          mobile: r == null ? void 0 : r({ name: "no-results-found", subFolder: "images/small" })
        }
      } : {
        title: "common.errors.somethingWentWrong",
        message: ["payByLink.overview.errors.couldNotLoadLinks", "common.errors.retry"],
        onContactSupport: e
      };
    case "00_500":
    case void 0:
      return {
        title: "common.errors.somethingWentWrong",
        message: ["payByLink.overview.errors.couldNotLoadLinks", s],
        translationValues: n,
        refreshComponent: !0
      };
    default:
      return {
        title: "common.errors.somethingWentWrong",
        message: [t, s],
        translationValues: n,
        onContactSupport: e,
        refreshComponent: !0
      };
  }
};
export {
  u as ACCOUNT_MISCONFIGURATION,
  y as WRONG_STORE_IDS,
  N as getPaymentLinksErrorMessage
};
