// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/BalanceItem/BalanceItem.js
import { jsx as d } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import h from "../../../../../core/Context/useCoreContext.js";
import { useRef as s, useMemo as C } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { SummaryItem as b } from "../SummaryItem/SummaryItem.js";
const R = ({
  balance: e,
  hiddenField: r,
  isHeader: o = !1,
  showLabelUnderline: a = !1,
  isSkeleton: l = !1,
  isLoading: n = !1,
  widths: i,
  onWidthsSet: u,
  availableBalanceElemId: m,
  reservedBalanceElemId: f
}) => {
  const { i18n: t } = h(), v = s(null), c = s(null), g = s(null), y = C(() => {
    const p = {
      elemId: m,
      labelKey: "transactions.overview.balances.tags.available",
      ref: v,
      skeletonWidth: 80,
      getValue: () => e && t.amount(e.value, e.currency),
      get ariaLabel() {
        return `${t.get(this.labelKey)}: ${this.getValue()}`;
      }
    }, V = {
      elemId: f,
      labelKey: "transactions.overview.balances.tags.reserved",
      ref: c,
      skeletonWidth: 80,
      getValue: () => e && t.amount(e.reservedValue, e.currency),
      tooltipLabel: "transactions.overview.balances.tags.reserved.description",
      get ariaLabel() {
        return `${t.get(this.labelKey)}: ${this.getValue()}`;
      }
    };
    return [
      ...r !== "available" ? [p] : [],
      ...r !== "reserved" ? [V] : [],
      {
        ref: g,
        skeletonWidth: 40,
        valueHasLabelStyle: !0,
        getValue: () => e == null ? void 0 : e.currency
      }
    ];
  }, [e, r, t]);
  return /* @__PURE__ */ d(
    b,
    {
      showLabelUnderline: a,
      isEmpty: !e,
      columnConfigs: y,
      isHeader: o,
      isSkeletonVisible: l,
      isLoading: n,
      widths: i,
      onWidthsSet: u
    }
  );
};
export {
  R as BalanceItem
};
