// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkSettings/hooks/useSaveAction.js
import P from "../../../../hooks/useMutation/useMutation.js";
import { useRef as N, useEffect as O, useCallback as T } from "../../../../external/preact/hooks/dist/hooks.module.js";
import { MenuItem as U } from "../components/PaymentLinkSettingsContainer/context/constants.js";
import { getThemePayload as R } from "../components/PaymentLinkSettingsContainer/utils/getThemePayload.js";
import { isThemePayload as A } from "../components/PaymentLinkThemeContainer/types.js";
import { useConfigContext as W } from "../../../../core/ConfigContext/context.js";
import { isFunction as j, isUndefined as q } from "../../../../utils/value/is.js";
const K = (r, f, m, e, n, i, t, l, c, s, p) => {
  const { updatePayByLinkTheme: F, savePayByLinkSettings: k } = W().endpoints, h = N(null);
  O(() => () => {
    h.current && clearTimeout(h.current);
  }, []);
  const x = T(() => {
    p && j(p) ? h.current = setTimeout(() => {
      p();
    }, 500) : r(!1);
  }, [p, r]), y = P({
    queryFn: F,
    options: {
      onSuccess: (o) => {
        const u = { brandName: o == null ? void 0 : o.brandName, logo: o == null ? void 0 : o.logoUrl, fullWidthLogo: o == null ? void 0 : o.fullWidthLogoUrl };
        c(u), s(R(u)), f(!1), m(!0), r(!1);
      },
      onError: () => {
        f(!0), m(!1), r(!1);
      }
    }
  }), C = T(() => {
    !e || q(n) || !t() || A(n) && (r(!0), y.mutate({ contentType: "multipart/form-data", body: n }, { path: { storeId: e } }));
  }, [y, e, n, t, r]), L = P({
    queryFn: k,
    options: {
      onSuccess: (o) => {
        const u = !o || !(o != null && o.termsOfServiceUrl) ? { termsOfServiceUrl: "" } : o;
        c(u), s(u), f(!1), m(!0), x();
      },
      onError: () => {
        f(!0), m(!1), r(!1);
      }
    }
  }), b = T(() => {
    !e || q(n) || !t() || A(n) || (r(!0), L.mutate(
      {
        contentType: "application/json",
        body: n
      },
      { path: { storeId: e } }
    ));
  }, [L, e, n, t, r]);
  return { onSave: T(() => {
    if (i) {
      if (l(!0), i === U.theme) return C();
      if (i === U.termsAndConditions) return b();
    }
  }, [i, b, C, l]) };
};
export {
  K as useSaveAction
};
