// @adyen/adyen-platform-experience-web/dist/es/components/utils/disputes/constants.js
const s = 10, e = {
  mc: "Mastercard",
  visa: "Visa",
  ach: "ACH Direct Debit",
  amex: "American Express",
  discover: "Discover",
  elo: "Elo",
  jcb: "JCB",
  pulse: "PULSE",
  sepadirectdebit: "SEPA Direct Debit",
  others: "Others"
}, o = {
  FRAUD: "disputes.common.reasonCategories.fraud",
  CONSUMER_DISPUTE: "disputes.common.reasonCategories.consumerDispute",
  PROCESSING_ERROR: "disputes.common.reasonCategories.processingError",
  REQUEST_FOR_INFORMATION: "disputes.common.reasonCategories.requestForInformation",
  AUTHORISATION_ERROR: "disputes.common.reasonCategories.authorisationError",
  ADJUSTMENT: "disputes.common.reasonCategories.adjustment",
  OTHER: "disputes.common.reasonCategories.other"
}, t = {
  CHARGEBACKS: "disputes.overview.common.statusGroups.chargebacks",
  FRAUD_ALERTS: "disputes.overview.common.statusGroups.fraudAlerts",
  ONGOING_AND_CLOSED: "disputes.overview.common.statusGroups.ongoingAndClosed"
}, n = {
  ACCEPTED: "disputes.common.statuses.accepted",
  EXPIRED: "disputes.common.statuses.expired",
  LOST: "disputes.common.statuses.lost",
  PENDING: "disputes.common.statuses.pending",
  RESPONDED: "disputes.common.statuses.responded",
  UNDEFENDED: "disputes.common.statuses.undefended",
  UNRESPONDED: "disputes.common.statuses.unresponded",
  WON: "disputes.common.statuses.won"
}, i = {
  CHARGEBACK: "disputes.management.details.types.chargeback",
  NOTIFICATION_OF_FRAUD: "disputes.management.details.types.notificationOfFraud",
  REQUEST_FOR_INFORMATION: "disputes.management.details.types.requestForInformation"
};
export {
  s as DISPUTE_ACTION_NEEDED_URGENTLY_THRESHOLD_DAYS,
  e as DISPUTE_PAYMENT_SCHEMES,
  o as DISPUTE_REASON_CATEGORIES,
  n as DISPUTE_STATUSES,
  t as DISPUTE_STATUS_GROUPS,
  i as DISPUTE_TYPES
};
