// @adyen/adyen-platform-experience-web/dist/es/components/utils/disputes/getDisputesErrorMessage.js
import { jsx as n } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { UNDEFINED_ERROR as t } from "../getCommonErrorCode.js";
import m from "../../internal/CopyText/CopyText.js";
const c = (e, o, r) => {
  if (!e) return t;
  switch (e.errorCode) {
    case void 0:
      return {
        title: "common.errors.somethingWentWrong",
        message: [o, "common.errors.retry"],
        refreshComponent: !0
      };
    case "00_500": {
      const s = r ? "common.errors.errorCode" : "common.errors.errorCodeSupport";
      return {
        title: "common.errors.somethingWentWrong",
        message: [o, s],
        translationValues: {
          [s]: e.requestId ? /* @__PURE__ */ n(m, { isUnderlineVisible: !0, copyButtonAriaLabelKey: "common.actions.copy.labels.errorCode", textToCopy: e.requestId }) : null
        },
        onContactSupport: r
      };
    }
    case "30_112":
      return {
        title: "common.errors.notFound",
        message: ["disputes.management.common.errors.notFound"],
        onContactSupport: r
      };
    default:
      return t;
  }
};
export {
  c as getDisputesErrorMessage
};
