// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/InputBase.js
import { jsx as i } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { ARIA_ERROR_SUFFIX as Z } from "../../../core/Errors/constants.js";
import T from "classnames";
import { forwardRef as D } from "../../../external/preact/compat/dist/compat.module.js";
import { useCallback as t } from "../../../external/preact/hooks/dist/hooks.module.js";
import { filterDisallowedCharacters as K } from "./utils.js";
/* empty css                */
import { ButtonVariant as U } from "../Button/types.js";
import { FieldError as B } from "./FieldError/FieldError.js";
import H from "./Select/Select.js";
import { Fragment as w } from "../../../external/preact/dist/preact.module.js";
import { hasOwnProperty as aa } from "../../../utils/struct/property.js";
function q({
  onInput: s,
  onKeyUp: h,
  trimOnBlur: C,
  onBlurHandler: l,
  onBlur: r,
  onFocusHandler: n,
  errorMessage: N,
  iconBeforeSlot: y,
  iconAfterSlot: v,
  dropdown: e,
  dropdownPosition: E = "start",
  onDropdownInput: c,
  onKeyDown: m,
  onUpdateDropdown: u,
  ...p
}, A) {
  const { classNameModifiers: _, isInvalid: f, isValid: k, readonly: R = !1, type: d, uniqueId: g, isCollatingErrors: o, disabled: P } = p;
  aa(p, "onChange") && console.error("Error: Form fields that rely on InputBase may not have an onChange property");
  const j = t(
    (a) => {
      s == null || s(a);
    },
    [s]
  ), M = t(
    (a) => {
      h && h(a);
    },
    [h]
  ), X = t(
    (a) => {
      l == null || l(a), C && (a.target.value = a.target.value.trim()), r == null || r(a);
    },
    [r, l, C]
  ), Y = t(
    (a) => {
      n == null || n(a);
    },
    [n]
  ), I = t(
    (a) => {
      var x;
      const $ = (x = a.target) == null ? void 0 : x.value;
      c == null || c($), e && (u == null || u({ ...e, value: $ }));
    },
    [e, c, u]
  ), z = T(
    "adyen-pe-input",
    [`adyen-pe-input--${d}`],
    p.className,
    {
      "adyen-pe-input--invalid": f,
      "adyen-pe-input--valid": k
    },
    _ == null ? void 0 : _.map((a) => `adyen-pe-input--${a}`)
  ), G = t(
    (a) => {
      K({ event: a, inputType: d, onValidInput: () => m == null ? void 0 : m(a) });
    },
    [d, m]
  ), {
    classNameModifiers: ea,
    uniqueId: ia,
    isInvalid: ta,
    isValid: sa,
    isCollatingErrors: la,
    autoFocus: ra,
    autofocus: na,
    ...J
  } = p, L = y || v, b = !!e && e.items.length > 0, Q = b && E === "start", S = b && E === "end", W = L || b, V = R || (e == null ? void 0 : e.readonly), F = /* @__PURE__ */ i(
    "input",
    {
      id: g,
      ...J,
      type: d,
      className: z,
      readOnly: R,
      "aria-describedby": o ? void 0 : `${g}${Z}`,
      "aria-invalid": f,
      onInput: j,
      onBlurCapture: X,
      onFocus: Y,
      onKeyDown: G,
      onKeyUp: M,
      disabled: P,
      ref: A,
      autoFocus: !1
    }
  ), O = t(
    () => e ? /* @__PURE__ */ i(
      H,
      {
        name: e.name,
        buttonVariant: U.TERTIARY,
        items: e.items,
        selected: e.value,
        onChange: I,
        readonly: V,
        filterable: e.filterable,
        "aria-label": e["aria-label"],
        classNameModifiers: ["input-field"],
        isCollatingErrors: o,
        disableToggleFocusOnClose: !0
      }
    ) : null,
    [e, I, o, V]
  );
  return /* @__PURE__ */ i(w, { children: [
    W ? /* @__PURE__ */ i(
      "div",
      {
        className: T("adyen-pe-input__container", {
          "adyen-pe-input--invalid": f
        }),
        children: [
          Q && /* @__PURE__ */ i("div", { role: "presentation", className: "adyen-pe-input__dropdown adyen-pe-input__dropdown--start", children: O() }),
          y && /* @__PURE__ */ i("span", { className: "adyen-pe-input__slot-before", children: y }),
          F,
          v && /* @__PURE__ */ i("span", { className: "adyen-pe-input__slot-after", children: v }),
          S && /* @__PURE__ */ i("div", { role: "presentation", className: "adyen-pe-input__dropdown adyen-pe-input__dropdown--end", children: O() })
        ]
      }
    ) : F,
    f && N && /* @__PURE__ */ i(B, { id: g, errorMessage: N })
  ] });
}
q.defaultProps = {
  type: "text",
  classNameModifiers: [],
  onInput: () => {
  }
};
const ba = D(q);
export {
  ba as default
};
