// @adyen/adyen-platform-experience-web/dist/es/components/external/PayoutDetails/components/PayoutData.js
import { jsx as e } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import A from "classnames";
import { useMemo as N } from "../../../../external/preact/hooks/dist/hooks.module.js";
import x from "../../../../core/Context/useCoreContext.js";
import I from "../../../internal/Accordion/Accordion.js";
import S from "../../../internal/Card/Card.js";
import { TypographyVariant as o, TypographyElement as b } from "../../../internal/Typography/types.js";
import c from "../../../internal/Typography/Typography.js";
import F from "../../../internal/DataOverviewDetails/DataOverviewDetailsSkeleton.js";
import X from "../../../../hooks/useTimezoneAwareDateFormatting.js";
/* empty css                */
import { PD_BASE_CLASS as $, PD_TITLE_CONTAINER_CLASS as H, PD_SECTION_CLASS as p, PD_TITLE_BA_CLASS as j, PD_TITLE_CLASS as z, PD_TITLE_CLASS_WITH_EXTRA_DETAILS as G, PD_EXTRA_DETAILS_CLASS as J, PD_EXTRA_DETAILS_ICON as V, PD_EXTRA_DETAILS_LABEL as W, PD_CONTENT_CLASS as q, PD_CARD_CLASS as L, PD_SECTION_AMOUNT_CLASS as P, PD_SECTION_GROSS_AMOUNT_CLASS as Y, PD_CARD_TITLE_CLASS as R, PD_SECTION_NET_AMOUNT_CLASS as K, PD_UNPAID_AMOUNT as Q, PD_BUTTON_ACTIONS as Z } from "./constants.js";
import { Tag as w } from "../../../internal/Tag/Tag.js";
import { TagVariant as ee } from "../../../internal/Tag/types.js";
import re from "../../../internal/Link/Link.js";
import te from "../../../internal/DataGrid/components/Icon.js";
import { isCustomDataObject as g } from "../../../internal/DataGrid/components/TableCells.js";
import { ButtonVariant as ne } from "../../../internal/Button/types.js";
import { ButtonActionsLayoutBasic as ae } from "../../../internal/Button/ButtonActions/types.js";
import ie from "../../../internal/Button/ButtonActions/ButtonActions.js";
import { getPayoutFundsCapturedType as se, getPayoutAdjustmentType as oe } from "../../../utils/translation/getters.js";
import { DATE_FORMAT_PAYOUT_DETAILS as ce } from "../../../../constants/dateFormats.js";
import _ from "../../../internal/StructuredList/StructuredList.js";
import { Fragment as ue } from "../../../../external/preact/dist/preact.module.js";
import { EMPTY_OBJECT as le } from "../../../../utils/value/constants.js";
const Ue = ({
  balanceAccountId: k,
  balanceAccountDescription: B,
  payout: d,
  extraFields: T
}) => {
  var f;
  const { payout: i } = d ?? le, { dateFormat: U } = X("UTC"), { i18n: s } = x(), n = N(() => {
    var l, u;
    const a = (u = (l = d == null ? void 0 : d.amountBreakdowns) == null ? void 0 : l.adjustmentBreakdown) == null ? void 0 : u.reduce(
      (r, t) => {
        var m;
        if (t.category && ((m = t.amount) != null && m.value)) {
          const { currency: v, value: h } = t.amount;
          r[h && h < 0 ? "subtractions" : "additions"].push({
            key: t.category,
            value: s.amount(h, v, { hideCurrency: !0 })
          });
        }
        return r;
      },
      { subtractions: [], additions: [] }
    );
    return a == null || a.subtractions.sort((r, t) => r.key.localeCompare(t.key)), a == null || a.additions.sort((r, t) => r.key.localeCompare(t.key)), a;
  }, [s, d]), C = N(() => {
    var l, u;
    const a = (u = (l = d == null ? void 0 : d.amountBreakdowns) == null ? void 0 : l.fundsCapturedBreakdown) == null ? void 0 : u.reduce(
      (r, t) => {
        var m, v, h, O;
        return ((m = t == null ? void 0 : t.amount) == null ? void 0 : m.value) === 0 || (v = t == null ? void 0 : t.amount) != null && v.value && t.category && r.push({
          key: t.category,
          value: s.amount((h = t == null ? void 0 : t.amount) == null ? void 0 : h.value, (O = t == null ? void 0 : t.amount) == null ? void 0 : O.currency, { hideCurrency: !0 })
        }), r;
      },
      []
    );
    return a == null || a.sort((r, t) => {
      const m = "capture";
      return r.key === m ? -1 : t.key === m ? 1 : r.key.localeCompare(t.key);
    }), a;
  }, [d, s]), { payoutAdjustmentTypeLabelRenderer: D, payoutFundsCapturedTypeLabelRenderer: M } = N(() => {
    const a = (l) => (u, r) => /* @__PURE__ */ e(c, { el: b.SPAN, variant: o.BODY, children: l(s, r) });
    return {
      payoutAdjustmentTypeLabelRenderer: a(oe),
      payoutFundsCapturedTypeLabelRenderer: a(se)
    };
  }, [s]), y = Object.entries(T || {}).filter(([, a]) => a.type !== "button" && a.visibility !== "hidden").map(([a, l]) => ({
    key: a,
    value: g(l) ? l.value : l,
    type: g(l) ? l.type : "text",
    config: g(l) ? l.config : void 0
  })) || [], E = N(() => [...T ? Object.values(T).filter((u) => u.type === "button").map((u) => {
    var r, t, m;
    return {
      title: u.value,
      variant: ne.SECONDARY,
      event: (r = u.config) == null ? void 0 : r.action,
      classNames: (t = u == null ? void 0 : u.config) != null && t.className ? [(m = u == null ? void 0 : u.config) == null ? void 0 : m.className] : []
    };
  }) : []].filter(Boolean), [T]);
  return /* @__PURE__ */ e(ue, { children: i ? /* @__PURE__ */ e("div", { className: $, children: [
    /* @__PURE__ */ e(
      "div",
      {
        className: A(z, {
          [G]: y.length
        }),
        children: [
          /* @__PURE__ */ e("div", { className: H, children: [
            /* @__PURE__ */ e(c, { variant: o.SUBTITLE, stronger: !0, children: s.get("payouts.details.tags.netPayout") }),
            i.isSumOfSameDayPayouts && /* @__PURE__ */ e(w, { variant: ee.BLUE, label: s.get("payouts.details.tags.sameDaySum") })
          ] }),
          /* @__PURE__ */ e(c, { variant: o.TITLE, large: !0, children: `${s.amount(i.payoutAmount.value, i.payoutAmount.currency, {
            hideCurrency: !0
          })} ${i.payoutAmount.currency}` }),
          (i == null ? void 0 : i.createdAt) && /* @__PURE__ */ e("time", { dateTime: i.createdAt, children: /* @__PURE__ */ e(c, { el: b.SPAN, variant: o.BODY, children: U(i.createdAt, ce) }) }),
          /* @__PURE__ */ e("div", { className: p, children: [
            B && /* @__PURE__ */ e(c, { variant: o.CAPTION, stronger: !0, wide: !0, children: `${B}` }),
            /* @__PURE__ */ e(c, { variant: o.CAPTION, className: j, children: `${k}` })
          ] })
        ]
      }
    ),
    y && y.length > 0 ? /* @__PURE__ */ e("div", { children: /* @__PURE__ */ e(
      _,
      {
        classNames: J,
        items: y,
        align: "start",
        layout: "4-8",
        renderLabel: (a) => /* @__PURE__ */ e("div", { className: W, children: a }),
        renderValue: (a, l, u, r) => {
          if (u === "link" && r)
            return /* @__PURE__ */ e(re, { classNames: [A(r == null ? void 0 : r.className)], href: r.href, target: r.target || "_blank", children: a });
          if (u === "icon" && r) {
            const t = { url: r.src, alt: r.alt || a };
            return /* @__PURE__ */ e("div", { className: A(V, r == null ? void 0 : r.className), children: [
              /* @__PURE__ */ e(te, { ...t }),
              /* @__PURE__ */ e(c, { variant: o.BODY, children: a })
            ] });
          }
          return /* @__PURE__ */ e(c, { className: A(r == null ? void 0 : r.className), variant: o.BODY, children: a });
        }
      }
    ) }) : null,
    /* @__PURE__ */ e("div", { className: q, children: [
      /* @__PURE__ */ e("div", { className: p, children: (i == null ? void 0 : i.fundsCapturedAmount) && (C && Object.keys(C).length > 0 ? /* @__PURE__ */ e(
        I,
        {
          header: /* @__PURE__ */ e(c, { variant: o.BODY, children: s.get("payouts.details.breakdown.fields.fundsCaptured") }),
          headerInformation: /* @__PURE__ */ e(c, { variant: o.BODY, children: s.amount(i.fundsCapturedAmount.value, i.fundsCapturedAmount.currency) }),
          children: /* @__PURE__ */ e("div", { className: p, children: /* @__PURE__ */ e("div", { className: L, children: /* @__PURE__ */ e(S, { noPadding: !0, children: /* @__PURE__ */ e(_, { items: C, renderLabel: M }) }) }) })
        }
      ) : /* @__PURE__ */ e("div", { className: A(P, Y), children: [
        /* @__PURE__ */ e(c, { variant: o.BODY, children: s.get("payouts.details.breakdown.fields.fundsCaptured") }),
        /* @__PURE__ */ e(c, { variant: o.BODY, children: s.amount(i.fundsCapturedAmount.value, i.fundsCapturedAmount.currency) })
      ] })) }),
      /* @__PURE__ */ e("div", { className: p, children: n != null && n.subtractions && Object.keys(n == null ? void 0 : n.subtractions).length > 0 || n != null && n.additions && Object.keys(n == null ? void 0 : n.additions).length > 0 ? /* @__PURE__ */ e(
        I,
        {
          header: /* @__PURE__ */ e(c, { variant: o.BODY, children: s.get("payouts.details.breakdown.fields.adjustments") }),
          headerInformation: /* @__PURE__ */ e(c, { variant: o.BODY, children: s.amount(i.adjustmentAmount.value, i.adjustmentAmount.currency) }),
          children: [
            (n == null ? void 0 : n.additions) && Object.keys(n == null ? void 0 : n.additions).length > 0 && /* @__PURE__ */ e("div", { className: L, children: /* @__PURE__ */ e(
              S,
              {
                noPadding: !0,
                renderHeader: /* @__PURE__ */ e(c, { className: R, variant: o.CAPTION, stronger: !0, children: s.get("payouts.details.breakdown.fields.additions") }),
                children: /* @__PURE__ */ e(_, { items: n == null ? void 0 : n.additions, renderLabel: D })
              }
            ) }),
            (n == null ? void 0 : n.subtractions) && Object.keys(n == null ? void 0 : n.subtractions).length > 0 && /* @__PURE__ */ e("div", { className: L, children: /* @__PURE__ */ e(
              S,
              {
                noPadding: !0,
                renderHeader: /* @__PURE__ */ e(c, { className: R, variant: o.CAPTION, stronger: !0, children: s.get("payouts.details.breakdown.fields.subtractions") }),
                children: /* @__PURE__ */ e(_, { items: n == null ? void 0 : n.subtractions, renderLabel: D })
              }
            ) })
          ]
        }
      ) : /* @__PURE__ */ e("div", { className: A(P, Y), children: [
        /* @__PURE__ */ e(c, { variant: o.BODY, children: s.get("payouts.details.breakdown.fields.adjustments") }),
        /* @__PURE__ */ e(c, { variant: o.BODY, children: s.amount(i.adjustmentAmount.value, i.adjustmentAmount.currency) })
      ] }) }),
      /* @__PURE__ */ e("div", { className: A(p), children: /* @__PURE__ */ e("div", { className: A(P, K), children: [
        /* @__PURE__ */ e(c, { variant: o.BODY, stronger: !0, children: s.get("payouts.details.breakdown.fields.netPayout") }),
        /* @__PURE__ */ e(c, { variant: o.BODY, stronger: !0, children: s.amount(i.payoutAmount.value, i.payoutAmount.currency) })
      ] }) })
    ] }),
    ((f = d == null ? void 0 : d.payout) == null ? void 0 : f.unpaidAmount) && /* @__PURE__ */ e("div", { className: Q, children: [
      /* @__PURE__ */ e(c, { variant: o.BODY, children: s.get("payouts.details.breakdown.fields.remainingAmount") }),
      /* @__PURE__ */ e(c, { variant: o.BODY, children: s.amount(d.payout.unpaidAmount.value, d.payout.unpaidAmount.currency) })
    ] }),
    E.length ? /* @__PURE__ */ e("div", { className: Z, children: /* @__PURE__ */ e(ie, { actions: E, layout: ae.BUTTONS_END }) }) : null
  ] }) : /* @__PURE__ */ e(F, { skeletonRowNumber: 6 }) });
};
export {
  Ue as PayoutData
};
