// @adyen/adyen-platform-experience-web/dist/es/hooks/useCommitAction/useCommitAction.js
import { useState as B, useCallback as u, useMemo as i, useEffect as Y } from "../../external/preact/hooks/dist/hooks.module.js";
import { ButtonVariant as d } from "../../components/internal/Button/types.js";
import { CommitAction as t } from "./types.js";
import v from "../../core/Context/useCoreContext.js";
import { EMPTY_OBJECT as L } from "../../utils/value/constants.js";
import { boolOrFalse as O } from "../../utils/value/bool.js";
const h = ({
  applyDisabled: A,
  applyTitle: n,
  resetDisabled: e,
  resetTitle: c,
  onResetAction: a
} = L) => {
  const { i18n: s } = v(), [m, o] = B(t.NONE), f = u(() => o(t.APPLY), [o]), C = u(() => {
    o(t.CLEAR), a && a();
  }, [o, a]), r = u(() => o(t.NONE), [o]), E = i(
    () => ({
      disabled: O(A),
      event: f,
      title: (n == null ? void 0 : n.trim()) || s.get("common.actions.apply.labels.default"),
      variant: d.PRIMARY
    }),
    [s, f, A, n]
  ), N = i(
    () => ({
      disabled: O(e),
      event: C,
      title: (c == null ? void 0 : c.trim()) || s.get("common.actions.reset.labels.default"),
      variant: d.SECONDARY
    }),
    [s, C, e, c]
  ), P = i(() => [E, N], [E, N]), b = i(() => m !== t.NONE, [m]);
  return Y(() => {
    switch (m) {
      case t.APPLY:
      case t.CLEAR:
        r();
        break;
    }
  }, [m, r]), { commitAction: m, commitActionButtons: P, committing: b, resetCommitAction: r };
};
export {
  h as default
};
