// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/Select/BalanceAccountSelector/BalanceAccountSelector.js
import { jsx as s } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { ALL_BALANCE_ACCOUNTS_SELECTION_ID as _ } from "../../../../../hooks/useBalanceAccountSelection.js";
import { memo as S } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useMemo as d, useCallback as f } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import L from "../../../../../core/Context/useCoreContext.js";
import { useResponsiveContainer as a, containerQueries as h } from "../../../../../hooks/useResponsiveContainer.js";
import { renderDefaultSingleSelectionCheckedness as E } from "../components/SelectListItem.js";
/* empty css                            */
import { Fragment as N } from "../../../../../external/preact/dist/preact.module.js";
import T from "../Select.js";
const m = "adyen-pe-balance-account-selector", u = `${m}__account-id`, l = `${m}__account-label`, D = S(
  ({
    activeBalanceAccount: r,
    balanceAccountSelectionOptions: o,
    onBalanceAccountSelection: n
  }) => {
    const { i18n: t } = L(), C = a(h.down.xs), i = d(() => t.get("common.filters.types.account.label"), [t]), p = f(
      (e) => /* @__PURE__ */ s(N, { children: [
        /* @__PURE__ */ s("div", { className: e.contentClassName, children: [
          e.item.name && /* @__PURE__ */ s("span", { className: l, children: e.item.name }),
          e.item.id !== _ && /* @__PURE__ */ s("span", { className: e.item.name ? u : l, children: e.item.id })
        ] }),
        E(e)
      ] }),
      []
    );
    return o && o.length > 1 && /* @__PURE__ */ s(
      T,
      {
        popoverClassNameModifiers: [m],
        onChange: n,
        filterable: !1,
        multiSelect: !1,
        placeholder: (r == null ? void 0 : r.id) || i,
        selected: r == null ? void 0 : r.id,
        withoutCollapseIndicator: !0,
        items: o,
        renderListItem: p,
        showOverlay: C,
        "aria-label": i
      }
    );
  }
);
export {
  D as default
};
