// @adyen/adyen-platform-experience-web/dist/es/components/external/ReportsOverview/components/ReportsOverview/ReportsOverview.js
import { jsx as s } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useCallback as A, useMemo as n, useEffect as Q } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import V from "../../../../../hooks/useBalanceAccountSelection.js";
import W from "../../../../../hooks/useDefaultOverviewFilterParams.js";
import X from "../../../../internal/FilterBar/filters/DateFilter/DateFilter.js";
import { LIMIT_OPTIONS as Z, DEFAULT_PAGE_LIMIT as $ } from "../../../../internal/Pagination/constants.js";
import { FilterParam as l } from "../../../../types.js";
import { ReportsTable as C, FIELDS as a } from "../ReportsTable/ReportsTable.js";
import { BASE_CLASS as z, EARLIEST_PAYOUT_SINCE_DATE as ee } from "./constants.js";
/* empty css                     */
import { useCustomColumnsData as re } from "../../../../../hooks/useCustomColumnsData.js";
import { hasCustomField as te } from "../../../../utils/customData/hasCustomField.js";
import { mergeRecords as se } from "../../../../utils/customData/mergeRecords.js";
import oe from "../../../../internal/Pagination/hooks/useCursorPaginatedRecords.js";
import { useFilterBarState as le, FilterBarMobileSwitch as ie, FilterBar as pe } from "../../../../internal/FilterBar/FilterBar.js";
import me from "../../../../internal/FormFields/Select/BalanceAccountSelector/BalanceAccountSelector.js";
import { useConfigContext as fe } from "../../../../../core/ConfigContext/context.js";
import { isFunction as ne } from "../../../../../utils/value/is.js";
import { Header as ce } from "../../../../internal/Header/Header.js";
const Ne = ({
  onFiltersChanged: i,
  balanceAccounts: c,
  allowLimitSelection: d = !0,
  preferredLimit: L = $,
  isLoadingBalanceAccount: _,
  onContactSupport: P,
  hideTitle: g,
  dataCustomization: e
}) => {
  var T, D, R, u;
  const { getReports: p } = fe().endpoints, { activeBalanceAccount: r, balanceAccountSelectionOptions: h, onBalanceAccountSelection: v } = V({ balanceAccounts: c }), { defaultParams: o, nowTimestamp: O, refreshNowTimestamp: m } = W("reports", r), b = A(
    async (t, f) => p({ signal: f, errorLevel: "error" }, {
      query: {
        ...t,
        type: "payout",
        createdSince: t[l.CREATED_SINCE] ?? o.current.defaultFilterParams[l.CREATED_SINCE],
        createdUntil: t[l.CREATED_UNTIL] ?? o.current.defaultFilterParams[l.CREATED_UNTIL],
        balanceAccountId: (r == null ? void 0 : r.id) ?? ""
      }
    }),
    [r == null ? void 0 : r.id, o, p]
  ), E = le(), w = n(() => ne(i) ? i : void 0, [i]), y = n(() => d ? Z : void 0, [d]), { canResetFilters: N, error: U, fetching: M, filters: F, limit: x, limitOptions: K, records: S, resetFilters: de, updateFilters: q, updateLimit: j, ...k } = oe({
    fetchRecords: b,
    dataField: "data",
    filterParams: o.current.defaultFilterParams,
    initialFiltersSameAsDefault: !0,
    onFiltersChanged: w,
    preferredLimit: L,
    preferredLimitOptions: y,
    enabled: !!(r != null && r.id) && !!p
  }), B = A(
    ({ records: t, retrievedData: f }) => se(t, f, (I, J) => I.createdAt === J.createdAt),
    []
  ), G = n(() => {
    var t;
    return te((t = e == null ? void 0 : e.list) == null ? void 0 : t.fields, a);
  }, [(T = e == null ? void 0 : e.list) == null ? void 0 : T.fields]), { customRecords: H, loadingCustomRecords: Y } = re({
    records: S,
    hasCustomColumn: G,
    onDataRetrieve: (D = e == null ? void 0 : e.list) == null ? void 0 : D.onDataRetrieve,
    mergeCustomData: B
  });
  return Q(() => {
    m();
  }, [F, m]), /* @__PURE__ */ s("div", { className: z, children: [
    /* @__PURE__ */ s(ce, { hideTitle: g, titleKey: "reports.overview.title", subtitleKey: "reports.overview.generateInfo", children: /* @__PURE__ */ s(ie, { ...E }) }),
    /* @__PURE__ */ s(pe, { ...E, ariaLabelKey: "reports.overview.filters.label", children: [
      /* @__PURE__ */ s(
        me,
        {
          activeBalanceAccount: r,
          balanceAccountSelectionOptions: h,
          onBalanceAccountSelection: v
        }
      ),
      /* @__PURE__ */ s(
        X,
        {
          canResetFilters: N,
          defaultParams: o,
          filters: F,
          nowTimestamp: O,
          refreshNowTimestamp: m,
          sinceDate: ee,
          timezone: "UTC",
          updateFilters: q
        }
      )
    ] }),
    /* @__PURE__ */ s(
      C,
      {
        balanceAccountId: r == null ? void 0 : r.id,
        loading: M || _ || !c || !r || Y,
        data: (R = e == null ? void 0 : e.list) != null && R.onDataRetrieve ? H : S,
        showPagination: !0,
        limit: x,
        limitOptions: K,
        onContactSupport: P,
        onLimitSelection: j,
        error: U,
        customColumns: (u = e == null ? void 0 : e.list) == null ? void 0 : u.fields,
        ...k
      }
    )
  ] });
};
export {
  Ne as ReportsOverview
};
