// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkDetails/components/PaymentLinkDetails/PaymentLinkDetails.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import K from "classnames";
import $ from "../../../../../core/Context/useCoreContext.js";
import { useMemo as F, useCallback as r, useState as P, useEffect as z } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import b from "../../../../internal/Button/ButtonActions/ButtonActions.js";
import { ButtonVariant as p } from "../../../../internal/Button/types.js";
import { useFetch as _ } from "../../../../../hooks/useFetch.js";
import { useModalContext as G } from "../../../../internal/Modal/Modal.js";
import { PaymentLinkSummary as H } from "../PaymentLinkSummary/PaymentLinkSummary.js";
import { PaymentLinkTabs as Q } from "../PaymentLinkTabs/PaymentLinkTabs.js";
import { PaymentLinkExpiration as W } from "../PaymentLinkExpiration/PaymentLinkExpiration.js";
/* empty css                        */
import { PaymentLinkSkeleton as X } from "../PaymentLinkSkeleton/PaymentLinkSkeleton.js";
import { PaymentLinkError as Z } from "../PaymentLinkError/PaymentLinkError.js";
import w from "../../../../internal/Typography/Typography.js";
import { TypographyVariant as O } from "../../../../internal/Typography/types.js";
import { Icon as R } from "../../../../internal/Icon/Icon.js";
import { Fragment as U } from "../../../../../external/preact/dist/preact.module.js";
import { Header as D } from "../../../../internal/Header/Header.js";
import { useConfigContext as tt } from "../../../../../core/ConfigContext/context.js";
import { EMPTY_OBJECT as et } from "../../../../../utils/value/constants.js";
const d = {
  root: "adyen-pe-payment-link-details",
  content: "adyen-pe-payment-link-details__content",
  expirationSuccessContainer: "adyen-pe-payment-link-details__expiration-success-container",
  expirationSuccessIcon: "adyen-pe-payment-link-details__expiration-success-icon"
}, vt = ({ id: u, onUpdate: l, hideTitle: Y, onContactSupport: C, onDismiss: a, isDismissButtonHidden: g }) => {
  const { i18n: i, getCdnDataset: f } = $(), { getPayByLinkPaymentLinkById: k } = tt().endpoints, {
    data: n,
    isFetching: M,
    error: h,
    refetch: L
  } = _(
    F(
      () => ({
        fetchOptions: {
          enabled: !!u && !!k
        },
        queryFn: async () => k(et, {
          path: {
            paymentLinkId: u
          }
        })
      }),
      [k, u]
    )
  ), { data: m, isFetching: V } = _({
    queryFn: r(async () => {
      const e = `${i.locale ?? "en-US"}`;
      return f ? await f({
        name: e,
        extension: "json",
        subFolder: "countries",
        fallback: []
      }) ?? [] : [];
    }, [f, i.locale])
  }), S = r(
    (e) => {
      const x = m == null ? void 0 : m.find((J) => J.id === e);
      return x == null ? void 0 : x.name;
    },
    [m]
  ), v = M || V, o = F(
    () => {
      var e;
      return n && {
        ...n,
        shopperInformation: {
          ...n == null ? void 0 : n.shopperInformation,
          shopperCountry: ((e = n == null ? void 0 : n.shopperInformation) == null ? void 0 : e.shopperCountry) && S(n == null ? void 0 : n.shopperInformation.shopperCountry)
        }
      };
    },
    [n, S]
  ), [c, y] = P("details"), [s, B] = P(!1);
  z(() => {
    let e;
    return s && (e = setTimeout(() => B(!1), 3e3)), () => clearTimeout(e);
  }, [s]);
  const E = r(async () => {
    if (o)
      try {
        await navigator.clipboard.writeText(o.linkInformation.paymentLink), B(!0);
      } catch (e) {
        console.error("Failed to copy link:", e);
      }
  }, [o]), I = r(() => {
    y("expirationConfirmation");
  }, []), N = r(() => {
    y("expirationSuccess"), l == null || l();
  }, [l]), T = r(() => {
    y("details"), L();
  }, [L]), A = r(() => {
    a == null || a();
  }, [a]), { withinModal: q } = G(), j = r(() => {
    if (v)
      return /* @__PURE__ */ t(X, {});
    if (!o || h)
      return /* @__PURE__ */ t(
        Z,
        {
          error: h,
          onDismiss: a,
          onContactSupport: C
        }
      );
    if (c === "expirationConfirmation")
      return /* @__PURE__ */ t(
        W,
        {
          paymentLink: o,
          onCancel: () => y("details"),
          onExpirationSuccess: N
        }
      );
    if (c === "expirationSuccess") {
      const e = [
        ...a ? [
          {
            title: i.get("payByLink.details.expirationSuccess.actions.goBackToList"),
            event: A,
            variant: p.SECONDARY
          }
        ] : [],
        {
          title: i.get("payByLink.details.expirationSuccess.actions.showDetails"),
          event: T,
          variant: p.SECONDARY
        }
      ];
      return /* @__PURE__ */ t("div", { className: d.expirationSuccessContainer, children: [
        /* @__PURE__ */ t(R, { name: "checkmark-circle-fill", className: d.expirationSuccessIcon }),
        /* @__PURE__ */ t(w, { variant: O.TITLE, children: i.get("payByLink.details.expirationSuccess.title") }),
        /* @__PURE__ */ t(w, { variant: O.BODY, children: i.get("payByLink.details.expirationSuccess.description") }),
        /* @__PURE__ */ t(b, { actions: e })
      ] });
    }
    if (c === "details") {
      const e = [
        {
          title: i.get(s ? "payByLink.details.actions.copied" : "payByLink.details.actions.copyLink"),
          event: E,
          variant: p.PRIMARY,
          iconLeft: /* @__PURE__ */ t(
            R,
            {
              className: "adyen-pe-payment-link-creation-form-success__button-icon",
              name: s ? "checkmark" : "copy"
            }
          )
        },
        ...o.linkInformation.status !== "expired" && o.linkInformation.status !== "completed" ? [
          {
            title: i.get("payByLink.details.actions.expire"),
            event: I,
            variant: p.SECONDARY
          }
        ] : [],
        ...!g && a ? [
          {
            title: i.get("payByLink.common.actions.goBack"),
            event: a,
            variant: p.SECONDARY
          }
        ] : []
      ];
      return /* @__PURE__ */ t(U, { children: [
        /* @__PURE__ */ t(H, { paymentLink: o }),
        /* @__PURE__ */ t(Q, { paymentLink: o }),
        /* @__PURE__ */ t(b, { actions: e })
      ] });
    }
  }, [
    v,
    o,
    h,
    c,
    a,
    C,
    N,
    i,
    A,
    T,
    s,
    E,
    I,
    g
  ]);
  return /* @__PURE__ */ t("div", { className: d.root, children: [
    /* @__PURE__ */ t("div", { className: K({ "adyen-pe-visually-hidden": c !== "details" }), children: /* @__PURE__ */ t(D, { hideTitle: Y, forwardedToRoot: !q, titleKey: "payByLink.details.title" }) }),
    /* @__PURE__ */ t("div", { className: d.content, children: j() })
  ] });
};
export {
  vt as PaymentLinkDetails
};
