// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/components/GrantAdjustmentDetails/GrantAdjustmentDetails.js
import { jsx as t } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import n from "classnames";
/* empty css                            */
import a from "../../../../internal/Button/Button.js";
import { ButtonVariant as c } from "../../../../internal/Button/types.js";
import l from "../../../../../core/Context/useCoreContext.js";
import { Icon as d } from "../../../../internal/Icon/Icon.js";
import { Header as p } from "../../../../internal/Header/Header.js";
const C = ({
  children: r,
  className: o,
  headerTitleKey: i,
  headerSubtitleKey: e,
  onDetailsClose: s
}) => {
  const { i18n: m } = l();
  return /* @__PURE__ */ t("div", { className: n("adyen-pe-grant-adjustment-details", o), children: [
    /* @__PURE__ */ t(p, { titleKey: i, subtitleKey: e, children: /* @__PURE__ */ t(
      a,
      {
        onClick: s,
        variant: c.TERTIARY,
        iconButton: !0,
        classNameModifiers: ["circle"],
        "aria-label": m.get("common.actions.dismiss.labels.dismiss"),
        children: /* @__PURE__ */ t(d, { name: "cross" })
      }
    ) }),
    r
  ] });
};
export {
  C as GrantAdjustmentDetails
};
