// @adyen/adyen-platform-experience-web/dist/es/constants/dateFormats.js
const t = "2-digit", T = "long", n = "numeric", o = "short", E = "shortOffset", A = {
  month: T,
  day: n,
  year: n
}, O = {
  hour: t,
  minute: t
}, _ = {
  ...A,
  ...O,
  month: o,
  hour12: !1
}, e = { ..._, year: void 0 }, R = { ...A, month: "short" }, M = { ...A, month: "long" }, I = A, D = {
  ..._,
  hour12: !0,
  timeZoneName: E
}, S = _, s = _, c = {
  ..._,
  month: T,
  ...O,
  timeZoneName: E
}, N = {
  ..._,
  month: T,
  weekday: T,
  timeZoneName: E
}, m = A, F = e, i = { ...A, weekday: T }, a = { ...A, month: o }, h = _, r = e, L = {
  ...A,
  ...O,
  weekday: T,
  timeZoneName: E
}, P = {
  month: o,
  weekday: T,
  hour: t,
  minute: n,
  year: void 0,
  timeZoneName: E
}, d = {
  month: o,
  weekday: T,
  hour: void 0,
  minute: void 0,
  year: "numeric",
  timeZoneName: void 0
};
export {
  R as DATE_FORMAT_CAPITAL_OVERVIEW,
  d as DATE_FORMAT_DELIVERY_DATE,
  I as DATE_FORMAT_DISPUTES,
  D as DATE_FORMAT_DISPUTE_DETAILS,
  M as DATE_FORMAT_MISSING_ACTION,
  S as DATE_FORMAT_PAYMENT_LINKS_OVERVIEW,
  s as DATE_FORMAT_PAYMENT_LINKS_OVERVIEW_EXPIRATION_DATE,
  N as DATE_FORMAT_PAYMENT_LINK_DETAILS_SUMMARY,
  c as DATE_FORMAT_PAYMENT_LINK_DETAILS_TABS,
  m as DATE_FORMAT_PAYOUTS,
  F as DATE_FORMAT_PAYOUTS_MOBILE,
  i as DATE_FORMAT_PAYOUT_DETAILS,
  a as DATE_FORMAT_REPORTS,
  P as DATE_FORMAT_RESPONSE_DEADLINE,
  h as DATE_FORMAT_TRANSACTIONS,
  r as DATE_FORMAT_TRANSACTIONS_MOBILE,
  L as DATE_FORMAT_TRANSACTION_DETAILS
};
