// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/CalendarInput/CalendarInput.js
import { jsx as r } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useState as u, useRef as R, useMemo as i, useCallback as m } from "../../../../external/preact/hooks/dist/hooks.module.js";
import x from "../../../../core/Context/useCoreContext.js";
import F from "../../../../hooks/useTimezoneAwareDateFormatting.js";
import { CalendarInputButton as L } from "./components/CalendarInputButton.js";
import { CalendarInputPopover as S } from "./components/CalendarInputPopover.js";
import { DATE_FORMAT_DELIVERY_DATE as _ } from "../../../../constants/dateFormats.js";
function j({ clearable: l, value: o, onInput: n, isInvalid: D, timezone: T }) {
  const { i18n: e } = x(), { dateFormat: p } = F(T), [c, a] = u(!1), f = R(null), h = i(() => o ? [new Date(o)] : void 0, [o]), O = i(
    () => o ? p(o, _) : e.get("common.inputs.select.placeholder"),
    [p, e, o]
  ), [d, g] = u(o), b = m(
    (t) => e.get("common.filters.types.date.calendar.label", { values: { monthOfYear: t.label } }),
    [e]
  ), A = m(
    (t) => {
      if (t) {
        const s = new Date(t).toISOString();
        n(s), g(s), s !== d && a(!1);
      }
    },
    [d, n]
  ), E = i(() => !!l && !!o, [l, o]), w = m(
    (t) => {
      var s, C;
      (s = t == null ? void 0 : t.preventDefault) == null || s.call(t), (C = t == null ? void 0 : t.stopPropagation) == null || C.call(t), n(""), g(void 0), a(!1);
    },
    [n]
  );
  return /* @__PURE__ */ r("div", { children: [
    /* @__PURE__ */ r(
      L,
      {
        label: O,
        isOpen: c,
        isInvalid: D,
        onClick: () => a((t) => !t),
        showClearButton: E,
        onClear: w,
        buttonRef: f
      }
    ),
    /* @__PURE__ */ r(
      S,
      {
        isOpen: c,
        onClose: () => a(!1),
        originDate: h,
        onHighlight: A,
        getGridLabel: b,
        targetElement: f
      }
    )
  ] });
}
export {
  j as CalendarInput
};
