// @adyen/adyen-platform-experience-web/dist/es/components/internal/DataGrid/components/InteractiveBody.js
import { jsx as a } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { TableCells as b } from "./TableCells.js";
import { useCallback as v } from "../../../../external/preact/hooks/dist/hooks.module.js";
import { useInteractiveDataGrid as f } from "../hooks/useInteractiveDataGrid.js";
import { Fragment as F } from "../../../../external/preact/dist/preact.module.js";
const B = ({
  data: r,
  columns: p,
  onRowClick: e,
  customCells: c,
  onRowHover: s
}) => {
  const { currentIndex: d, listeners: u, ref: m } = f({ totalRows: (r == null ? void 0 : r.length) ?? 0 }), i = v(
    (t) => () => e == null ? void 0 : e.callback(e != null && e.retrievedField ? t[e.retrievedField] : t),
    [e]
  );
  return /* @__PURE__ */ a(F, { children: r == null ? void 0 : r.map((t, n) => {
    const l = s && s.bind(null, n), o = s && s.bind(null, void 0);
    return /* @__PURE__ */ a(
      "div",
      {
        role: "row",
        tabIndex: 0,
        onMouseEnter: l,
        onFocus: l,
        onMouseLeave: o,
        onBlur: o,
        ref: m,
        "aria-selected": n === d,
        "data-index": n,
        className: "adyen-pe-data-grid__row adyen-pe-data-grid__row--clickable",
        onClick: i(t),
        onFocusCapture: u.onFocusCapture(n),
        onKeyDownCapture: u.onKeyDownCapture,
        children: /* @__PURE__ */ a(b, { columns: p, customCells: c, item: t, rowIndex: n })
      },
      t
    );
  }) });
};
export {
  B as InteractiveBody
};
