// @adyen/adyen-platform-experience-web/dist/es/components/internal/DataGrid/components/TableCells.js
import { jsx as c } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import y from "../DataGridCell.js";
import T from "./Icon.js";
import B from "../../Link/Link.js";
import { ButtonVariant as D } from "../../Button/types.js";
import a from "classnames";
import S from "../../Button/Button.js";
import { Fragment as I } from "../../../../external/preact/dist/preact.module.js";
const L = (o) => !!o && typeof o == "object" && "value" in o, _ = (o) => !!o && typeof o == "object" && o.type === "icon", O = (o) => !!o && typeof o == "object" && o.type === "button", w = (o) => !!o && typeof o == "object" && o.type === "link", Y = ({
  columns: o,
  customCells: t,
  item: l,
  rowIndex: j
}) => /* @__PURE__ */ c(I, { children: o.map(({ key: r, position: s }) => {
  var g, u, m, b, d, h, v;
  if (t != null && t[r])
    return /* @__PURE__ */ c(y, { "aria-labelledby": String(r), column: r, position: s, children: /* @__PURE__ */ c("div", {
      style: { width: "min-content" },
      // TODO create safeguard to remove "as any" assertion
      children: t[r]({
        key: r,
        value: l[r],
        item: l,
        rowIndex: j
      })
    }) }, r);
  const n = l[r], { value: i, type: f } = L(n) ? n : { value: n, type: "text" }, e = _(n) ? { url: (g = n == null ? void 0 : n.config) == null ? void 0 : g.src, alt: ((u = n == null ? void 0 : n.config) == null ? void 0 : u.alt) !== void 0 && ((m = n == null ? void 0 : n.config) == null ? void 0 : m.alt) !== null ? (b = n == null ? void 0 : n.config) == null ? void 0 : b.alt : n.value } : void 0, p = O(n) ? (x) => {
    var N;
    x.stopPropagation(), (N = n == null ? void 0 : n.config) == null || N.action();
  } : void 0;
  return /* @__PURE__ */ c(y, { "aria-labelledby": String(r), column: r, position: s, children: /* @__PURE__ */ c("div", { className: "adyen-pe-data-grid__cell-value", children: [
    _(n) && n.config && (e == null ? void 0 : e.url) && /* @__PURE__ */ c("div", { className: a("adyen-pe-data-grid__icon-cell", (d = n == null ? void 0 : n.config) == null ? void 0 : d.className), children: [
      /* @__PURE__ */ c(T, { ...e }),
      i.trim() && /* @__PURE__ */ c("span", { children: i })
    ] }),
    f === "text" && /* @__PURE__ */ c("span", { className: a((h = n == null ? void 0 : n.config) == null ? void 0 : h.className), children: i }),
    f === "button" && n.config && p && /* @__PURE__ */ c(S, { className: a((v = n.config) == null ? void 0 : v.className), onClick: p, variant: D.SECONDARY, children: i }),
    w(n) && n.config && /* @__PURE__ */ c(
      B,
      {
        classNames: n.config.className ? [n.config.className] : [],
        href: n.config.href,
        target: n.config.target,
        children: i
      }
    )
  ] }) }, r);
}) });
export {
  Y as TableCells,
  L as isCustomDataObject
};
