// @adyen/adyen-platform-experience-web/dist/es/components/internal/DataGrid/components/TableHeaderCell.js
import { jsx as n } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import l from "classnames";
import { useDataGridContext as c } from "../hooks/useDataGridContext.js";
import { useRef as o, useEffect as s } from "../../../../external/preact/hooks/dist/hooks.module.js";
const f = ({ cellKey: e, position: t, label: i }) => {
  const { registerCells: a } = c(), r = o(null);
  return s(() => {
    var d;
    r.current && a({
      column: e,
      width: (d = r.current) == null ? void 0 : d.getBoundingClientRect().width
    });
  }, [e, a]), /* @__PURE__ */ n(
    "div",
    {
      role: "columnheader",
      id: String(e),
      className: l("adyen-pe-data-grid__cell adyen-pe-data-grid__cell--heading", {
        "adyen-pe-data-grid__cell--right": t === "right",
        "adyen-pe-data-grid__cell--center": t === "center"
      }),
      children: /* @__PURE__ */ n("div", { ref: r, className: "adyen-pe-data-grid__cell--heading-content", children: i })
    }
  );
};
export {
  f as TableHeaderCell
};
