// @adyen/adyen-platform-experience-web/dist/es/components/internal/Button/DownloadButton/DownloadButton.js
import { jsx as o } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import h from "classnames";
import { useState as N, useEffect as l, useMemo as _ } from "../../../../external/preact/hooks/dist/hooks.module.js";
import S from "../../../../core/Context/useCoreContext.js";
import { useResponsiveContainer as F, containerQueries as Y } from "../../../../hooks/useResponsiveContainer.js";
import y from "../Button.js";
import { ButtonVariant as c } from "../types.js";
import k from "./useDownload.js";
/* empty css                    */
import A from "../../Spinner/Spinner.js";
import { Icon as E } from "../../Icon/Icon.js";
import { Fragment as L } from "../../../../external/preact/dist/preact.module.js";
import { downloadBlob as R } from "../../../../utils/file/download.js";
function P({
  className: C,
  disabled: g,
  endpointName: v,
  requestParams: x,
  setError: r,
  errorDisplay: f,
  onDownloadRequested: i,
  iconButton: n = !1,
  errorMessage: p,
  ...u
}) {
  const { i18n: m } = S(), [s, b] = N(!1), T = F(Y.down.xs), { data: d, error: t, isFetching: a } = k(v, x, s);
  l(() => {
    s && b(!1);
  }, [s]), l(() => {
    d && R(d);
  }, [d]), l(() => {
    r && t && r(t);
  }, [t, r]);
  const w = () => {
    b(!0), i == null || i();
  }, e = _(() => a ? /* @__PURE__ */ o(A, { size: "small" }) : /* @__PURE__ */ o(E, { name: "download" }), [a]), I = _(() => n ? e : a ? `${m.get("common.actions.download.labels.inProgress")}..` : m.get("common.actions.download.labels.default"), [e, m, a, n]);
  return /* @__PURE__ */ o(L, { children: [
    /* @__PURE__ */ o(
      "div",
      {
        className: h("adyen-pe-download", {
          "adyen-pe-download-icon-button-container": n
        }),
        children: [
          T ? /* @__PURE__ */ o(y, { iconButton: !0, variant: c.TERTIARY, onClick: w, ...u, children: e }) : /* @__PURE__ */ o(
            y,
            {
              className: h(
                "adyen-pe-download__button",
                { "adyen-pe-download__button--loading": a, "adyen-pe-download__button--icon": n },
                C
              ),
              disabled: g || a,
              variant: n ? c.TERTIARY : c.SECONDARY,
              onClick: w,
              ...!n && { iconLeft: e },
              ...u,
              children: I
            }
          ),
          t && f && /* @__PURE__ */ o("div", { className: "adyen-pe-download__error", children: f })
        ]
      }
    ),
    t && p && p(t)
  ] });
}
export {
  P as default
};
