// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/CalendarInput/components/CalendarInputButton.js
import { jsx as n } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { InteractionKeyCode as r } from "../../../../types.js";
import { ButtonVariant as u } from "../../../Button/types.js";
import s from "classnames";
import { Fragment as f } from "../../../../../external/preact/dist/preact.module.js";
import { Icon as e } from "../../../Icon/Icon.js";
import l from "../../../Button/Button.js";
function x({ label: i, isOpen: p, isInvalid: a, onClick: d, onClear: o, showClearButton: c, buttonRef: m }) {
  return /* @__PURE__ */ n(
    l,
    {
      ref: m,
      onClick: (t) => {
        t.preventDefault(), d();
      },
      variant: u.SECONDARY,
      "aria-haspopup": "dialog",
      "aria-expanded": p,
      "aria-invalid": a ? "true" : void 0,
      className: s("adyen-pe-button adyen-pe-dropdown__button", {
        "adyen-pe-button--invalid": a
      }),
      iconRight: c ? /* @__PURE__ */ n(f, { children: /* @__PURE__ */ n(
        "span",
        {
          role: "button",
          tabIndex: 0,
          className: "adyen-pe-dropdown__button-clear",
          onClick: (t) => {
            t.preventDefault(), t.stopPropagation(), o == null || o(t);
          },
          onKeyDown: (t) => {
            (t.code === r.ENTER || t.code === r.SPACE) && (t.preventDefault(), t.stopPropagation(), o == null || o(t));
          },
          children: /* @__PURE__ */ n(e, { name: "cross-circle-fill" })
        }
      ) }) : /* @__PURE__ */ n(e, { className: "adyen-pe-dropdown__button-collapse-indicator", name: "chevron-down" }),
      children: /* @__PURE__ */ n("span", { children: i })
    }
  );
}
export {
  x as CalendarInputButton
};
