// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/CalendarInput/components/CalendarInputPopover.js
import { jsx as o } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import p from "../../../Popover/Popover.js";
import { PopoverContainerPosition as c, PopoverContainerSize as f, PopoverContainerVariant as C } from "../../../Popover/types.js";
import { DEFAULT_FIRST_WEEK_DAY as u } from "../../../Calendar/calendar/timerange/presets/shared/offsetWeek.js";
import P from "../../../Calendar/hooks/useCalendarControlsRendering.js";
import g from "../../../../../core/Context/useCoreContext.js";
import h from "../../../Calendar/Calendar.js";
import r from "../../../Calendar/calendar/facade/index.js";
function x({ isOpen: e, onClose: n, originDate: i, onHighlight: t, getGridLabel: a, targetElement: l }) {
  const { i18n: s } = g(), [m, d] = P();
  return /* @__PURE__ */ o(
    p,
    {
      open: e,
      dismiss: () => {
        n();
      },
      variant: C.POPOVER,
      containerSize: f.MEDIUM,
      targetElement: l,
      withContentPadding: !0,
      position: c.TOP,
      fixedPositioning: !1,
      dismissible: !1,
      children: [
        /* @__PURE__ */ o(
          "div",
          {
            ref: d,
            role: "group",
            className: "adyen-pe-datepicker__controls",
            "aria-label": s.get("common.filters.types.date.calendar.navigation.label")
          }
        ),
        /* @__PURE__ */ o("div", { children: /* @__PURE__ */ o(
          h,
          {
            getGridLabel: a,
            firstWeekDay: u,
            controls: r.controls.MINIMAL,
            highlight: r.highlight.ONE,
            dynamicBlockRows: !0,
            onlyCellsWithin: !0,
            originDate: i,
            onHighlight: t,
            renderControl: m,
            trackCurrentDay: !0
          }
        ) })
      ]
    }
  );
}
export {
  x as CalendarInputPopover
};
