// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinksOverview/components/constants.js
const n = "adyen-pe-payment-links-overview", t = "adyen-pe-payment-link-table", i = `${n}--xs`, c = `${n}__tabs-container`, S = `${t}__mobile-cell`, s = `${t}__mobile-amount-cell`, a = `${t}__mobile-expire-date-cell`, A = `${n}__filters-container`, L = `${n}__action-buttons-container`, T = `${n}__filters-alert-container`, E = `${n}__action-button`, N = 90, l = "active", o = {
  active: "payByLink.overview.list.statusGroups.active",
  inactive: "payByLink.overview.list.statusGroups.inactive"
}, p = Object.entries(o).map(([_, e]) => ({
  id: _,
  label: e,
  content: null
})), y = {
  active: "active",
  inactive: "inactive"
}, m = {
  active: "payByLink.common.status.active",
  expired: "payByLink.common.status.expired",
  completed: "payByLink.common.status.completed",
  paymentPending: "payByLink.common.status.paymentPending"
}, I = {
  open: "payByLink.common.linkType.open",
  singleUse: "payByLink.common.linkType.singleUse"
};
export {
  L as ACTION_BUTTONS_CONTAINER_CLASS,
  E as ACTION_BUTTON_CLASS,
  n as BASE_CLASS,
  t as BASE_TABLE_GRID_CLASS,
  i as BASE_XS_CLASS,
  l as DEFAULT_PAYMENT_LINK_STATUS_GROUP,
  N as EARLIEST_PAYMENT_LINK_DATE,
  T as FILTERS_ALERT_CONTAINER_CLASS,
  A as FILTERS_CONTAINER_CLASS,
  s as MOBILE_AMOUNT_CELL_CLASS,
  a as MOBILE_EXPIRE_DATE_CELL_CLASS,
  S as MOBILE_TABLE_CELL_CLASS,
  m as PAYMENT_LINK_STATUSES,
  o as PAYMENT_LINK_STATUS_GROUPS,
  y as PAYMENT_LINK_STATUS_GROUPS_FILTER_MAPPING,
  p as PAYMENT_LINK_STATUS_GROUPS_TABS,
  I as PAYMENT_LINK_TYPES,
  c as TABS_CONTAINER_CLASS
};
