// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinksOverview/components/PaymentLinksTable.js
import { jsx as r } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useCallback as G, useMemo as T } from "../../../../external/preact/hooks/dist/hooks.module.js";
import V from "../../../../core/Context/useCoreContext.js";
import W from "../../../../hooks/useTimezoneAwareDateFormatting.js";
import { useResponsiveContainer as g, containerQueries as k } from "../../../../hooks/useResponsiveContainer.js";
import { useTableColumns as q } from "../../../../hooks/useTableColumns.js";
import { BASE_TABLE_GRID_CLASS as X, MOBILE_TABLE_CELL_CLASS as N, MOBILE_EXPIRE_DATE_CELL_CLASS as $, PAYMENT_LINK_STATUSES as B, MOBILE_AMOUNT_CELL_CLASS as j } from "./constants.js";
import { Tag as v } from "../../../internal/Tag/Tag.js";
import { TagVariant as m } from "../../../internal/Tag/types.js";
import s from "../../../internal/Typography/Typography.js";
import { TypographyVariant as c, TypographyElement as l } from "../../../internal/Typography/types.js";
import { DAY_MS as z } from "../../../internal/Calendar/calendar/constants.js";
import { Tooltip as Q } from "../../../internal/Tooltip/Tooltip.js";
import { isActionNeededUrgently as H } from "../../../utils/payByLink/actionLevel.js";
import J from "classnames";
import { PaymentLinksErrors as Z } from "./PaymentLinksErrors.js";
import { FRONTEND_REDACTED_DATA_MARKER as ee, BACKEND_REDACTED_DATA_MARKER as re } from "../../../constants.js";
import { ACCOUNT_MISCONFIGURATION as ie, WRONG_STORE_IDS as ne } from "../utils/getPaymentLinksErrorMessage.js";
import C from "../../../internal/DataGrid/DataGrid.js";
import te from "../../../internal/Pagination/Pagination.js";
import { Fragment as I } from "../../../../external/preact/dist/preact.module.js";
import { DATE_FORMAT_RESPONSE_DEADLINE as ae, DATE_FORMAT_PAYMENT_LINKS_OVERVIEW_EXPIRATION_DATE as A, DATE_FORMAT_PAYMENT_LINKS_OVERVIEW as oe } from "../../../../constants/dateFormats.js";
const S = (d) => {
  switch (d) {
    case "completed":
      return m.SUCCESS;
    case "expired":
      return m.DEFAULT;
    case "paymentPending":
      return m.WARNING;
    case "active":
      return m.BLUE;
    default:
      return m.DEFAULT;
  }
}, se = [
  "paymentLinkId",
  "merchantReference",
  "storeCode",
  "currency",
  "amount",
  "status",
  "expirationDate",
  "creationDate",
  "linkType",
  "shopperEmail"
], R = {
  paymentLinkId: "payByLink.overview.list.fields.id",
  amount: "payByLink.overview.list.fields.amount",
  currency: "payByLink.overview.list.fields.currency",
  status: "payByLink.overview.list.fields.status",
  expirationDate: "payByLink.overview.list.fields.expirationDate",
  creationDate: "payByLink.overview.list.fields.createdAt",
  linkType: "payByLink.overview.list.fields.linkType",
  merchantReference: "payByLink.overview.list.fields.merchantReference",
  shopperEmail: "payByLink.overview.list.fields.shopperEmail",
  storeCode: "payByLink.overview.list.fields.store"
}, ce = "payByLink.overview.errors.couldNotLoadLinks", we = ({
  error: d,
  loading: w,
  onContactSupport: L,
  onRowClick: O,
  showDetails: le,
  showPagination: M,
  paymentLinks: b,
  stores: o,
  allStores: t,
  storeError: f,
  ...P
}) => {
  const { i18n: a, getImageAsset: D } = V(), { dateFormat: p } = W(), x = g(k.up.sm), F = G(
    (e) => {
      if (!e) return "";
      const y = new Date(e).getTime() - Date.now(), _ = Math.ceil(y / z), h = p(e, { ...ae, weekday: void 0 });
      return _ <= 1 ? a.get("payByLink.overview.common.actionNeeded.expiresToday", { values: { date: h } }) : a.get("payByLink.overview.common.actionNeeded.expiresDays", { values: { days: _, date: h } });
    },
    [p, a]
  ), i = g(k.down.xs), U = q({
    fields: se,
    fieldsKeys: R,
    columnConfig: {
      amount: {
        position: "right",
        flex: x ? 1.5 : void 0
      },
      linkType: {
        label: a.get(R.linkType),
        visible: !i
      },
      storeCode: {
        visible: o && (o == null ? void 0 : o.length) > 1 && !i
      },
      merchantReference: {
        visible: !i
      },
      currency: {
        visible: !i
      },
      status: {
        visible: !i
      },
      expirationDate: {
        visible: !i
      },
      creationDate: {
        visible: !i
      },
      shopperEmail: {
        visible: !i
      }
    }
  }), Y = {
    title: "payByLink.overview.errors.listEmpty",
    message: ["payByLink.overview.errors.listEmpty.message"]
  }, u = T(() => {
    if (!((t == null ? void 0 : t.length) !== 0 || f))
      return {
        message: "No stores configured",
        name: "Account misconfiguration",
        errorCode: ie,
        type: "error",
        requestId: ""
      };
  }, [t, f]), E = T(() => {
    if (!(t && (t == null ? void 0 : t.length) > 0 && (o == null ? void 0 : o.length) !== 0))
      return {
        errorCode: ne,
        type: "error",
        requestId: ""
      };
  }, [t, o]), K = T(
    () => () => /* @__PURE__ */ r(
      Z,
      {
        getImageAsset: D,
        error: u || d || E,
        onContactSupport: L,
        errorMessage: ce
      }
    ),
    [d, D, L, u, E]
  );
  return /* @__PURE__ */ r("div", { className: X, children: /* @__PURE__ */ r(
    C,
    {
      narrowColumns: i,
      errorDisplay: K,
      error: u || d || E,
      columns: U,
      data: b,
      loading: w,
      outline: !1,
      onRowClick: { callback: O },
      emptyTableMessage: Y,
      customCells: {
        currency: ({ item: e }) => {
          var n;
          if ((n = e == null ? void 0 : e.amount) != null && n.currency)
            return /* @__PURE__ */ r(v, { label: `${e.amount.currency}`, variant: m.DEFAULT });
        },
        amount: ({ value: e, item: n }) => {
          const y = a.amount(e.value, e.currency, { hideCurrency: !0 });
          return i ? /* @__PURE__ */ r("div", { className: J(N, j), children: [
            /* @__PURE__ */ r(s, { el: l.SPAN, variant: c.BODY, strongest: !0, children: y }),
            /* @__PURE__ */ r("span", { children: n.status && /* @__PURE__ */ r(v, { label: a.get(B[n.status]), variant: S(n.status) }) })
          ] }) : /* @__PURE__ */ r(s, { el: l.SPAN, variant: c.BODY, children: y });
        },
        status: ({ value: e }) => {
          if (e)
            return /* @__PURE__ */ r(v, { label: a.get(`${B[e]}`), variant: S(e) });
        },
        linkType: ({ item: e }) => {
          if (!(e != null && e.linkType)) return null;
          const n = (e == null ? void 0 : e.linkType) === "open" ? "payByLink.common.linkType.open" : "payByLink.common.linkType.singleUse";
          return /* @__PURE__ */ r(s, { el: l.SPAN, variant: c.BODY, children: a.get(n) });
        },
        creationDate: ({ value: e }) => /* @__PURE__ */ r("time", { dateTime: e, children: /* @__PURE__ */ r(s, { el: l.SPAN, variant: c.BODY, children: p(e, oe) }) }),
        expirationDate: ({ value: e }) => H(e) ? /* @__PURE__ */ r(Q, { content: F(e), children: /* @__PURE__ */ r("span", { children: /* @__PURE__ */ r("time", { dateTime: e, children: p(e, A) }) }) }) : /* @__PURE__ */ r("time", { dateTime: e, children: /* @__PURE__ */ r(s, { el: l.SPAN, variant: c.BODY, children: p(e, A) }) }),
        paymentLinkId: ({ item: e }) => i ? /* @__PURE__ */ r("div", { className: N, children: [
          /* @__PURE__ */ r(s, { strongest: !0, el: l.SPAN, variant: c.BODY, children: e.paymentLinkId }),
          /* @__PURE__ */ r("time", { dateTime: e.expirationDate, children: /* @__PURE__ */ r(
            s,
            {
              className: $,
              el: l.SPAN,
              variant: c.CAPTION,
              children: a.get("payByLink.overview.common.actionNeeded.expiresAt", {
                values: { date: p(e.expirationDate, A) }
              })
            }
          ) })
        ] }) : /* @__PURE__ */ r(I, { children: e.paymentLinkId }),
        shopperEmail: ({ item: e }) => /* @__PURE__ */ r(I, { children: e.shopperEmail === re ? ee : e.shopperEmail })
      },
      children: M && /* @__PURE__ */ r(C.Footer, { children: /* @__PURE__ */ r(
        te,
        {
          ...P,
          ariaLabelKey: "payByLink.overview.pagination.label",
          limitSelectAriaLabelKey: "payByLink.overview.pagination.controls.limitSelect.label"
        }
      ) })
    }
  ) });
};
export {
  se as PAYMENT_LINKS_TABLE_FIELDS,
  we as PaymentLinksTable
};
