// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinksOverview/components/PaymentLinksOverviewModal.js
import { jsx as l } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import h from "../../../../core/Context/useCoreContext.js";
import { useState as B, useCallback as m, useEffect as R } from "../../../../external/preact/hooks/dist/hooks.module.js";
import { popoverUtil as S } from "../../../internal/Popover/utils/popoverUtil.js";
import a from "../../PaymentLinkCreation/components/PaymentLinkCreationContainer/PaymentLinkCreationContainer.js";
import i from "../../PaymentLinkSettings/components/PaymentLinkSettingsContainer/PaymentLinkSettingsContainer.js";
import j from "../../../internal/Modal/Modal.js";
const U = ({
  isModalVisible: r,
  onCloseModal: u,
  modalType: s,
  paymentLinkCreation: e,
  paymentLinkSettings: g,
  storeIds: f,
  onContactSupport: t,
  refreshPaymentLinkList: o
}) => {
  const { i18n: w } = h(), [c, v] = B(!1), O = m(() => {
    u(), c && (o == null || o(), v(!1));
  }, [u, c, o]);
  R(() => {
    r && S.closeAll();
  }, [r]);
  const b = m(
    (x) => {
      var d;
      (d = e == null ? void 0 : e.onPaymentLinkCreated) == null || d.call(e, x), v(!0);
    },
    [e]
  );
  return !r || !s ? null : /* @__PURE__ */ l("div", { children: /* @__PURE__ */ l(
    j,
    {
      isOpen: r,
      "aria-label": w.get("payByLink.overview.title"),
      onClose: O,
      isDismissible: !0,
      headerWithBorder: !1,
      size: "large",
      children: [
        s === "Creation" ? /* @__PURE__ */ l(
          a,
          {
            ...e,
            onPaymentLinkCreated: b,
            storeIds: f,
            onContactSupport: t,
            embeddedInOverview: !0
          }
        ) : null,
        s === "Settings" ? /* @__PURE__ */ l(
          i,
          {
            ...g,
            storeIds: f,
            onContactSupport: t,
            embeddedInOverview: !0
          }
        ) : null
      ]
    }
  ) });
};
export {
  U as PaymentLinksOverviewModal
};
