// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/hooks/useTransaction/transactionNavigator/createDuplexTransactionNavigator.js
import { EMPTY_ARRAY as T } from "../../../../../../utils/value/constants.js";
import { struct as l } from "../../../../../../utils/struct/main.js";
import { enumerable as d, getter as e } from "../../../../../../utils/struct/property.js";
import { isNullish as k, isFunction as b } from "../../../../../../utils/value/is.js";
import { isEmptyString as g } from "../../../../../../utils/value/string.js";
const A = () => {
  let [t, i, a, o] = T;
  const m = () => !!(t && t === o && a), u = () => !!(t && t === a && o);
  let n = null;
  const v = () => {
    m() && (i = t, t = a, c());
  }, p = () => {
    u() && (i = t, t = o, c());
  }, w = (r, s) => {
    const f = t;
    t = i = a = o = void 0, !g(r) && !g(s) && (a = r, o = s, t = f === a || f === o ? f : a, f !== t && c());
  }, N = (r) => {
    k(r) ? n = null : b(r) && n !== (n = r) && c();
  }, c = () => {
    const r = i, s = t;
    i = void 0, n == null || n(
      l({
        from: e(() => r),
        to: e(() => s)
      })
    );
  };
  return l({
    backward: d(v),
    forward: d(p),
    canNavigateBackward: e(m),
    canNavigateForward: e(u),
    currentTransaction: e(() => t),
    onNavigation: { set: N },
    reset: d(w)
  });
};
export {
  A as createDuplexTransactionNavigator,
  A as default
};
