// @adyen/adyen-platform-experience-web/dist/es/components/internal/CopyText/CopyText.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import B from "classnames";
import { useState as C, useCallback as f, useMemo as b } from "../../../external/preact/hooks/dist/hooks.module.js";
import I from "../../../core/Context/useCoreContext.js";
import { Tooltip as _ } from "../Tooltip/Tooltip.js";
import { ButtonVariant as S } from "../Button/types.js";
/* empty css              */
import T from "../Button/Button.js";
import { Icon as A } from "../Icon/Icon.js";
const o = "adyen-pe-copy-text", a = {
  base: o,
  container: o + "__container",
  icon: o + "__icon",
  information: o + "__information",
  label: o + "__label",
  text: o + "__text",
  stronger: o + "--stronger"
}, V = ({
  copyButtonAriaLabelKey: c,
  isUnderlineVisible: h,
  textToCopy: t,
  visibleText: r,
  onCopyText: s,
  showCopyTextTooltip: v = !0,
  type: n = "Trimmed",
  stronger: l,
  ...g
}) => {
  const { i18n: i } = I(), [m, d] = C(!1), p = f(() => d(!1), []), N = b(() => i.get(c ?? "common.actions.copy.labels.default"), [i, c]), x = f(async () => {
    if (t)
      try {
        await navigator.clipboard.writeText(t), d(!0), s && s();
      } catch (y) {
        console.error(y);
      }
  }, [t, s]), u = b(
    () => /* @__PURE__ */ e(
      "span",
      {
        className: B({
          [a.label]: n !== "Default",
          [a.information]: n === "Trimmed",
          [a.text]: n === "Text",
          [a.stronger]: l
        }),
        children: r || t
      }
    ),
    [n, l, r, t]
  );
  return /* @__PURE__ */ e("span", { className: a.container, ...g, children: [
    v ? /* @__PURE__ */ e(_, { content: t, isUnderlineVisible: h, children: u }) : u,
    /* @__PURE__ */ e(_, { content: i.get(m ? "common.actions.copy.labels.done" : "common.actions.copy.labels.default"), children: /* @__PURE__ */ e(
      T,
      {
        variant: S.TERTIARY,
        className: a.base,
        onClick: x,
        onBlur: p,
        onMouseLeaveCapture: p,
        "aria-label": N,
        "data-testid": "copyText",
        children: /* @__PURE__ */ e("div", { className: a.icon, children: /* @__PURE__ */ e(A, { name: "copy", "data-testid": "copy-icon" }) })
      }
    ) }),
    /* @__PURE__ */ e("div", { className: "adyen-pe-visually-hidden", "aria-atomic": "true", "aria-live": "polite", children: m && i.get("common.actions.copy.labels.done") })
  ] });
};
export {
  V as default
};
