// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/components/GrantRepaymentDetails/GrantRepaymentDetails.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as h } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { useUniqueId as m } from "../../../../../hooks/useUniqueId.js";
import f from "../../../../../core/Context/useCoreContext.js";
import t from "../../../../internal/Typography/Typography.js";
import { TypographyVariant as r, TypographyElement as n } from "../../../../internal/Typography/types.js";
import { AccountDetails as N } from "../AccountDetails/AccountDetails.js";
import { GrantAdjustmentDetails as I } from "../GrantAdjustmentDetails/GrantAdjustmentDetails.js";
/* empty css                           */
import { Divider as P } from "../../../../internal/Divider/Divider.js";
import { Fragment as g } from "../../../../../external/preact/dist/preact.module.js";
import { Translation as S } from "../../../../internal/Translation/Translation.js";
const i = "adyen-pe-grant-repayment-details", s = {
  instructionList: `${i}__instruction-list`,
  notice: `${i}__notice`,
  repaymentAccount: `${i}__repayment-account`,
  transferInstrumentItem: `${i}__transfer-instrument-item`,
  transferInstrumentList: `${i}__transfer-instrument-list`,
  verifiedBankAccountDetails: `${i}__verified-bank-account-details`
}, x = ({ grant: c, onDetailsClose: A }) => {
  var v, y;
  const { i18n: a } = f(), d = `list-${m()}`, u = `list-${m()}`, p = `list-${m()}`, o = h(() => {
    var l;
    return (l = c.unscheduledRepaymentAccounts) == null ? void 0 : l[0];
  }, [c.unscheduledRepaymentAccounts]);
  return o ? /* @__PURE__ */ e(
    I,
    {
      className: i,
      onDetailsClose: A,
      headerTitleKey: "capital.overview.repayment.title",
      headerSubtitleKey: "capital.overview.repayment.subtitle",
      children: [
        /* @__PURE__ */ e("div", { className: s.repaymentAccount, children: [
          /* @__PURE__ */ e(t, { id: d, el: n.SPAN, variant: r.BODY, stronger: !0, children: a.get("capital.overview.repayment.accountDetails.title") }),
          /* @__PURE__ */ e(N, { bankAccount: o, "aria-labelledby": d })
        ] }),
        /* @__PURE__ */ e("div", { className: s.notice, children: [
          !!((v = c.transferInstruments) != null && v.length) && /* @__PURE__ */ e(g, { children: [
            /* @__PURE__ */ e("div", { children: [
              /* @__PURE__ */ e(t, { id: p, el: n.SPAN, variant: r.CAPTION, stronger: !0, children: a.get("capital.overview.repayment.transferInstruments") }),
              /* @__PURE__ */ e("ul", { className: s.transferInstrumentList, "aria-labelledby": p, children: (y = c.transferInstruments) == null ? void 0 : y.map(({ accountIdentifier: l }) => /* @__PURE__ */ e("li", { className: s.transferInstrumentItem, children: /* @__PURE__ */ e(t, { el: n.SPAN, variant: r.CAPTION, children: l }) }, l)) })
            ] }),
            /* @__PURE__ */ e(P, {})
          ] }),
          /* @__PURE__ */ e("div", { children: [
            /* @__PURE__ */ e(t, { id: u, el: n.SPAN, variant: r.CAPTION, stronger: !0, children: a.get("capital.overview.repayment.instructions.title") }),
            /* @__PURE__ */ e("ol", { className: s.instructionList, "aria-labelledby": u, children: [
              /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(t, { el: n.SPAN, variant: r.CAPTION, children: /* @__PURE__ */ e(
                S,
                {
                  translationKey: "capital.overview.repayment.instructions.addingBeneficiary",
                  fills: {
                    beneficiaryName: /* @__PURE__ */ e(t, { el: n.SPAN, variant: r.CAPTION, stronger: !0, children: o.beneficiaryName })
                  }
                }
              ) }) }),
              /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(t, { el: n.SPAN, variant: r.CAPTION, children: a.get("capital.overview.repayment.instructions.sendingPayment") }) }),
              /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(t, { el: n.SPAN, variant: r.CAPTION, children: a.get("capital.overview.repayment.instructions.waiting") }) })
            ] })
          ] }),
          /* @__PURE__ */ e(t, { className: s.verifiedBankAccountDetails, el: n.SPAN, variant: r.CAPTION, children: a.get("capital.overview.repayment.instructions.verifiedAccount") })
        ] })
      ]
    }
  ) : null;
};
export {
  x as GrantRepaymentDetails
};
