// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/TransactionsOverview/TransactionsOverview.js
import { jsx as r } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import I from "classnames";
import H from "../../hooks/useTransactionsList.js";
import W from "../../hooks/useTransactionsTotals.js";
import j from "../../../../../hooks/useAccountBalances.js";
import k from "../../../../../core/Context/useCoreContext.js";
import G from "./TransactionsOverviewList.js";
import K from "./TransactionsOverviewInsights.js";
import q from "../TransactionFilters/TransactionFilters.js";
import D from "../TransactionsExport/TransactionsExport.js";
import J from "../../../../internal/SegmentedControl/SegmentedControl.js";
import { INITIAL_FILTERS as P, TRANSACTIONS_VIEW_TABS as C, classes as a } from "../../constants.js";
import { TransactionsView as B } from "../../types.js";
import { useRef as Q, useState as u, useMemo as p, useEffect as U } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { compareTransactionsFilters as X } from "../utils.js";
/* empty css                          */
import { Fragment as d } from "../../../../../external/preact/dist/preact.module.js";
import { useFilterBarState as Y, FilterBarMobileSwitch as Z } from "../../../../internal/FilterBar/FilterBar.js";
import { Header as $ } from "../../../../internal/Header/Header.js";
const Sr = ({
  onFiltersChanged: F,
  balanceAccounts: h,
  allowLimitSelection: N,
  preferredLimit: R,
  onRecordSelection: b,
  showDetails: x,
  isLoadingBalanceAccount: y,
  onContactSupport: E,
  hideTitle: A,
  dataCustomization: T
}) => {
  const c = Q(P), l = Y(), [t, L] = u(c.current), [m, O] = u(B.TRANSACTIONS), [V, M] = u(), { balanceAccount: e } = t, { isMobileContainer: s } = l, { i18n: v } = k(), i = m !== B.INSIGHTS, f = p(
    () => i && !!(e != null && e.id) && X(t, c.current),
    [i, e, t]
  ), n = j({ balanceAccount: e }), o = H({
    fetchEnabled: f,
    allowLimitSelection: N,
    dataCustomization: T,
    filters: t,
    onFiltersChanged: F,
    preferredLimit: R
  }), S = W({
    currencies: n.currencies,
    fetchEnabled: !!(e != null && e.id),
    loadingBalances: n.isWaiting,
    filters: t
  }), g = p(
    () => i ? /* @__PURE__ */ r(D, { disabled: o.records.length < 1 || o.fetching, filters: t }) : null,
    [t, i, o.fetching, o.records]
  ), w = p(
    () => C.length > 1 ? /* @__PURE__ */ r(
      J,
      {
        "aria-label": v.get("transactions.overview.viewSelect.a11y.label"),
        activeItem: m,
        items: C,
        onChange: ({ id: _ }) => O(_)
      }
    ) : null,
    [m, v]
  );
  return U(() => {
    f && (c.current = t);
  }, [f, t]), /* @__PURE__ */ r("div", { className: I(a.root, { [a.rootSmall]: s }), children: [
    /* @__PURE__ */ r($, { hideTitle: A, titleKey: "transactions.overview.title", children: /* @__PURE__ */ r("div", { className: I({ [a.filterBarSmall]: s }), children: [
      s && g,
      /* @__PURE__ */ r(Z, { ...l }),
      !s && /* @__PURE__ */ r(d, { children: w })
    ] }) }),
    s && /* @__PURE__ */ r(d, { children: w }),
    /* @__PURE__ */ r("div", { role: "toolbar", className: a.toolbar, children: [
      /* @__PURE__ */ r(
        q,
        {
          ...l,
          availableCurrencies: n.currencies,
          balanceAccounts: h,
          isTransactionsView: i,
          setInsightsCurrency: M,
          onChange: L
        }
      ),
      !s && /* @__PURE__ */ r(d, { children: g })
    ] }),
    i ? /* @__PURE__ */ r(
      G,
      {
        accountBalancesResult: n,
        balanceAccount: e,
        balanceAccounts: h,
        dataCustomization: T,
        isLoadingBalanceAccount: y,
        onContactSupport: E,
        onRecordSelection: b,
        showDetails: x,
        transactionsListResult: o,
        transactionsTotalsResult: S
      }
    ) : /* @__PURE__ */ r(K, { currency: V, transactionsTotalsResult: S })
  ] });
};
export {
  Sr as TransactionsOverview
};
