// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/TransactionsOverview/TransactionsOverviewList.js
import { jsx as r } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useCallback as E, useMemo as d } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { TRANSACTION_ANALYTICS_SUBCATEGORY_LIST as S, TRANSACTION_ANALYTICS_CATEGORY as B, classes as s } from "../../constants.js";
import { useLandedPageEvent as L } from "../../../../../hooks/useAnalytics/useLandedPageEvent.js";
import { useDurationEvent as O } from "../../../../../hooks/useAnalytics/useDurationEvent.js";
import R from "../TransactionsList/TransactionsList.js";
import Y from "../TransactionTotals/TransactionTotals.js";
import _ from "../Balances/Balances.js";
import { Alert as k } from "../../../../internal/Alert/Alert.js";
import x from "../../../../../core/Context/useCoreContext.js";
import { AlertTypeOption as F } from "../../../../internal/Alert/types.js";
import { ButtonVariant as G } from "../../../../internal/Button/types.js";
import W from "../../../../internal/Button/Button.js";
import { Fragment as w } from "../../../../../external/preact/dist/preact.module.js";
const f = {
  category: B,
  subCategory: S
}, Z = ({
  accountBalancesResult: a,
  balanceAccount: t,
  balanceAccounts: p,
  isLoadingBalanceAccount: u,
  transactionsListResult: T,
  transactionsTotalsResult: C,
  ...v
}) => {
  const { i18n: i, updateCore: N } = x(), { isWaiting: g, balances: n } = a, { isWaiting: y, totals: m, error: A } = C, o = t == null ? void 0 : t.defaultCurrencyCode, e = E(
    ({ currency: l }, { currency: c }) => {
      if (o) {
        if (l === o) return -1;
        if (c === o) return 1;
      }
      return l.localeCompare(c);
    },
    [o]
  ), I = d(() => [...n].sort(e), [n, e]), h = d(() => [...m].sort(e), [m, e]);
  return L(f), O(f), /* @__PURE__ */ r(w, { children: [
    /* @__PURE__ */ r("div", { className: s.summary, children: [
      A ? /* @__PURE__ */ r(
        k,
        {
          className: s.totalsError,
          type: F.WARNING,
          title: i.get("transactions.overview.totals.error"),
          description: /* @__PURE__ */ r("div", { children: /* @__PURE__ */ r(W, { variant: G.TERTIARY, onClick: N, children: i.get("common.actions.refresh.labels.default") }) })
        }
      ) : /* @__PURE__ */ r("div", { className: s.summaryItem, children: /* @__PURE__ */ r(Y, { totals: h, loadingTotals: y }) }),
      /* @__PURE__ */ r("div", { className: s.summaryItem, children: /* @__PURE__ */ r(_, { balances: I, loadingBalances: g }) })
    ] }),
    /* @__PURE__ */ r(
      R,
      {
        accountBalancesResult: a,
        balanceAccount: t,
        loadingBalanceAccounts: u || !p,
        transactionsListResult: T,
        ...v
      }
    )
  ] });
};
export {
  Z as default
};
