// @adyen/adyen-platform-experience-web/dist/es/external/preact/dist/preact.module.js
var D, d, X, se, x, K, Y, Z, ee, O, $, j, _e, M = {}, ne = [], ce = /acit|ex(?:s|g|n|p|$)|rph|grid|ows|mnc|ntw|ine[ch]|zoo|^ord|itera/i, L = Array.isArray;
function k(_, e) {
  for (var n in e) _[n] = e[n];
  return _;
}
function V(_) {
  _ && _.parentNode && _.parentNode.removeChild(_);
}
function pe(_, e, n) {
  var o, l, t, i = {};
  for (t in e) t == "key" ? o = e[t] : t == "ref" ? l = e[t] : i[t] = e[t];
  if (arguments.length > 2 && (i.children = arguments.length > 3 ? D.call(arguments, 2) : n), typeof _ == "function" && _.defaultProps != null) for (t in _.defaultProps) i[t] === void 0 && (i[t] = _.defaultProps[t]);
  return E(_, i, o, l, null);
}
function E(_, e, n, o, l) {
  var t = { type: _, props: e, key: n, ref: o, __k: null, __: null, __b: 0, __e: null, __c: null, constructor: void 0, __v: l ?? ++X, __i: -1, __u: 0 };
  return l == null && d.vnode != null && d.vnode(t), t;
}
function me() {
  return { current: null };
}
function I(_) {
  return _.children;
}
function N(_, e) {
  this.props = _, this.context = e;
}
function C(_, e) {
  if (e == null) return _.__ ? C(_.__, _.__i + 1) : null;
  for (var n; e < _.__k.length; e++) if ((n = _.__k[e]) != null && n.__e != null) return n.__e;
  return typeof _.type == "function" ? C(_) : null;
}
function te(_) {
  var e, n;
  if ((_ = _.__) != null && _.__c != null) {
    for (_.__e = _.__c.base = null, e = 0; e < _.__k.length; e++) if ((n = _.__k[e]) != null && n.__e != null) {
      _.__e = _.__c.base = n.__e;
      break;
    }
    return te(_);
  }
}
function B(_) {
  (!_.__d && (_.__d = !0) && x.push(_) && !H.__r++ || K !== d.debounceRendering) && ((K = d.debounceRendering) || Y)(H);
}
function H() {
  for (var _, e, n, o, l, t, i, s = 1; x.length; ) x.length > s && x.sort(Z), _ = x.shift(), s = x.length, _.__d && (n = void 0, l = (o = (e = _).__v).__e, t = [], i = [], e.__P && ((n = k({}, o)).__v = o.__v + 1, d.vnode && d.vnode(n), z(e.__P, n, o, e.__n, e.__P.namespaceURI, 32 & o.__u ? [l] : null, t, l ?? C(o), !!(32 & o.__u), i), n.__v = o.__v, n.__.__k[n.__i] = n, le(t, n, i), n.__e != l && te(n)));
  H.__r = 0;
}
function re(_, e, n, o, l, t, i, s, p, u, f) {
  var r, v, c, m, g, y, a = o && o.__k || ne, h = e.length;
  for (p = fe(n, e, a, p, h), r = 0; r < h; r++) (c = n.__k[r]) != null && (v = c.__i === -1 ? M : a[c.__i] || M, c.__i = r, y = z(_, c, v, l, t, i, s, p, u, f), m = c.__e, c.ref && v.ref != c.ref && (v.ref && q(v.ref, null, c), f.push(c.ref, c.__c || m, c)), g == null && m != null && (g = m), 4 & c.__u || v.__k === c.__k ? p = oe(c, p, _) : typeof c.type == "function" && y !== void 0 ? p = y : m && (p = m.nextSibling), c.__u &= -7);
  return n.__e = g, p;
}
function fe(_, e, n, o, l) {
  var t, i, s, p, u, f = n.length, r = f, v = 0;
  for (_.__k = new Array(l), t = 0; t < l; t++) (i = e[t]) != null && typeof i != "boolean" && typeof i != "function" ? (p = t + v, (i = _.__k[t] = typeof i == "string" || typeof i == "number" || typeof i == "bigint" || i.constructor == String ? E(null, i, null, null, null) : L(i) ? E(I, { children: i }, null, null, null) : i.constructor === void 0 && i.__b > 0 ? E(i.type, i.props, i.key, i.ref ? i.ref : null, i.__v) : i).__ = _, i.__b = _.__b + 1, s = null, (u = i.__i = he(i, n, p, r)) !== -1 && (r--, (s = n[u]) && (s.__u |= 2)), s == null || s.__v === null ? (u == -1 && (l > f ? v-- : l < f && v++), typeof i.type != "function" && (i.__u |= 4)) : u != p && (u == p - 1 ? v-- : u == p + 1 ? v++ : (u > p ? v-- : v++, i.__u |= 4))) : _.__k[t] = null;
  if (r) for (t = 0; t < f; t++) (s = n[t]) != null && (2 & s.__u) == 0 && (s.__e == o && (o = C(s)), ue(s, s));
  return o;
}
function oe(_, e, n) {
  var o, l;
  if (typeof _.type == "function") {
    for (o = _.__k, l = 0; o && l < o.length; l++) o[l] && (o[l].__ = _, e = oe(o[l], e, n));
    return e;
  }
  _.__e != e && (e && _.type && !n.contains(e) && (e = C(_)), n.insertBefore(_.__e, e || null), e = _.__e);
  do
    e = e && e.nextSibling;
  while (e != null && e.nodeType == 8);
  return e;
}
function ae(_, e) {
  return e = e || [], _ == null || typeof _ == "boolean" || (L(_) ? _.some(function(n) {
    ae(n, e);
  }) : e.push(_)), e;
}
function he(_, e, n, o) {
  var l, t, i = _.key, s = _.type, p = e[n];
  if (p === null && _.key == null || p && i == p.key && s === p.type && (2 & p.__u) == 0) return n;
  if (o > (p != null && (2 & p.__u) == 0 ? 1 : 0)) for (l = n - 1, t = n + 1; l >= 0 || t < e.length; ) {
    if (l >= 0) {
      if ((p = e[l]) && (2 & p.__u) == 0 && i == p.key && s === p.type) return l;
      l--;
    }
    if (t < e.length) {
      if ((p = e[t]) && (2 & p.__u) == 0 && i == p.key && s === p.type) return t;
      t++;
    }
  }
  return -1;
}
function G(_, e, n) {
  e[0] == "-" ? _.setProperty(e, n ?? "") : _[e] = n == null ? "" : typeof n != "number" || ce.test(e) ? n : n + "px";
}
function F(_, e, n, o, l) {
  var t;
  e: if (e == "style") if (typeof n == "string") _.style.cssText = n;
  else {
    if (typeof o == "string" && (_.style.cssText = o = ""), o) for (e in o) n && e in n || G(_.style, e, "");
    if (n) for (e in n) o && n[e] === o[e] || G(_.style, e, n[e]);
  }
  else if (e[0] == "o" && e[1] == "n") t = e != (e = e.replace(ee, "$1")), e = e.toLowerCase() in _ || e == "onFocusOut" || e == "onFocusIn" ? e.toLowerCase().slice(2) : e.slice(2), _.l || (_.l = {}), _.l[e + t] = n, n ? o ? n.t = o.t : (n.t = O, _.addEventListener(e, t ? j : $, t)) : _.removeEventListener(e, t ? j : $, t);
  else {
    if (l == "http://www.w3.org/2000/svg") e = e.replace(/xlink(H|:h)/, "h").replace(/sName$/, "s");
    else if (e != "width" && e != "height" && e != "href" && e != "list" && e != "form" && e != "tabIndex" && e != "download" && e != "rowSpan" && e != "colSpan" && e != "role" && e != "popover" && e in _) try {
      _[e] = n ?? "";
      break e;
    } catch {
    }
    typeof n == "function" || (n == null || n === !1 && e[4] != "-" ? _.removeAttribute(e) : _.setAttribute(e, e == "popover" && n == 1 ? "" : n));
  }
}
function Q(_) {
  return function(e) {
    if (this.l) {
      var n = this.l[e.type + _];
      if (e.u == null) e.u = O++;
      else if (e.u < n.t) return;
      return n(d.event ? d.event(e) : e);
    }
  };
}
function z(_, e, n, o, l, t, i, s, p, u) {
  var f, r, v, c, m, g, y, a, h, P, w, W, S, J, A, U, R, b = e.type;
  if (e.constructor !== void 0) return null;
  128 & n.__u && (p = !!(32 & n.__u), t = [s = e.__e = n.__e]), (f = d.__b) && f(e);
  e: if (typeof b == "function") try {
    if (a = e.props, h = "prototype" in b && b.prototype.render, P = (f = b.contextType) && o[f.__c], w = f ? P ? P.props.value : f.__ : o, n.__c ? y = (r = e.__c = n.__c).__ = r.__E : (h ? e.__c = r = new b(a, w) : (e.__c = r = new N(a, w), r.constructor = b, r.render = ve), P && P.sub(r), r.props = a, r.state || (r.state = {}), r.context = w, r.__n = o, v = r.__d = !0, r.__h = [], r._sb = []), h && r.__s == null && (r.__s = r.state), h && b.getDerivedStateFromProps != null && (r.__s == r.state && (r.__s = k({}, r.__s)), k(r.__s, b.getDerivedStateFromProps(a, r.__s))), c = r.props, m = r.state, r.__v = e, v) h && b.getDerivedStateFromProps == null && r.componentWillMount != null && r.componentWillMount(), h && r.componentDidMount != null && r.__h.push(r.componentDidMount);
    else {
      if (h && b.getDerivedStateFromProps == null && a !== c && r.componentWillReceiveProps != null && r.componentWillReceiveProps(a, w), !r.__e && (r.shouldComponentUpdate != null && r.shouldComponentUpdate(a, r.__s, w) === !1 || e.__v == n.__v)) {
        for (e.__v != n.__v && (r.props = a, r.state = r.__s, r.__d = !1), e.__e = n.__e, e.__k = n.__k, e.__k.some(function(T) {
          T && (T.__ = e);
        }), W = 0; W < r._sb.length; W++) r.__h.push(r._sb[W]);
        r._sb = [], r.__h.length && i.push(r);
        break e;
      }
      r.componentWillUpdate != null && r.componentWillUpdate(a, r.__s, w), h && r.componentDidUpdate != null && r.__h.push(function() {
        r.componentDidUpdate(c, m, g);
      });
    }
    if (r.context = w, r.props = a, r.__P = _, r.__e = !1, S = d.__r, J = 0, h) {
      for (r.state = r.__s, r.__d = !1, S && S(e), f = r.render(r.props, r.state, r.context), A = 0; A < r._sb.length; A++) r.__h.push(r._sb[A]);
      r._sb = [];
    } else do
      r.__d = !1, S && S(e), f = r.render(r.props, r.state, r.context), r.state = r.__s;
    while (r.__d && ++J < 25);
    r.state = r.__s, r.getChildContext != null && (o = k(k({}, o), r.getChildContext())), h && !v && r.getSnapshotBeforeUpdate != null && (g = r.getSnapshotBeforeUpdate(c, m)), U = f, f != null && f.type === I && f.key == null && (U = ie(f.props.children)), s = re(_, L(U) ? U : [U], e, n, o, l, t, i, s, p, u), r.base = e.__e, e.__u &= -161, r.__h.length && i.push(r), y && (r.__E = r.__ = null);
  } catch (T) {
    if (e.__v = null, p || t != null) if (T.then) {
      for (e.__u |= p ? 160 : 128; s && s.nodeType == 8 && s.nextSibling; ) s = s.nextSibling;
      t[t.indexOf(s)] = null, e.__e = s;
    } else for (R = t.length; R--; ) V(t[R]);
    else e.__e = n.__e, e.__k = n.__k;
    d.__e(T, e, n);
  }
  else t == null && e.__v == n.__v ? (e.__k = n.__k, e.__e = n.__e) : s = e.__e = de(n.__e, e, n, o, l, t, i, p, u);
  return (f = d.diffed) && f(e), 128 & e.__u ? void 0 : s;
}
function le(_, e, n) {
  for (var o = 0; o < n.length; o++) q(n[o], n[++o], n[++o]);
  d.__c && d.__c(e, _), _.some(function(l) {
    try {
      _ = l.__h, l.__h = [], _.some(function(t) {
        t.call(l);
      });
    } catch (t) {
      d.__e(t, l.__v);
    }
  });
}
function ie(_) {
  return typeof _ != "object" || _ == null ? _ : L(_) ? _.map(ie) : k({}, _);
}
function de(_, e, n, o, l, t, i, s, p) {
  var u, f, r, v, c, m, g, y = n.props, a = e.props, h = e.type;
  if (h == "svg" ? l = "http://www.w3.org/2000/svg" : h == "math" ? l = "http://www.w3.org/1998/Math/MathML" : l || (l = "http://www.w3.org/1999/xhtml"), t != null) {
    for (u = 0; u < t.length; u++) if ((c = t[u]) && "setAttribute" in c == !!h && (h ? c.localName == h : c.nodeType == 3)) {
      _ = c, t[u] = null;
      break;
    }
  }
  if (_ == null) {
    if (h == null) return document.createTextNode(a);
    _ = document.createElementNS(l, h, a.is && a), s && (d.__m && d.__m(e, t), s = !1), t = null;
  }
  if (h === null) y === a || s && _.data === a || (_.data = a);
  else {
    if (t = t && D.call(_.childNodes), y = n.props || M, !s && t != null) for (y = {}, u = 0; u < _.attributes.length; u++) y[(c = _.attributes[u]).name] = c.value;
    for (u in y) if (c = y[u], u != "children") {
      if (u == "dangerouslySetInnerHTML") r = c;
      else if (!(u in a)) {
        if (u == "value" && "defaultValue" in a || u == "checked" && "defaultChecked" in a) continue;
        F(_, u, null, c, l);
      }
    }
    for (u in a) c = a[u], u == "children" ? v = c : u == "dangerouslySetInnerHTML" ? f = c : u == "value" ? m = c : u == "checked" ? g = c : s && typeof c != "function" || y[u] === c || F(_, u, c, y[u], l);
    if (f) s || r && (f.__html === r.__html || f.__html === _.innerHTML) || (_.innerHTML = f.__html), e.__k = [];
    else if (r && (_.innerHTML = ""), re(e.type === "template" ? _.content : _, L(v) ? v : [v], e, n, o, h == "foreignObject" ? "http://www.w3.org/1999/xhtml" : l, t, i, t ? t[0] : n.__k && C(n, 0), s, p), t != null) for (u = t.length; u--; ) V(t[u]);
    s || (u = "value", h == "progress" && m == null ? _.removeAttribute("value") : m !== void 0 && (m !== _[u] || h == "progress" && !m || h == "option" && m !== y[u]) && F(_, u, m, y[u], l), u = "checked", g !== void 0 && g !== _[u] && F(_, u, g, y[u], l));
  }
  return _;
}
function q(_, e, n) {
  try {
    if (typeof _ == "function") {
      var o = typeof _.__u == "function";
      o && _.__u(), o && e == null || (_.__u = _(e));
    } else _.current = e;
  } catch (l) {
    d.__e(l, n);
  }
}
function ue(_, e, n) {
  var o, l;
  if (d.unmount && d.unmount(_), (o = _.ref) && (o.current && o.current !== _.__e || q(o, null, e)), (o = _.__c) != null) {
    if (o.componentWillUnmount) try {
      o.componentWillUnmount();
    } catch (t) {
      d.__e(t, e);
    }
    o.base = o.__P = null;
  }
  if (o = _.__k) for (l = 0; l < o.length; l++) o[l] && ue(o[l], e, n || typeof _.type != "function");
  n || V(_.__e), _.__c = _.__ = _.__e = void 0;
}
function ve(_, e, n) {
  return this.constructor(_, n);
}
function ye(_, e, n) {
  var o, l, t, i;
  e == document && (e = document.documentElement), d.__ && d.__(_, e), l = (o = !1) ? null : e.__k, t = [], i = [], z(e, _ = e.__k = pe(I, null, [_]), l || M, M, e.namespaceURI, l ? null : e.firstChild ? D.call(e.childNodes) : null, t, l ? l.__e : e.firstChild, o, i), le(t, _, i);
}
function ge(_, e, n) {
  var o, l, t, i, s = k({}, _.props);
  for (t in _.type && _.type.defaultProps && (i = _.type.defaultProps), e) t == "key" ? o = e[t] : t == "ref" ? l = e[t] : s[t] = e[t] === void 0 && i !== void 0 ? i[t] : e[t];
  return arguments.length > 2 && (s.children = arguments.length > 3 ? D.call(arguments, 2) : n), E(_.type, s, o || _.key, l || _.ref, null);
}
function be(_) {
  function e(n) {
    var o, l;
    return this.getChildContext || (o = /* @__PURE__ */ new Set(), (l = {})[e.__c] = this, this.getChildContext = function() {
      return l;
    }, this.componentWillUnmount = function() {
      o = null;
    }, this.shouldComponentUpdate = function(t) {
      this.props.value !== t.value && o.forEach(function(i) {
        i.__e = !0, B(i);
      });
    }, this.sub = function(t) {
      o.add(t);
      var i = t.componentWillUnmount;
      t.componentWillUnmount = function() {
        o && o.delete(t), i && i.call(t);
      };
    }), n.children;
  }
  return e.__c = "__cC" + _e++, e.__ = _, e.Provider = e.__l = (e.Consumer = function(n, o) {
    return n.children(o);
  }).contextType = e, e;
}
D = ne.slice, d = { __e: function(_, e, n, o) {
  for (var l, t, i; e = e.__; ) if ((l = e.__c) && !l.__) try {
    if ((t = l.constructor) && t.getDerivedStateFromError != null && (l.setState(t.getDerivedStateFromError(_)), i = l.__d), l.componentDidCatch != null && (l.componentDidCatch(_, o || {}), i = l.__d), i) return l.__E = l;
  } catch (s) {
    _ = s;
  }
  throw _;
} }, X = 0, se = function(_) {
  return _ != null && _.constructor == null;
}, N.prototype.setState = function(_, e) {
  var n;
  n = this.__s != null && this.__s !== this.state ? this.__s : this.__s = k({}, this.state), typeof _ == "function" && (_ = _(k({}, n), this.props)), _ && k(n, _), _ != null && this.__v && (e && this._sb.push(e), B(this));
}, N.prototype.forceUpdate = function(_) {
  this.__v && (this.__e = !0, _ && this.__h.push(_), B(this));
}, N.prototype.render = I, x = [], Y = typeof Promise == "function" ? Promise.prototype.then.bind(Promise.resolve()) : setTimeout, Z = function(_, e) {
  return _.__v.__b - e.__v.__b;
}, H.__r = 0, ee = /(PointerCapture)$|Capture$/i, O = 0, $ = Q(!1), j = Q(!0), _e = 0;
export {
  N as Component,
  I as Fragment,
  ge as cloneElement,
  be as createContext,
  pe as createElement,
  me as createRef,
  pe as h,
  se as isValidElement,
  d as options,
  ye as render,
  ae as toChildArray
};
