// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/components/PreQualifiedIntro.js
import { jsx as e } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useCallback as s } from "../../../../external/preact/hooks/dist/hooks.module.js";
import { sharedCapitalOverviewAnalyticsEventProperties as c, CAPITAL_OVERVIEW_CLASS_NAMES as n } from "../constants.js";
import { useLandedPageEvent as u } from "../../../../hooks/useAnalytics/useLandedPageEvent.js";
import d from "../../../internal/Button/Button.js";
import f from "../../../../core/Context/analytics/useAnalyticsContext.js";
import v from "../../../../core/Context/useCoreContext.js";
import y from "../../../internal/Typography/Typography.js";
import { TypographyVariant as h } from "../../../internal/Typography/types.js";
import C from "../../../internal/InfoBox/InfoBox.js";
import { Fragment as g } from "../../../../external/preact/dist/preact.module.js";
import { CapitalHeader as A } from "../../../internal/CapitalHeader/CapitalHeader.js";
const l = {
  ...c,
  subCategory: "Prequalified"
}, L = ({
  dynamicOfferConfig: a,
  hideTitle: m,
  onOfferOptionsRequest: i
}) => {
  const { i18n: t } = v(), r = f(), p = s(() => {
    var o;
    try {
      return i();
    } finally {
      (o = r.addEvent) == null || o.call(r, "Clicked button", { ...l, label: "See options" });
    }
  }, [i, r]);
  return u({ ...l, label: "Capital overview" }), /* @__PURE__ */ e(g, { children: [
    /* @__PURE__ */ e(A, { hideTitle: m, titleKey: "capital.overview.common.titles.qualificationIntro" }),
    /* @__PURE__ */ e("div", { className: n.preQualifiedGrant, children: [
      /* @__PURE__ */ e(C, { children: /* @__PURE__ */ e(y, { variant: h.BODY, children: [
        t.get("capital.overview.prequalified.alreadyQualifyInfo.part1"),
        /* @__PURE__ */ e("strong", { children: t.get("capital.overview.prequalified.alreadyQualifyInfo.part2", {
          values: {
            amount: t.amount(a.maxAmount.value, a.maxAmount.currency, {
              minimumFractionDigits: 0
            })
          }
        }) })
      ] }) }),
      /* @__PURE__ */ e(d, { className: n.preQualifiedGrantButton, onClick: p, children: t.get("capital.overview.prequalified.actions.seeOptions") })
    ] })
  ] });
};
export {
  L as default
};
