// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOffer/components/CapitalOffer/CapitalOffer.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as y, useState as c, useCallback as p } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { isCapitalRegionSupported as q } from "../../../../internal/CapitalHeader/helpers.js";
import { CapitalErrorMessageDisplay as F } from "../utils/CapitalErrorMessageDisplay.js";
import { CAPITAL_OFFER_CLASS_NAMES as C, sharedCapitalOfferAnalyticsEventProperties as M } from "./constants.js";
import { CapitalOfferSelection as _ } from "../CapitalOfferSelection/CapitalOfferSelection.js";
import { useFetch as x } from "../../../../../hooks/useFetch.js";
import { CapitalOfferSummary as D } from "../CapitalOfferSummary/CapitalOfferSummary.js";
import { useDurationEvent as L } from "../../../../../hooks/useAnalytics/useDurationEvent.js";
import { useLandedPageEvent as N } from "../../../../../hooks/useAnalytics/useLandedPageEvent.js";
/* empty css                  */
import { CapitalHeader as S } from "../../../../internal/CapitalHeader/CapitalHeader.js";
import { useConfigContext as g } from "../../../../../core/ConfigContext/context.js";
import { EMPTY_OBJECT as k } from "../../../../../utils/value/constants.js";
const O = {
  ...M,
  subCategory: "Capital offer"
}, B = ({
  externalDynamicOffersConfig: r,
  hideTitle: i,
  onContactSupport: n,
  onFundsRequest: f,
  onOfferDismiss: a,
  onOfferSelect: s
}) => {
  const [E, u] = c(!1), [h, v] = c(), [l, d] = c(), { getDynamicGrantOffersConfiguration: o } = g().endpoints, A = p((t) => {
    u(!t);
  }, []), { data: R, error: T } = x({
    fetchOptions: {
      enabled: !r && !!o,
      onSuccess: A
    },
    queryFn: p(async () => o == null ? void 0 : o(k), [o])
  }), b = r || R, P = p(
    (t) => {
      s ? s(t) : (v(t == null ? void 0 : t.grantAmount.value), d(t));
    },
    [s]
  ), m = y(() => l ? "OfferSummary" : "OfferSelection", [l]);
  return N({ ...O, label: "Capital offer" }), L(O), /* @__PURE__ */ e("div", { className: C.base, children: [
    /* @__PURE__ */ e(
      S,
      {
        hasDivider: !0,
        hideTitle: i,
        titleKey: m === "OfferSummary" ? "capital.offer.summary.title" : "capital.offer.selection.title"
      }
    ),
    m === "OfferSelection" && /* @__PURE__ */ e(
      _,
      {
        requestedAmount: h,
        dynamicOffersConfig: b,
        dynamicOffersConfigError: T,
        onOfferDismiss: a,
        onOfferSelect: P,
        emptyGrantOffer: E,
        onContactSupport: n
      }
    ),
    m === "OfferSummary" && /* @__PURE__ */ e(
      D,
      {
        grantOffer: l,
        onBack: () => d(void 0),
        onFundsRequest: f,
        onContactSupport: n
      }
    )
  ] });
}, Z = (r) => {
  var f, a;
  const i = (a = (f = g()) == null ? void 0 : f.extraConfig) == null ? void 0 : a.legalEntity;
  return y(() => q(i), [i]) ? /* @__PURE__ */ e(B, { ...r }) : /* @__PURE__ */ e("div", { className: C.errorContainer, children: [
    /* @__PURE__ */ e(S, { hideTitle: r.hideTitle, titleKey: "capital.common.title" }),
    /* @__PURE__ */ e(F, { unsupportedRegion: !0 })
  ] });
};
export {
  Z as CapitalOffer
};
