// @adyen/adyen-platform-experience-web/dist/es/components/internal/FilterBar/filters/BaseFilter/BaseFilter.js
import { jsx as n } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useResponsiveContainer as C, containerQueries as M } from "../../../../../hooks/useResponsiveContainer.js";
import L from "../../components/FilterButton/FilterButton.js";
import j from "../../../Popover/Popover.js";
import { PopoverContainerPosition as Q, PopoverContainerVariant as G } from "../../../Popover/types.js";
import { TypographyVariant as _, TypographyElement as D } from "../../../Typography/types.js";
import I from "../../../Typography/Typography.js";
import { memo as J } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useMemo as E, useRef as V, useCallback as R, useEffect as p, useState as K } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import m from "../../../../../hooks/useBooleanState.js";
import { useUniqueId as W } from "../../../../../hooks/useUniqueId.js";
import X from "../../../FormFields/InputText.js";
import Z from "../../../../../hooks/useCommitAction/useCommitAction.js";
import { Fragment as ee } from "../../../../../external/preact/dist/preact.module.js";
import { isEmptyString as te } from "../../../../../utils/value/string.js";
import { isNull as ie } from "../../../../../utils/value/is.js";
import { CommitAction as T } from "../../../../../hooks/useCommitAction/types.js";
const ae = (r) => !r || te(r), ne = /* @__PURE__ */ (() => {
  const r = (d) => {
    const { editAction: e, name: b, onChange: t, onValueUpdated: s } = d, [c, u] = K(d.value), i = V(null), f = R(
      (a) => {
        const y = a.target.value.trim();
        u(y), s(y);
      },
      [s]
    );
    return p(() => {
      i.current && i.current.focus();
    }, [i]), p(() => {
      if (e === T.CLEAR) {
        const a = "";
        u(a), s(a), t(a);
      }
      e === T.APPLY && t(c ?? "");
    }, [c, e, t, s]), /* @__PURE__ */ n(X, { ref: i, name: b, value: c, onInput: f });
  };
  return (d) => /* @__PURE__ */ n(r, { ...d });
})(), oe = ({ render: r, ["aria-label"]: d, ...e }) => {
  var S;
  const b = C(M.down.xs), [t, s] = m(!1), [c, u] = m(!1), i = E(() => e.isValueEmpty ?? ae, [e.isValueEmpty]), [f, a] = m(i(e.value)), [y, h] = m(!1), [w, A] = m(!1), [k, P] = m(i(e.value)), g = V(null), x = `elem-${W()}`, U = E(() => r ?? ne, [r]), Y = R(
    (o) => {
      const l = i(o ?? void 0);
      a(l), P(ie(o)), A(y ? o !== e.value : !l);
    },
    [i, a, P, A, y, e.value]
  ), { commitAction: $, commitActionButtons: q, committing: B, resetCommitAction: F } = Z({
    applyDisabled: k || !w,
    resetDisabled: f,
    onResetAction: e == null ? void 0 : e.onResetAction
  }), N = V(!1), [v, O] = E(() => {
    const o = (l) => () => {
      l === t || l && N.current || (l && (F(), A(!1), h(!1)), s(l), u(l));
    };
    return [o(!1), o(!0)];
  }, [s, t, F, u, h, A]);
  p(() => {
    if (c) {
      const o = i(e.value);
      u(!1), a(o), h(!o);
    }
  }, [e.value, c, i, u, a, h]), p(() => {
    B && v();
  }, [B, v]), p(() => {
    N.current = t;
  }, [t]);
  const z = C(M.only.sm), H = C(M.only.md);
  return /* @__PURE__ */ n(ee, { children: [
    /* @__PURE__ */ n("div", { className: `adyen-pe-filter adyen-pe-filter--${e.type}`, children: E(
      () => /* @__PURE__ */ n(
        L,
        {
          classNameModifiers: [
            ...e.appliedFilterAmount ? ["with-counter"] : [],
            ...e.classNameModifiers ?? [],
            ...t ? ["active"] : [],
            ...f ? [] : ["has-selection"]
          ],
          "aria-label": d,
          id: x,
          onClick: t ? v : O,
          ref: g,
          tabIndex: 0,
          children: /* @__PURE__ */ n("div", { className: "adyen-pe-filter-button__default-container", children: [
            /* @__PURE__ */ n(I, { el: D.SPAN, variant: _.BODY, className: "adyen-pe-filter-button__label", children: e.label }),
            !!e.appliedFilterAmount && /* @__PURE__ */ n("div", { className: "adyen-pe-filter-button__counter-wrapper", children: /* @__PURE__ */ n(
              I,
              {
                el: D.SPAN,
                variant: _.BODY,
                stronger: !0,
                className: "adyen-pe-filter-button__counter",
                children: e.appliedFilterAmount
              }
            ) })
          ] })
        }
      ),
      [
        e.appliedFilterAmount,
        e.classNameModifiers,
        e.label,
        t,
        f,
        v,
        O,
        g
      ]
    ) }),
    t && /* @__PURE__ */ n(
      j,
      {
        actions: q,
        title: (S = e.title) == null ? void 0 : S.trim(),
        variant: G.POPOVER,
        modifiers: ["filter"],
        open: t,
        dismiss: v,
        dismissible: !1,
        withContentPadding: e.withContentPadding ?? !0,
        divider: !0,
        targetElement: g,
        disableFocusTrap: !1,
        position: Q.BOTTOM,
        containerSize: e.containerSize,
        showOverlay: b,
        fitPosition: z || H,
        children: U({ ...e, editAction: $, onValueUpdated: Y })
      }
    )
  ] });
}, Me = J(oe);
export {
  Me as default
};
