// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/hooks/useTransaction/useTransaction.js
import y from "../../../../../hooks/useBalanceAccounts.js";
import { createDuplexTransactionNavigator as A } from "./transactionNavigator/createDuplexTransactionNavigator.js";
import { useRef as l, useState as T, useMemo as h, useCallback as C, useEffect as u } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { useFetch as x } from "../../../../../hooks/useFetch.js";
import { useConfigContext as B } from "../../../../../core/ConfigContext/context.js";
import { isFunction as E } from "../../../../../utils/value/is.js";
import { EMPTY_OBJECT as M } from "../../../../../utils/value/constants.js";
const v = (o) => {
  const { onNavigation: n, reset: e, ...r } = o;
  return r;
}, J = (o) => {
  const n = l(A()).current, [e, r] = T(o), [t, N] = T(), [F, I] = T(() => v(n)), { balanceAccounts: i } = y(), { getTransaction: f } = B().endpoints, {
    data: d,
    error: g,
    isFetching: c
  } = x(
    h(() => {
      const a = E(f) && !!e, s = { transactionId: e };
      return {
        fetchOptions: { enabled: a },
        queryFn: () => f(M, { path: s })
      };
    }, [f, e])
  ), p = l(c), m = l(e), S = h(() => {
    if (!t) return;
    const a = i == null ? void 0 : i.find((s) => s.id === t.balanceAccountId);
    return { ...t, balanceAccount: a };
  }, [i, t]), b = C(() => r(void 0), []);
  return u(() => {
    var a;
    switch ((t == null ? void 0 : t.id) === o && (t == null ? void 0 : t.category)) {
      case "Refund": {
        n.reset(t == null ? void 0 : t.id, (a = t == null ? void 0 : t.refundMetadata) == null ? void 0 : a.originalPaymentId), n.onNavigation = ({ to: s }) => r(s);
        break;
      }
    }
    I(v(n));
  }, [o, t, n]), u(() => {
    p.current !== c && ((p.current = c) || (!d || g ? r(m.current) : (N(d), m.current = e)));
  }, [d, g, c, e]), u(() => {
    e || r(m.current);
  }, [t, e]), u(() => () => {
    n.onNavigation = null, n.reset();
  }, [n]), {
    error: g,
    fetchingTransaction: c,
    refreshTransaction: b,
    transaction: S,
    transactionNavigator: F
  };
};
export {
  J as default,
  J as useTransaction
};
