// @adyen/adyen-platform-experience-web/dist/es/components/external/PaymentLinkSettings/components/PaymentLinkSettingsContainer/context/context.js
import { jsx as ae } from "../../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { memo as ie } from "../../../../../../external/preact/compat/dist/compat.module.js";
import { createContext as de } from "../../../../../../external/preact/dist/preact.module.js";
import { useContext as ce, useState as t, useMemo as V, useRef as le, useEffect as l, useCallback as d } from "../../../../../../external/preact/hooks/dist/hooks.module.js";
import { DEFAULT_MENU_ITEM as me, MenuItem as m } from "./constants.js";
import { useStores as ue } from "../../../../../../hooks/useStores.js";
import { useStoreTheme as ve } from "../../../hooks/useStoreTheme.js";
import { useStoreTermsAndConditions as Se } from "../../../hooks/useStoreTermsAndConditions.js";
import { useSaveAction as fe } from "../../../hooks/useSaveAction.js";
import { useResponsiveContainer as ge, containerQueries as he } from "../../../../../../hooks/useResponsiveContainer.js";
import { noop as n } from "../../../../../../utils/common.js";
const U = de({
  activeMenuItem: m.theme,
  payload: void 0,
  setPayload: n,
  selectedStore: void 0,
  setSelectedMenuItem: n,
  saveActionCalled: void 0,
  setIsValid: n,
  getIsValid: () => !1,
  setSaveActionCalled: n,
  filteredStores: void 0,
  allStores: void 0,
  setSelectedStore: n,
  savedData: void 0,
  setSavedData: () => {
  },
  menuItems: void 0,
  isLoadingContent: !0,
  isSaveError: void 0,
  isSaveSuccess: void 0,
  isSaving: void 0,
  isShowingRequirements: !1,
  onSave: n,
  setIsSaveError: n,
  setIsSaveSuccess: n,
  setIsShowingRequirements: n,
  isLoadingStores: !1,
  storesError: void 0,
  termsAndConditionsError: void 0,
  themeError: void 0
}), De = ie(
  ({
    children: _,
    selectedMenuItems: j,
    storeIds: K,
    embeddedInOverview: N,
    navigateBack: Q
  }) => {
    const [z, B] = t(!1), [r] = t(j), [G, a] = t(!1), v = ge(he.down.xs), S = V(() => {
      if (!(v && r.length > 1))
        return r.length > 0 && r[0] ? r[0].value : me;
    }, [r, v]), [s, P] = t(null), [x, u] = t(void 0), [H, i] = t(void 0), f = le(!1), [J, g] = t(!1), { filteredStores: c, selectedStore: o, setSelectedStore: h, isFetching: y, error: O, allStores: W } = ue(K), [X, D] = t(!1), [Y, C] = t(!1), [Z, I] = t(!1);
    l(() => {
      S && P(S);
    }, [S]);
    const b = d(() => f.current, []), { onSave: $ } = fe(
      D,
      C,
      I,
      o,
      x,
      s,
      b,
      g,
      i,
      u,
      Q
    ), [ee, w] = t(!1), [te, k] = t(!1), F = d(() => {
      D(!1), C(!1), I(!1), g(!1), i(void 0), u(void 0);
    }, []);
    l(() => {
      switch (F(), s) {
        case m.theme:
          a(!0), o && w(!0);
          break;
        case m.termsAndConditions:
          a(!0), o && k(!0);
          break;
      }
    }, [s, o, F]);
    const E = d(
      (e) => {
        e && u(e);
      },
      [u]
    ), {
      theme: R,
      isFetching: A,
      error: p
    } = ve(o, ee, w, E, a), {
      data: q,
      isFetching: T,
      error: L
    } = Se(o, te, k, E, a), M = V(() => {
      switch (s) {
        case m.theme:
          return A ? void 0 : R;
        case m.termsAndConditions:
          return T ? void 0 : q;
        default:
          return;
      }
    }, [s, R, q, A, T]);
    l(() => {
      M && i(M);
    }, [M, i]), l(() => {
      var e;
      o || h((e = c == null ? void 0 : c[0]) == null ? void 0 : e.id);
    }, [c, o, h]);
    const se = d(
      (e) => {
        i(e);
      },
      [i]
    );
    l(() => {
      (p || L) && a(!1);
    }, [p, L]);
    const oe = d((e) => {
      f.current !== e && (f.current = e);
    }, []), ne = d(
      (e) => {
        s !== e.value && (a(!0), P(e.value));
      },
      [s]
    ), re = G || A || T || y;
    return /* @__PURE__ */ ae(
      U.Provider,
      {
        value: {
          isLoadingContent: re,
          setPayload: E,
          menuItems: r,
          payload: x,
          activeMenuItem: s,
          selectedStore: o,
          setSelectedMenuItem: ne,
          getIsValid: b,
          setIsValid: oe,
          saveActionCalled: J,
          setSaveActionCalled: g,
          filteredStores: c,
          setSelectedStore: h,
          savedData: H,
          setSavedData: se,
          isSaveError: Y,
          isSaveSuccess: Z,
          isSaving: X,
          isShowingRequirements: z,
          setIsShowingRequirements: B,
          setIsSaveError: C,
          setIsSaveSuccess: I,
          onSave: $,
          allStores: W,
          isLoadingStores: y,
          storesError: O,
          embeddedInOverview: N,
          themeError: p,
          termsAndConditionsError: L
        },
        children: !s && !v ? null : _
      }
    );
  }
), be = () => ce(U);
export {
  U as PaymentLinkSettingsContext,
  De as PaymentLinkSettingsProvider,
  be as default,
  be as usePaymentLinkSettingsContext
};
