// @adyen/adyen-platform-experience-web/dist/es/components/internal/Alert/Alert.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import I from "classnames";
import { ButtonVariant as p } from "../Button/types.js";
import N from "../../../core/Context/useCoreContext.js";
import { TypographyVariant as l, TypographyElement as s } from "../Typography/types.js";
import c from "../Typography/Typography.js";
import { AlertIcon as A } from "./AlertIcon.js";
import { AlertVariantOption as t } from "./types.js";
/* empty css           */
import y from "../Button/Button.js";
import { Icon as g } from "../Icon/Icon.js";
const Y = ({
  className: _,
  description: n,
  closeButton: h,
  title: o,
  type: i,
  children: u,
  onClose: m,
  actions: r,
  variant: a = t.DEFAULT
}) => {
  const { i18n: T } = N();
  return /* @__PURE__ */ e("div", { className: I("adyen-pe-alert", `adyen-pe-alert--${i}`, `adyen-pe-alert--${a}`, _), role: "alert", children: [
    /* @__PURE__ */ e(A, { type: i, className: "adyen-pe-alert__icon" }),
    /* @__PURE__ */ e("div", { className: "adyen-pe-alert__content-container", children: [
      /* @__PURE__ */ e("div", { className: "adyen-pe-alert__content", children: [
        o && a !== t.TIP && /* @__PURE__ */ e(c, { className: "adyen-pe-alert__title", el: s.DIV, variant: l.BODY, wide: !0, strongest: !0, children: o }),
        n && /* @__PURE__ */ e(
          c,
          {
            className: "adyen-pe-alert__description",
            el: s.DIV,
            variant: a === t.TIP ? l.CAPTION : l.BODY,
            wide: !0,
            children: n
          }
        ),
        u
      ] }),
      (r == null ? void 0 : r.length) && /* @__PURE__ */ e("div", { className: "adyen-pe-alert__actions", children: r == null ? void 0 : r.map((d, f) => /* @__PURE__ */ e(y, { onClick: d.onClick, variant: p.TERTIARY, children: d.label }, f)) })
    ] }),
    m && (a !== t.TIP || h) && /* @__PURE__ */ e("div", { className: "adyen-pe-alert__close-button", children: /* @__PURE__ */ e(
      y,
      {
        iconButton: !0,
        variant: p.TERTIARY,
        onClick: m,
        "aria-label": T.get("common.actions.dismiss.labels.dismiss"),
        children: /* @__PURE__ */ e(g, { name: "cross" })
      }
    ) })
  ] });
};
export {
  Y as Alert,
  Y as default
};
